{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "LBNBl2exUYWu"
   },
   "source": [
    "# Ag3.7\n",
    "\n",
    "The **[Ag3.7](ag3.7):  _Anopheles gambiae_ data resource** contains single nucleotide polymorphism (SNP) calls, copy number variant (CNV) calls and SNP haplotypes from whole-genome sequencing of 2546 mosquitoes.\n",
    "\n",
    "More information about this release can be found in the [data resource website](https://www.malariagen.net/data/ag37-anopheles-gambiae-data-resource).  \n",
    "\n",
    "This page provides an introduction to open data resources released as part of `Ag3.7`. \n",
    "\n",
    "If you have any questions about this guide or how to use the data, please [start a new discussion](https://github.com/malariagen/vector-public-data/discussions/new) on the malariagen/vector-open-data repo on GitHub. If you find any bugs, please [raise an issue](https://github.com/malariagen/vector-public-data/issues/new/choose)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "kJqs4cXppk8j"
   },
   "source": [
    "## Terms of use\n",
    "\n",
    "Data from this project will be made publicly available before journal publication. Unless otherwise stated, analyses of project data are ongoing and publications are in preparation by project partners, and it is not permitted to use project data for publication (including any type of communication with the general public) without prior permission from the originating partner studies. \n",
    "\n",
    "Although malaria is generally an endemic rather than an epidemic disease, and the focus of this project is on surveillance of disease vectors rather than pathogens, our data terms of use build on MalariaGEN's approach to data sharing, and adopt norms which have been established for rapid sharing of pathogen genomic data during disease outbreaks. The primary rationale for this approach is that malaria remains a public health emergency, where ethically appropriate and rapid sharing of genomic surveillance data can help to detect and respond to biological threats such as new forms of insecticide resistance, and to adapt malaria vector control strategies to different settings and changing circumstances.\n",
    "\n",
    "The publication embargo for all data on this release will expire on the **17th of November 2025**. \n",
    "\n",
    "If you have any questions about the terms of use, please email [data@malariagen.net](mailto:data@malariagen.net)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "iNSicUCtpk8j"
   },
   "source": [
    "## Partner studies\n",
    "\n",
    "- [1190-VO-GH-AMENGA-ETEGO](https://www.malariagen.net/network/where-we-work/1190-VO-GH-AMENGA-ETEGO) -  _Population genetics, parasite diversity and ecology of the major malaria vectors in the Kassena-Nankana Districts of Ghana_\n",
    "- [1237-VO-BJ-DJOGBENOU](https://www.malariagen.net/network/where-we-work/1237-VO-BJ-DJOGBENOU) -  _Genomics for African Anopheles Resistance Diagnostics/GAARDian (Benin)_\n",
    "- [1246-VO-TZ-KABULA](https://www.malariagen.net/network/where-we-work/1246-VO-TZ-KABULA) -   _Genome-based diagnostics for monitoring and evaluation of insecticide resistance in Anopheles gambiae in Tanzania_\n",
    "- [1270-VO-MULTI-PAMGEN](https://www.malariagen.net/network/where-we-work/1270-VO-MULTI-PAMGEN) -  _Pan-African Malaria Genetic Epidemiology Network (PAMGEN) Anopheles gambiae samples (Cameroon)_\n",
    "- [1281-VO-CM-CHRISTOPHE](https://www.malariagen.net/network/where-we-work/1281-VO-CM-CHRISTOPHE) -  _PAMCA Anopheles genomics programme - Malaria vector surveillance, population genetics and insecticide resistance characterization in four eco-epidemiological settings in Cameroon_\n",
    "\n",
    "This release also includes data from 2 studies openly available in the literature: [barron-2019](https://doi.org/10.1038/s41598-019-49065-5) and [crawford-2016](https://doi.org/10.1111/mec.13572)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "5RHbe7N6pk8k"
   },
   "source": [
    "## Whole-genome sequencing and variant calling\n",
    "\n",
    "All samples in `Ag3.7` have been sequenced individually to high coverage using Illumina technology at the Wellcome Sanger Institute. These sequence data have then been analysed to identify genetic variants such as single nucleotide polymorphisms (SNPs). After variant calling, both the samples and the variants have been through a range of quality control analyses, to ensure the data are of high quality. Both the raw sequence data and the curated variant calls are openly available for download and analysis. \n",
    "\n",
    "\n",
    "For further information about the sequencing and variant calling methods used, please see the [methods page](methods)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "9Hfchko2pk8l"
   },
   "source": [
    "## Data hosting\n",
    "\n",
    "Data from `Ag3.7` are hosted by several different services. \n",
    "\n",
    "The SNP data have also been uploaded to Google Cloud, and can be analysed directly within the cloud without having to download or copy any data, including via free interactive computing services such as [MyBinder](https://gke.mybinder.org/) and [Google Colab](https://colab.research.google.com/). Further information about analysing these data in the cloud is provided in the [cloud data access guide](cloud)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "lTJ_EnvOpk8l"
   },
   "source": [
    "## Sample sets\n",
    "\n",
    "The samples included in `Ag3.7` have been organised into 9 sample sets. \n",
    "\n",
    "Each sample set corresponds to a set of mosquito specimens from a contributing study. Study details can be found in the partner studies webpages listed above."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "hGA4d7Yrpk8m",
    "outputId": "c29827c1-0361-4926-c227-8f6e76c2a497",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "!pip install -qq malariagen_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "id": "AnmzLmEgpk8n",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  const force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "const JS_MIME_TYPE = 'application/javascript';\n",
       "  const HTML_MIME_TYPE = 'text/html';\n",
       "  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  const CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    const script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    function drop(id) {\n",
       "      const view = Bokeh.index.get_by_id(id)\n",
       "      if (view != null) {\n",
       "        view.model.document.clear()\n",
       "        Bokeh.index.delete(view)\n",
       "      }\n",
       "    }\n",
       "\n",
       "    const cell = handle.cell;\n",
       "\n",
       "    const id = cell.output_area._bokeh_element_id;\n",
       "    const server_id = cell.output_area._bokeh_server_id;\n",
       "\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null) {\n",
       "      drop(id)\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd_clean, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            const id = msg.content.text.trim()\n",
       "            drop(id)\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd_destroy);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    const output_area = handle.output_area;\n",
       "    const output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      const bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      const script_attrs = bk_div.children[0].attributes;\n",
       "      for (let i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      const toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    const events = require('base/js/events');\n",
       "    const OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  const NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    const el = document.getElementById(null);\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error(url) {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < css_urls.length; i++) {\n",
       "      const url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < js_urls.length; i++) {\n",
       "      const url = js_urls[i];\n",
       "      const element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.3.0.min.js\"];\n",
       "  const css_urls = [];\n",
       "\n",
       "  const inline_js = [    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "function(Bokeh) {\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "          for (let i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n\n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    const el = document.getElementById(null);\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.3.0.min.js\"];\n  const css_urls = [];\n\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {\n    }\n  ];\n\n  function run_inline_js() {\n    if (root.Bokeh !== undefined || force === true) {\n          for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import malariagen_data\n",
    "ag3 = malariagen_data.Ag3()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 927
    },
    "id": "qsElasBepk8n",
    "outputId": "4bf80a06-c2e8-4d2d-b4a6-99c8c66da7db",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_set</th>\n",
       "      <th>sample_count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1190-VO-GH-AMENGA-ETEGO</th>\n",
       "      <td>1190-VO-GH-AMENGA-ETEGO-VMF00088</td>\n",
       "      <td>255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1190-VO-GH-AMENGA-ETEGO</th>\n",
       "      <td>1190-VO-GH-AMENGA-ETEGO-VMF00102</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1237-VO-BJ-DJOGBENOU</th>\n",
       "      <td>1237-VO-BJ-DJOGBENOU-VMF00212</td>\n",
       "      <td>451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246-VO-TZ-KABULA</th>\n",
       "      <td>1246-VO-TZ-KABULA-VMF00185</td>\n",
       "      <td>467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246-VO-TZ-KABULA</th>\n",
       "      <td>1246-VO-TZ-KABULA-VMF00197</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1270-VO-MULTI-PAMGEN</th>\n",
       "      <td>1270-VO-MULTI-PAMGEN-VMF00193-VMF00202</td>\n",
       "      <td>276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1281-VO-CM-CHRISTOPHE</th>\n",
       "      <td>1281-VO-CM-CHRISTOPHE-VMF00208</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>barron-2019</th>\n",
       "      <td>barron-2019</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>crawford-2016</th>\n",
       "      <td>crawford-2016</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     sample_set  sample_count\n",
       "study_id                                                                     \n",
       "1190-VO-GH-AMENGA-ETEGO        1190-VO-GH-AMENGA-ETEGO-VMF00088           255\n",
       "1190-VO-GH-AMENGA-ETEGO        1190-VO-GH-AMENGA-ETEGO-VMF00102           273\n",
       "1237-VO-BJ-DJOGBENOU              1237-VO-BJ-DJOGBENOU-VMF00212           451\n",
       "1246-VO-TZ-KABULA                    1246-VO-TZ-KABULA-VMF00185           467\n",
       "1246-VO-TZ-KABULA                    1246-VO-TZ-KABULA-VMF00197           327\n",
       "1270-VO-MULTI-PAMGEN     1270-VO-MULTI-PAMGEN-VMF00193-VMF00202           276\n",
       "1281-VO-CM-CHRISTOPHE            1281-VO-CM-CHRISTOPHE-VMF00208           468\n",
       "barron-2019                                         barron-2019             4\n",
       "crawford-2016                                     crawford-2016            25"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sample_sets = ag3.sample_sets(release=\"3.7\")\n",
    "df_sample_sets[['study_id','sample_set', 'sample_count']].set_index('study_id')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "yJ16OQ0Hpk8o"
   },
   "source": [
    "Here is a more detailed breakdown of the samples contained within this sample set, summarised by country, year of collection, and species:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "a1OMvuTxUWpJ",
    "outputId": "9f872334-fd50-4649-990a-df60ea71c12c",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>taxon</th>\n",
       "      <th>arabiensis</th>\n",
       "      <th>coluzzii</th>\n",
       "      <th>fontenillei</th>\n",
       "      <th>gambiae</th>\n",
       "      <th>gcx4</th>\n",
       "      <th>melas</th>\n",
       "      <th>unassigned</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_id</th>\n",
       "      <th>sample_set</th>\n",
       "      <th>country</th>\n",
       "      <th>year</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">1190-VO-GH-AMENGA-ETEGO</th>\n",
       "      <th>1190-VO-GH-AMENGA-ETEGO-VMF00088</th>\n",
       "      <th>Ghana</th>\n",
       "      <th>2018</th>\n",
       "      <td>2</td>\n",
       "      <td>246</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1190-VO-GH-AMENGA-ETEGO-VMF00102</th>\n",
       "      <th>Ghana</th>\n",
       "      <th>2018</th>\n",
       "      <td>2</td>\n",
       "      <td>268</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1237-VO-BJ-DJOGBENOU</th>\n",
       "      <th>1237-VO-BJ-DJOGBENOU-VMF00212</th>\n",
       "      <th>Benin</th>\n",
       "      <th>2018</th>\n",
       "      <td>0</td>\n",
       "      <td>380</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">1246-VO-TZ-KABULA</th>\n",
       "      <th>1246-VO-TZ-KABULA-VMF00185</th>\n",
       "      <th>Tanzania</th>\n",
       "      <th>2018</th>\n",
       "      <td>467</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1246-VO-TZ-KABULA-VMF00197</th>\n",
       "      <th>Tanzania</th>\n",
       "      <th>2019</th>\n",
       "      <td>97</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>227</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">1270-VO-MULTI-PAMGEN</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">1270-VO-MULTI-PAMGEN-VMF00193-VMF00202</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Cameroon</th>\n",
       "      <th>2020</th>\n",
       "      <td>0</td>\n",
       "      <td>69</td>\n",
       "      <td>0</td>\n",
       "      <td>173</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2021</th>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">1281-VO-CM-CHRISTOPHE</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">1281-VO-CM-CHRISTOPHE-VMF00208</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">Cameroon</th>\n",
       "      <th>2019</th>\n",
       "      <td>8</td>\n",
       "      <td>173</td>\n",
       "      <td>0</td>\n",
       "      <td>85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2020</th>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2021</th>\n",
       "      <td>13</td>\n",
       "      <td>52</td>\n",
       "      <td>0</td>\n",
       "      <td>95</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>barron-2019</th>\n",
       "      <th>barron-2019</th>\n",
       "      <th>Gabon</th>\n",
       "      <th>2014</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">crawford-2016</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">crawford-2016</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Burkina Faso</th>\n",
       "      <th>2007</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2008</th>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Guinea</th>\n",
       "      <th>2012</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "taxon                                                                             arabiensis  \\\n",
       "study_id                sample_set                             country      year               \n",
       "1190-VO-GH-AMENGA-ETEGO 1190-VO-GH-AMENGA-ETEGO-VMF00088       Ghana        2018           2   \n",
       "                        1190-VO-GH-AMENGA-ETEGO-VMF00102       Ghana        2018           2   \n",
       "1237-VO-BJ-DJOGBENOU    1237-VO-BJ-DJOGBENOU-VMF00212          Benin        2018           0   \n",
       "1246-VO-TZ-KABULA       1246-VO-TZ-KABULA-VMF00185             Tanzania     2018         467   \n",
       "                        1246-VO-TZ-KABULA-VMF00197             Tanzania     2019          97   \n",
       "1270-VO-MULTI-PAMGEN    1270-VO-MULTI-PAMGEN-VMF00193-VMF00202 Cameroon     2020           0   \n",
       "                                                                            2021           0   \n",
       "1281-VO-CM-CHRISTOPHE   1281-VO-CM-CHRISTOPHE-VMF00208         Cameroon     2019           8   \n",
       "                                                                            2020           1   \n",
       "                                                                            2021          13   \n",
       "barron-2019             barron-2019                            Gabon        2014           0   \n",
       "crawford-2016           crawford-2016                          Burkina Faso 2007           0   \n",
       "                                                                            2008           5   \n",
       "                                                               Guinea       2012           0   \n",
       "\n",
       "taxon                                                                             coluzzii  \\\n",
       "study_id                sample_set                             country      year             \n",
       "1190-VO-GH-AMENGA-ETEGO 1190-VO-GH-AMENGA-ETEGO-VMF00088       Ghana        2018       246   \n",
       "                        1190-VO-GH-AMENGA-ETEGO-VMF00102       Ghana        2018       268   \n",
       "1237-VO-BJ-DJOGBENOU    1237-VO-BJ-DJOGBENOU-VMF00212          Benin        2018       380   \n",
       "1246-VO-TZ-KABULA       1246-VO-TZ-KABULA-VMF00185             Tanzania     2018         0   \n",
       "                        1246-VO-TZ-KABULA-VMF00197             Tanzania     2019         0   \n",
       "1270-VO-MULTI-PAMGEN    1270-VO-MULTI-PAMGEN-VMF00193-VMF00202 Cameroon     2020        69   \n",
       "                                                                            2021        24   \n",
       "1281-VO-CM-CHRISTOPHE   1281-VO-CM-CHRISTOPHE-VMF00208         Cameroon     2019       173   \n",
       "                                                                            2020        25   \n",
       "                                                                            2021        52   \n",
       "barron-2019             barron-2019                            Gabon        2014         0   \n",
       "crawford-2016           crawford-2016                          Burkina Faso 2007         0   \n",
       "                                                                            2008         7   \n",
       "                                                               Guinea       2012         0   \n",
       "\n",
       "taxon                                                                             fontenillei  \\\n",
       "study_id                sample_set                             country      year                \n",
       "1190-VO-GH-AMENGA-ETEGO 1190-VO-GH-AMENGA-ETEGO-VMF00088       Ghana        2018            0   \n",
       "                        1190-VO-GH-AMENGA-ETEGO-VMF00102       Ghana        2018            0   \n",
       "1237-VO-BJ-DJOGBENOU    1237-VO-BJ-DJOGBENOU-VMF00212          Benin        2018            0   \n",
       "1246-VO-TZ-KABULA       1246-VO-TZ-KABULA-VMF00185             Tanzania     2018            0   \n",
       "                        1246-VO-TZ-KABULA-VMF00197             Tanzania     2019            0   \n",
       "1270-VO-MULTI-PAMGEN    1270-VO-MULTI-PAMGEN-VMF00193-VMF00202 Cameroon     2020            0   \n",
       "                                                                            2021            0   \n",
       "1281-VO-CM-CHRISTOPHE   1281-VO-CM-CHRISTOPHE-VMF00208         Cameroon     2019            0   \n",
       "                                                                            2020            0   \n",
       "                                                                            2021            0   \n",
       "barron-2019             barron-2019                            Gabon        2014            4   \n",
       "crawford-2016           crawford-2016                          Burkina Faso 2007            0   \n",
       "                                                                            2008            0   \n",
       "                                                               Guinea       2012            0   \n",
       "\n",
       "taxon                                                                             gambiae  \\\n",
       "study_id                sample_set                             country      year            \n",
       "1190-VO-GH-AMENGA-ETEGO 1190-VO-GH-AMENGA-ETEGO-VMF00088       Ghana        2018        6   \n",
       "                        1190-VO-GH-AMENGA-ETEGO-VMF00102       Ghana        2018        3   \n",
       "1237-VO-BJ-DJOGBENOU    1237-VO-BJ-DJOGBENOU-VMF00212          Benin        2018       21   \n",
       "1246-VO-TZ-KABULA       1246-VO-TZ-KABULA-VMF00185             Tanzania     2018        0   \n",
       "                        1246-VO-TZ-KABULA-VMF00197             Tanzania     2019      227   \n",
       "1270-VO-MULTI-PAMGEN    1270-VO-MULTI-PAMGEN-VMF00193-VMF00202 Cameroon     2020      173   \n",
       "                                                                            2021        5   \n",
       "1281-VO-CM-CHRISTOPHE   1281-VO-CM-CHRISTOPHE-VMF00208         Cameroon     2019       85   \n",
       "                                                                            2020       16   \n",
       "                                                                            2021       95   \n",
       "barron-2019             barron-2019                            Gabon        2014        0   \n",
       "crawford-2016           crawford-2016                          Burkina Faso 2007        0   \n",
       "                                                                            2008        0   \n",
       "                                                               Guinea       2012        1   \n",
       "\n",
       "taxon                                                                             gcx4  \\\n",
       "study_id                sample_set                             country      year         \n",
       "1190-VO-GH-AMENGA-ETEGO 1190-VO-GH-AMENGA-ETEGO-VMF00088       Ghana        2018     0   \n",
       "                        1190-VO-GH-AMENGA-ETEGO-VMF00102       Ghana        2018     0   \n",
       "1237-VO-BJ-DJOGBENOU    1237-VO-BJ-DJOGBENOU-VMF00212          Benin        2018     0   \n",
       "1246-VO-TZ-KABULA       1246-VO-TZ-KABULA-VMF00185             Tanzania     2018     0   \n",
       "                        1246-VO-TZ-KABULA-VMF00197             Tanzania     2019     0   \n",
       "1270-VO-MULTI-PAMGEN    1270-VO-MULTI-PAMGEN-VMF00193-VMF00202 Cameroon     2020     0   \n",
       "                                                                            2021     0   \n",
       "1281-VO-CM-CHRISTOPHE   1281-VO-CM-CHRISTOPHE-VMF00208         Cameroon     2019     0   \n",
       "                                                                            2020     0   \n",
       "                                                                            2021     0   \n",
       "barron-2019             barron-2019                            Gabon        2014     0   \n",
       "crawford-2016           crawford-2016                          Burkina Faso 2007     4   \n",
       "                                                                            2008     8   \n",
       "                                                               Guinea       2012     0   \n",
       "\n",
       "taxon                                                                             melas  \\\n",
       "study_id                sample_set                             country      year          \n",
       "1190-VO-GH-AMENGA-ETEGO 1190-VO-GH-AMENGA-ETEGO-VMF00088       Ghana        2018      0   \n",
       "                        1190-VO-GH-AMENGA-ETEGO-VMF00102       Ghana        2018      0   \n",
       "1237-VO-BJ-DJOGBENOU    1237-VO-BJ-DJOGBENOU-VMF00212          Benin        2018     50   \n",
       "1246-VO-TZ-KABULA       1246-VO-TZ-KABULA-VMF00185             Tanzania     2018      0   \n",
       "                        1246-VO-TZ-KABULA-VMF00197             Tanzania     2019      0   \n",
       "1270-VO-MULTI-PAMGEN    1270-VO-MULTI-PAMGEN-VMF00193-VMF00202 Cameroon     2020      0   \n",
       "                                                                            2021      4   \n",
       "1281-VO-CM-CHRISTOPHE   1281-VO-CM-CHRISTOPHE-VMF00208         Cameroon     2019      0   \n",
       "                                                                            2020      0   \n",
       "                                                                            2021      0   \n",
       "barron-2019             barron-2019                            Gabon        2014      0   \n",
       "crawford-2016           crawford-2016                          Burkina Faso 2007      0   \n",
       "                                                                            2008      0   \n",
       "                                                               Guinea       2012      0   \n",
       "\n",
       "taxon                                                                             unassigned  \n",
       "study_id                sample_set                             country      year              \n",
       "1190-VO-GH-AMENGA-ETEGO 1190-VO-GH-AMENGA-ETEGO-VMF00088       Ghana        2018           1  \n",
       "                        1190-VO-GH-AMENGA-ETEGO-VMF00102       Ghana        2018           0  \n",
       "1237-VO-BJ-DJOGBENOU    1237-VO-BJ-DJOGBENOU-VMF00212          Benin        2018           0  \n",
       "1246-VO-TZ-KABULA       1246-VO-TZ-KABULA-VMF00185             Tanzania     2018           0  \n",
       "                        1246-VO-TZ-KABULA-VMF00197             Tanzania     2019           3  \n",
       "1270-VO-MULTI-PAMGEN    1270-VO-MULTI-PAMGEN-VMF00193-VMF00202 Cameroon     2020           1  \n",
       "                                                                            2021           0  \n",
       "1281-VO-CM-CHRISTOPHE   1281-VO-CM-CHRISTOPHE-VMF00208         Cameroon     2019           0  \n",
       "                                                                            2020           0  \n",
       "                                                                            2021           0  \n",
       "barron-2019             barron-2019                            Gabon        2014           0  \n",
       "crawford-2016           crawford-2016                          Burkina Faso 2007           0  \n",
       "                                                                            2008           0  \n",
       "                                                               Guinea       2012           0  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_samples = ag3.sample_metadata(sample_sets=\"3.7\")\n",
    "df_summary = df_samples.pivot_table(\n",
    "    index=[\"study_id\",\"sample_set\", \"country\", \"year\"], \n",
    "    columns=[\"taxon\"],\n",
    "    values=\"sample_id\", \n",
    "    aggfunc=len,\n",
    "    fill_value=0)\n",
    "df_summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "dLiU0ulIpk8p"
   },
   "source": [
    "Note that there can be multiple sampling sites represented within the same sample set."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "OToX5vhfpk8p"
   },
   "source": [
    "## Further reading\n",
    "\n",
    "We hope this page has provided a useful introduction to the `Ag3.7` data resource. If you would like to start working with these data, please visit the [cloud data access guide](cloud) or the [data download guide](download) or continue browsing the other documentation on this site.\n",
    "\n",
    "If you have any questions about the data and how to use them, please do get in touch by [starting a new discussion](https://github.com/malariagen/vector-data/discussions/new) on the malariagen/vector-data repository on GitHub."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Tags",
  "colab": {
   "name": "Ag3.0-intro.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
