{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "LBNBl2exUYWu"
   },
   "source": [
    "# Ag3.8\n",
    "\n",
    "The **[Ag3.8](ag3.8):  _Anopheles gambiae_ data resource** contains single nucleotide polymorphism (SNP) calls, copy number variant (CNV) calls and SNP haplotypes from whole-genome sequencing of 2261 mosquitoes.\n",
    "\n",
    "More information about this release can be found in the [data resource website](https://www.malariagen.net/data_package/ag38-anopheles-gambiae-data-resource).  \n",
    "\n",
    "This page provides an introduction to open data resources released as part of `Ag3.8`. \n",
    "\n",
    "If you have any questions about this guide or how to use the data, please [start a new discussion](https://github.com/malariagen/vector-public-data/discussions/new) on the malariagen/vector-open-data repo on GitHub. If you find any bugs, please [raise an issue](https://github.com/malariagen/vector-public-data/issues/new/choose)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "kJqs4cXppk8j"
   },
   "source": [
    "## Terms of use\n",
    "\n",
    "Data from this project will be made publicly available before journal publication, subject to the following publication embargo: unless otherwise stated, analyses of project data are ongoing and publications are in preparation by project partners, and it is not permitted to use project data for publication (including any type of communication with the general public) without prior permission from the originating partner studies. The publication embargo will expire 24 months after the data is integrated into the Malaria Genome Vector Observatory data repository, or earlier, if the project partner agrees to remove the embargo before the expiry date.\n",
    "\n",
    "Although malaria is generally an endemic rather than an epidemic disease, and the focus of this project is on surveillance of disease vectors rather than pathogens, our data terms of use build on MalariaGEN's approach to data sharing, and adopt norms which have been established for rapid sharing of pathogen genomic data during disease outbreaks. The primary rationale for this approach is that malaria remains a public health emergency, where ethically appropriate and rapid sharing of genomic surveillance data can help to detect and respond to biological threats such as new forms of insecticide resistance, and to adapt malaria vector control strategies to different settings and changing circumstances.\n",
    "\n",
    "The publication embargo for all data on this release will expire on the **17th of November 2025**. \n",
    "\n",
    "If you have any questions about the terms of use, please email support@malariagen.net"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "iNSicUCtpk8j"
   },
   "source": [
    "## Partner studies\n",
    "\n",
    "- [1230-VO-MULTI-AYALA](https://www.malariagen.net/network/where-we-work/1230-VO-MULTI-AYALA) -  _ANOSPP screening of Anopheles species and Plasmodium presence in malaria vectors in West-Central Africa (Gabon)_\n",
    "- [1288-VO-UG-DONNELLY](https://www.malariagen.net/network/where-we-work/1288-VO-UG-DONNELLY) -  _LLIN Evaluation in Uganda Project (LLINEUP)_\n",
    "- [1314-VO-BF-KIENTEGA](https://www.malariagen.net/network/where-we-work/1314-VO-BF-KIENTEGA) -   _ANOSPP screening of Anopheles species and Plasmodium presence in malaria vectors in Burkina Faso_\n",
    "- [1315-VO-NG-OMITOLA](https://www.malariagen.net/network/where-we-work/1315-VO-NG-OMITOLA) -  _ANOSPP screening of Anopheles species and Plasmodium presence in malaria vectors in Nigeria_\n",
    "- [1326-VO-UG-KAYONDO](https://www.malariagen.net/network/where-we-work/1326-VO-UG-KAYONDO) -  _ANOSPP screening of Anopheles species and Plasmodium presence in malaria vectors in Uganda_\n",
    "\n",
    "This release also includes data from one study openly available in the literature: [tennessen-2021](https://doi.org/10.1111/mec.15756)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "5RHbe7N6pk8k"
   },
   "source": [
    "## Whole-genome sequencing and variant calling\n",
    "\n",
    "All samples in `Ag3.8` have been sequenced individually to high coverage using Illumina technology at the Wellcome Sanger Institute. These sequence data have then been analysed to identify genetic variants such as single nucleotide polymorphisms (SNPs). After variant calling, both the samples and the variants have been through a range of quality control analyses, to ensure the data are of high quality. Both the raw sequence data and the curated variant calls are openly available for download and analysis. \n",
    "\n",
    "\n",
    "For further information about the sequencing and variant calling methods used, please see the [methods page](methods)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "9Hfchko2pk8l"
   },
   "source": [
    "## Data hosting\n",
    "\n",
    "Data from `Ag3.8` are hosted by several different services. \n",
    "\n",
    "The SNP data have also been uploaded to Google Cloud, and can be analysed directly within the cloud without having to download or copy any data, including via free interactive computing services such as [Google Colab](https://colab.research.google.com/). Further information about analysing these data in the cloud is provided in the [cloud data access guide](cloud)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "lTJ_EnvOpk8l"
   },
   "source": [
    "## Sample sets\n",
    "\n",
    "The samples included in `Ag3.8` have been organised into 6 sample sets. \n",
    "\n",
    "Each sample set corresponds to a set of mosquito specimens from a contributing study. Study details can be found in the partner studies webpages listed above."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "hGA4d7Yrpk8m",
    "outputId": "c29827c1-0361-4926-c227-8f6e76c2a497",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "%pip install -qq malariagen_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "id": "AnmzLmEgpk8n",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  const force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "const JS_MIME_TYPE = 'application/javascript';\n",
       "  const HTML_MIME_TYPE = 'text/html';\n",
       "  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  const CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    const script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    function drop(id) {\n",
       "      const view = Bokeh.index.get_by_id(id)\n",
       "      if (view != null) {\n",
       "        view.model.document.clear()\n",
       "        Bokeh.index.delete(view)\n",
       "      }\n",
       "    }\n",
       "\n",
       "    const cell = handle.cell;\n",
       "\n",
       "    const id = cell.output_area._bokeh_element_id;\n",
       "    const server_id = cell.output_area._bokeh_server_id;\n",
       "\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null) {\n",
       "      drop(id)\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd_clean, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            const id = msg.content.text.trim()\n",
       "            drop(id)\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd_destroy);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    const output_area = handle.output_area;\n",
       "    const output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      const bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      const script_attrs = bk_div.children[0].attributes;\n",
       "      for (let i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      const toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    const events = require('base/js/events');\n",
       "    const OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  const NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    const el = document.getElementById(null);\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error(url) {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < css_urls.length; i++) {\n",
       "      const url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < js_urls.length; i++) {\n",
       "      const url = js_urls[i];\n",
       "      const element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.3.0.min.js\"];\n",
       "  const css_urls = [];\n",
       "\n",
       "  const inline_js = [    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "function(Bokeh) {\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "          for (let i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n\n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    const el = document.getElementById(null);\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.3.0.min.js\"];\n  const css_urls = [];\n\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {\n    }\n  ];\n\n  function run_inline_js() {\n    if (root.Bokeh !== undefined || force === true) {\n          for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import malariagen_data\n",
    "ag3 = malariagen_data.Ag3()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 927
    },
    "id": "qsElasBepk8n",
    "outputId": "4bf80a06-c2e8-4d2d-b4a6-99c8c66da7db",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_set</th>\n",
       "      <th>sample_count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1230-VO-MULTI-AYALA</th>\n",
       "      <td>1230-VO-MULTI-AYALA-AYDI-GA-2204</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1288-VO-UG-DONNELLY</th>\n",
       "      <td>1288-VO-UG-DONNELLY-VMF00219</td>\n",
       "      <td>739</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1314-VO-BF-KIENTEGA</th>\n",
       "      <td>1314-VO-BF-KIENTEGA-KIMA-BF-2104</td>\n",
       "      <td>179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1315-VO-NG-OMITOLA</th>\n",
       "      <td>1315-VO-NG-OMITOLA-OMOL-NG-2008</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1326-VO-UG-KAYONDO</th>\n",
       "      <td>1326-VO-UG-KAYONDO-KAJO-UG-2203</td>\n",
       "      <td>975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tennessen-2021</th>\n",
       "      <td>tennessen-2021</td>\n",
       "      <td>208</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           sample_set  sample_count\n",
       "study_id                                                           \n",
       "1230-VO-MULTI-AYALA  1230-VO-MULTI-AYALA-AYDI-GA-2204            43\n",
       "1288-VO-UG-DONNELLY      1288-VO-UG-DONNELLY-VMF00219           739\n",
       "1314-VO-BF-KIENTEGA  1314-VO-BF-KIENTEGA-KIMA-BF-2104           179\n",
       "1315-VO-NG-OMITOLA    1315-VO-NG-OMITOLA-OMOL-NG-2008           117\n",
       "1326-VO-UG-KAYONDO    1326-VO-UG-KAYONDO-KAJO-UG-2203           975\n",
       "tennessen-2021                         tennessen-2021           208"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sample_sets = ag3.sample_sets(release=\"3.8\")\n",
    "df_sample_sets[['study_id','sample_set', 'sample_count']].set_index('study_id')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "yJ16OQ0Hpk8o"
   },
   "source": [
    "Here is a more detailed breakdown of the samples contained within this sample set, summarised by country, year of collection, and species:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "a1OMvuTxUWpJ",
    "outputId": "9f872334-fd50-4649-990a-df60ea71c12c",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>taxon</th>\n",
       "      <th>arabiensis</th>\n",
       "      <th>coluzzii</th>\n",
       "      <th>fontenillei</th>\n",
       "      <th>gambiae</th>\n",
       "      <th>gcx4</th>\n",
       "      <th>unassigned</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_id</th>\n",
       "      <th>sample_set</th>\n",
       "      <th>country</th>\n",
       "      <th>year</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">1230-VO-MULTI-AYALA</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">1230-VO-MULTI-AYALA-AYDI-GA-2204</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Gabon</th>\n",
       "      <th>2019</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2020</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">1288-VO-UG-DONNELLY</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">1288-VO-UG-DONNELLY-VMF00219</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">Uganda</th>\n",
       "      <th>2017</th>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>323</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2018</th>\n",
       "      <td>59</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>70</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2019</th>\n",
       "      <td>26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>243</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">1314-VO-BF-KIENTEGA</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">1314-VO-BF-KIENTEGA-KIMA-BF-2104</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">Burkina Faso</th>\n",
       "      <th>2017</th>\n",
       "      <td>7</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2018</th>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2019</th>\n",
       "      <td>36</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1315-VO-NG-OMITOLA</th>\n",
       "      <th>1315-VO-NG-OMITOLA-OMOL-NG-2008</th>\n",
       "      <th>Nigeria</th>\n",
       "      <th>2018</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>101</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">1326-VO-UG-KAYONDO</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">1326-VO-UG-KAYONDO-KAJO-UG-2203</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">Uganda</th>\n",
       "      <th>2013</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2014</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>153</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2015</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>94</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2016</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>131</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2017</th>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>190</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2018</th>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>327</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">tennessen-2021</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">tennessen-2021</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">Burkina Faso</th>\n",
       "      <th>2011</th>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2012</th>\n",
       "      <td>0</td>\n",
       "      <td>63</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2015</th>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2016</th>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "taxon                                                                   arabiensis  \\\n",
       "study_id            sample_set                       country      year               \n",
       "1230-VO-MULTI-AYALA 1230-VO-MULTI-AYALA-AYDI-GA-2204 Gabon        2019           0   \n",
       "                                                                  2020           0   \n",
       "1288-VO-UG-DONNELLY 1288-VO-UG-DONNELLY-VMF00219     Uganda       2017          17   \n",
       "                                                                  2018          59   \n",
       "                                                                  2019          26   \n",
       "1314-VO-BF-KIENTEGA 1314-VO-BF-KIENTEGA-KIMA-BF-2104 Burkina Faso 2017           7   \n",
       "                                                                  2018           9   \n",
       "                                                                  2019          36   \n",
       "1315-VO-NG-OMITOLA  1315-VO-NG-OMITOLA-OMOL-NG-2008  Nigeria      2018           0   \n",
       "1326-VO-UG-KAYONDO  1326-VO-UG-KAYONDO-KAJO-UG-2203  Uganda       2013           1   \n",
       "                                                                  2014           0   \n",
       "                                                                  2015           2   \n",
       "                                                                  2016           0   \n",
       "                                                                  2017          19   \n",
       "                                                                  2018          12   \n",
       "tennessen-2021      tennessen-2021                   Burkina Faso 2011           0   \n",
       "                                                                  2012           0   \n",
       "                                                                  2015           0   \n",
       "                                                                  2016           0   \n",
       "\n",
       "taxon                                                                   coluzzii  \\\n",
       "study_id            sample_set                       country      year             \n",
       "1230-VO-MULTI-AYALA 1230-VO-MULTI-AYALA-AYDI-GA-2204 Gabon        2019         0   \n",
       "                                                                  2020         0   \n",
       "1288-VO-UG-DONNELLY 1288-VO-UG-DONNELLY-VMF00219     Uganda       2017         0   \n",
       "                                                                  2018         0   \n",
       "                                                                  2019         0   \n",
       "1314-VO-BF-KIENTEGA 1314-VO-BF-KIENTEGA-KIMA-BF-2104 Burkina Faso 2017        19   \n",
       "                                                                  2018        16   \n",
       "                                                                  2019        43   \n",
       "1315-VO-NG-OMITOLA  1315-VO-NG-OMITOLA-OMOL-NG-2008  Nigeria      2018        16   \n",
       "1326-VO-UG-KAYONDO  1326-VO-UG-KAYONDO-KAJO-UG-2203  Uganda       2013         0   \n",
       "                                                                  2014         0   \n",
       "                                                                  2015         0   \n",
       "                                                                  2016         0   \n",
       "                                                                  2017         0   \n",
       "                                                                  2018         0   \n",
       "tennessen-2021      tennessen-2021                   Burkina Faso 2011        18   \n",
       "                                                                  2012        63   \n",
       "                                                                  2015        33   \n",
       "                                                                  2016        53   \n",
       "\n",
       "taxon                                                                   fontenillei  \\\n",
       "study_id            sample_set                       country      year                \n",
       "1230-VO-MULTI-AYALA 1230-VO-MULTI-AYALA-AYDI-GA-2204 Gabon        2019           24   \n",
       "                                                                  2020           15   \n",
       "1288-VO-UG-DONNELLY 1288-VO-UG-DONNELLY-VMF00219     Uganda       2017            0   \n",
       "                                                                  2018            0   \n",
       "                                                                  2019            0   \n",
       "1314-VO-BF-KIENTEGA 1314-VO-BF-KIENTEGA-KIMA-BF-2104 Burkina Faso 2017            0   \n",
       "                                                                  2018            0   \n",
       "                                                                  2019            0   \n",
       "1315-VO-NG-OMITOLA  1315-VO-NG-OMITOLA-OMOL-NG-2008  Nigeria      2018            0   \n",
       "1326-VO-UG-KAYONDO  1326-VO-UG-KAYONDO-KAJO-UG-2203  Uganda       2013            0   \n",
       "                                                                  2014            0   \n",
       "                                                                  2015            0   \n",
       "                                                                  2016            0   \n",
       "                                                                  2017            0   \n",
       "                                                                  2018            0   \n",
       "tennessen-2021      tennessen-2021                   Burkina Faso 2011            0   \n",
       "                                                                  2012            0   \n",
       "                                                                  2015            0   \n",
       "                                                                  2016            0   \n",
       "\n",
       "taxon                                                                   gambiae  \\\n",
       "study_id            sample_set                       country      year            \n",
       "1230-VO-MULTI-AYALA 1230-VO-MULTI-AYALA-AYDI-GA-2204 Gabon        2019        1   \n",
       "                                                                  2020        0   \n",
       "1288-VO-UG-DONNELLY 1288-VO-UG-DONNELLY-VMF00219     Uganda       2017      323   \n",
       "                                                                  2018       70   \n",
       "                                                                  2019      243   \n",
       "1314-VO-BF-KIENTEGA 1314-VO-BF-KIENTEGA-KIMA-BF-2104 Burkina Faso 2017       15   \n",
       "                                                                  2018       18   \n",
       "                                                                  2019       16   \n",
       "1315-VO-NG-OMITOLA  1315-VO-NG-OMITOLA-OMOL-NG-2008  Nigeria      2018      101   \n",
       "1326-VO-UG-KAYONDO  1326-VO-UG-KAYONDO-KAJO-UG-2203  Uganda       2013       46   \n",
       "                                                                  2014      153   \n",
       "                                                                  2015       94   \n",
       "                                                                  2016      131   \n",
       "                                                                  2017      190   \n",
       "                                                                  2018      327   \n",
       "tennessen-2021      tennessen-2021                   Burkina Faso 2011        0   \n",
       "                                                                  2012        0   \n",
       "                                                                  2015        0   \n",
       "                                                                  2016        0   \n",
       "\n",
       "taxon                                                                   gcx4  \\\n",
       "study_id            sample_set                       country      year         \n",
       "1230-VO-MULTI-AYALA 1230-VO-MULTI-AYALA-AYDI-GA-2204 Gabon        2019     0   \n",
       "                                                                  2020     0   \n",
       "1288-VO-UG-DONNELLY 1288-VO-UG-DONNELLY-VMF00219     Uganda       2017     0   \n",
       "                                                                  2018     0   \n",
       "                                                                  2019     0   \n",
       "1314-VO-BF-KIENTEGA 1314-VO-BF-KIENTEGA-KIMA-BF-2104 Burkina Faso 2017     0   \n",
       "                                                                  2018     0   \n",
       "                                                                  2019     0   \n",
       "1315-VO-NG-OMITOLA  1315-VO-NG-OMITOLA-OMOL-NG-2008  Nigeria      2018     0   \n",
       "1326-VO-UG-KAYONDO  1326-VO-UG-KAYONDO-KAJO-UG-2203  Uganda       2013     0   \n",
       "                                                                  2014     0   \n",
       "                                                                  2015     0   \n",
       "                                                                  2016     0   \n",
       "                                                                  2017     0   \n",
       "                                                                  2018     0   \n",
       "tennessen-2021      tennessen-2021                   Burkina Faso 2011    41   \n",
       "                                                                  2012     0   \n",
       "                                                                  2015     0   \n",
       "                                                                  2016     0   \n",
       "\n",
       "taxon                                                                   unassigned  \n",
       "study_id            sample_set                       country      year              \n",
       "1230-VO-MULTI-AYALA 1230-VO-MULTI-AYALA-AYDI-GA-2204 Gabon        2019           2  \n",
       "                                                                  2020           1  \n",
       "1288-VO-UG-DONNELLY 1288-VO-UG-DONNELLY-VMF00219     Uganda       2017           0  \n",
       "                                                                  2018           1  \n",
       "                                                                  2019           0  \n",
       "1314-VO-BF-KIENTEGA 1314-VO-BF-KIENTEGA-KIMA-BF-2104 Burkina Faso 2017           0  \n",
       "                                                                  2018           0  \n",
       "                                                                  2019           0  \n",
       "1315-VO-NG-OMITOLA  1315-VO-NG-OMITOLA-OMOL-NG-2008  Nigeria      2018           0  \n",
       "1326-VO-UG-KAYONDO  1326-VO-UG-KAYONDO-KAJO-UG-2203  Uganda       2013           0  \n",
       "                                                                  2014           0  \n",
       "                                                                  2015           0  \n",
       "                                                                  2016           0  \n",
       "                                                                  2017           0  \n",
       "                                                                  2018           0  \n",
       "tennessen-2021      tennessen-2021                   Burkina Faso 2011           0  \n",
       "                                                                  2012           0  \n",
       "                                                                  2015           0  \n",
       "                                                                  2016           0  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_samples = ag3.sample_metadata(sample_sets=\"3.8\")\n",
    "df_summary = df_samples.pivot_table(\n",
    "    index=[\"study_id\",\"sample_set\", \"country\", \"year\"], \n",
    "    columns=[\"taxon\"],\n",
    "    values=\"sample_id\", \n",
    "    aggfunc=len,\n",
    "    fill_value=0)\n",
    "df_summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "dLiU0ulIpk8p"
   },
   "source": [
    "Note that there can be multiple sampling sites represented within the same sample set."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "OToX5vhfpk8p"
   },
   "source": [
    "## Further reading\n",
    "\n",
    "We hope this page has provided a useful introduction to the `Ag3.8` data resource. If you would like to start working with these data, please visit the [cloud data access guide](cloud) or the [data download guide](download) or continue browsing the other documentation on this site.\n",
    "\n",
    "If you have any questions about the data and how to use them, please do get in touch by [starting a new discussion](https://github.com/malariagen/vector-data/discussions/new) on the malariagen/vector-data repository on GitHub."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Tags",
  "colab": {
   "name": "Ag3.0-intro.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "global-global-mgenv-6.0.6",
   "language": "python",
   "name": "conda-env-global-global-mgenv-6.0.6-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
