{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "LBNBl2exUYWu"
   },
   "source": [
    "# Ag3.0 (Ag1000G phase 3)\n",
    "\n",
    "The [*Anopheles gambiae* 1000 Genomes Project (Ag1000G)](https://www.malariagen.net/ag1000g) is a collaborative project using whole-genome sequencing to study genetic variation and evolution in natural populations of mosquitoes in the *Anopheles gambiae* species complex. \n",
    "\n",
    "This page provides an introduction to open data resources released as part of the third phase of the Ag1000G project, known as `Ag3.0` for short. We hope the data from `Ag3.0` will be a valuable resource for research and surveillance of malaria vectors. If you have any questions about this guide or how to use the data, please [start a new discussion](https://github.com/malariagen/vector-public-data/discussions/new) on the malariagen/vector-open-data repo on GitHub. If you find any bugs, please [raise an issue](https://github.com/malariagen/vector-public-data/issues/new/choose)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "kJqs4cXppk8j"
   },
   "source": [
    "## Terms of use\n",
    "\n",
    "Data from `Ag3.0` are released openly and can be downloaded and analysed for any purpose. The data have been released prior to publication by the Ag1000G Consortium, and are currently subject to a publication embargo described further in the [Ag1000G terms of use](https://www.malariagen.net/data/terms-use/ag1000g-terms-use). If you have any questions about the terms of use, please email data@malariagen.net."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "iNSicUCtpk8j"
   },
   "source": [
    "## Partner studies\n",
    "\n",
    "The Ag1000G project is coordinated by a consortium of partners from a range of different research institutions and countries. This includes consortium members who are carrying out independent research studies in malaria-endemic regions, and who have contributed mosquito specimens or mosquito DNA samples collected in the course of their own research. In total, 26 studies contributed samples to `Ag3.0`, including wild-caught specimens from 19 countries. For further information about these contributing studies, the researchers involved, and the collection sites and methods, see the {doc}`../studies-ag1000g` page."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "Ijw7S0-Ypk8k"
   },
   "source": [
    "## Population sampling\n",
    "\n",
    "`Ag3.0` includes data from 3,081 individual mosquitoes, including 2,784 mosquitoes collected from natural populations in 19 countries. Three species are represented within the cohort: *Anopheles gambiae*, *Anopheles coluzzii* and *Anopheles arabiensis*. The map below provides an overview of the collection locations and the numbers of samples broken down by species. \n",
    "\n",
    "```{image} ../images/ag3-map.png\n",
    ":alt: Ag3 map of sampling sites\n",
    ":class: bg-primary\n",
    ":width: 700px\n",
    ":align: center\n",
    "```\n",
    "\n",
    "In addition to these wild-caught samples, a further 297 samples are included from 15 lab crosses."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "5RHbe7N6pk8k"
   },
   "source": [
    "## Whole-genome sequencing and variant calling\n",
    "\n",
    "All samples in `Ag3.0` have been sequenced individually to high coverage using Illumina technology at the Wellcome Sanger Institute. These sequence data have then been analysed to identify genetic variants such as single nucleotide polymorphisms (SNPs). After variant calling, both the samples and the variants have been through a range of quality control analyses, to ensure the data are of high quality. Both the raw sequence data and the curated variant calls are openly available for download and analysis. \n",
    "\n",
    "\n",
    "For further information about the sequencing and variant calling methods used, please see the [methods page](methods)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "9Hfchko2pk8l"
   },
   "source": [
    "## Data hosting\n",
    "\n",
    "Data from `Ag3.0` are hosted by several different services. \n",
    "\n",
    "Raw sequence reads, sequence read alignments and SNP calls are available for download from the European Nucleotide Archive (ENA). Further information on how to find and download these data is provided in the [data download guide](download).\n",
    "\n",
    "The SNP data have also been uploaded to Google Cloud, and can be analysed directly within the cloud without having to download or copy any data, including via free interactive computing services such as [MyBinder](https://gke.mybinder.org/) and [Google Colab](https://colab.research.google.com/). Further information about analysing these data in the cloud is provided in the [cloud data access guide](cloud)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "lTJ_EnvOpk8l"
   },
   "source": [
    "## Sample sets\n",
    "\n",
    "The samples included in `Ag3.0` have been organised into 28 sample sets. Each of these sample sets corresponds to a set of mosquito specimens from a contributing study. Depending on your objectives, you may want to access data from only specific sample sets, or all sample sets. Here is a list of the sample sets included:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "hGA4d7Yrpk8m",
    "outputId": "c29827c1-0361-4926-c227-8f6e76c2a497",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "%pip install -q malariagen_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "id": "AnmzLmEgpk8n",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  const force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "const JS_MIME_TYPE = 'application/javascript';\n",
       "  const HTML_MIME_TYPE = 'text/html';\n",
       "  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  const CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    const script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    function drop(id) {\n",
       "      const view = Bokeh.index.get_by_id(id)\n",
       "      if (view != null) {\n",
       "        view.model.document.clear()\n",
       "        Bokeh.index.delete(view)\n",
       "      }\n",
       "    }\n",
       "\n",
       "    const cell = handle.cell;\n",
       "\n",
       "    const id = cell.output_area._bokeh_element_id;\n",
       "    const server_id = cell.output_area._bokeh_server_id;\n",
       "\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null) {\n",
       "      drop(id)\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd_clean, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            const id = msg.content.text.trim()\n",
       "            drop(id)\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd_destroy);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    const output_area = handle.output_area;\n",
       "    const output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      const bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      const script_attrs = bk_div.children[0].attributes;\n",
       "      for (let i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      const toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    const events = require('base/js/events');\n",
       "    const OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  const NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    const el = document.getElementById(null);\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error(url) {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < css_urls.length; i++) {\n",
       "      const url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < js_urls.length; i++) {\n",
       "      const url = js_urls[i];\n",
       "      const element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.3.0.min.js\"];\n",
       "  const css_urls = [];\n",
       "\n",
       "  const inline_js = [    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "function(Bokeh) {\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "          for (let i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n\n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    const el = document.getElementById(null);\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.3.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.3.0.min.js\"];\n  const css_urls = [];\n\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {\n    }\n  ];\n\n  function run_inline_js() {\n    if (root.Bokeh !== undefined || force === true) {\n          for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import malariagen_data\n",
    "ag3 = malariagen_data.Ag3()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 927
    },
    "id": "qsElasBepk8n",
    "outputId": "4bf80a06-c2e8-4d2d-b4a6-99c8c66da7db",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_set</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AG1000G-AO</th>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-BF-A</th>\n",
       "      <td>181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-BF-B</th>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-BF-C</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CD</th>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CF</th>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CI</th>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CM-A</th>\n",
       "      <td>303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CM-B</th>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CM-C</th>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-FR</th>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GA-A</th>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GH</th>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GM-A</th>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GM-B</th>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GM-C</th>\n",
       "      <td>174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GN-A</th>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GN-B</th>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GQ</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GW</th>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-KE</th>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-ML-A</th>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-ML-B</th>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-MW</th>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-MZ</th>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-TZ</th>\n",
       "      <td>300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-UG</th>\n",
       "      <td>290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-X</th>\n",
       "      <td>669</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              sample_count\n",
       "sample_set                \n",
       "AG1000G-AO              81\n",
       "AG1000G-BF-A           181\n",
       "AG1000G-BF-B           102\n",
       "AG1000G-BF-C            13\n",
       "AG1000G-CD              76\n",
       "AG1000G-CF              73\n",
       "AG1000G-CI              80\n",
       "AG1000G-CM-A           303\n",
       "AG1000G-CM-B            97\n",
       "AG1000G-CM-C            44\n",
       "AG1000G-FR              23\n",
       "AG1000G-GA-A            69\n",
       "AG1000G-GH             100\n",
       "AG1000G-GM-A            74\n",
       "AG1000G-GM-B            31\n",
       "AG1000G-GM-C           174\n",
       "AG1000G-GN-A            45\n",
       "AG1000G-GN-B           185\n",
       "AG1000G-GQ              10\n",
       "AG1000G-GW             101\n",
       "AG1000G-KE              86\n",
       "AG1000G-ML-A            60\n",
       "AG1000G-ML-B            71\n",
       "AG1000G-MW              41\n",
       "AG1000G-MZ              74\n",
       "AG1000G-TZ             300\n",
       "AG1000G-UG             290\n",
       "AG1000G-X              669"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sample_sets = ag3.sample_sets(release=\"3.0\")\n",
    "df_sample_sets[['sample_set', 'sample_count']].set_index('sample_set')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "yJ16OQ0Hpk8o"
   },
   "source": [
    "The sample set identifiers all start with \"AG1000G-\" followed by the two-letter code of the country from which samples were collected (e.g., \"AO\" is Angola). Where there are multiple sample sets from the same country, these have been given alphabetical suffixes, e.g., \"AG1000G-BF-A\", \"AG1000G-BF-B\" and \"AG1000G-BF-C\" are three sample sets from Burkina Faso.\n",
    "\n",
    "These country code suffixes are just a convenience to help remember which sample sets contain which data, please see the sample metadata for more precise location information. Note also that sample set AG1000G-GN-B contains samples from both Guinea and Mali.\n",
    "\n",
    "Here is a more detailed breakdown of the samples contained within each sample set, summarised by country, year of collection, and species:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "a1OMvuTxUWpJ",
    "outputId": "9f872334-fd50-4649-990a-df60ea71c12c",
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>taxon</th>\n",
       "      <th>arabiensis</th>\n",
       "      <th>coluzzii</th>\n",
       "      <th>gambiae</th>\n",
       "      <th>gcx1</th>\n",
       "      <th>gcx2</th>\n",
       "      <th>gcx3</th>\n",
       "      <th>unassigned</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>study_id</th>\n",
       "      <th>sample_set</th>\n",
       "      <th>country</th>\n",
       "      <th>year</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AG1000G-AO</th>\n",
       "      <th>AG1000G-AO</th>\n",
       "      <th>Angola</th>\n",
       "      <th>2009</th>\n",
       "      <td>0</td>\n",
       "      <td>81</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">AG1000G-BF-1</th>\n",
       "      <th>AG1000G-BF-A</th>\n",
       "      <th>Burkina Faso</th>\n",
       "      <th>2012</th>\n",
       "      <td>0</td>\n",
       "      <td>82</td>\n",
       "      <td>99</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-BF-B</th>\n",
       "      <th>Burkina Faso</th>\n",
       "      <th>2014</th>\n",
       "      <td>3</td>\n",
       "      <td>53</td>\n",
       "      <td>46</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-BF-2</th>\n",
       "      <th>AG1000G-BF-C</th>\n",
       "      <th>Burkina Faso</th>\n",
       "      <th>2004</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CD</th>\n",
       "      <th>AG1000G-CD</th>\n",
       "      <th>Democratic Republic of the Congo</th>\n",
       "      <th>2015</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>76</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">AG1000G-CF</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">AG1000G-CF</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Central African Republic</th>\n",
       "      <th>1993</th>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1994</th>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>53</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CI</th>\n",
       "      <th>AG1000G-CI</th>\n",
       "      <th>Cote d'Ivoire</th>\n",
       "      <th>2012</th>\n",
       "      <td>0</td>\n",
       "      <td>80</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CM-1</th>\n",
       "      <th>AG1000G-CM-A</th>\n",
       "      <th>Cameroon</th>\n",
       "      <th>2009</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>303</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CM-2</th>\n",
       "      <th>AG1000G-CM-B</th>\n",
       "      <th>Cameroon</th>\n",
       "      <th>2005</th>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>90</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-CM-3</th>\n",
       "      <th>AG1000G-CM-C</th>\n",
       "      <th>Cameroon</th>\n",
       "      <th>2013</th>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-FR</th>\n",
       "      <th>AG1000G-FR</th>\n",
       "      <th>Mayotte</th>\n",
       "      <th>2011</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GA-1</th>\n",
       "      <th>AG1000G-GA-A</th>\n",
       "      <th>Gabon</th>\n",
       "      <th>2000</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>69</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GH</th>\n",
       "      <th>AG1000G-GH</th>\n",
       "      <th>Ghana</th>\n",
       "      <th>2012</th>\n",
       "      <td>0</td>\n",
       "      <td>64</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GM-1</th>\n",
       "      <th>AG1000G-GM-A</th>\n",
       "      <th>Gambia, The</th>\n",
       "      <th>2011</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>68</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GM-2</th>\n",
       "      <th>AG1000G-GM-B</th>\n",
       "      <th>Gambia, The</th>\n",
       "      <th>2006</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GM-3</th>\n",
       "      <th>AG1000G-GM-C</th>\n",
       "      <th>Gambia, The</th>\n",
       "      <th>2012</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>172</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">AG1000G-GN-ML</th>\n",
       "      <th>AG1000G-GN-A</th>\n",
       "      <th>Guinea</th>\n",
       "      <th>2012</th>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">AG1000G-GN-B</th>\n",
       "      <th>Guinea</th>\n",
       "      <th>2012</th>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>84</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mali</th>\n",
       "      <th>2012</th>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>65</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GQ</th>\n",
       "      <th>AG1000G-GQ</th>\n",
       "      <th>Equatorial Guinea</th>\n",
       "      <th>2002</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-GW</th>\n",
       "      <th>AG1000G-GW</th>\n",
       "      <th>Guinea-Bissau</th>\n",
       "      <th>2010</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>93</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">AG1000G-KE</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">AG1000G-KE</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">Kenya</th>\n",
       "      <th>2000</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2007</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2012</th>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-ML-1</th>\n",
       "      <th>AG1000G-ML-A</th>\n",
       "      <th>Mali</th>\n",
       "      <th>2014</th>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-ML-2</th>\n",
       "      <th>AG1000G-ML-B</th>\n",
       "      <th>Mali</th>\n",
       "      <th>2004</th>\n",
       "      <td>2</td>\n",
       "      <td>36</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-MW</th>\n",
       "      <th>AG1000G-MW</th>\n",
       "      <th>Malawi</th>\n",
       "      <th>2015</th>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">AG1000G-MZ</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">AG1000G-MZ</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Mozambique</th>\n",
       "      <th>2003</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2004</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>71</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">AG1000G-TZ</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">AG1000G-TZ</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">Tanzania</th>\n",
       "      <th>2012</th>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2013</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2015</th>\n",
       "      <td>137</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-UG</th>\n",
       "      <th>AG1000G-UG</th>\n",
       "      <th>Uganda</th>\n",
       "      <th>2012</th>\n",
       "      <td>82</td>\n",
       "      <td>0</td>\n",
       "      <td>207</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AG1000G-X</th>\n",
       "      <th>AG1000G-X</th>\n",
       "      <th>Lab Cross</th>\n",
       "      <th>-1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>297</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "taxon                                                              arabiensis  \\\n",
       "study_id      sample_set   country                          year                \n",
       "AG1000G-AO    AG1000G-AO   Angola                            2009           0   \n",
       "AG1000G-BF-1  AG1000G-BF-A Burkina Faso                      2012           0   \n",
       "              AG1000G-BF-B Burkina Faso                      2014           3   \n",
       "AG1000G-BF-2  AG1000G-BF-C Burkina Faso                      2004           0   \n",
       "AG1000G-CD    AG1000G-CD   Democratic Republic of the Congo  2015           0   \n",
       "AG1000G-CF    AG1000G-CF   Central African Republic          1993           0   \n",
       "                                                             1994           0   \n",
       "AG1000G-CI    AG1000G-CI   Cote d'Ivoire                     2012           0   \n",
       "AG1000G-CM-1  AG1000G-CM-A Cameroon                          2009           0   \n",
       "AG1000G-CM-2  AG1000G-CM-B Cameroon                          2005           0   \n",
       "AG1000G-CM-3  AG1000G-CM-C Cameroon                          2013           2   \n",
       "AG1000G-FR    AG1000G-FR   Mayotte                           2011           0   \n",
       "AG1000G-GA-1  AG1000G-GA-A Gabon                             2000           0   \n",
       "AG1000G-GH    AG1000G-GH   Ghana                             2012           0   \n",
       "AG1000G-GM-1  AG1000G-GM-A Gambia, The                       2011           0   \n",
       "AG1000G-GM-2  AG1000G-GM-B Gambia, The                       2006           0   \n",
       "AG1000G-GM-3  AG1000G-GM-C Gambia, The                       2012           0   \n",
       "AG1000G-GN-ML AG1000G-GN-A Guinea                            2012           0   \n",
       "              AG1000G-GN-B Guinea                            2012           0   \n",
       "                           Mali                              2012           0   \n",
       "AG1000G-GQ    AG1000G-GQ   Equatorial Guinea                 2002           0   \n",
       "AG1000G-GW    AG1000G-GW   Guinea-Bissau                     2010           0   \n",
       "AG1000G-KE    AG1000G-KE   Kenya                             2000           0   \n",
       "                                                             2007           3   \n",
       "                                                             2012          10   \n",
       "AG1000G-ML-1  AG1000G-ML-A Mali                              2014           0   \n",
       "AG1000G-ML-2  AG1000G-ML-B Mali                              2004           2   \n",
       "AG1000G-MW    AG1000G-MW   Malawi                            2015          41   \n",
       "AG1000G-MZ    AG1000G-MZ   Mozambique                        2003           0   \n",
       "                                                             2004           0   \n",
       "AG1000G-TZ    AG1000G-TZ   Tanzania                          2012          87   \n",
       "                                                             2013           1   \n",
       "                                                             2015         137   \n",
       "AG1000G-UG    AG1000G-UG   Uganda                            2012          82   \n",
       "AG1000G-X     AG1000G-X    Lab Cross                        -1              0   \n",
       "\n",
       "taxon                                                              coluzzii  \\\n",
       "study_id      sample_set   country                          year              \n",
       "AG1000G-AO    AG1000G-AO   Angola                            2009        81   \n",
       "AG1000G-BF-1  AG1000G-BF-A Burkina Faso                      2012        82   \n",
       "              AG1000G-BF-B Burkina Faso                      2014        53   \n",
       "AG1000G-BF-2  AG1000G-BF-C Burkina Faso                      2004         0   \n",
       "AG1000G-CD    AG1000G-CD   Democratic Republic of the Congo  2015         0   \n",
       "AG1000G-CF    AG1000G-CF   Central African Republic          1993         5   \n",
       "                                                             1994        13   \n",
       "AG1000G-CI    AG1000G-CI   Cote d'Ivoire                     2012        80   \n",
       "AG1000G-CM-1  AG1000G-CM-A Cameroon                          2009         0   \n",
       "AG1000G-CM-2  AG1000G-CM-B Cameroon                          2005         7   \n",
       "AG1000G-CM-3  AG1000G-CM-C Cameroon                          2013        19   \n",
       "AG1000G-FR    AG1000G-FR   Mayotte                           2011         0   \n",
       "AG1000G-GA-1  AG1000G-GA-A Gabon                             2000         0   \n",
       "AG1000G-GH    AG1000G-GH   Ghana                             2012        64   \n",
       "AG1000G-GM-1  AG1000G-GM-A Gambia, The                       2011         0   \n",
       "AG1000G-GM-2  AG1000G-GM-B Gambia, The                       2006         0   \n",
       "AG1000G-GM-3  AG1000G-GM-C Gambia, The                       2012         0   \n",
       "AG1000G-GN-ML AG1000G-GN-A Guinea                            2012         4   \n",
       "              AG1000G-GN-B Guinea                            2012         7   \n",
       "                           Mali                              2012        27   \n",
       "AG1000G-GQ    AG1000G-GQ   Equatorial Guinea                 2002         0   \n",
       "AG1000G-GW    AG1000G-GW   Guinea-Bissau                     2010         0   \n",
       "AG1000G-KE    AG1000G-KE   Kenya                             2000         0   \n",
       "                                                             2007         0   \n",
       "                                                             2012         0   \n",
       "AG1000G-ML-1  AG1000G-ML-A Mali                              2014        27   \n",
       "AG1000G-ML-2  AG1000G-ML-B Mali                              2004        36   \n",
       "AG1000G-MW    AG1000G-MW   Malawi                            2015         0   \n",
       "AG1000G-MZ    AG1000G-MZ   Mozambique                        2003         0   \n",
       "                                                             2004         0   \n",
       "AG1000G-TZ    AG1000G-TZ   Tanzania                          2012         0   \n",
       "                                                             2013         0   \n",
       "                                                             2015         0   \n",
       "AG1000G-UG    AG1000G-UG   Uganda                            2012         0   \n",
       "AG1000G-X     AG1000G-X    Lab Cross                        -1            0   \n",
       "\n",
       "taxon                                                              gambiae  \\\n",
       "study_id      sample_set   country                          year             \n",
       "AG1000G-AO    AG1000G-AO   Angola                            2009        0   \n",
       "AG1000G-BF-1  AG1000G-BF-A Burkina Faso                      2012       99   \n",
       "              AG1000G-BF-B Burkina Faso                      2014       46   \n",
       "AG1000G-BF-2  AG1000G-BF-C Burkina Faso                      2004       13   \n",
       "AG1000G-CD    AG1000G-CD   Democratic Republic of the Congo  2015       76   \n",
       "AG1000G-CF    AG1000G-CF   Central African Republic          1993        2   \n",
       "                                                             1994       53   \n",
       "AG1000G-CI    AG1000G-CI   Cote d'Ivoire                     2012        0   \n",
       "AG1000G-CM-1  AG1000G-CM-A Cameroon                          2009      303   \n",
       "AG1000G-CM-2  AG1000G-CM-B Cameroon                          2005       90   \n",
       "AG1000G-CM-3  AG1000G-CM-C Cameroon                          2013       23   \n",
       "AG1000G-FR    AG1000G-FR   Mayotte                           2011       23   \n",
       "AG1000G-GA-1  AG1000G-GA-A Gabon                             2000       69   \n",
       "AG1000G-GH    AG1000G-GH   Ghana                             2012       36   \n",
       "AG1000G-GM-1  AG1000G-GM-A Gambia, The                       2011        0   \n",
       "AG1000G-GM-2  AG1000G-GM-B Gambia, The                       2006        0   \n",
       "AG1000G-GM-3  AG1000G-GM-C Gambia, The                       2012        2   \n",
       "AG1000G-GN-ML AG1000G-GN-A Guinea                            2012       41   \n",
       "              AG1000G-GN-B Guinea                            2012       84   \n",
       "                           Mali                              2012       65   \n",
       "AG1000G-GQ    AG1000G-GQ   Equatorial Guinea                 2002       10   \n",
       "AG1000G-GW    AG1000G-GW   Guinea-Bissau                     2010        8   \n",
       "AG1000G-KE    AG1000G-KE   Kenya                             2000       19   \n",
       "                                                             2007        0   \n",
       "                                                             2012        0   \n",
       "AG1000G-ML-1  AG1000G-ML-A Mali                              2014       33   \n",
       "AG1000G-ML-2  AG1000G-ML-B Mali                              2004       33   \n",
       "AG1000G-MW    AG1000G-MW   Malawi                            2015        0   \n",
       "AG1000G-MZ    AG1000G-MZ   Mozambique                        2003        3   \n",
       "                                                             2004       71   \n",
       "AG1000G-TZ    AG1000G-TZ   Tanzania                          2012        0   \n",
       "                                                             2013       32   \n",
       "                                                             2015       32   \n",
       "AG1000G-UG    AG1000G-UG   Uganda                            2012      207   \n",
       "AG1000G-X     AG1000G-X    Lab Cross                        -1           0   \n",
       "\n",
       "taxon                                                              gcx1  gcx2  \\\n",
       "study_id      sample_set   country                          year                \n",
       "AG1000G-AO    AG1000G-AO   Angola                            2009     0     0   \n",
       "AG1000G-BF-1  AG1000G-BF-A Burkina Faso                      2012     0     0   \n",
       "              AG1000G-BF-B Burkina Faso                      2014     0     0   \n",
       "AG1000G-BF-2  AG1000G-BF-C Burkina Faso                      2004     0     0   \n",
       "AG1000G-CD    AG1000G-CD   Democratic Republic of the Congo  2015     0     0   \n",
       "AG1000G-CF    AG1000G-CF   Central African Republic          1993     0     0   \n",
       "                                                             1994     0     0   \n",
       "AG1000G-CI    AG1000G-CI   Cote d'Ivoire                     2012     0     0   \n",
       "AG1000G-CM-1  AG1000G-CM-A Cameroon                          2009     0     0   \n",
       "AG1000G-CM-2  AG1000G-CM-B Cameroon                          2005     0     0   \n",
       "AG1000G-CM-3  AG1000G-CM-C Cameroon                          2013     0     0   \n",
       "AG1000G-FR    AG1000G-FR   Mayotte                           2011     0     0   \n",
       "AG1000G-GA-1  AG1000G-GA-A Gabon                             2000     0     0   \n",
       "AG1000G-GH    AG1000G-GH   Ghana                             2012     0     0   \n",
       "AG1000G-GM-1  AG1000G-GM-A Gambia, The                       2011    68     6   \n",
       "AG1000G-GM-2  AG1000G-GM-B Gambia, The                       2006     9    22   \n",
       "AG1000G-GM-3  AG1000G-GM-C Gambia, The                       2012     0   172   \n",
       "AG1000G-GN-ML AG1000G-GN-A Guinea                            2012     0     0   \n",
       "              AG1000G-GN-B Guinea                            2012     0     0   \n",
       "                           Mali                              2012     0     0   \n",
       "AG1000G-GQ    AG1000G-GQ   Equatorial Guinea                 2002     0     0   \n",
       "AG1000G-GW    AG1000G-GW   Guinea-Bissau                     2010    93     0   \n",
       "AG1000G-KE    AG1000G-KE   Kenya                             2000     0     0   \n",
       "                                                             2007     0     0   \n",
       "                                                             2012     0     0   \n",
       "AG1000G-ML-1  AG1000G-ML-A Mali                              2014     0     0   \n",
       "AG1000G-ML-2  AG1000G-ML-B Mali                              2004     0     0   \n",
       "AG1000G-MW    AG1000G-MW   Malawi                            2015     0     0   \n",
       "AG1000G-MZ    AG1000G-MZ   Mozambique                        2003     0     0   \n",
       "                                                             2004     0     0   \n",
       "AG1000G-TZ    AG1000G-TZ   Tanzania                          2012     0     0   \n",
       "                                                             2013     0     0   \n",
       "                                                             2015     0     0   \n",
       "AG1000G-UG    AG1000G-UG   Uganda                            2012     0     0   \n",
       "AG1000G-X     AG1000G-X    Lab Cross                        -1        0     0   \n",
       "\n",
       "taxon                                                              gcx3  \\\n",
       "study_id      sample_set   country                          year          \n",
       "AG1000G-AO    AG1000G-AO   Angola                            2009     0   \n",
       "AG1000G-BF-1  AG1000G-BF-A Burkina Faso                      2012     0   \n",
       "              AG1000G-BF-B Burkina Faso                      2014     0   \n",
       "AG1000G-BF-2  AG1000G-BF-C Burkina Faso                      2004     0   \n",
       "AG1000G-CD    AG1000G-CD   Democratic Republic of the Congo  2015     0   \n",
       "AG1000G-CF    AG1000G-CF   Central African Republic          1993     0   \n",
       "                                                             1994     0   \n",
       "AG1000G-CI    AG1000G-CI   Cote d'Ivoire                     2012     0   \n",
       "AG1000G-CM-1  AG1000G-CM-A Cameroon                          2009     0   \n",
       "AG1000G-CM-2  AG1000G-CM-B Cameroon                          2005     0   \n",
       "AG1000G-CM-3  AG1000G-CM-C Cameroon                          2013     0   \n",
       "AG1000G-FR    AG1000G-FR   Mayotte                           2011     0   \n",
       "AG1000G-GA-1  AG1000G-GA-A Gabon                             2000     0   \n",
       "AG1000G-GH    AG1000G-GH   Ghana                             2012     0   \n",
       "AG1000G-GM-1  AG1000G-GM-A Gambia, The                       2011     0   \n",
       "AG1000G-GM-2  AG1000G-GM-B Gambia, The                       2006     0   \n",
       "AG1000G-GM-3  AG1000G-GM-C Gambia, The                       2012     0   \n",
       "AG1000G-GN-ML AG1000G-GN-A Guinea                            2012     0   \n",
       "              AG1000G-GN-B Guinea                            2012     0   \n",
       "                           Mali                              2012     0   \n",
       "AG1000G-GQ    AG1000G-GQ   Equatorial Guinea                 2002     0   \n",
       "AG1000G-GW    AG1000G-GW   Guinea-Bissau                     2010     0   \n",
       "AG1000G-KE    AG1000G-KE   Kenya                             2000     0   \n",
       "                                                             2007     0   \n",
       "                                                             2012    54   \n",
       "AG1000G-ML-1  AG1000G-ML-A Mali                              2014     0   \n",
       "AG1000G-ML-2  AG1000G-ML-B Mali                              2004     0   \n",
       "AG1000G-MW    AG1000G-MW   Malawi                            2015     0   \n",
       "AG1000G-MZ    AG1000G-MZ   Mozambique                        2003     0   \n",
       "                                                             2004     0   \n",
       "AG1000G-TZ    AG1000G-TZ   Tanzania                          2012     0   \n",
       "                                                             2013    10   \n",
       "                                                             2015     1   \n",
       "AG1000G-UG    AG1000G-UG   Uganda                            2012     0   \n",
       "AG1000G-X     AG1000G-X    Lab Cross                        -1        0   \n",
       "\n",
       "taxon                                                              unassigned  \n",
       "study_id      sample_set   country                          year               \n",
       "AG1000G-AO    AG1000G-AO   Angola                            2009           0  \n",
       "AG1000G-BF-1  AG1000G-BF-A Burkina Faso                      2012           0  \n",
       "              AG1000G-BF-B Burkina Faso                      2014           0  \n",
       "AG1000G-BF-2  AG1000G-BF-C Burkina Faso                      2004           0  \n",
       "AG1000G-CD    AG1000G-CD   Democratic Republic of the Congo  2015           0  \n",
       "AG1000G-CF    AG1000G-CF   Central African Republic          1993           0  \n",
       "                                                             1994           0  \n",
       "AG1000G-CI    AG1000G-CI   Cote d'Ivoire                     2012           0  \n",
       "AG1000G-CM-1  AG1000G-CM-A Cameroon                          2009           0  \n",
       "AG1000G-CM-2  AG1000G-CM-B Cameroon                          2005           0  \n",
       "AG1000G-CM-3  AG1000G-CM-C Cameroon                          2013           0  \n",
       "AG1000G-FR    AG1000G-FR   Mayotte                           2011           0  \n",
       "AG1000G-GA-1  AG1000G-GA-A Gabon                             2000           0  \n",
       "AG1000G-GH    AG1000G-GH   Ghana                             2012           0  \n",
       "AG1000G-GM-1  AG1000G-GM-A Gambia, The                       2011           0  \n",
       "AG1000G-GM-2  AG1000G-GM-B Gambia, The                       2006           0  \n",
       "AG1000G-GM-3  AG1000G-GM-C Gambia, The                       2012           0  \n",
       "AG1000G-GN-ML AG1000G-GN-A Guinea                            2012           0  \n",
       "              AG1000G-GN-B Guinea                            2012           0  \n",
       "                           Mali                              2012           2  \n",
       "AG1000G-GQ    AG1000G-GQ   Equatorial Guinea                 2002           0  \n",
       "AG1000G-GW    AG1000G-GW   Guinea-Bissau                     2010           0  \n",
       "AG1000G-KE    AG1000G-KE   Kenya                             2000           0  \n",
       "                                                             2007           0  \n",
       "                                                             2012           0  \n",
       "AG1000G-ML-1  AG1000G-ML-A Mali                              2014           0  \n",
       "AG1000G-ML-2  AG1000G-ML-B Mali                              2004           0  \n",
       "AG1000G-MW    AG1000G-MW   Malawi                            2015           0  \n",
       "AG1000G-MZ    AG1000G-MZ   Mozambique                        2003           0  \n",
       "                                                             2004           0  \n",
       "AG1000G-TZ    AG1000G-TZ   Tanzania                          2012           0  \n",
       "                                                             2013           0  \n",
       "                                                             2015           0  \n",
       "AG1000G-UG    AG1000G-UG   Uganda                            2012           1  \n",
       "AG1000G-X     AG1000G-X    Lab Cross                        -1            297  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_samples = ag3.sample_metadata(sample_sets=\"3.0\")\n",
    "df_summary = df_samples.pivot_table(\n",
    "    index=[\"study_id\",\"sample_set\", \"country\", \"year\"], \n",
    "    columns=[\"taxon\"],\n",
    "    values=\"sample_id\", \n",
    "    aggfunc=len,\n",
    "    fill_value=0)\n",
    "df_summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "dLiU0ulIpk8p"
   },
   "source": [
    "Note that there are also multiple sampling sites represented within some sample sets."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "OToX5vhfpk8p"
   },
   "source": [
    "## Further reading\n",
    "\n",
    "Hopefully this page has provided a useful introduction to the `Ag3.0` data resource. If you would like to start working with these data, please visit the [cloud data access guide](cloud) or the [data download guide](download) or continue browsing the other documentation on this site.\n",
    "\n",
    "If you have any questions about the data and how to use them, please do get in touch by [starting a new discussion](https://github.com/malariagen/vector-data/discussions/new) on the malariagen/vector-data repo on GitHub."
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Tags",
  "colab": {
   "name": "Ag3.0-intro.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
