
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Adir1 cloud data access &#8212; MalariaGEN vector data user guide</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css?v=a1e01515" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=4a39c7ea"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "malariagen/vector-data");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'adir1/cloud';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Adir1.0 data downloads" href="download.html" />
    <link rel="prev" title="Adir1.0 (Vector Observatory - Asia Project Anopheles dirus Phase 1 Data Release)" href="adir1.0.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../landing-page.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/logo.png" class="logo__image only-light" alt="MalariaGEN vector data user guide - Home"/>
    <script>document.write(`<img src="../_static/logo.png" class="logo__image only-dark" alt="MalariaGEN vector data user guide - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../landing-page.html">
                    MalariaGEN vector data user guide
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Malaria Vector Genome Observatory</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../vobs/vobs-data-access.html">Vector Observatory Data Access</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Anopheles gambiae complex</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.0.html">Ag3.0 (Ag1000G phase 3)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.1.html">Ag3.1</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.2.html">Ag3.2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.3.html">Ag3.3</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.4.html">Ag3.4</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.5.html">Ag3.5</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.6.html">Ag3.6</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.7.html">Ag3.7</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.8.html">Ag3.8</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.9.html">Ag3.9</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.10.html">Ag3.10</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.11.html">Ag3.11</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.12.html">Ag3.12</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.13.html">Ag3.13</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/ag3.14.html">Ag3.14</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/cloud.html">Ag3 cloud data access</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/download.html">Ag3 data downloads</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/methods.html">Ag3 sequencing and variant calling methods</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/api.html">Ag3 API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ag3/analysis.html">Ag3 analysis</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Anopheles funestus</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../af1/af1.0.html">Af1.0  (<em>Anopheles funestus</em> Project Phase 1 Data Release)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/af1.1.html">Af1.1</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/af1.2.html">Af1.2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/af1.3.html">Af1.3</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/af1.4.html">Af1.4</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/cloud.html">Af1 cloud data access</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/download.html">Af1 data downloads</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/api.html">Af1 API</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Anopheles minimus</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../amin1/intro.html">Amin1.0 (Vector Observatory - Asia Project <em>Anopheles minimus</em> Phase 1 Data Release)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../amin1/cloud.html">Amin1.0 cloud data access</a></li>
<li class="toctree-l2"><a class="reference internal" href="../amin1/download.html">Amin1.0 data downloads</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../amin1/api.html">Amin1 API</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Anopheles dirus</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active has-children"><a class="reference internal" href="adir1.0.html">Adir1.0 (Vector Observatory - Asia Project <em>Anopheles dirus</em> Phase 1 Data Release)</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Adir1 cloud data access</a></li>
<li class="toctree-l2"><a class="reference internal" href="download.html">Adir1.0 data downloads</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="api.html">Adir1 API</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Partners and community</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../studies-ag1000g.html">Ag1000G partner studies</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Release notes</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../release-notes.html">Release notes</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/malariagen/vector-data" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/malariagen/vector-data/issues/new?title=Issue%20on%20page%20%2Fadir1/cloud.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/adir1/cloud.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Adir1 cloud data access</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-hosting">Data hosting</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#setup">Setup</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#sample-sets">Sample sets</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#sample-metadata">Sample metadata</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#snp-calls">SNP calls</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#snp-sites-and-alleles">SNP sites and alleles</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#site-filters">Site filters</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#snp-genotypes">SNP genotypes</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#example-computation">Example computation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#running-larger-computations">Running larger computations</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#feedback-and-suggestions">Feedback and suggestions</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="adir1-cloud-data-access">
<h1>Adir1 cloud data access<a class="headerlink" href="#adir1-cloud-data-access" title="Link to this heading">#</a></h1>
<p>This notebook provides information about how to download data from the <a class="reference external" href="https://www.malariagen.net/project/vector-observatory-asia/">MalariaGEN Vector Observatory Asian Vector Genomic Surveillance Project</a>, for <em>Anopheles dirus</em> via Google Cloud. This includes sample metadata, raw sequence reads, sequence read alignments, and single nucleotide polymorphism (SNP) calls.</p>
<p>This notebook illustrates how to read data directly from the cloud, without having to first download any data locally. This notebook can be run from any computer, but will work best when run from a compute node within Google Cloud, because it will be physically closer to the data and so data transfer is faster. For example, this notebook can be run via <a class="reference external" href="https://colab.research.google.com/">Google Colab</a> which are free interactive computing service running in the cloud.</p>
<p>To launch this notebook in the cloud and run it for yourself, click the launch icon (<i class="fas fa-rocket"></i>) at the top of the page and select one of the cloud computing services available.</p>
<section id="data-hosting">
<h2>Data hosting<a class="headerlink" href="#data-hosting" title="Link to this heading">#</a></h2>
<p>All data required for this notebook is hosted on Google Cloud Storage (GCS). Data are hosted in the <code class="docutils literal notranslate"><span class="pre">vo_adir_release_master_us_central1</span></code> bucket, which is a single-region bucket located in the United States. All data hosted in GCS are publicly accessible and do not require any authentication to access.</p>
</section>
<section id="setup">
<h2>Setup<a class="headerlink" href="#setup" title="Link to this heading">#</a></h2>
<p>Running this notebook requires some Python packages to be installed:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input above-output-prompt docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%</span><span class="k">pip</span> install -q malariagen_data
</pre></div>
</div>
</div>
<details class="admonition hide below-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell output</p>
<p class="expanded admonition-title">Hide code cell output</p>
</summary>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Note: you may need to restart the kernel to use updated packages.
</pre></div>
</div>
</div>
</details>
</div>
<p>To make accessing these data more convenient, we’ve created the <a class="reference external" href="https://github.com/malariagen/malariagen-data-python">malariagen_data</a> Python package. This is experimental so please let us know if you find any bugs or have any suggestions. See the <a class="reference external" href="https://malariagen.github.io/malariagen-data-python/latest/Adir1.0.html">Adir1.0 API docs</a> for documentation of all functions available from this package.</p>
<p>Import other packages we’ll need to use here.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">dask</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">dask.array</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">da</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">dask.diagnostics.progress</span><span class="w"> </span><span class="kn">import</span> <span class="n">ProgressBar</span>
<span class="c1"># silence some warnings</span>
<span class="n">dask</span><span class="o">.</span><span class="n">config</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="o">**</span><span class="p">{</span><span class="s1">&#39;array.slicing.split_large_chunks&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">})</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">allel</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">malariagen_data</span>
</pre></div>
</div>
</div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">Adir1</span></code> data access from Google Cloud is set up with the following code:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">adir1</span> <span class="o">=</span> <span class="n">malariagen_data</span><span class="o">.</span><span class="n">Adir1</span><span class="p">()</span>
<span class="n">adir1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<script type="application/javascript">'use strict';
(function(root) {
  function now() {
    return new Date();
  }

  const force = true;

  if (typeof root._bokeh_onload_callbacks === "undefined" || force === true) {
    root._bokeh_onload_callbacks = [];
    root._bokeh_is_loading = undefined;
  }

const JS_MIME_TYPE = 'application/javascript';
  const HTML_MIME_TYPE = 'text/html';
  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';
  const CLASS_NAME = 'output_bokeh rendered_html';

  /**
   * Render data to the DOM node
   */
  function render(props, node) {
    const script = document.createElement("script");
    node.appendChild(script);
  }

  /**
   * Handle when an output is cleared or removed
   */
  function handleClearOutput(event, handle) {
    function drop(id) {
      const view = Bokeh.index.get_by_id(id)
      if (view != null) {
        view.model.document.clear()
        Bokeh.index.delete(view)
      }
    }

    const cell = handle.cell;

    const id = cell.output_area._bokeh_element_id;
    const server_id = cell.output_area._bokeh_server_id;

    // Clean up Bokeh references
    if (id != null) {
      drop(id)
    }

    if (server_id !== undefined) {
      // Clean up Bokeh references
      const cmd_clean = "from bokeh.io.state import curstate; print(curstate().uuid_to_server['" + server_id + "'].get_sessions()[0].document.roots[0]._id)";
      cell.notebook.kernel.execute(cmd_clean, {
        iopub: {
          output: function(msg) {
            const id = msg.content.text.trim()
            drop(id)
          }
        }
      });
      // Destroy server and session
      const cmd_destroy = "import bokeh.io.notebook as ion; ion.destroy_server('" + server_id + "')";
      cell.notebook.kernel.execute(cmd_destroy);
    }
  }

  /**
   * Handle when a new output is added
   */
  function handleAddOutput(event, handle) {
    const output_area = handle.output_area;
    const output = handle.output;

    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only
    if ((output.output_type != "display_data") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {
      return
    }

    const toinsert = output_area.element.find("." + CLASS_NAME.split(' ')[0]);

    if (output.metadata[EXEC_MIME_TYPE]["id"] !== undefined) {
      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];
      // store reference to embed id on output_area
      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE]["id"];
    }
    if (output.metadata[EXEC_MIME_TYPE]["server_id"] !== undefined) {
      const bk_div = document.createElement("div");
      bk_div.innerHTML = output.data[HTML_MIME_TYPE];
      const script_attrs = bk_div.children[0].attributes;
      for (let i = 0; i < script_attrs.length; i++) {
        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);
        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent
      }
      // store reference to server id on output_area
      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE]["server_id"];
    }
  }

  function register_renderer(events, OutputArea) {

    function append_mime(data, metadata, element) {
      // create a DOM node to render to
      const toinsert = this.create_output_subarea(
        metadata,
        CLASS_NAME,
        EXEC_MIME_TYPE
      );
      this.keyboard_manager.register_events(toinsert);
      // Render to node
      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};
      render(props, toinsert[toinsert.length - 1]);
      element.append(toinsert);
      return toinsert
    }

    /* Handle when an output is cleared or removed */
    events.on('clear_output.CodeCell', handleClearOutput);
    events.on('delete.Cell', handleClearOutput);

    /* Handle when a new output is added */
    events.on('output_added.OutputArea', handleAddOutput);

    /**
     * Register the mime type and append_mime function with output_area
     */
    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {
      /* Is output safe? */
      safe: true,
      /* Index of renderer in `output_area.display_order` */
      index: 0
    });
  }

  // register the mime type if in Jupyter Notebook environment and previously unregistered
  if (root.Jupyter !== undefined) {
    const events = require('base/js/events');
    const OutputArea = require('notebook/js/outputarea').OutputArea;

    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {
      register_renderer(events, OutputArea);
    }
  }
  if (typeof (root._bokeh_timeout) === "undefined" || force === true) {
    root._bokeh_timeout = Date.now() + 5000;
    root._bokeh_failed_load = false;
  }

  const NB_LOAD_WARNING = {'data': {'text/html':
     "<div style='background-color: #fdd'>\n"+
     "<p>\n"+
     "BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \n"+
     "may be due to a slow or bad network connection. Possible fixes:\n"+
     "</p>\n"+
     "<ul>\n"+
     "<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\n"+
     "<li>use INLINE resources instead, as so:</li>\n"+
     "</ul>\n"+
     "<code>\n"+
     "from bokeh.resources import INLINE\n"+
     "output_notebook(resources=INLINE)\n"+
     "</code>\n"+
     "</div>"}};

  function display_loaded(error = null) {
    const el = document.getElementById(null);
    if (el != null) {
      const html = (() => {
        if (typeof root.Bokeh === "undefined") {
          if (error == null) {
            return "BokehJS is loading ...";
          } else {
            return "BokehJS failed to load.";
          }
        } else {
          const prefix = `BokehJS ${root.Bokeh.version}`;
          if (error == null) {
            return `${prefix} successfully loaded.`;
          } else {
            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;
          }
        }
      })();
      el.innerHTML = html;

      if (error != null) {
        const wrapper = document.createElement("div");
        wrapper.style.overflow = "auto";
        wrapper.style.height = "5em";
        wrapper.style.resize = "vertical";
        const content = document.createElement("div");
        content.style.fontFamily = "monospace";
        content.style.whiteSpace = "pre-wrap";
        content.style.backgroundColor = "rgb(255, 221, 221)";
        content.textContent = error.stack ?? error.toString();
        wrapper.append(content);
        el.append(wrapper);
      }
    } else if (Date.now() < root._bokeh_timeout) {
      setTimeout(() => display_loaded(error), 100);
    }
  }

  function run_callbacks() {
    try {
      root._bokeh_onload_callbacks.forEach(function(callback) {
        if (callback != null)
          callback();
      });
    } finally {
      delete root._bokeh_onload_callbacks
    }
    console.debug("Bokeh: all callbacks have finished");
  }

  function load_libs(css_urls, js_urls, callback) {
    if (css_urls == null) css_urls = [];
    if (js_urls == null) js_urls = [];

    root._bokeh_onload_callbacks.push(callback);
    if (root._bokeh_is_loading > 0) {
      console.debug("Bokeh: BokehJS is being loaded, scheduling callback at", now());
      return null;
    }
    if (js_urls == null || js_urls.length === 0) {
      run_callbacks();
      return null;
    }
    console.debug("Bokeh: BokehJS not loaded, scheduling load and callback at", now());
    root._bokeh_is_loading = css_urls.length + js_urls.length;

    function on_load() {
      root._bokeh_is_loading--;
      if (root._bokeh_is_loading === 0) {
        console.debug("Bokeh: all BokehJS libraries/stylesheets loaded");
        run_callbacks()
      }
    }

    function on_error(url) {
      console.error("failed to load " + url);
    }

    for (let i = 0; i < css_urls.length; i++) {
      const url = css_urls[i];
      const element = document.createElement("link");
      element.onload = on_load;
      element.onerror = on_error.bind(null, url);
      element.rel = "stylesheet";
      element.type = "text/css";
      element.href = url;
      console.debug("Bokeh: injecting link tag for BokehJS stylesheet: ", url);
      document.body.appendChild(element);
    }

    for (let i = 0; i < js_urls.length; i++) {
      const url = js_urls[i];
      const element = document.createElement('script');
      element.onload = on_load;
      element.onerror = on_error.bind(null, url);
      element.async = false;
      element.src = url;
      console.debug("Bokeh: injecting script tag for BokehJS library: ", url);
      document.head.appendChild(element);
    }
  };

  function inject_raw_css(css) {
    const element = document.createElement("style");
    element.appendChild(document.createTextNode(css));
    document.body.appendChild(element);
  }

  const js_urls = ["https://cdn.bokeh.org/bokeh/release/bokeh-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.6.3.min.js"];
  const css_urls = [];

  const inline_js = [    function(Bokeh) {
      Bokeh.set_log_level("info");
    },
function(Bokeh) {
    }
  ];

  function run_inline_js() {
    if (root.Bokeh !== undefined || force === true) {
      try {
            for (let i = 0; i < inline_js.length; i++) {
      inline_js[i].call(root, root.Bokeh);
    }

      } catch (error) {throw error;
      }} else if (Date.now() < root._bokeh_timeout) {
      setTimeout(run_inline_js, 100);
    } else if (!root._bokeh_failed_load) {
      console.log("Bokeh: BokehJS failed to load within specified timeout.");
      root._bokeh_failed_load = true;
    } else if (force !== true) {
      const cell = $(document.getElementById(null)).parents('.cell').data().cell;
      cell.output_area.append_execute_result(NB_LOAD_WARNING)
    }
  }

  if (root._bokeh_is_loading === 0) {
    console.debug("Bokeh: BokehJS loaded, going straight to plotting");
    run_inline_js();
  } else {
    load_libs(css_urls, js_urls, function() {
      console.debug("Bokeh: BokehJS plotting callback run at", now());
      run_inline_js();
    });
  }
}(window));</script><div class="output text_html">
            <table class="malariagen-adir1">
                <thead>
                    <tr>
                        <th style="text-align: left" colspan="2">MalariaGEN Adir1 API client</th>
                    </tr>
                    <tr><td colspan="2" style="text-align: left">
                        Please note that data are subject to terms of use,
                        for more information see <a href="https://www.malariagen.net/data">
                        the MalariaGEN website</a> or contact support@malariagen.net.
                        See also the <a href="https://malariagen.github.io/malariagen-data-python/v0.0.0/Adir1.html">Adir1 API docs</a>.
                    </td></tr>
                </thead>
                <tbody>
                    <tr>
                        <th style="text-align: left">
                            Storage URL
                        </th>
                        <td>gs://vo_adir_production_us_central1/release/</td>
                    </tr>
                    <tr>
                        <th style="text-align: left">
                            Data releases available
                        </th>
                        <td>1.0</td>
                    </tr>
                    <tr>
                        <th style="text-align: left">
                            Results cache
                        </th>
                        <td>None</td>
                    </tr>
                    <tr>
                        <th style="text-align: left">
                            Cohorts analysis
                        </th>
                        <td>20250710</td>
                    </tr>
                    <tr>
                        <th style="text-align: left">
                            Site filters analysis
                        </th>
                        <td>sc_20250610</td>
                    </tr>
                    <tr>
                        <th style="text-align: left">
                            Software version
                        </th>
                        <td>malariagen_data 0.0.0</td>
                    </tr>
                    <tr>
                        <th style="text-align: left">
                            Client location
                        </th>
                        <td>Queensland, Australia</td>
                    </tr>
                </tbody>
            </table>
        </div></div>
</div>
</section>
<section id="sample-sets">
<h2>Sample sets<a class="headerlink" href="#sample-sets" title="Link to this heading">#</a></h2>
<p>Data are organised into different releases. As an example, data in Adir1.0 are organised into 4 sample sets. Each of these sample sets corresponds to a set of mosquito specimens contributed by a collaborating study. Depending on your objectives, you may want to access data from only specific sample sets, or all sample sets.</p>
<p>To see which sample sets are available, load the sample set manifest into a pandas dataframe:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_sample_sets</span> <span class="o">=</span> <span class="n">adir1</span><span class="o">.</span><span class="n">sample_sets</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s2">&quot;1.0&quot;</span><span class="p">)</span>
<span class="n">df_sample_sets</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_set</th>
      <th>sample_count</th>
      <th>study_id</th>
      <th>study_url</th>
      <th>terms_of_use_expiry_date</th>
      <th>terms_of_use_url</th>
      <th>release</th>
      <th>unrestricted_use</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1276-AD-BD-ALAM-VMF00156</td>
      <td>47</td>
      <td>1276-AD-BD-ALAM</td>
      <td>https://www.malariagen.net/partner_study/1276-...</td>
      <td>2027-11-30</td>
      <td>https://www.malariagen.net/data/our-approach-s...</td>
      <td>1.0</td>
      <td>False</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1277-VO-KH-WITKOWSKI-VMF00151</td>
      <td>26</td>
      <td>1277-VO-KH-WITKOWSKI</td>
      <td>https://www.malariagen.net/partner_study/1277-...</td>
      <td>2027-11-30</td>
      <td>https://www.malariagen.net/data/our-approach-s...</td>
      <td>1.0</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2</th>
      <td>1277-VO-KH-WITKOWSKI-VMF00183</td>
      <td>248</td>
      <td>1277-VO-KH-WITKOWSKI</td>
      <td>https://www.malariagen.net/partner_study/1277-...</td>
      <td>2027-11-30</td>
      <td>https://www.malariagen.net/data/our-approach-s...</td>
      <td>1.0</td>
      <td>False</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1278-VO-TH-KOBYLINSKI-VMF00153</td>
      <td>219</td>
      <td>1278-VO-TH-KOBYLINSKI</td>
      <td>https://www.malariagen.net/partner_study/1278-...</td>
      <td>2027-11-30</td>
      <td>https://www.malariagen.net/data/our-approach-s...</td>
      <td>1.0</td>
      <td>False</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>For more information about these sample sets, you can read about each sample set from the URLs under the field <code class="docutils literal notranslate"><span class="pre">study_url</span></code>.</p>
</section>
<section id="sample-metadata">
<h2>Sample metadata<a class="headerlink" href="#sample-metadata" title="Link to this heading">#</a></h2>
<p>Data about the samples that were sequenced to generate this data resource are available, including the time and place of collection, the gender of the specimen, and our call regarding the species of the specimen. These are organised by sample set.</p>
<p>E.g., load sample metadata for all samples in the Adir1.0 release into a <a class="reference external" href="https://pandas.pydata.org/pandas-docs/stable/user_guide/dsintro.html#dataframe">pandas DataFrame</a>:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_samples</span> <span class="o">=</span> <span class="n">adir1</span><span class="o">.</span><span class="n">sample_metadata</span><span class="p">(</span><span class="n">sample_sets</span><span class="o">=</span><span class="s2">&quot;1.0&quot;</span><span class="p">)</span>
<span class="n">df_samples</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>                                     
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>derived_sample_id</th>
      <th>partner_sample_id</th>
      <th>contributor</th>
      <th>country</th>
      <th>location</th>
      <th>year</th>
      <th>month</th>
      <th>latitude</th>
      <th>longitude</th>
      <th>...</th>
      <th>admin1_name</th>
      <th>admin1_iso</th>
      <th>admin2_name</th>
      <th>taxon</th>
      <th>cohort_admin1_year</th>
      <th>cohort_admin1_month</th>
      <th>cohort_admin1_quarter</th>
      <th>cohort_admin2_year</th>
      <th>cohort_admin2_month</th>
      <th>cohort_admin2_quarter</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>VBS46299-6321STDY9453299</td>
      <td>VBS46299-6321STDY9453299</td>
      <td>158</td>
      <td>Shiaful Alam</td>
      <td>Bangladesh</td>
      <td>Bangladesh_2</td>
      <td>2018</td>
      <td>5</td>
      <td>22.287</td>
      <td>92.194</td>
      <td>...</td>
      <td>Chittagong Division</td>
      <td>BD-B</td>
      <td>Bandarban</td>
      <td>baimaii</td>
      <td>BD-B_baim_2018</td>
      <td>BD-B_baim_2018_05</td>
      <td>BD-B_baim_2018_Q2</td>
      <td>BD-B_Bandarban_baim_2018</td>
      <td>BD-B_Bandarban_baim_2018_05</td>
      <td>BD-B_Bandarban_baim_2018_Q2</td>
    </tr>
    <tr>
      <th>1</th>
      <td>VBS46307-6321STDY9453307</td>
      <td>VBS46307-6321STDY9453307</td>
      <td>2973</td>
      <td>Shiaful Alam</td>
      <td>Bangladesh</td>
      <td>Bangladesh_1</td>
      <td>2018</td>
      <td>6</td>
      <td>22.254</td>
      <td>92.203</td>
      <td>...</td>
      <td>Chittagong Division</td>
      <td>BD-B</td>
      <td>Bandarban</td>
      <td>baimaii</td>
      <td>BD-B_baim_2018</td>
      <td>BD-B_baim_2018_06</td>
      <td>BD-B_baim_2018_Q2</td>
      <td>BD-B_Bandarban_baim_2018</td>
      <td>BD-B_Bandarban_baim_2018_06</td>
      <td>BD-B_Bandarban_baim_2018_Q2</td>
    </tr>
    <tr>
      <th>2</th>
      <td>VBS46315-6321STDY9453315</td>
      <td>VBS46315-6321STDY9453315</td>
      <td>2340</td>
      <td>Shiaful Alam</td>
      <td>Bangladesh</td>
      <td>Bangladesh_1</td>
      <td>2018</td>
      <td>7</td>
      <td>22.254</td>
      <td>92.203</td>
      <td>...</td>
      <td>Chittagong Division</td>
      <td>BD-B</td>
      <td>Bandarban</td>
      <td>baimaii</td>
      <td>BD-B_baim_2018</td>
      <td>BD-B_baim_2018_07</td>
      <td>BD-B_baim_2018_Q3</td>
      <td>BD-B_Bandarban_baim_2018</td>
      <td>BD-B_Bandarban_baim_2018_07</td>
      <td>BD-B_Bandarban_baim_2018_Q3</td>
    </tr>
    <tr>
      <th>3</th>
      <td>VBS46323-6321STDY9453323</td>
      <td>VBS46323-6321STDY9453323</td>
      <td>2525</td>
      <td>Shiaful Alam</td>
      <td>Bangladesh</td>
      <td>Bangladesh_2</td>
      <td>2018</td>
      <td>7</td>
      <td>22.287</td>
      <td>92.194</td>
      <td>...</td>
      <td>Chittagong Division</td>
      <td>BD-B</td>
      <td>Bandarban</td>
      <td>baimaii</td>
      <td>BD-B_baim_2018</td>
      <td>BD-B_baim_2018_07</td>
      <td>BD-B_baim_2018_Q3</td>
      <td>BD-B_Bandarban_baim_2018</td>
      <td>BD-B_Bandarban_baim_2018_07</td>
      <td>BD-B_Bandarban_baim_2018_Q3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>VBS46331-6321STDY9453331</td>
      <td>VBS46331-6321STDY9453331</td>
      <td>5249</td>
      <td>Shiaful Alam</td>
      <td>Bangladesh</td>
      <td>Bangladesh_1</td>
      <td>2018</td>
      <td>9</td>
      <td>22.254</td>
      <td>92.203</td>
      <td>...</td>
      <td>Chittagong Division</td>
      <td>BD-B</td>
      <td>Bandarban</td>
      <td>baimaii</td>
      <td>BD-B_baim_2018</td>
      <td>BD-B_baim_2018_09</td>
      <td>BD-B_baim_2018_Q3</td>
      <td>BD-B_Bandarban_baim_2018</td>
      <td>BD-B_Bandarban_baim_2018_09</td>
      <td>BD-B_Bandarban_baim_2018_Q3</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>535</th>
      <td>VBS46203-6296STDY10244759</td>
      <td>VBS46203-6296STDY10244759</td>
      <td>5895</td>
      <td>Kevin Kobylinski</td>
      <td>Thailand</td>
      <td>Khirirat Nikhom, nine, Q</td>
      <td>2019</td>
      <td>10</td>
      <td>9.127</td>
      <td>98.905</td>
      <td>...</td>
      <td>Surat Thani Province</td>
      <td>TH-84</td>
      <td>Khiri Rat Nikhom</td>
      <td>dirus</td>
      <td>TH-84_diru_2019</td>
      <td>TH-84_diru_2019_10</td>
      <td>TH-84_diru_2019_Q4</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019_10</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019_Q4</td>
    </tr>
    <tr>
      <th>536</th>
      <td>VBS46204-6296STDY10244760</td>
      <td>VBS46204-6296STDY10244760</td>
      <td>5972</td>
      <td>Kevin Kobylinski</td>
      <td>Thailand</td>
      <td>Khirirat Nikhom, eight, M</td>
      <td>2019</td>
      <td>10</td>
      <td>9.104</td>
      <td>98.891</td>
      <td>...</td>
      <td>Surat Thani Province</td>
      <td>TH-84</td>
      <td>Khiri Rat Nikhom</td>
      <td>dirus</td>
      <td>TH-84_diru_2019</td>
      <td>TH-84_diru_2019_10</td>
      <td>TH-84_diru_2019_Q4</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019_10</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019_Q4</td>
    </tr>
    <tr>
      <th>537</th>
      <td>VBS46205-6296STDY10244761</td>
      <td>VBS46205-6296STDY10244761</td>
      <td>6024</td>
      <td>Kevin Kobylinski</td>
      <td>Thailand</td>
      <td>Khirirat Nikhom, eight, M</td>
      <td>2019</td>
      <td>10</td>
      <td>9.104</td>
      <td>98.891</td>
      <td>...</td>
      <td>Surat Thani Province</td>
      <td>TH-84</td>
      <td>Khiri Rat Nikhom</td>
      <td>dirus</td>
      <td>TH-84_diru_2019</td>
      <td>TH-84_diru_2019_10</td>
      <td>TH-84_diru_2019_Q4</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019_10</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019_Q4</td>
    </tr>
    <tr>
      <th>538</th>
      <td>VBS46206-6296STDY10244762</td>
      <td>VBS46206-6296STDY10244762</td>
      <td>6036</td>
      <td>Kevin Kobylinski</td>
      <td>Thailand</td>
      <td>Khirirat Nikhom, eight, N</td>
      <td>2019</td>
      <td>10</td>
      <td>9.106</td>
      <td>98.887</td>
      <td>...</td>
      <td>Surat Thani Province</td>
      <td>TH-84</td>
      <td>Khiri Rat Nikhom</td>
      <td>dirus</td>
      <td>TH-84_diru_2019</td>
      <td>TH-84_diru_2019_10</td>
      <td>TH-84_diru_2019_Q4</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019_10</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019_Q4</td>
    </tr>
    <tr>
      <th>539</th>
      <td>VBS46207-6296STDY10244763</td>
      <td>VBS46207-6296STDY10244763</td>
      <td>6037</td>
      <td>Kevin Kobylinski</td>
      <td>Thailand</td>
      <td>Khirirat Nikhom, eight, N</td>
      <td>2019</td>
      <td>10</td>
      <td>9.106</td>
      <td>98.887</td>
      <td>...</td>
      <td>Surat Thani Province</td>
      <td>TH-84</td>
      <td>Khiri Rat Nikhom</td>
      <td>dirus</td>
      <td>TH-84_diru_2019</td>
      <td>TH-84_diru_2019_10</td>
      <td>TH-84_diru_2019_Q4</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019_10</td>
      <td>TH-84_Khiri-Rat-Nikhom_diru_2019_Q4</td>
    </tr>
  </tbody>
</table>
<p>540 rows × 50 columns</p>
</div></div></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">sample_id</span></code> column gives the sample identifier used throughout all Adir1.0 analyses.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">country</span></code>, <code class="docutils literal notranslate"><span class="pre">location</span></code>, <code class="docutils literal notranslate"><span class="pre">latitude</span></code> and <code class="docutils literal notranslate"><span class="pre">longitude</span></code> columns give the location where the specimen was collected.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">year</span></code> and <code class="docutils literal notranslate"><span class="pre">month</span></code> columns give the approximate date when the specimen was collected.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">sex_call</span></code> column gives the gender as determined from the sequence data.</p>
<p><a class="reference external" href="https://pandas.pydata.org/">Pandas</a> can be used to explore and query the sample metadata in various ways. E.g., here is a summary of the numbers of samples by species:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_samples</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s2">&quot;taxon&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">size</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>taxon
baimaii     47
dirus      493
dtype: int64
</pre></div>
</div>
</div>
</div>
</section>
<section id="snp-calls">
<h2>SNP calls<a class="headerlink" href="#snp-calls" title="Link to this heading">#</a></h2>
<p>Data on SNP calls, including the SNP positions, alleles, site filters, and genotypes, can be accessed as an <a class="reference external" href="http://xarray.pydata.org/en/stable/user-guide/data-structures.html#dataset">xarray Dataset</a>.</p>
<p>E.g., access SNP calls for contig KB672490 for all samples in <code class="docutils literal notranslate"><span class="pre">Adir1.0</span></code>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ds_snps</span> <span class="o">=</span> <span class="n">adir1</span><span class="o">.</span><span class="n">snp_calls</span><span class="p">(</span><span class="n">region</span><span class="o">=</span><span class="s2">&quot;KB672490&quot;</span><span class="p">,</span> <span class="n">sample_sets</span><span class="o">=</span><span class="s2">&quot;1.0&quot;</span><span class="p">)</span>
<span class="n">ds_snps</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>                                 
</pre></div>
</div>
<div class="output text_html"><div><svg style="position: absolute; width: 0; height: 0; overflow: hidden">
<defs>
<symbol id="icon-database" viewBox="0 0 32 32">
<path d="M16 0c-8.837 0-16 2.239-16 5v4c0 2.761 7.163 5 16 5s16-2.239 16-5v-4c0-2.761-7.163-5-16-5z"></path>
<path d="M16 17c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z"></path>
<path d="M16 26c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z"></path>
</symbol>
<symbol id="icon-file-text2" viewBox="0 0 32 32">
<path d="M28.681 7.159c-0.694-0.947-1.662-2.053-2.724-3.116s-2.169-2.030-3.116-2.724c-1.612-1.182-2.393-1.319-2.841-1.319h-15.5c-1.378 0-2.5 1.121-2.5 2.5v27c0 1.378 1.122 2.5 2.5 2.5h23c1.378 0 2.5-1.122 2.5-2.5v-19.5c0-0.448-0.137-1.23-1.319-2.841zM24.543 5.457c0.959 0.959 1.712 1.825 2.268 2.543h-4.811v-4.811c0.718 0.556 1.584 1.309 2.543 2.268zM28 29.5c0 0.271-0.229 0.5-0.5 0.5h-23c-0.271 0-0.5-0.229-0.5-0.5v-27c0-0.271 0.229-0.5 0.5-0.5 0 0 15.499-0 15.5 0v7c0 0.552 0.448 1 1 1h7v19.5z"></path>
<path d="M23 26h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z"></path>
<path d="M23 22h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z"></path>
<path d="M23 18h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z"></path>
</symbol>
</defs>
</svg>
<style>/* CSS stylesheet for displaying xarray objects in jupyterlab.
 *
 */

:root {
  --xr-font-color0: var(
    --jp-content-font-color0,
    var(--pst-color-text-base rgba(0, 0, 0, 1))
  );
  --xr-font-color2: var(
    --jp-content-font-color2,
    var(--pst-color-text-base, rgba(0, 0, 0, 0.54))
  );
  --xr-font-color3: var(
    --jp-content-font-color3,
    var(--pst-color-text-base, rgba(0, 0, 0, 0.38))
  );
  --xr-border-color: var(
    --jp-border-color2,
    hsl(from var(--pst-color-on-background, white) h s calc(l - 10))
  );
  --xr-disabled-color: var(
    --jp-layout-color3,
    hsl(from var(--pst-color-on-background, white) h s calc(l - 40))
  );
  --xr-background-color: var(
    --jp-layout-color0,
    var(--pst-color-on-background, white)
  );
  --xr-background-color-row-even: var(
    --jp-layout-color1,
    hsl(from var(--pst-color-on-background, white) h s calc(l - 5))
  );
  --xr-background-color-row-odd: var(
    --jp-layout-color2,
    hsl(from var(--pst-color-on-background, white) h s calc(l - 15))
  );
}

html[theme="dark"],
html[data-theme="dark"],
body[data-theme="dark"],
body.vscode-dark {
  --xr-font-color0: var(
    --jp-content-font-color0,
    var(--pst-color-text-base, rgba(255, 255, 255, 1))
  );
  --xr-font-color2: var(
    --jp-content-font-color2,
    var(--pst-color-text-base, rgba(255, 255, 255, 0.54))
  );
  --xr-font-color3: var(
    --jp-content-font-color3,
    var(--pst-color-text-base, rgba(255, 255, 255, 0.38))
  );
  --xr-border-color: var(
    --jp-border-color2,
    hsl(from var(--pst-color-on-background, #111111) h s calc(l + 10))
  );
  --xr-disabled-color: var(
    --jp-layout-color3,
    hsl(from var(--pst-color-on-background, #111111) h s calc(l + 40))
  );
  --xr-background-color: var(
    --jp-layout-color0,
    var(--pst-color-on-background, #111111)
  );
  --xr-background-color-row-even: var(
    --jp-layout-color1,
    hsl(from var(--pst-color-on-background, #111111) h s calc(l + 5))
  );
  --xr-background-color-row-odd: var(
    --jp-layout-color2,
    hsl(from var(--pst-color-on-background, #111111) h s calc(l + 15))
  );
}

.xr-wrap {
  display: block !important;
  min-width: 300px;
  max-width: 700px;
}

.xr-text-repr-fallback {
  /* fallback to plain text repr when CSS is not injected (untrusted notebook) */
  display: none;
}

.xr-header {
  padding-top: 6px;
  padding-bottom: 6px;
  margin-bottom: 4px;
  border-bottom: solid 1px var(--xr-border-color);
}

.xr-header > div,
.xr-header > ul {
  display: inline;
  margin-top: 0;
  margin-bottom: 0;
}

.xr-obj-type,
.xr-array-name {
  margin-left: 2px;
  margin-right: 10px;
}

.xr-obj-type {
  color: var(--xr-font-color2);
}

.xr-sections {
  padding-left: 0 !important;
  display: grid;
  grid-template-columns: 150px auto auto 1fr 0 20px 0 20px;
}

.xr-section-item {
  display: contents;
}

.xr-section-item input {
  display: inline-block;
  opacity: 0;
  height: 0;
}

.xr-section-item input + label {
  color: var(--xr-disabled-color);
  border: 2px solid transparent !important;
}

.xr-section-item input:enabled + label {
  cursor: pointer;
  color: var(--xr-font-color2);
}

.xr-section-item input:focus + label {
  border: 2px solid var(--xr-font-color0) !important;
}

.xr-section-item input:enabled + label:hover {
  color: var(--xr-font-color0);
}

.xr-section-summary {
  grid-column: 1;
  color: var(--xr-font-color2);
  font-weight: 500;
}

.xr-section-summary > span {
  display: inline-block;
  padding-left: 0.5em;
}

.xr-section-summary-in:disabled + label {
  color: var(--xr-font-color2);
}

.xr-section-summary-in + label:before {
  display: inline-block;
  content: "►";
  font-size: 11px;
  width: 15px;
  text-align: center;
}

.xr-section-summary-in:disabled + label:before {
  color: var(--xr-disabled-color);
}

.xr-section-summary-in:checked + label:before {
  content: "▼";
}

.xr-section-summary-in:checked + label > span {
  display: none;
}

.xr-section-summary,
.xr-section-inline-details {
  padding-top: 4px;
  padding-bottom: 4px;
}

.xr-section-inline-details {
  grid-column: 2 / -1;
}

.xr-section-details {
  display: none;
  grid-column: 1 / -1;
  margin-bottom: 5px;
}

.xr-section-summary-in:checked ~ .xr-section-details {
  display: contents;
}

.xr-array-wrap {
  grid-column: 1 / -1;
  display: grid;
  grid-template-columns: 20px auto;
}

.xr-array-wrap > label {
  grid-column: 1;
  vertical-align: top;
}

.xr-preview {
  color: var(--xr-font-color3);
}

.xr-array-preview,
.xr-array-data {
  padding: 0 5px !important;
  grid-column: 2;
}

.xr-array-data,
.xr-array-in:checked ~ .xr-array-preview {
  display: none;
}

.xr-array-in:checked ~ .xr-array-data,
.xr-array-preview {
  display: inline-block;
}

.xr-dim-list {
  display: inline-block !important;
  list-style: none;
  padding: 0 !important;
  margin: 0;
}

.xr-dim-list li {
  display: inline-block;
  padding: 0;
  margin: 0;
}

.xr-dim-list:before {
  content: "(";
}

.xr-dim-list:after {
  content: ")";
}

.xr-dim-list li:not(:last-child):after {
  content: ",";
  padding-right: 5px;
}

.xr-has-index {
  font-weight: bold;
}

.xr-var-list,
.xr-var-item {
  display: contents;
}

.xr-var-item > div,
.xr-var-item label,
.xr-var-item > .xr-var-name span {
  background-color: var(--xr-background-color-row-even);
  border-color: var(--xr-background-color-row-odd);
  margin-bottom: 0;
  padding-top: 2px;
}

.xr-var-item > .xr-var-name:hover span {
  padding-right: 5px;
}

.xr-var-list > li:nth-child(odd) > div,
.xr-var-list > li:nth-child(odd) > label,
.xr-var-list > li:nth-child(odd) > .xr-var-name span {
  background-color: var(--xr-background-color-row-odd);
  border-color: var(--xr-background-color-row-even);
}

.xr-var-name {
  grid-column: 1;
}

.xr-var-dims {
  grid-column: 2;
}

.xr-var-dtype {
  grid-column: 3;
  text-align: right;
  color: var(--xr-font-color2);
}

.xr-var-preview {
  grid-column: 4;
}

.xr-index-preview {
  grid-column: 2 / 5;
  color: var(--xr-font-color2);
}

.xr-var-name,
.xr-var-dims,
.xr-var-dtype,
.xr-preview,
.xr-attrs dt {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  padding-right: 10px;
}

.xr-var-name:hover,
.xr-var-dims:hover,
.xr-var-dtype:hover,
.xr-attrs dt:hover {
  overflow: visible;
  width: auto;
  z-index: 1;
}

.xr-var-attrs,
.xr-var-data,
.xr-index-data {
  display: none;
  border-top: 2px dotted var(--xr-background-color);
  padding-bottom: 20px !important;
  padding-top: 10px !important;
}

.xr-var-attrs-in + label,
.xr-var-data-in + label,
.xr-index-data-in + label {
  padding: 0 1px;
}

.xr-var-attrs-in:checked ~ .xr-var-attrs,
.xr-var-data-in:checked ~ .xr-var-data,
.xr-index-data-in:checked ~ .xr-index-data {
  display: block;
}

.xr-var-data > table {
  float: right;
}

.xr-var-data > pre,
.xr-index-data > pre,
.xr-var-data > table > tbody > tr {
  background-color: transparent !important;
}

.xr-var-name span,
.xr-var-data,
.xr-index-name div,
.xr-index-data,
.xr-attrs {
  padding-left: 25px !important;
}

.xr-attrs,
.xr-var-attrs,
.xr-var-data,
.xr-index-data {
  grid-column: 1 / -1;
}

dl.xr-attrs {
  padding: 0;
  margin: 0;
  display: grid;
  grid-template-columns: 125px auto;
}

.xr-attrs dt,
.xr-attrs dd {
  padding: 0;
  margin: 0;
  float: left;
  padding-right: 10px;
  width: auto;
}

.xr-attrs dt {
  font-weight: normal;
  grid-column: 1;
}

.xr-attrs dt:hover span {
  display: inline-block;
  background: var(--xr-background-color);
  padding-right: 10px;
}

.xr-attrs dd {
  grid-column: 2;
  white-space: pre-wrap;
  word-break: break-all;
}

.xr-icon-database,
.xr-icon-file-text2,
.xr-no-icon {
  display: inline-block;
  vertical-align: middle;
  width: 1em;
  height: 1.5em !important;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
}

.xr-var-attrs-in:checked + label > .xr-icon-file-text2,
.xr-var-data-in:checked + label > .xr-icon-database,
.xr-index-data-in:checked + label > .xr-icon-database {
  color: var(--xr-font-color0);
  filter: drop-shadow(1px 1px 5px var(--xr-font-color2));
  stroke-width: 0.8px;
}
</style><pre class='xr-text-repr-fallback'>&lt;xarray.Dataset&gt; Size: 202GB
Dimensions:                    (variants: 21967539, alleles: 4, samples: 540,
                                ploidy: 2)
Coordinates:
    variant_position           (variants) int32 88MB dask.array&lt;chunksize=(65536,), meta=np.ndarray&gt;
    variant_contig             (variants) uint8 22MB dask.array&lt;chunksize=(65536,), meta=np.ndarray&gt;
    sample_id                  (samples) &lt;U36 78kB dask.array&lt;chunksize=(47,), meta=np.ndarray&gt;
Dimensions without coordinates: variants, alleles, samples, ploidy
Data variables:
    variant_allele             (variants, alleles) object 703MB dask.array&lt;chunksize=(65536, 4), meta=np.ndarray&gt;
    variant_filter_pass_dirus  (variants) bool 22MB dask.array&lt;chunksize=(300000,), meta=np.ndarray&gt;
    call_genotype              (variants, samples, ploidy) int8 24GB dask.array&lt;chunksize=(300000, 47, 2), meta=np.ndarray&gt;
    call_GQ                    (variants, samples) int8 12GB dask.array&lt;chunksize=(300000, 47), meta=np.ndarray&gt;
    call_MQ                    (variants, samples) float32 47GB dask.array&lt;chunksize=(300000, 47), meta=np.ndarray&gt;
    call_AD                    (variants, samples, alleles) int16 95GB dask.array&lt;chunksize=(300000, 47, 4), meta=np.ndarray&gt;
    call_genotype_mask         (variants, samples, ploidy) bool 24GB dask.array&lt;chunksize=(300000, 47, 2), meta=np.ndarray&gt;
Attributes:
    contigs:  (&#x27;KB672490&#x27;, &#x27;KB672868&#x27;, &#x27;KB672979&#x27;, &#x27;KB673090&#x27;, &#x27;KB673201&#x27;, &#x27;K...</pre><div class='xr-wrap' style='display:none'><div class='xr-header'><div class='xr-obj-type'>xarray.Dataset</div></div><ul class='xr-sections'><li class='xr-section-item'><input id='section-f004a81f-3086-42b3-bb48-45d529de9c58' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-f004a81f-3086-42b3-bb48-45d529de9c58' class='xr-section-summary'  title='Expand/collapse section'>Dimensions:</label><div class='xr-section-inline-details'><ul class='xr-dim-list'><li><span>variants</span>: 21967539</li><li><span>alleles</span>: 4</li><li><span>samples</span>: 540</li><li><span>ploidy</span>: 2</li></ul></div><div class='xr-section-details'></div></li><li class='xr-section-item'><input id='section-03bf44fd-6f45-422d-9ba7-388366151a78' class='xr-section-summary-in' type='checkbox'  checked><label for='section-03bf44fd-6f45-422d-9ba7-388366151a78' class='xr-section-summary' >Coordinates: <span>(3)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'><li class='xr-var-item'><div class='xr-var-name'><span>variant_position</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>int32</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(65536,), meta=np.ndarray&gt;</div><input id='attrs-dda8f0d0-0f04-42c0-b84e-b0de38df6724' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-dda8f0d0-0f04-42c0-b84e-b0de38df6724' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-2c50ff82-d096-423a-a0a6-622550572dc4' class='xr-var-data-in' type='checkbox'><label for='data-2c50ff82-d096-423a-a0a6-622550572dc4' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 83.80 MiB </td>
                        <td> 256.00 kiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539,) </td>
                        <td> (65536,) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 336 chunks in 1 graph layer </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> int32 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="6" y1="0" x2="6" y2="25" />
  <line x1="12" y1="0" x2="12" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="25" y1="0" x2="25" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="44" y1="0" x2="44" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="56" y1="0" x2="56" y2="25" />
  <line x1="63" y1="0" x2="63" y2="25" />
  <line x1="69" y1="0" x2="69" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="81" y1="0" x2="81" y2="25" />
  <line x1="88" y1="0" x2="88" y2="25" />
  <line x1="94" y1="0" x2="94" y2="25" />
  <line x1="100" y1="0" x2="100" y2="25" />
  <line x1="107" y1="0" x2="107" y2="25" />
  <line x1="113" y1="0" x2="113" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >21967539</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_contig</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>uint8</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(65536,), meta=np.ndarray&gt;</div><input id='attrs-44281565-9bbd-4d04-97c7-16a7972603b8' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-44281565-9bbd-4d04-97c7-16a7972603b8' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-f697c29e-72c2-4b0f-ad26-64e3bb1fa24c' class='xr-var-data-in' type='checkbox'><label for='data-f697c29e-72c2-4b0f-ad26-64e3bb1fa24c' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 20.95 MiB </td>
                        <td> 64.00 kiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539,) </td>
                        <td> (65536,) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 336 chunks in 1 graph layer </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> uint8 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="6" y1="0" x2="6" y2="25" />
  <line x1="12" y1="0" x2="12" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="25" y1="0" x2="25" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="44" y1="0" x2="44" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="56" y1="0" x2="56" y2="25" />
  <line x1="63" y1="0" x2="63" y2="25" />
  <line x1="69" y1="0" x2="69" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="81" y1="0" x2="81" y2="25" />
  <line x1="88" y1="0" x2="88" y2="25" />
  <line x1="94" y1="0" x2="94" y2="25" />
  <line x1="100" y1="0" x2="100" y2="25" />
  <line x1="107" y1="0" x2="107" y2="25" />
  <line x1="113" y1="0" x2="113" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >21967539</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>sample_id</span></div><div class='xr-var-dims'>(samples)</div><div class='xr-var-dtype'>&lt;U36</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(47,), meta=np.ndarray&gt;</div><input id='attrs-4177ecf6-deba-4b37-8c25-47a262ee3824' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-4177ecf6-deba-4b37-8c25-47a262ee3824' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-34b3d1d7-fc50-4e63-aab2-4eb15c57422d' class='xr-var-data-in' type='checkbox'><label for='data-34b3d1d7-fc50-4e63-aab2-4eb15c57422d' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 75.94 kiB </td>
                        <td> 34.88 kiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (540,) </td>
                        <td> (248,) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 4 chunks in 9 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> <U36 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="10" y1="0" x2="10" y2="25" />
  <line x1="16" y1="0" x2="16" y2="25" />
  <line x1="71" y1="0" x2="71" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >540</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
        </td>
    </tr>
</table></div></li></ul></div></li><li class='xr-section-item'><input id='section-db4bf193-9f7f-4449-9eaa-1935975cf295' class='xr-section-summary-in' type='checkbox'  checked><label for='section-db4bf193-9f7f-4449-9eaa-1935975cf295' class='xr-section-summary' >Data variables: <span>(7)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'><li class='xr-var-item'><div class='xr-var-name'><span>variant_allele</span></div><div class='xr-var-dims'>(variants, alleles)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(65536, 4), meta=np.ndarray&gt;</div><input id='attrs-0d172708-1069-4236-81fa-dc6b275d54a4' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-0d172708-1069-4236-81fa-dc6b275d54a4' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-bf997842-60d1-4bd3-9c1d-699a69d31665' class='xr-var-data-in' type='checkbox'><label for='data-bf997842-60d1-4bd3-9c1d-699a69d31665' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 670.40 MiB </td>
                        <td> 2.00 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 4) </td>
                        <td> (65536, 4) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 336 chunks in 5 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> object numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="75" height="170" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="25" y2="0" style="stroke-width:2" />
  <line x1="0" y1="6" x2="25" y2="6" />
  <line x1="0" y1="12" x2="25" y2="12" />
  <line x1="0" y1="18" x2="25" y2="18" />
  <line x1="0" y1="25" x2="25" y2="25" />
  <line x1="0" y1="31" x2="25" y2="31" />
  <line x1="0" y1="37" x2="25" y2="37" />
  <line x1="0" y1="44" x2="25" y2="44" />
  <line x1="0" y1="50" x2="25" y2="50" />
  <line x1="0" y1="56" x2="25" y2="56" />
  <line x1="0" y1="63" x2="25" y2="63" />
  <line x1="0" y1="69" x2="25" y2="69" />
  <line x1="0" y1="75" x2="25" y2="75" />
  <line x1="0" y1="81" x2="25" y2="81" />
  <line x1="0" y1="88" x2="25" y2="88" />
  <line x1="0" y1="94" x2="25" y2="94" />
  <line x1="0" y1="100" x2="25" y2="100" />
  <line x1="0" y1="107" x2="25" y2="107" />
  <line x1="0" y1="113" x2="25" y2="113" />
  <line x1="0" y1="120" x2="25" y2="120" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="120" style="stroke-width:2" />
  <line x1="25" y1="0" x2="25" y2="120" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="12.706308" y="140.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" >4</text>
  <text x="45.412617" y="60.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,45.412617,60.000000)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_filter_pass_dirus</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>bool</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000,), meta=np.ndarray&gt;</div><input id='attrs-369463e8-dced-4871-9677-da1fe3912496' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-369463e8-dced-4871-9677-da1fe3912496' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-05cf72aa-bbc8-4647-9f45-aa0448a95766' class='xr-var-data-in' type='checkbox'><label for='data-05cf72aa-bbc8-4647-9f45-aa0448a95766' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 20.95 MiB </td>
                        <td> 292.97 kiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539,) </td>
                        <td> (300000,) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 74 chunks in 1 graph layer </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> bool numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="4" y1="0" x2="4" y2="25" />
  <line x1="11" y1="0" x2="11" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="24" y1="0" x2="24" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="44" y1="0" x2="44" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="57" y1="0" x2="57" y2="25" />
  <line x1="62" y1="0" x2="62" y2="25" />
  <line x1="68" y1="0" x2="68" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="81" y1="0" x2="81" y2="25" />
  <line x1="88" y1="0" x2="88" y2="25" />
  <line x1="95" y1="0" x2="95" y2="25" />
  <line x1="101" y1="0" x2="101" y2="25" />
  <line x1="108" y1="0" x2="108" y2="25" />
  <line x1="114" y1="0" x2="114" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >21967539</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_genotype</span></div><div class='xr-var-dims'>(variants, samples, ploidy)</div><div class='xr-var-dtype'>int8</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000, 47, 2), meta=np.ndarray&gt;</div><input id='attrs-cd572fd0-1da9-475e-a04b-c5e18fd831d4' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-cd572fd0-1da9-475e-a04b-c5e18fd831d4' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-00140ca4-31c8-449f-aacd-ddf53393be45' class='xr-var-data-in' type='checkbox'><label for='data-00140ca4-31c8-449f-aacd-ddf53393be45' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 22.10 GiB </td>
                        <td> 28.61 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 540, 2) </td>
                        <td> (300000, 50, 2) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 888 chunks in 5 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> int8 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="2" x2="80" y2="72" />
  <line x1="10" y1="3" x2="80" y2="74" />
  <line x1="10" y1="5" x2="80" y2="76" />
  <line x1="10" y1="8" x2="80" y2="78" />
  <line x1="10" y1="10" x2="80" y2="81" />
  <line x1="10" y1="12" x2="80" y2="83" />
  <line x1="10" y1="15" x2="80" y2="85" />
  <line x1="10" y1="17" x2="80" y2="88" />
  <line x1="10" y1="19" x2="80" y2="90" />
  <line x1="10" y1="22" x2="80" y2="92" />
  <line x1="10" y1="24" x2="80" y2="95" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="12" y1="2" x2="12" y2="28" />
  <line x1="16" y1="6" x2="16" y2="32" />
  <line x1="20" y1="10" x2="20" y2="36" />
  <line x1="24" y1="14" x2="24" y2="39" />
  <line x1="28" y1="18" x2="28" y2="43" />
  <line x1="32" y1="22" x2="32" y2="47" />
  <line x1="36" y1="26" x2="36" y2="51" />
  <line x1="39" y1="29" x2="39" y2="55" />
  <line x1="43" y1="33" x2="43" y2="59" />
  <line x1="46" y1="36" x2="46" y2="62" />
  <line x1="50" y1="40" x2="50" y2="65" />
  <line x1="54" y1="44" x2="54" y2="69" />
  <line x1="58" y1="48" x2="58" y2="73" />
  <line x1="62" y1="52" x2="62" y2="77" />
  <line x1="65" y1="55" x2="65" y2="81" />
  <line x1="69" y1="59" x2="69" y2="85" />
  <line x1="73" y1="63" x2="73" y2="89" />
  <line x1="77" y1="67" x2="77" y2="92" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="12" y1="2" x2="38" y2="2" />
  <line x1="16" y1="6" x2="42" y2="6" />
  <line x1="20" y1="10" x2="46" y2="10" />
  <line x1="24" y1="14" x2="49" y2="14" />
  <line x1="28" y1="18" x2="53" y2="18" />
  <line x1="32" y1="22" x2="57" y2="22" />
  <line x1="36" y1="26" x2="61" y2="26" />
  <line x1="39" y1="29" x2="65" y2="29" />
  <line x1="43" y1="33" x2="69" y2="33" />
  <line x1="46" y1="36" x2="72" y2="36" />
  <line x1="50" y1="40" x2="75" y2="40" />
  <line x1="54" y1="44" x2="79" y2="44" />
  <line x1="58" y1="48" x2="83" y2="48" />
  <line x1="62" y1="52" x2="87" y2="52" />
  <line x1="65" y1="55" x2="91" y2="55" />
  <line x1="69" y1="59" x2="95" y2="59" />
  <line x1="73" y1="63" x2="99" y2="63" />
  <line x1="77" y1="67" x2="102" y2="67" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="72" x2="106" y2="72" />
  <line x1="80" y1="74" x2="106" y2="74" />
  <line x1="80" y1="76" x2="106" y2="76" />
  <line x1="80" y1="78" x2="106" y2="78" />
  <line x1="80" y1="81" x2="106" y2="81" />
  <line x1="80" y1="83" x2="106" y2="83" />
  <line x1="80" y1="85" x2="106" y2="85" />
  <line x1="80" y1="88" x2="106" y2="88" />
  <line x1="80" y1="90" x2="106" y2="90" />
  <line x1="80" y1="92" x2="106" y2="92" />
  <line x1="80" y1="95" x2="106" y2="95" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,126.000852,83.294544)">540</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_GQ</span></div><div class='xr-var-dims'>(variants, samples)</div><div class='xr-var-dtype'>int8</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000, 47), meta=np.ndarray&gt;</div><input id='attrs-6327aab4-bd0c-45b9-a546-fc76ae5aa312' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-6327aab4-bd0c-45b9-a546-fc76ae5aa312' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-0c9716f7-b020-442d-ab50-f4fd648c4957' class='xr-var-data-in' type='checkbox'><label for='data-0c9716f7-b020-442d-ab50-f4fd648c4957' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 11.05 GiB </td>
                        <td> 14.31 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 540) </td>
                        <td> (300000, 50) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 888 chunks in 5 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> int8 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="75" height="170" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="25" y2="0" style="stroke-width:2" />
  <line x1="0" y1="4" x2="25" y2="4" />
  <line x1="0" y1="11" x2="25" y2="11" />
  <line x1="0" y1="18" x2="25" y2="18" />
  <line x1="0" y1="24" x2="25" y2="24" />
  <line x1="0" y1="31" x2="25" y2="31" />
  <line x1="0" y1="37" x2="25" y2="37" />
  <line x1="0" y1="44" x2="25" y2="44" />
  <line x1="0" y1="50" x2="25" y2="50" />
  <line x1="0" y1="57" x2="25" y2="57" />
  <line x1="0" y1="62" x2="25" y2="62" />
  <line x1="0" y1="68" x2="25" y2="68" />
  <line x1="0" y1="75" x2="25" y2="75" />
  <line x1="0" y1="81" x2="25" y2="81" />
  <line x1="0" y1="88" x2="25" y2="88" />
  <line x1="0" y1="95" x2="25" y2="95" />
  <line x1="0" y1="101" x2="25" y2="101" />
  <line x1="0" y1="108" x2="25" y2="108" />
  <line x1="0" y1="114" x2="25" y2="114" />
  <line x1="0" y1="120" x2="25" y2="120" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="120" style="stroke-width:2" />
  <line x1="2" y1="0" x2="2" y2="120" />
  <line x1="3" y1="0" x2="3" y2="120" />
  <line x1="5" y1="0" x2="5" y2="120" />
  <line x1="8" y1="0" x2="8" y2="120" />
  <line x1="10" y1="0" x2="10" y2="120" />
  <line x1="12" y1="0" x2="12" y2="120" />
  <line x1="15" y1="0" x2="15" y2="120" />
  <line x1="17" y1="0" x2="17" y2="120" />
  <line x1="19" y1="0" x2="19" y2="120" />
  <line x1="22" y1="0" x2="22" y2="120" />
  <line x1="24" y1="0" x2="24" y2="120" />
  <line x1="25" y1="0" x2="25" y2="120" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="12.706308" y="140.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" >540</text>
  <text x="45.412617" y="60.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,45.412617,60.000000)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_MQ</span></div><div class='xr-var-dims'>(variants, samples)</div><div class='xr-var-dtype'>float32</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000, 47), meta=np.ndarray&gt;</div><input id='attrs-10de8cba-1585-45e9-88fa-c31482a4d7e7' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-10de8cba-1585-45e9-88fa-c31482a4d7e7' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-89a33c5b-41fd-4c4d-b7e1-df8e7d071543' class='xr-var-data-in' type='checkbox'><label for='data-89a33c5b-41fd-4c4d-b7e1-df8e7d071543' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 44.19 GiB </td>
                        <td> 57.22 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 540) </td>
                        <td> (300000, 50) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 888 chunks in 5 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> float32 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="75" height="170" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="25" y2="0" style="stroke-width:2" />
  <line x1="0" y1="4" x2="25" y2="4" />
  <line x1="0" y1="11" x2="25" y2="11" />
  <line x1="0" y1="18" x2="25" y2="18" />
  <line x1="0" y1="24" x2="25" y2="24" />
  <line x1="0" y1="31" x2="25" y2="31" />
  <line x1="0" y1="37" x2="25" y2="37" />
  <line x1="0" y1="44" x2="25" y2="44" />
  <line x1="0" y1="50" x2="25" y2="50" />
  <line x1="0" y1="57" x2="25" y2="57" />
  <line x1="0" y1="62" x2="25" y2="62" />
  <line x1="0" y1="68" x2="25" y2="68" />
  <line x1="0" y1="75" x2="25" y2="75" />
  <line x1="0" y1="81" x2="25" y2="81" />
  <line x1="0" y1="88" x2="25" y2="88" />
  <line x1="0" y1="95" x2="25" y2="95" />
  <line x1="0" y1="101" x2="25" y2="101" />
  <line x1="0" y1="108" x2="25" y2="108" />
  <line x1="0" y1="114" x2="25" y2="114" />
  <line x1="0" y1="120" x2="25" y2="120" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="120" style="stroke-width:2" />
  <line x1="2" y1="0" x2="2" y2="120" />
  <line x1="3" y1="0" x2="3" y2="120" />
  <line x1="5" y1="0" x2="5" y2="120" />
  <line x1="8" y1="0" x2="8" y2="120" />
  <line x1="10" y1="0" x2="10" y2="120" />
  <line x1="12" y1="0" x2="12" y2="120" />
  <line x1="15" y1="0" x2="15" y2="120" />
  <line x1="17" y1="0" x2="17" y2="120" />
  <line x1="19" y1="0" x2="19" y2="120" />
  <line x1="22" y1="0" x2="22" y2="120" />
  <line x1="24" y1="0" x2="24" y2="120" />
  <line x1="25" y1="0" x2="25" y2="120" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="12.706308" y="140.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" >540</text>
  <text x="45.412617" y="60.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,45.412617,60.000000)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_AD</span></div><div class='xr-var-dims'>(variants, samples, alleles)</div><div class='xr-var-dtype'>int16</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000, 47, 4), meta=np.ndarray&gt;</div><input id='attrs-ab7b2846-27c9-4521-8bb1-87aa794920d5' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-ab7b2846-27c9-4521-8bb1-87aa794920d5' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-9383fd26-f4ed-497f-a9b8-b657c0333c92' class='xr-var-data-in' type='checkbox'><label for='data-9383fd26-f4ed-497f-a9b8-b657c0333c92' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 88.38 GiB </td>
                        <td> 114.44 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 540, 4) </td>
                        <td> (300000, 50, 4) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 888 chunks in 5 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> int16 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="2" x2="80" y2="72" />
  <line x1="10" y1="3" x2="80" y2="74" />
  <line x1="10" y1="5" x2="80" y2="76" />
  <line x1="10" y1="8" x2="80" y2="78" />
  <line x1="10" y1="10" x2="80" y2="81" />
  <line x1="10" y1="12" x2="80" y2="83" />
  <line x1="10" y1="15" x2="80" y2="85" />
  <line x1="10" y1="17" x2="80" y2="88" />
  <line x1="10" y1="19" x2="80" y2="90" />
  <line x1="10" y1="22" x2="80" y2="92" />
  <line x1="10" y1="24" x2="80" y2="95" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="12" y1="2" x2="12" y2="28" />
  <line x1="16" y1="6" x2="16" y2="32" />
  <line x1="20" y1="10" x2="20" y2="36" />
  <line x1="24" y1="14" x2="24" y2="39" />
  <line x1="28" y1="18" x2="28" y2="43" />
  <line x1="32" y1="22" x2="32" y2="47" />
  <line x1="36" y1="26" x2="36" y2="51" />
  <line x1="39" y1="29" x2="39" y2="55" />
  <line x1="43" y1="33" x2="43" y2="59" />
  <line x1="46" y1="36" x2="46" y2="62" />
  <line x1="50" y1="40" x2="50" y2="65" />
  <line x1="54" y1="44" x2="54" y2="69" />
  <line x1="58" y1="48" x2="58" y2="73" />
  <line x1="62" y1="52" x2="62" y2="77" />
  <line x1="65" y1="55" x2="65" y2="81" />
  <line x1="69" y1="59" x2="69" y2="85" />
  <line x1="73" y1="63" x2="73" y2="89" />
  <line x1="77" y1="67" x2="77" y2="92" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="12" y1="2" x2="38" y2="2" />
  <line x1="16" y1="6" x2="42" y2="6" />
  <line x1="20" y1="10" x2="46" y2="10" />
  <line x1="24" y1="14" x2="49" y2="14" />
  <line x1="28" y1="18" x2="53" y2="18" />
  <line x1="32" y1="22" x2="57" y2="22" />
  <line x1="36" y1="26" x2="61" y2="26" />
  <line x1="39" y1="29" x2="65" y2="29" />
  <line x1="43" y1="33" x2="69" y2="33" />
  <line x1="46" y1="36" x2="72" y2="36" />
  <line x1="50" y1="40" x2="75" y2="40" />
  <line x1="54" y1="44" x2="79" y2="44" />
  <line x1="58" y1="48" x2="83" y2="48" />
  <line x1="62" y1="52" x2="87" y2="52" />
  <line x1="65" y1="55" x2="91" y2="55" />
  <line x1="69" y1="59" x2="95" y2="59" />
  <line x1="73" y1="63" x2="99" y2="63" />
  <line x1="77" y1="67" x2="102" y2="67" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="72" x2="106" y2="72" />
  <line x1="80" y1="74" x2="106" y2="74" />
  <line x1="80" y1="76" x2="106" y2="76" />
  <line x1="80" y1="78" x2="106" y2="78" />
  <line x1="80" y1="81" x2="106" y2="81" />
  <line x1="80" y1="83" x2="106" y2="83" />
  <line x1="80" y1="85" x2="106" y2="85" />
  <line x1="80" y1="88" x2="106" y2="88" />
  <line x1="80" y1="90" x2="106" y2="90" />
  <line x1="80" y1="92" x2="106" y2="92" />
  <line x1="80" y1="95" x2="106" y2="95" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >4</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,126.000852,83.294544)">540</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_genotype_mask</span></div><div class='xr-var-dims'>(variants, samples, ploidy)</div><div class='xr-var-dtype'>bool</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000, 47, 2), meta=np.ndarray&gt;</div><input id='attrs-7fc013c8-a49f-41c8-82e7-7997540624c4' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-7fc013c8-a49f-41c8-82e7-7997540624c4' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-e6a0f898-6991-4e16-9a14-3521bb499953' class='xr-var-data-in' type='checkbox'><label for='data-e6a0f898-6991-4e16-9a14-3521bb499953' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 22.10 GiB </td>
                        <td> 28.61 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 540, 2) </td>
                        <td> (300000, 50, 2) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 888 chunks in 6 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> bool numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="2" x2="80" y2="72" />
  <line x1="10" y1="3" x2="80" y2="74" />
  <line x1="10" y1="5" x2="80" y2="76" />
  <line x1="10" y1="8" x2="80" y2="78" />
  <line x1="10" y1="10" x2="80" y2="81" />
  <line x1="10" y1="12" x2="80" y2="83" />
  <line x1="10" y1="15" x2="80" y2="85" />
  <line x1="10" y1="17" x2="80" y2="88" />
  <line x1="10" y1="19" x2="80" y2="90" />
  <line x1="10" y1="22" x2="80" y2="92" />
  <line x1="10" y1="24" x2="80" y2="95" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="12" y1="2" x2="12" y2="28" />
  <line x1="16" y1="6" x2="16" y2="32" />
  <line x1="20" y1="10" x2="20" y2="36" />
  <line x1="24" y1="14" x2="24" y2="39" />
  <line x1="28" y1="18" x2="28" y2="43" />
  <line x1="32" y1="22" x2="32" y2="47" />
  <line x1="36" y1="26" x2="36" y2="51" />
  <line x1="39" y1="29" x2="39" y2="55" />
  <line x1="43" y1="33" x2="43" y2="59" />
  <line x1="46" y1="36" x2="46" y2="62" />
  <line x1="50" y1="40" x2="50" y2="65" />
  <line x1="54" y1="44" x2="54" y2="69" />
  <line x1="58" y1="48" x2="58" y2="73" />
  <line x1="62" y1="52" x2="62" y2="77" />
  <line x1="65" y1="55" x2="65" y2="81" />
  <line x1="69" y1="59" x2="69" y2="85" />
  <line x1="73" y1="63" x2="73" y2="89" />
  <line x1="77" y1="67" x2="77" y2="92" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="12" y1="2" x2="38" y2="2" />
  <line x1="16" y1="6" x2="42" y2="6" />
  <line x1="20" y1="10" x2="46" y2="10" />
  <line x1="24" y1="14" x2="49" y2="14" />
  <line x1="28" y1="18" x2="53" y2="18" />
  <line x1="32" y1="22" x2="57" y2="22" />
  <line x1="36" y1="26" x2="61" y2="26" />
  <line x1="39" y1="29" x2="65" y2="29" />
  <line x1="43" y1="33" x2="69" y2="33" />
  <line x1="46" y1="36" x2="72" y2="36" />
  <line x1="50" y1="40" x2="75" y2="40" />
  <line x1="54" y1="44" x2="79" y2="44" />
  <line x1="58" y1="48" x2="83" y2="48" />
  <line x1="62" y1="52" x2="87" y2="52" />
  <line x1="65" y1="55" x2="91" y2="55" />
  <line x1="69" y1="59" x2="95" y2="59" />
  <line x1="73" y1="63" x2="99" y2="63" />
  <line x1="77" y1="67" x2="102" y2="67" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="72" x2="106" y2="72" />
  <line x1="80" y1="74" x2="106" y2="74" />
  <line x1="80" y1="76" x2="106" y2="76" />
  <line x1="80" y1="78" x2="106" y2="78" />
  <line x1="80" y1="81" x2="106" y2="81" />
  <line x1="80" y1="83" x2="106" y2="83" />
  <line x1="80" y1="85" x2="106" y2="85" />
  <line x1="80" y1="88" x2="106" y2="88" />
  <line x1="80" y1="90" x2="106" y2="90" />
  <line x1="80" y1="92" x2="106" y2="92" />
  <line x1="80" y1="95" x2="106" y2="95" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,126.000852,83.294544)">540</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li></ul></div></li><li class='xr-section-item'><input id='section-70383ce0-8284-4f36-8d2b-0d4b92c59e6a' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-70383ce0-8284-4f36-8d2b-0d4b92c59e6a' class='xr-section-summary'  title='Expand/collapse section'>Indexes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'></ul></div></li><li class='xr-section-item'><input id='section-7aaf1625-43aa-4811-98fc-3c36310b576b' class='xr-section-summary-in' type='checkbox'  checked><label for='section-7aaf1625-43aa-4811-98fc-3c36310b576b' class='xr-section-summary' >Attributes: <span>(1)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><dl class='xr-attrs'><dt><span>contigs :</span></dt><dd>(&#x27;KB672490&#x27;, &#x27;KB672868&#x27;, &#x27;KB672979&#x27;, &#x27;KB673090&#x27;, &#x27;KB673201&#x27;, &#x27;KB673312&#x27;, &#x27;KB673423&#x27;, &#x27;KB673534&#x27;, &#x27;KB673645&#x27;, &#x27;KB672491&#x27;, &#x27;KB672602&#x27;, &#x27;KB672791&#x27;, &#x27;KB672713&#x27;, &#x27;KB672802&#x27;, &#x27;KB672813&#x27;, &#x27;KB672824&#x27;, &#x27;KB672846&#x27;, &#x27;KB672835&#x27;, &#x27;KB672857&#x27;, &#x27;KB672869&#x27;, &#x27;KB672880&#x27;, &#x27;KB672891&#x27;, &#x27;KB672902&#x27;, &#x27;KB672913&#x27;, &#x27;KB672924&#x27;, &#x27;KB672935&#x27;, &#x27;KB672957&#x27;, &#x27;KB672946&#x27;, &#x27;KB672968&#x27;, &#x27;KB672980&#x27;, &#x27;KB672991&#x27;, &#x27;KB673002&#x27;, &#x27;KB673013&#x27;, &#x27;KB673024&#x27;, &#x27;KB673046&#x27;, &#x27;KB673035&#x27;, &#x27;KB673057&#x27;, &#x27;KB673068&#x27;, &#x27;KB673079&#x27;, &#x27;KB673091&#x27;, &#x27;KB673102&#x27;, &#x27;KB673124&#x27;, &#x27;KB673113&#x27;, &#x27;KB673157&#x27;, &#x27;KB673146&#x27;, &#x27;KB673135&#x27;, &#x27;KB673168&#x27;, &#x27;KB673190&#x27;, &#x27;KB673179&#x27;, &#x27;KB673202&#x27;, &#x27;KB673213&#x27;, &#x27;KB673235&#x27;, &#x27;KB673224&#x27;, &#x27;KB673268&#x27;, &#x27;KB673246&#x27;, &#x27;KB673257&#x27;, &#x27;KB673279&#x27;, &#x27;KB673301&#x27;, &#x27;KB673290&#x27;, &#x27;KB673313&#x27;, &#x27;KB673324&#x27;, &#x27;KB673357&#x27;, &#x27;KB673379&#x27;, &#x27;KB673346&#x27;, &#x27;KB673457&#x27;, &#x27;KB673335&#x27;, &#x27;KB673535&#x27;, &#x27;KB673546&#x27;, &#x27;KB673435&#x27;, &#x27;KB673446&#x27;, &#x27;KB673368&#x27;, &#x27;KB673424&#x27;, &#x27;KB673390&#x27;, &#x27;KB673401&#x27;, &#x27;KB673468&#x27;, &#x27;KB673479&#x27;, &#x27;KB673412&#x27;, &#x27;KB673646&#x27;, &#x27;KB673490&#x27;)</dd></dl></div></li></ul></div></div></div></div>
</div>
<p>The arrays within this dataset are backed by <a class="reference external" href="https://docs.dask.org/en/latest/array.html">Dask arrays</a>, and can be accessed as shown below.</p>
<section id="snp-sites-and-alleles">
<h3>SNP sites and alleles<a class="headerlink" href="#snp-sites-and-alleles" title="Link to this heading">#</a></h3>
<p>We have called SNP genotypes in all samples at all positions in the genome where the reference allele is not “N”. Data on this set of genomic positions and alleles for a given chromosome (e.g., 2RL) can be accessed as <a class="reference external" href="https://docs.dask.org/en/latest/array.html">Dask arrays</a> as follows.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pos</span> <span class="o">=</span> <span class="n">ds_snps</span><span class="p">[</span><span class="s2">&quot;variant_position&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">data</span>
<span class="n">pos</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 83.80 MiB </td>
                        <td> 256.00 kiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539,) </td>
                        <td> (65536,) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 336 chunks in 1 graph layer </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> int32 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="6" y1="0" x2="6" y2="25" />
  <line x1="12" y1="0" x2="12" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="25" y1="0" x2="25" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="44" y1="0" x2="44" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="56" y1="0" x2="56" y2="25" />
  <line x1="63" y1="0" x2="63" y2="25" />
  <line x1="69" y1="0" x2="69" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="81" y1="0" x2="81" y2="25" />
  <line x1="88" y1="0" x2="88" y2="25" />
  <line x1="94" y1="0" x2="94" y2="25" />
  <line x1="100" y1="0" x2="100" y2="25" />
  <line x1="107" y1="0" x2="107" y2="25" />
  <line x1="113" y1="0" x2="113" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >21967539</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
        </td>
    </tr>
</table></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">alleles</span> <span class="o">=</span> <span class="n">ds_snps</span><span class="p">[</span><span class="s2">&quot;variant_allele&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">data</span>
<span class="n">alleles</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 670.40 MiB </td>
                        <td> 2.00 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 4) </td>
                        <td> (65536, 4) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 336 chunks in 5 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> object numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="75" height="170" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="25" y2="0" style="stroke-width:2" />
  <line x1="0" y1="6" x2="25" y2="6" />
  <line x1="0" y1="12" x2="25" y2="12" />
  <line x1="0" y1="18" x2="25" y2="18" />
  <line x1="0" y1="25" x2="25" y2="25" />
  <line x1="0" y1="31" x2="25" y2="31" />
  <line x1="0" y1="37" x2="25" y2="37" />
  <line x1="0" y1="44" x2="25" y2="44" />
  <line x1="0" y1="50" x2="25" y2="50" />
  <line x1="0" y1="56" x2="25" y2="56" />
  <line x1="0" y1="63" x2="25" y2="63" />
  <line x1="0" y1="69" x2="25" y2="69" />
  <line x1="0" y1="75" x2="25" y2="75" />
  <line x1="0" y1="81" x2="25" y2="81" />
  <line x1="0" y1="88" x2="25" y2="88" />
  <line x1="0" y1="94" x2="25" y2="94" />
  <line x1="0" y1="100" x2="25" y2="100" />
  <line x1="0" y1="107" x2="25" y2="107" />
  <line x1="0" y1="113" x2="25" y2="113" />
  <line x1="0" y1="120" x2="25" y2="120" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="120" style="stroke-width:2" />
  <line x1="25" y1="0" x2="25" y2="120" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="12.706308" y="140.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" >4</text>
  <text x="45.412617" y="60.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,45.412617,60.000000)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></div>
</div>
<p>Data can be loaded into memory as a <a class="reference external" href="https://numpy.org/doc/stable/user/absolute_beginners.html">NumPy array</a> as shown in the following examples.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read first 10 SNP positions into a numpy array</span>
<span class="n">p</span> <span class="o">=</span> <span class="n">pos</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">p</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10], dtype=int32)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read first 10 SNP alleles into a numpy array</span>
<span class="n">a</span> <span class="o">=</span> <span class="n">alleles</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">a</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([[&#39;A&#39;, &#39;C&#39;, &#39;G&#39;, &#39;T&#39;],
       [&#39;A&#39;, &#39;C&#39;, &#39;G&#39;, &#39;T&#39;],
       [&#39;A&#39;, &#39;C&#39;, &#39;G&#39;, &#39;T&#39;],
       [&#39;T&#39;, &#39;A&#39;, &#39;C&#39;, &#39;G&#39;],
       [&#39;T&#39;, &#39;A&#39;, &#39;C&#39;, &#39;G&#39;],
       [&#39;C&#39;, &#39;A&#39;, &#39;G&#39;, &#39;T&#39;],
       [&#39;A&#39;, &#39;C&#39;, &#39;G&#39;, &#39;T&#39;],
       [&#39;A&#39;, &#39;C&#39;, &#39;G&#39;, &#39;T&#39;],
       [&#39;A&#39;, &#39;C&#39;, &#39;G&#39;, &#39;T&#39;],
       [&#39;A&#39;, &#39;C&#39;, &#39;G&#39;, &#39;T&#39;]], dtype=object)
</pre></div>
</div>
</div>
</div>
<p>Here the first column contains the reference alleles, and the remaining columns contain the alternate alleles.</p>
<p>Note that a byte string data type is used here for efficiency. E.g., the Python code <code class="docutils literal notranslate"><span class="pre">b'T'</span></code> represents a byte string containing the letter “T”, which here stands for the nucleotide thymine.</p>
<p>Note that we have chosen to genotype all samples at all sites in the genome, assuming all possible SNP alleles. Not all of these alternate alleles will actually have been observed in the <code class="docutils literal notranslate"><span class="pre">Adir1</span></code> samples. To determine which sites and alleles are segregating, an allele count can be performed over the samples you are interested in. See the example below.</p>
</section>
<section id="site-filters">
<h3>Site filters<a class="headerlink" href="#site-filters" title="Link to this heading">#</a></h3>
<p>SNP calling is not always reliable, and we have created some site filters to allow excluding low quality SNPs.</p>
<p>Each set of site filters provides a “filter_pass” Boolean mask for each chromosome arm, where True indicates that the site passed the filter and is accessible to high quality SNP calling.</p>
<p>The site filters data can be accessed as dask arrays as shown in the examples below.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># access gamb_colu_arab site filters as a dask array</span>
<span class="n">filter_pass</span> <span class="o">=</span> <span class="n">ds_snps</span><span class="p">[</span><span class="s1">&#39;variant_filter_pass_dirus&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">data</span>
<span class="n">filter_pass</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 20.95 MiB </td>
                        <td> 292.97 kiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539,) </td>
                        <td> (300000,) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 74 chunks in 1 graph layer </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> bool numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="4" y1="0" x2="4" y2="25" />
  <line x1="11" y1="0" x2="11" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="24" y1="0" x2="24" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="44" y1="0" x2="44" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="57" y1="0" x2="57" y2="25" />
  <line x1="62" y1="0" x2="62" y2="25" />
  <line x1="68" y1="0" x2="68" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="81" y1="0" x2="81" y2="25" />
  <line x1="88" y1="0" x2="88" y2="25" />
  <line x1="95" y1="0" x2="95" y2="25" />
  <line x1="101" y1="0" x2="101" y2="25" />
  <line x1="108" y1="0" x2="108" y2="25" />
  <line x1="114" y1="0" x2="114" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >21967539</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
        </td>
    </tr>
</table></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read filter values for first 10 SNPs (True means the site passes filters)</span>
<span class="n">f</span> <span class="o">=</span> <span class="n">filter_pass</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">f</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([False, False, False, False, False, False, False, False, False,
       False])
</pre></div>
</div>
</div>
</div>
</section>
<section id="snp-genotypes">
<h3>SNP genotypes<a class="headerlink" href="#snp-genotypes" title="Link to this heading">#</a></h3>
<p>SNP genotypes for individual samples are available. Genotypes are stored as a three-dimensional array, where the first dimension corresponds to genomic positions, the second dimension is samples, and the third dimension is ploidy (2). Values are coded as integers, where -1 represents a missing value, 0 represents the reference allele, and 1, 2, and 3 represent alternate alleles.</p>
<p>SNP genotypes can be accessed as dask arrays as shown below.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt</span> <span class="o">=</span> <span class="n">ds_snps</span><span class="p">[</span><span class="s2">&quot;call_genotype&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">data</span>
<span class="n">gt</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 22.10 GiB </td>
                        <td> 28.61 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 540, 2) </td>
                        <td> (300000, 50, 2) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 888 chunks in 5 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> int8 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="2" x2="80" y2="72" />
  <line x1="10" y1="3" x2="80" y2="74" />
  <line x1="10" y1="5" x2="80" y2="76" />
  <line x1="10" y1="8" x2="80" y2="78" />
  <line x1="10" y1="10" x2="80" y2="81" />
  <line x1="10" y1="12" x2="80" y2="83" />
  <line x1="10" y1="15" x2="80" y2="85" />
  <line x1="10" y1="17" x2="80" y2="88" />
  <line x1="10" y1="19" x2="80" y2="90" />
  <line x1="10" y1="22" x2="80" y2="92" />
  <line x1="10" y1="24" x2="80" y2="95" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="12" y1="2" x2="12" y2="28" />
  <line x1="16" y1="6" x2="16" y2="32" />
  <line x1="20" y1="10" x2="20" y2="36" />
  <line x1="24" y1="14" x2="24" y2="39" />
  <line x1="28" y1="18" x2="28" y2="43" />
  <line x1="32" y1="22" x2="32" y2="47" />
  <line x1="36" y1="26" x2="36" y2="51" />
  <line x1="39" y1="29" x2="39" y2="55" />
  <line x1="43" y1="33" x2="43" y2="59" />
  <line x1="46" y1="36" x2="46" y2="62" />
  <line x1="50" y1="40" x2="50" y2="65" />
  <line x1="54" y1="44" x2="54" y2="69" />
  <line x1="58" y1="48" x2="58" y2="73" />
  <line x1="62" y1="52" x2="62" y2="77" />
  <line x1="65" y1="55" x2="65" y2="81" />
  <line x1="69" y1="59" x2="69" y2="85" />
  <line x1="73" y1="63" x2="73" y2="89" />
  <line x1="77" y1="67" x2="77" y2="92" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="12" y1="2" x2="38" y2="2" />
  <line x1="16" y1="6" x2="42" y2="6" />
  <line x1="20" y1="10" x2="46" y2="10" />
  <line x1="24" y1="14" x2="49" y2="14" />
  <line x1="28" y1="18" x2="53" y2="18" />
  <line x1="32" y1="22" x2="57" y2="22" />
  <line x1="36" y1="26" x2="61" y2="26" />
  <line x1="39" y1="29" x2="65" y2="29" />
  <line x1="43" y1="33" x2="69" y2="33" />
  <line x1="46" y1="36" x2="72" y2="36" />
  <line x1="50" y1="40" x2="75" y2="40" />
  <line x1="54" y1="44" x2="79" y2="44" />
  <line x1="58" y1="48" x2="83" y2="48" />
  <line x1="62" y1="52" x2="87" y2="52" />
  <line x1="65" y1="55" x2="91" y2="55" />
  <line x1="69" y1="59" x2="95" y2="59" />
  <line x1="73" y1="63" x2="99" y2="63" />
  <line x1="77" y1="67" x2="102" y2="67" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="72" x2="106" y2="72" />
  <line x1="80" y1="74" x2="106" y2="74" />
  <line x1="80" y1="76" x2="106" y2="76" />
  <line x1="80" y1="78" x2="106" y2="78" />
  <line x1="80" y1="81" x2="106" y2="81" />
  <line x1="80" y1="83" x2="106" y2="83" />
  <line x1="80" y1="85" x2="106" y2="85" />
  <line x1="80" y1="88" x2="106" y2="88" />
  <line x1="80" y1="90" x2="106" y2="90" />
  <line x1="80" y1="92" x2="106" y2="92" />
  <line x1="80" y1="95" x2="106" y2="95" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,126.000852,83.294544)">540</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></div>
</div>
<p>Note that the columns of this array (second dimension) match the rows in the sample metadata, if the same sample sets were loaded. I.e.:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_samples</span> <span class="o">=</span> <span class="n">adir1</span><span class="o">.</span><span class="n">sample_metadata</span><span class="p">(</span><span class="n">sample_sets</span><span class="o">=</span><span class="s2">&quot;1.0&quot;</span><span class="p">)</span>
<span class="n">gt</span> <span class="o">=</span> <span class="n">ds_snps</span><span class="p">[</span><span class="s2">&quot;call_genotype&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">data</span>
<span class="nb">len</span><span class="p">(</span><span class="n">df_samples</span><span class="p">)</span> <span class="o">==</span> <span class="n">gt</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>True
</pre></div>
</div>
</div>
</div>
<p>You can use this correspondance to apply further subsetting operations to the genotypes by querying the sample metadata. E.g.:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">loc_funestus</span> <span class="o">=</span> <span class="n">df_samples</span><span class="o">.</span><span class="n">eval</span><span class="p">(</span><span class="s2">&quot;taxon == &#39;baimaii&#39;&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">values</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;found </span><span class="si">{</span><span class="n">np</span><span class="o">.</span><span class="n">count_nonzero</span><span class="p">(</span><span class="n">loc_funestus</span><span class="p">)</span><span class="si">}</span><span class="s2"> baimaii samples&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>found 47 baimaii samples
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ds_snps_funestus</span> <span class="o">=</span> <span class="n">ds_snps</span><span class="o">.</span><span class="n">isel</span><span class="p">(</span><span class="n">samples</span><span class="o">=</span><span class="n">loc_funestus</span><span class="p">)</span>
<span class="n">ds_snps_funestus</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div><svg style="position: absolute; width: 0; height: 0; overflow: hidden">
<defs>
<symbol id="icon-database" viewBox="0 0 32 32">
<path d="M16 0c-8.837 0-16 2.239-16 5v4c0 2.761 7.163 5 16 5s16-2.239 16-5v-4c0-2.761-7.163-5-16-5z"></path>
<path d="M16 17c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z"></path>
<path d="M16 26c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z"></path>
</symbol>
<symbol id="icon-file-text2" viewBox="0 0 32 32">
<path d="M28.681 7.159c-0.694-0.947-1.662-2.053-2.724-3.116s-2.169-2.030-3.116-2.724c-1.612-1.182-2.393-1.319-2.841-1.319h-15.5c-1.378 0-2.5 1.121-2.5 2.5v27c0 1.378 1.122 2.5 2.5 2.5h23c1.378 0 2.5-1.122 2.5-2.5v-19.5c0-0.448-0.137-1.23-1.319-2.841zM24.543 5.457c0.959 0.959 1.712 1.825 2.268 2.543h-4.811v-4.811c0.718 0.556 1.584 1.309 2.543 2.268zM28 29.5c0 0.271-0.229 0.5-0.5 0.5h-23c-0.271 0-0.5-0.229-0.5-0.5v-27c0-0.271 0.229-0.5 0.5-0.5 0 0 15.499-0 15.5 0v7c0 0.552 0.448 1 1 1h7v19.5z"></path>
<path d="M23 26h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z"></path>
<path d="M23 22h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z"></path>
<path d="M23 18h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z"></path>
</symbol>
</defs>
</svg>
<style>/* CSS stylesheet for displaying xarray objects in jupyterlab.
 *
 */

:root {
  --xr-font-color0: var(
    --jp-content-font-color0,
    var(--pst-color-text-base rgba(0, 0, 0, 1))
  );
  --xr-font-color2: var(
    --jp-content-font-color2,
    var(--pst-color-text-base, rgba(0, 0, 0, 0.54))
  );
  --xr-font-color3: var(
    --jp-content-font-color3,
    var(--pst-color-text-base, rgba(0, 0, 0, 0.38))
  );
  --xr-border-color: var(
    --jp-border-color2,
    hsl(from var(--pst-color-on-background, white) h s calc(l - 10))
  );
  --xr-disabled-color: var(
    --jp-layout-color3,
    hsl(from var(--pst-color-on-background, white) h s calc(l - 40))
  );
  --xr-background-color: var(
    --jp-layout-color0,
    var(--pst-color-on-background, white)
  );
  --xr-background-color-row-even: var(
    --jp-layout-color1,
    hsl(from var(--pst-color-on-background, white) h s calc(l - 5))
  );
  --xr-background-color-row-odd: var(
    --jp-layout-color2,
    hsl(from var(--pst-color-on-background, white) h s calc(l - 15))
  );
}

html[theme="dark"],
html[data-theme="dark"],
body[data-theme="dark"],
body.vscode-dark {
  --xr-font-color0: var(
    --jp-content-font-color0,
    var(--pst-color-text-base, rgba(255, 255, 255, 1))
  );
  --xr-font-color2: var(
    --jp-content-font-color2,
    var(--pst-color-text-base, rgba(255, 255, 255, 0.54))
  );
  --xr-font-color3: var(
    --jp-content-font-color3,
    var(--pst-color-text-base, rgba(255, 255, 255, 0.38))
  );
  --xr-border-color: var(
    --jp-border-color2,
    hsl(from var(--pst-color-on-background, #111111) h s calc(l + 10))
  );
  --xr-disabled-color: var(
    --jp-layout-color3,
    hsl(from var(--pst-color-on-background, #111111) h s calc(l + 40))
  );
  --xr-background-color: var(
    --jp-layout-color0,
    var(--pst-color-on-background, #111111)
  );
  --xr-background-color-row-even: var(
    --jp-layout-color1,
    hsl(from var(--pst-color-on-background, #111111) h s calc(l + 5))
  );
  --xr-background-color-row-odd: var(
    --jp-layout-color2,
    hsl(from var(--pst-color-on-background, #111111) h s calc(l + 15))
  );
}

.xr-wrap {
  display: block !important;
  min-width: 300px;
  max-width: 700px;
}

.xr-text-repr-fallback {
  /* fallback to plain text repr when CSS is not injected (untrusted notebook) */
  display: none;
}

.xr-header {
  padding-top: 6px;
  padding-bottom: 6px;
  margin-bottom: 4px;
  border-bottom: solid 1px var(--xr-border-color);
}

.xr-header > div,
.xr-header > ul {
  display: inline;
  margin-top: 0;
  margin-bottom: 0;
}

.xr-obj-type,
.xr-array-name {
  margin-left: 2px;
  margin-right: 10px;
}

.xr-obj-type {
  color: var(--xr-font-color2);
}

.xr-sections {
  padding-left: 0 !important;
  display: grid;
  grid-template-columns: 150px auto auto 1fr 0 20px 0 20px;
}

.xr-section-item {
  display: contents;
}

.xr-section-item input {
  display: inline-block;
  opacity: 0;
  height: 0;
}

.xr-section-item input + label {
  color: var(--xr-disabled-color);
  border: 2px solid transparent !important;
}

.xr-section-item input:enabled + label {
  cursor: pointer;
  color: var(--xr-font-color2);
}

.xr-section-item input:focus + label {
  border: 2px solid var(--xr-font-color0) !important;
}

.xr-section-item input:enabled + label:hover {
  color: var(--xr-font-color0);
}

.xr-section-summary {
  grid-column: 1;
  color: var(--xr-font-color2);
  font-weight: 500;
}

.xr-section-summary > span {
  display: inline-block;
  padding-left: 0.5em;
}

.xr-section-summary-in:disabled + label {
  color: var(--xr-font-color2);
}

.xr-section-summary-in + label:before {
  display: inline-block;
  content: "►";
  font-size: 11px;
  width: 15px;
  text-align: center;
}

.xr-section-summary-in:disabled + label:before {
  color: var(--xr-disabled-color);
}

.xr-section-summary-in:checked + label:before {
  content: "▼";
}

.xr-section-summary-in:checked + label > span {
  display: none;
}

.xr-section-summary,
.xr-section-inline-details {
  padding-top: 4px;
  padding-bottom: 4px;
}

.xr-section-inline-details {
  grid-column: 2 / -1;
}

.xr-section-details {
  display: none;
  grid-column: 1 / -1;
  margin-bottom: 5px;
}

.xr-section-summary-in:checked ~ .xr-section-details {
  display: contents;
}

.xr-array-wrap {
  grid-column: 1 / -1;
  display: grid;
  grid-template-columns: 20px auto;
}

.xr-array-wrap > label {
  grid-column: 1;
  vertical-align: top;
}

.xr-preview {
  color: var(--xr-font-color3);
}

.xr-array-preview,
.xr-array-data {
  padding: 0 5px !important;
  grid-column: 2;
}

.xr-array-data,
.xr-array-in:checked ~ .xr-array-preview {
  display: none;
}

.xr-array-in:checked ~ .xr-array-data,
.xr-array-preview {
  display: inline-block;
}

.xr-dim-list {
  display: inline-block !important;
  list-style: none;
  padding: 0 !important;
  margin: 0;
}

.xr-dim-list li {
  display: inline-block;
  padding: 0;
  margin: 0;
}

.xr-dim-list:before {
  content: "(";
}

.xr-dim-list:after {
  content: ")";
}

.xr-dim-list li:not(:last-child):after {
  content: ",";
  padding-right: 5px;
}

.xr-has-index {
  font-weight: bold;
}

.xr-var-list,
.xr-var-item {
  display: contents;
}

.xr-var-item > div,
.xr-var-item label,
.xr-var-item > .xr-var-name span {
  background-color: var(--xr-background-color-row-even);
  border-color: var(--xr-background-color-row-odd);
  margin-bottom: 0;
  padding-top: 2px;
}

.xr-var-item > .xr-var-name:hover span {
  padding-right: 5px;
}

.xr-var-list > li:nth-child(odd) > div,
.xr-var-list > li:nth-child(odd) > label,
.xr-var-list > li:nth-child(odd) > .xr-var-name span {
  background-color: var(--xr-background-color-row-odd);
  border-color: var(--xr-background-color-row-even);
}

.xr-var-name {
  grid-column: 1;
}

.xr-var-dims {
  grid-column: 2;
}

.xr-var-dtype {
  grid-column: 3;
  text-align: right;
  color: var(--xr-font-color2);
}

.xr-var-preview {
  grid-column: 4;
}

.xr-index-preview {
  grid-column: 2 / 5;
  color: var(--xr-font-color2);
}

.xr-var-name,
.xr-var-dims,
.xr-var-dtype,
.xr-preview,
.xr-attrs dt {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  padding-right: 10px;
}

.xr-var-name:hover,
.xr-var-dims:hover,
.xr-var-dtype:hover,
.xr-attrs dt:hover {
  overflow: visible;
  width: auto;
  z-index: 1;
}

.xr-var-attrs,
.xr-var-data,
.xr-index-data {
  display: none;
  border-top: 2px dotted var(--xr-background-color);
  padding-bottom: 20px !important;
  padding-top: 10px !important;
}

.xr-var-attrs-in + label,
.xr-var-data-in + label,
.xr-index-data-in + label {
  padding: 0 1px;
}

.xr-var-attrs-in:checked ~ .xr-var-attrs,
.xr-var-data-in:checked ~ .xr-var-data,
.xr-index-data-in:checked ~ .xr-index-data {
  display: block;
}

.xr-var-data > table {
  float: right;
}

.xr-var-data > pre,
.xr-index-data > pre,
.xr-var-data > table > tbody > tr {
  background-color: transparent !important;
}

.xr-var-name span,
.xr-var-data,
.xr-index-name div,
.xr-index-data,
.xr-attrs {
  padding-left: 25px !important;
}

.xr-attrs,
.xr-var-attrs,
.xr-var-data,
.xr-index-data {
  grid-column: 1 / -1;
}

dl.xr-attrs {
  padding: 0;
  margin: 0;
  display: grid;
  grid-template-columns: 125px auto;
}

.xr-attrs dt,
.xr-attrs dd {
  padding: 0;
  margin: 0;
  float: left;
  padding-right: 10px;
  width: auto;
}

.xr-attrs dt {
  font-weight: normal;
  grid-column: 1;
}

.xr-attrs dt:hover span {
  display: inline-block;
  background: var(--xr-background-color);
  padding-right: 10px;
}

.xr-attrs dd {
  grid-column: 2;
  white-space: pre-wrap;
  word-break: break-all;
}

.xr-icon-database,
.xr-icon-file-text2,
.xr-no-icon {
  display: inline-block;
  vertical-align: middle;
  width: 1em;
  height: 1.5em !important;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
}

.xr-var-attrs-in:checked + label > .xr-icon-file-text2,
.xr-var-data-in:checked + label > .xr-icon-database,
.xr-index-data-in:checked + label > .xr-icon-database {
  color: var(--xr-font-color0);
  filter: drop-shadow(1px 1px 5px var(--xr-font-color2));
  stroke-width: 0.8px;
}
</style><pre class='xr-text-repr-fallback'>&lt;xarray.Dataset&gt; Size: 18GB
Dimensions:                    (variants: 21967539, alleles: 4, samples: 47,
                                ploidy: 2)
Coordinates:
    variant_position           (variants) int32 88MB dask.array&lt;chunksize=(65536,), meta=np.ndarray&gt;
    variant_contig             (variants) uint8 22MB dask.array&lt;chunksize=(65536,), meta=np.ndarray&gt;
    sample_id                  (samples) &lt;U36 7kB dask.array&lt;chunksize=(47,), meta=np.ndarray&gt;
Dimensions without coordinates: variants, alleles, samples, ploidy
Data variables:
    variant_allele             (variants, alleles) object 703MB dask.array&lt;chunksize=(65536, 4), meta=np.ndarray&gt;
    variant_filter_pass_dirus  (variants) bool 22MB dask.array&lt;chunksize=(300000,), meta=np.ndarray&gt;
    call_genotype              (variants, samples, ploidy) int8 2GB dask.array&lt;chunksize=(300000, 45, 2), meta=np.ndarray&gt;
    call_GQ                    (variants, samples) int8 1GB dask.array&lt;chunksize=(300000, 45), meta=np.ndarray&gt;
    call_MQ                    (variants, samples) float32 4GB dask.array&lt;chunksize=(300000, 45), meta=np.ndarray&gt;
    call_AD                    (variants, samples, alleles) int16 8GB dask.array&lt;chunksize=(300000, 45, 4), meta=np.ndarray&gt;
    call_genotype_mask         (variants, samples, ploidy) bool 2GB dask.array&lt;chunksize=(300000, 45, 2), meta=np.ndarray&gt;
Attributes:
    contigs:  (&#x27;KB672490&#x27;, &#x27;KB672868&#x27;, &#x27;KB672979&#x27;, &#x27;KB673090&#x27;, &#x27;KB673201&#x27;, &#x27;K...</pre><div class='xr-wrap' style='display:none'><div class='xr-header'><div class='xr-obj-type'>xarray.Dataset</div></div><ul class='xr-sections'><li class='xr-section-item'><input id='section-896415fe-81f5-44de-afe4-506291b86edb' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-896415fe-81f5-44de-afe4-506291b86edb' class='xr-section-summary'  title='Expand/collapse section'>Dimensions:</label><div class='xr-section-inline-details'><ul class='xr-dim-list'><li><span>variants</span>: 21967539</li><li><span>alleles</span>: 4</li><li><span>samples</span>: 47</li><li><span>ploidy</span>: 2</li></ul></div><div class='xr-section-details'></div></li><li class='xr-section-item'><input id='section-04b1591c-c2ab-4307-a25f-8b601b85c151' class='xr-section-summary-in' type='checkbox'  checked><label for='section-04b1591c-c2ab-4307-a25f-8b601b85c151' class='xr-section-summary' >Coordinates: <span>(3)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'><li class='xr-var-item'><div class='xr-var-name'><span>variant_position</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>int32</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(65536,), meta=np.ndarray&gt;</div><input id='attrs-108a3993-9a0b-4f58-b15c-9b30b10e28ae' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-108a3993-9a0b-4f58-b15c-9b30b10e28ae' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-2e05570a-72ce-4081-8f6a-53514c48b261' class='xr-var-data-in' type='checkbox'><label for='data-2e05570a-72ce-4081-8f6a-53514c48b261' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 83.80 MiB </td>
                        <td> 256.00 kiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539,) </td>
                        <td> (65536,) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 336 chunks in 1 graph layer </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> int32 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="6" y1="0" x2="6" y2="25" />
  <line x1="12" y1="0" x2="12" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="25" y1="0" x2="25" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="44" y1="0" x2="44" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="56" y1="0" x2="56" y2="25" />
  <line x1="63" y1="0" x2="63" y2="25" />
  <line x1="69" y1="0" x2="69" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="81" y1="0" x2="81" y2="25" />
  <line x1="88" y1="0" x2="88" y2="25" />
  <line x1="94" y1="0" x2="94" y2="25" />
  <line x1="100" y1="0" x2="100" y2="25" />
  <line x1="107" y1="0" x2="107" y2="25" />
  <line x1="113" y1="0" x2="113" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >21967539</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_contig</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>uint8</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(65536,), meta=np.ndarray&gt;</div><input id='attrs-b8229189-71e3-4ec4-93d9-3012d46a975c' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-b8229189-71e3-4ec4-93d9-3012d46a975c' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-c099c02d-00af-403e-9fb9-6d2d47134fa1' class='xr-var-data-in' type='checkbox'><label for='data-c099c02d-00af-403e-9fb9-6d2d47134fa1' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 20.95 MiB </td>
                        <td> 64.00 kiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539,) </td>
                        <td> (65536,) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 336 chunks in 1 graph layer </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> uint8 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="6" y1="0" x2="6" y2="25" />
  <line x1="12" y1="0" x2="12" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="25" y1="0" x2="25" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="44" y1="0" x2="44" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="56" y1="0" x2="56" y2="25" />
  <line x1="63" y1="0" x2="63" y2="25" />
  <line x1="69" y1="0" x2="69" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="81" y1="0" x2="81" y2="25" />
  <line x1="88" y1="0" x2="88" y2="25" />
  <line x1="94" y1="0" x2="94" y2="25" />
  <line x1="100" y1="0" x2="100" y2="25" />
  <line x1="107" y1="0" x2="107" y2="25" />
  <line x1="113" y1="0" x2="113" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >21967539</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>sample_id</span></div><div class='xr-var-dims'>(samples)</div><div class='xr-var-dtype'>&lt;U36</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(47,), meta=np.ndarray&gt;</div><input id='attrs-526767c4-9b85-4051-8f4a-8f8d44803aaa' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-526767c4-9b85-4051-8f4a-8f8d44803aaa' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-09f430af-dc12-41cb-b2ac-6770f51def90' class='xr-var-data-in' type='checkbox'><label for='data-09f430af-dc12-41cb-b2ac-6770f51def90' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 6.61 kiB </td>
                        <td> 6.61 kiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (47,) </td>
                        <td> (47,) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 1 chunks in 10 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> <U36 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="170" height="79" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="29" x2="120" y2="29" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="29" style="stroke-width:2" />
  <line x1="120" y1="0" x2="120" y2="29" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,29.38084118065451 0.0,29.38084118065451" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="49.380841" font-size="1.0rem" font-weight="100" text-anchor="middle" >47</text>
  <text x="140.000000" y="14.690421" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,14.690421)">1</text>
</svg>
        </td>
    </tr>
</table></div></li></ul></div></li><li class='xr-section-item'><input id='section-cf4ef662-b626-4aae-a06e-a5808c981ff6' class='xr-section-summary-in' type='checkbox'  checked><label for='section-cf4ef662-b626-4aae-a06e-a5808c981ff6' class='xr-section-summary' >Data variables: <span>(7)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'><li class='xr-var-item'><div class='xr-var-name'><span>variant_allele</span></div><div class='xr-var-dims'>(variants, alleles)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(65536, 4), meta=np.ndarray&gt;</div><input id='attrs-0c21ce6e-34e0-49e0-af22-f615b27a843b' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-0c21ce6e-34e0-49e0-af22-f615b27a843b' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-0e75cefa-a499-44b3-8cc8-862066783488' class='xr-var-data-in' type='checkbox'><label for='data-0e75cefa-a499-44b3-8cc8-862066783488' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 670.40 MiB </td>
                        <td> 2.00 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 4) </td>
                        <td> (65536, 4) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 336 chunks in 5 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> object numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="75" height="170" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="25" y2="0" style="stroke-width:2" />
  <line x1="0" y1="6" x2="25" y2="6" />
  <line x1="0" y1="12" x2="25" y2="12" />
  <line x1="0" y1="18" x2="25" y2="18" />
  <line x1="0" y1="25" x2="25" y2="25" />
  <line x1="0" y1="31" x2="25" y2="31" />
  <line x1="0" y1="37" x2="25" y2="37" />
  <line x1="0" y1="44" x2="25" y2="44" />
  <line x1="0" y1="50" x2="25" y2="50" />
  <line x1="0" y1="56" x2="25" y2="56" />
  <line x1="0" y1="63" x2="25" y2="63" />
  <line x1="0" y1="69" x2="25" y2="69" />
  <line x1="0" y1="75" x2="25" y2="75" />
  <line x1="0" y1="81" x2="25" y2="81" />
  <line x1="0" y1="88" x2="25" y2="88" />
  <line x1="0" y1="94" x2="25" y2="94" />
  <line x1="0" y1="100" x2="25" y2="100" />
  <line x1="0" y1="107" x2="25" y2="107" />
  <line x1="0" y1="113" x2="25" y2="113" />
  <line x1="0" y1="120" x2="25" y2="120" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="120" style="stroke-width:2" />
  <line x1="25" y1="0" x2="25" y2="120" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="12.706308" y="140.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" >4</text>
  <text x="45.412617" y="60.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,45.412617,60.000000)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_filter_pass_dirus</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>bool</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000,), meta=np.ndarray&gt;</div><input id='attrs-f6e5741b-599e-43cb-9a0c-2e8b0836ba4e' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-f6e5741b-599e-43cb-9a0c-2e8b0836ba4e' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-73e147d9-35e8-416b-b404-624b877accd0' class='xr-var-data-in' type='checkbox'><label for='data-73e147d9-35e8-416b-b404-624b877accd0' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 20.95 MiB </td>
                        <td> 292.97 kiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539,) </td>
                        <td> (300000,) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 74 chunks in 1 graph layer </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> bool numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="4" y1="0" x2="4" y2="25" />
  <line x1="11" y1="0" x2="11" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="24" y1="0" x2="24" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="44" y1="0" x2="44" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="57" y1="0" x2="57" y2="25" />
  <line x1="62" y1="0" x2="62" y2="25" />
  <line x1="68" y1="0" x2="68" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="81" y1="0" x2="81" y2="25" />
  <line x1="88" y1="0" x2="88" y2="25" />
  <line x1="95" y1="0" x2="95" y2="25" />
  <line x1="101" y1="0" x2="101" y2="25" />
  <line x1="108" y1="0" x2="108" y2="25" />
  <line x1="114" y1="0" x2="114" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >21967539</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_genotype</span></div><div class='xr-var-dims'>(variants, samples, ploidy)</div><div class='xr-var-dtype'>int8</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000, 45, 2), meta=np.ndarray&gt;</div><input id='attrs-fee2b005-f8d9-4d95-b400-6cf7746481bb' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-fee2b005-f8d9-4d95-b400-6cf7746481bb' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-74be3ddf-fb0d-4a74-9984-eaadb7dfd180' class='xr-var-data-in' type='checkbox'><label for='data-74be3ddf-fb0d-4a74-9984-eaadb7dfd180' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 1.92 GiB </td>
                        <td> 25.75 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 47, 2) </td>
                        <td> (300000, 45, 2) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 148 chunks in 6 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> int8 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="24" x2="80" y2="94" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="12" y1="2" x2="12" y2="28" />
  <line x1="16" y1="6" x2="16" y2="32" />
  <line x1="20" y1="10" x2="20" y2="36" />
  <line x1="24" y1="14" x2="24" y2="39" />
  <line x1="28" y1="18" x2="28" y2="43" />
  <line x1="32" y1="22" x2="32" y2="47" />
  <line x1="36" y1="26" x2="36" y2="51" />
  <line x1="39" y1="29" x2="39" y2="55" />
  <line x1="43" y1="33" x2="43" y2="59" />
  <line x1="46" y1="36" x2="46" y2="62" />
  <line x1="50" y1="40" x2="50" y2="65" />
  <line x1="54" y1="44" x2="54" y2="69" />
  <line x1="58" y1="48" x2="58" y2="73" />
  <line x1="62" y1="52" x2="62" y2="77" />
  <line x1="65" y1="55" x2="65" y2="81" />
  <line x1="69" y1="59" x2="69" y2="85" />
  <line x1="73" y1="63" x2="73" y2="89" />
  <line x1="77" y1="67" x2="77" y2="92" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="12" y1="2" x2="38" y2="2" />
  <line x1="16" y1="6" x2="42" y2="6" />
  <line x1="20" y1="10" x2="46" y2="10" />
  <line x1="24" y1="14" x2="49" y2="14" />
  <line x1="28" y1="18" x2="53" y2="18" />
  <line x1="32" y1="22" x2="57" y2="22" />
  <line x1="36" y1="26" x2="61" y2="26" />
  <line x1="39" y1="29" x2="65" y2="29" />
  <line x1="43" y1="33" x2="69" y2="33" />
  <line x1="46" y1="36" x2="72" y2="36" />
  <line x1="50" y1="40" x2="75" y2="40" />
  <line x1="54" y1="44" x2="79" y2="44" />
  <line x1="58" y1="48" x2="83" y2="48" />
  <line x1="62" y1="52" x2="87" y2="52" />
  <line x1="65" y1="55" x2="91" y2="55" />
  <line x1="69" y1="59" x2="95" y2="59" />
  <line x1="73" y1="63" x2="99" y2="63" />
  <line x1="77" y1="67" x2="102" y2="67" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="94" x2="106" y2="94" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,126.000852,83.294544)">47</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_GQ</span></div><div class='xr-var-dims'>(variants, samples)</div><div class='xr-var-dtype'>int8</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000, 45), meta=np.ndarray&gt;</div><input id='attrs-077e3963-054b-4db7-86e3-8edf09c205ec' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-077e3963-054b-4db7-86e3-8edf09c205ec' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-70134805-6747-402b-849b-59637bbd969a' class='xr-var-data-in' type='checkbox'><label for='data-70134805-6747-402b-849b-59637bbd969a' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 0.96 GiB </td>
                        <td> 12.87 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 47) </td>
                        <td> (300000, 45) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 148 chunks in 6 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> int8 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="75" height="170" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="25" y2="0" style="stroke-width:2" />
  <line x1="0" y1="4" x2="25" y2="4" />
  <line x1="0" y1="11" x2="25" y2="11" />
  <line x1="0" y1="18" x2="25" y2="18" />
  <line x1="0" y1="24" x2="25" y2="24" />
  <line x1="0" y1="31" x2="25" y2="31" />
  <line x1="0" y1="37" x2="25" y2="37" />
  <line x1="0" y1="44" x2="25" y2="44" />
  <line x1="0" y1="50" x2="25" y2="50" />
  <line x1="0" y1="57" x2="25" y2="57" />
  <line x1="0" y1="62" x2="25" y2="62" />
  <line x1="0" y1="68" x2="25" y2="68" />
  <line x1="0" y1="75" x2="25" y2="75" />
  <line x1="0" y1="81" x2="25" y2="81" />
  <line x1="0" y1="88" x2="25" y2="88" />
  <line x1="0" y1="95" x2="25" y2="95" />
  <line x1="0" y1="101" x2="25" y2="101" />
  <line x1="0" y1="108" x2="25" y2="108" />
  <line x1="0" y1="114" x2="25" y2="114" />
  <line x1="0" y1="120" x2="25" y2="120" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="120" style="stroke-width:2" />
  <line x1="24" y1="0" x2="24" y2="120" />
  <line x1="25" y1="0" x2="25" y2="120" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="12.706308" y="140.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" >47</text>
  <text x="45.412617" y="60.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,45.412617,60.000000)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_MQ</span></div><div class='xr-var-dims'>(variants, samples)</div><div class='xr-var-dtype'>float32</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000, 45), meta=np.ndarray&gt;</div><input id='attrs-52945d69-19e0-4ec4-80ea-0bd50ac33f9e' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-52945d69-19e0-4ec4-80ea-0bd50ac33f9e' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-00e26d69-6b03-4704-b97d-de3ed3cf91e9' class='xr-var-data-in' type='checkbox'><label for='data-00e26d69-6b03-4704-b97d-de3ed3cf91e9' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 3.85 GiB </td>
                        <td> 51.50 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 47) </td>
                        <td> (300000, 45) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 148 chunks in 6 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> float32 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="75" height="170" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="25" y2="0" style="stroke-width:2" />
  <line x1="0" y1="4" x2="25" y2="4" />
  <line x1="0" y1="11" x2="25" y2="11" />
  <line x1="0" y1="18" x2="25" y2="18" />
  <line x1="0" y1="24" x2="25" y2="24" />
  <line x1="0" y1="31" x2="25" y2="31" />
  <line x1="0" y1="37" x2="25" y2="37" />
  <line x1="0" y1="44" x2="25" y2="44" />
  <line x1="0" y1="50" x2="25" y2="50" />
  <line x1="0" y1="57" x2="25" y2="57" />
  <line x1="0" y1="62" x2="25" y2="62" />
  <line x1="0" y1="68" x2="25" y2="68" />
  <line x1="0" y1="75" x2="25" y2="75" />
  <line x1="0" y1="81" x2="25" y2="81" />
  <line x1="0" y1="88" x2="25" y2="88" />
  <line x1="0" y1="95" x2="25" y2="95" />
  <line x1="0" y1="101" x2="25" y2="101" />
  <line x1="0" y1="108" x2="25" y2="108" />
  <line x1="0" y1="114" x2="25" y2="114" />
  <line x1="0" y1="120" x2="25" y2="120" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="120" style="stroke-width:2" />
  <line x1="24" y1="0" x2="24" y2="120" />
  <line x1="25" y1="0" x2="25" y2="120" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="12.706308" y="140.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" >47</text>
  <text x="45.412617" y="60.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,45.412617,60.000000)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_AD</span></div><div class='xr-var-dims'>(variants, samples, alleles)</div><div class='xr-var-dtype'>int16</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000, 45, 4), meta=np.ndarray&gt;</div><input id='attrs-592f0f29-4935-413d-be93-6b06b00505f7' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-592f0f29-4935-413d-be93-6b06b00505f7' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-1f61181f-96a4-46fb-b5b5-0db1623f6269' class='xr-var-data-in' type='checkbox'><label for='data-1f61181f-96a4-46fb-b5b5-0db1623f6269' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 7.69 GiB </td>
                        <td> 103.00 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 47, 4) </td>
                        <td> (300000, 45, 4) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 148 chunks in 6 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> int16 numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="24" x2="80" y2="94" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="12" y1="2" x2="12" y2="28" />
  <line x1="16" y1="6" x2="16" y2="32" />
  <line x1="20" y1="10" x2="20" y2="36" />
  <line x1="24" y1="14" x2="24" y2="39" />
  <line x1="28" y1="18" x2="28" y2="43" />
  <line x1="32" y1="22" x2="32" y2="47" />
  <line x1="36" y1="26" x2="36" y2="51" />
  <line x1="39" y1="29" x2="39" y2="55" />
  <line x1="43" y1="33" x2="43" y2="59" />
  <line x1="46" y1="36" x2="46" y2="62" />
  <line x1="50" y1="40" x2="50" y2="65" />
  <line x1="54" y1="44" x2="54" y2="69" />
  <line x1="58" y1="48" x2="58" y2="73" />
  <line x1="62" y1="52" x2="62" y2="77" />
  <line x1="65" y1="55" x2="65" y2="81" />
  <line x1="69" y1="59" x2="69" y2="85" />
  <line x1="73" y1="63" x2="73" y2="89" />
  <line x1="77" y1="67" x2="77" y2="92" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="12" y1="2" x2="38" y2="2" />
  <line x1="16" y1="6" x2="42" y2="6" />
  <line x1="20" y1="10" x2="46" y2="10" />
  <line x1="24" y1="14" x2="49" y2="14" />
  <line x1="28" y1="18" x2="53" y2="18" />
  <line x1="32" y1="22" x2="57" y2="22" />
  <line x1="36" y1="26" x2="61" y2="26" />
  <line x1="39" y1="29" x2="65" y2="29" />
  <line x1="43" y1="33" x2="69" y2="33" />
  <line x1="46" y1="36" x2="72" y2="36" />
  <line x1="50" y1="40" x2="75" y2="40" />
  <line x1="54" y1="44" x2="79" y2="44" />
  <line x1="58" y1="48" x2="83" y2="48" />
  <line x1="62" y1="52" x2="87" y2="52" />
  <line x1="65" y1="55" x2="91" y2="55" />
  <line x1="69" y1="59" x2="95" y2="59" />
  <line x1="73" y1="63" x2="99" y2="63" />
  <line x1="77" y1="67" x2="102" y2="67" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="94" x2="106" y2="94" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >4</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,126.000852,83.294544)">47</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_genotype_mask</span></div><div class='xr-var-dims'>(variants, samples, ploidy)</div><div class='xr-var-dtype'>bool</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(300000, 45, 2), meta=np.ndarray&gt;</div><input id='attrs-4c3bafa0-712e-44be-88ce-acb4aadbfb81' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-4c3bafa0-712e-44be-88ce-acb4aadbfb81' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-c503575c-5f5c-4894-9d03-94b528df5c95' class='xr-var-data-in' type='checkbox'><label for='data-c503575c-5f5c-4894-9d03-94b528df5c95' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>
    <tr>
        <td>
            <table style="border-collapse: collapse;">
                <thead>
                    <tr>
                        <td> </td>
                        <th> Array </th>
                        <th> Chunk </th>
                    </tr>
                </thead>
                <tbody>
                    
                    <tr>
                        <th> Bytes </th>
                        <td> 1.92 GiB </td>
                        <td> 25.75 MiB </td>
                    </tr>
                    
                    <tr>
                        <th> Shape </th>
                        <td> (21967539, 47, 2) </td>
                        <td> (300000, 45, 2) </td>
                    </tr>
                    <tr>
                        <th> Dask graph </th>
                        <td colspan="2"> 148 chunks in 7 graph layers </td>
                    </tr>
                    <tr>
                        <th> Data type </th>
                        <td colspan="2"> bool numpy.ndarray </td>
                    </tr>
                </tbody>
            </table>
        </td>
        <td>
        <svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="24" x2="80" y2="94" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="12" y1="2" x2="12" y2="28" />
  <line x1="16" y1="6" x2="16" y2="32" />
  <line x1="20" y1="10" x2="20" y2="36" />
  <line x1="24" y1="14" x2="24" y2="39" />
  <line x1="28" y1="18" x2="28" y2="43" />
  <line x1="32" y1="22" x2="32" y2="47" />
  <line x1="36" y1="26" x2="36" y2="51" />
  <line x1="39" y1="29" x2="39" y2="55" />
  <line x1="43" y1="33" x2="43" y2="59" />
  <line x1="46" y1="36" x2="46" y2="62" />
  <line x1="50" y1="40" x2="50" y2="65" />
  <line x1="54" y1="44" x2="54" y2="69" />
  <line x1="58" y1="48" x2="58" y2="73" />
  <line x1="62" y1="52" x2="62" y2="77" />
  <line x1="65" y1="55" x2="65" y2="81" />
  <line x1="69" y1="59" x2="69" y2="85" />
  <line x1="73" y1="63" x2="73" y2="89" />
  <line x1="77" y1="67" x2="77" y2="92" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="12" y1="2" x2="38" y2="2" />
  <line x1="16" y1="6" x2="42" y2="6" />
  <line x1="20" y1="10" x2="46" y2="10" />
  <line x1="24" y1="14" x2="49" y2="14" />
  <line x1="28" y1="18" x2="53" y2="18" />
  <line x1="32" y1="22" x2="57" y2="22" />
  <line x1="36" y1="26" x2="61" y2="26" />
  <line x1="39" y1="29" x2="65" y2="29" />
  <line x1="43" y1="33" x2="69" y2="33" />
  <line x1="46" y1="36" x2="72" y2="36" />
  <line x1="50" y1="40" x2="75" y2="40" />
  <line x1="54" y1="44" x2="79" y2="44" />
  <line x1="58" y1="48" x2="83" y2="48" />
  <line x1="62" y1="52" x2="87" y2="52" />
  <line x1="65" y1="55" x2="91" y2="55" />
  <line x1="69" y1="59" x2="95" y2="59" />
  <line x1="73" y1="63" x2="99" y2="63" />
  <line x1="77" y1="67" x2="102" y2="67" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="94" x2="106" y2="94" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,126.000852,83.294544)">47</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">21967539</text>
</svg>
        </td>
    </tr>
</table></div></li></ul></div></li><li class='xr-section-item'><input id='section-a2f89015-cb25-4ce0-9ac5-4c9667c52440' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-a2f89015-cb25-4ce0-9ac5-4c9667c52440' class='xr-section-summary'  title='Expand/collapse section'>Indexes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'></ul></div></li><li class='xr-section-item'><input id='section-f54e219d-435c-4b5f-a892-ede420c2ca93' class='xr-section-summary-in' type='checkbox'  checked><label for='section-f54e219d-435c-4b5f-a892-ede420c2ca93' class='xr-section-summary' >Attributes: <span>(1)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><dl class='xr-attrs'><dt><span>contigs :</span></dt><dd>(&#x27;KB672490&#x27;, &#x27;KB672868&#x27;, &#x27;KB672979&#x27;, &#x27;KB673090&#x27;, &#x27;KB673201&#x27;, &#x27;KB673312&#x27;, &#x27;KB673423&#x27;, &#x27;KB673534&#x27;, &#x27;KB673645&#x27;, &#x27;KB672491&#x27;, &#x27;KB672602&#x27;, &#x27;KB672791&#x27;, &#x27;KB672713&#x27;, &#x27;KB672802&#x27;, &#x27;KB672813&#x27;, &#x27;KB672824&#x27;, &#x27;KB672846&#x27;, &#x27;KB672835&#x27;, &#x27;KB672857&#x27;, &#x27;KB672869&#x27;, &#x27;KB672880&#x27;, &#x27;KB672891&#x27;, &#x27;KB672902&#x27;, &#x27;KB672913&#x27;, &#x27;KB672924&#x27;, &#x27;KB672935&#x27;, &#x27;KB672957&#x27;, &#x27;KB672946&#x27;, &#x27;KB672968&#x27;, &#x27;KB672980&#x27;, &#x27;KB672991&#x27;, &#x27;KB673002&#x27;, &#x27;KB673013&#x27;, &#x27;KB673024&#x27;, &#x27;KB673046&#x27;, &#x27;KB673035&#x27;, &#x27;KB673057&#x27;, &#x27;KB673068&#x27;, &#x27;KB673079&#x27;, &#x27;KB673091&#x27;, &#x27;KB673102&#x27;, &#x27;KB673124&#x27;, &#x27;KB673113&#x27;, &#x27;KB673157&#x27;, &#x27;KB673146&#x27;, &#x27;KB673135&#x27;, &#x27;KB673168&#x27;, &#x27;KB673190&#x27;, &#x27;KB673179&#x27;, &#x27;KB673202&#x27;, &#x27;KB673213&#x27;, &#x27;KB673235&#x27;, &#x27;KB673224&#x27;, &#x27;KB673268&#x27;, &#x27;KB673246&#x27;, &#x27;KB673257&#x27;, &#x27;KB673279&#x27;, &#x27;KB673301&#x27;, &#x27;KB673290&#x27;, &#x27;KB673313&#x27;, &#x27;KB673324&#x27;, &#x27;KB673357&#x27;, &#x27;KB673379&#x27;, &#x27;KB673346&#x27;, &#x27;KB673457&#x27;, &#x27;KB673335&#x27;, &#x27;KB673535&#x27;, &#x27;KB673546&#x27;, &#x27;KB673435&#x27;, &#x27;KB673446&#x27;, &#x27;KB673368&#x27;, &#x27;KB673424&#x27;, &#x27;KB673390&#x27;, &#x27;KB673401&#x27;, &#x27;KB673468&#x27;, &#x27;KB673479&#x27;, &#x27;KB673412&#x27;, &#x27;KB673646&#x27;, &#x27;KB673490&#x27;)</dd></dl></div></li></ul></div></div></div></div>
</div>
<p>Data can be read into memory as numpy arrays, e.g., read genotypes for the first 5 SNPs and the first 3 samples:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">g</span> <span class="o">=</span> <span class="n">gt</span><span class="p">[:</span><span class="mi">5</span><span class="p">,</span> <span class="p">:</span><span class="mi">3</span><span class="p">,</span> <span class="p">:]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">g</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([[[-1, -1],
        [-1, -1],
        [-1, -1]],

       [[-1, -1],
        [-1, -1],
        [-1, -1]],

       [[-1, -1],
        [-1, -1],
        [-1, -1]],

       [[-1, -1],
        [-1, -1],
        [-1, -1]],

       [[-1, -1],
        [-1, -1],
        [-1, -1]]], dtype=int8)
</pre></div>
</div>
</div>
</div>
<p>If you want to work with the genotype calls, you may find it convenient to use <a class="reference external" href="http://scikit-allel.readthedocs.org/">scikit-allel</a>.
E.g., the code below sets up a genotype array.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># use the scikit-allel wrapper class for genotype calls</span>
<span class="n">gt</span> <span class="o">=</span> <span class="n">allel</span><span class="o">.</span><span class="n">GenotypeDaskArray</span><span class="p">(</span><span class="n">ds_snps</span><span class="p">[</span><span class="s2">&quot;call_genotype&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">data</span><span class="p">)</span>
<span class="n">gt</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div class="allel allel-DisplayAs2D"><span>&lt;GenotypeDaskArray shape=(21967539, 540, 2) dtype=int8&gt;</span><table><thead><tr><th></th><th style="text-align: center">0</th><th style="text-align: center">1</th><th style="text-align: center">2</th><th style="text-align: center">3</th><th style="text-align: center">4</th><th style="text-align: center">...</th><th style="text-align: center">535</th><th style="text-align: center">536</th><th style="text-align: center">537</th><th style="text-align: center">538</th><th style="text-align: center">539</th></tr></thead><tbody><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">0</th><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">...</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">1</th><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">...</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">2</th><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">...</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">...</th><td style="text-align: center" colspan="12">...</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">21967536</th><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">21967537</th><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">21967538</th><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr></tbody></table></div></div></div>
</div>
</section>
</section>
<section id="example-computation">
<h2>Example computation<a class="headerlink" href="#example-computation" title="Link to this heading">#</a></h2>
<p>Here’s an example computation to count the number of segregating SNPs on the longest contig (KB672490) that also pass site filters. This may take a minute or two, because it is scanning genotype calls at millions of SNPs in hundreds of samples.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># choose contig (longest contig)</span>
<span class="n">region</span> <span class="o">=</span> <span class="s2">&quot;KB672490&quot;</span>
<span class="c1"># choose site filter mask</span>

<span class="c1"># choose sample sets</span>
<span class="n">sample_sets</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;1278-VO-TH-KOBYLINSKI-VMF00153&quot;</span><span class="p">]</span>

<span class="c1"># access SNP calls</span>
<span class="n">ds_snps</span> <span class="o">=</span> <span class="n">adir1</span><span class="o">.</span><span class="n">snp_calls</span><span class="p">(</span><span class="n">region</span><span class="o">=</span><span class="n">region</span><span class="p">,</span> <span class="n">sample_sets</span><span class="o">=</span><span class="n">sample_sets</span><span class="p">)</span>

<span class="c1"># locate pass sites</span>
<span class="n">loc_pass</span> <span class="o">=</span> <span class="n">ds_snps</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;variant_filter_pass_dirus&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>

<span class="c1"># perform an allele count over genotypes</span>
<span class="n">gt</span> <span class="o">=</span> <span class="n">allel</span><span class="o">.</span><span class="n">GenotypeDaskArray</span><span class="p">(</span><span class="n">ds_snps</span><span class="p">[</span><span class="s2">&quot;call_genotype&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">data</span><span class="p">)</span>
<span class="k">with</span> <span class="n">ProgressBar</span><span class="p">():</span>
    <span class="n">ac</span> <span class="o">=</span> <span class="n">gt</span><span class="o">.</span><span class="n">count_alleles</span><span class="p">(</span><span class="n">max_allele</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>

<span class="c1"># locate segregating sites</span>
<span class="n">loc_seg</span> <span class="o">=</span> <span class="n">ac</span><span class="o">.</span><span class="n">is_segregating</span><span class="p">()</span>

<span class="c1"># count segregating and pass sites</span>
<span class="n">n_pass_seg</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">count_nonzero</span><span class="p">(</span><span class="n">loc_pass</span> <span class="o">&amp;</span> <span class="n">loc_seg</span><span class="p">)</span>
<span class="n">n_pass_seg</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[########################################] | 100% Completed | 139.73 s
</pre></div>
</div>
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2692310
</pre></div>
</div>
</div>
</div>
</section>
<section id="running-larger-computations">
<h2>Running larger computations<a class="headerlink" href="#running-larger-computations" title="Link to this heading">#</a></h2>
<p>Please note that free cloud computing services such as Google Colab and MyBinder provide only limited computing resources. Thus although these services are able to efficiently read <code class="docutils literal notranslate"><span class="pre">Adir1</span></code> data stored on Google Cloud, you may find that you run out of memory, or computations take a long time running on a single core. If you would like any suggestions regarding how to set up more powerful compute resources in the cloud, please feel free to get in touch via the <a class="reference external" href="https://github.com/malariagen/vector-data/discussions">malariagen/vector-data GitHub discussion board</a>.</p>
</section>
<section id="feedback-and-suggestions">
<h2>Feedback and suggestions<a class="headerlink" href="#feedback-and-suggestions" title="Link to this heading">#</a></h2>
<p>If there are particular analyses you would like to run, or if you have other suggestions for useful documentation we could add to this site, we would love to know, please get in touch via the <a class="reference external" href="https://github.com/malariagen/vector-data/discussions">malariagen/vector-data GitHub discussion board</a>.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "adir1.0-dev-env"
        },
        kernelOptions: {
            name: "adir1.0-dev-env",
            path: "./adir1"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'adir1.0-dev-env'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="adir1.0.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Adir1.0 (Vector Observatory - Asia Project <em>Anopheles dirus</em> Phase 1 Data Release)</p>
      </div>
    </a>
    <a class="right-next"
       href="download.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Adir1.0 data downloads</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-hosting">Data hosting</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#setup">Setup</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#sample-sets">Sample sets</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#sample-metadata">Sample metadata</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#snp-calls">SNP calls</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#snp-sites-and-alleles">SNP sites and alleles</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#site-filters">Site filters</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#snp-genotypes">SNP genotypes</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#example-computation">Example computation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#running-larger-computations">Running larger computations</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#feedback-and-suggestions">Feedback and suggestions</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By MalariaGEN
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2021.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>