
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Ag3.0 (Ag1000G phase 3) &#8212; MalariaGEN vector data user guide</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css?v=a1e01515" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=4a39c7ea"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "malariagen/vector-data");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'ag3/ag3.0';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ag3.1" href="ag3.1.html" />
    <link rel="prev" title="Vector Observatory Data Access" href="../vobs/vobs-data-access.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../landing-page.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/logo.png" class="logo__image only-light" alt="MalariaGEN vector data user guide - Home"/>
    <script>document.write(`<img src="../_static/logo.png" class="logo__image only-dark" alt="MalariaGEN vector data user guide - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../landing-page.html">
                    MalariaGEN vector data user guide
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Malaria Vector Genome Observatory</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../vobs/vobs-data-access.html">Vector Observatory Data Access</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Anopheles gambiae complex</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Ag3.0 (Ag1000G phase 3)</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.1.html">Ag3.1</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.2.html">Ag3.2</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.3.html">Ag3.3</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.4.html">Ag3.4</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.5.html">Ag3.5</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.6.html">Ag3.6</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.7.html">Ag3.7</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.8.html">Ag3.8</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.9.html">Ag3.9</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.10.html">Ag3.10</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.11.html">Ag3.11</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.12.html">Ag3.12</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.13.html">Ag3.13</a></li>
<li class="toctree-l1"><a class="reference internal" href="ag3.14.html">Ag3.14</a></li>
<li class="toctree-l1"><a class="reference internal" href="cloud.html">Ag3 cloud data access</a></li>
<li class="toctree-l1"><a class="reference internal" href="download.html">Ag3 data downloads</a></li>
<li class="toctree-l1"><a class="reference internal" href="methods.html">Ag3 sequencing and variant calling methods</a></li>
<li class="toctree-l1"><a class="reference internal" href="api.html">Ag3 API</a></li>
<li class="toctree-l1"><a class="reference internal" href="analysis.html">Ag3 analysis</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Anopheles funestus</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../af1/af1.0.html">Af1.0  (<em>Anopheles funestus</em> Project Phase 1 Data Release)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/af1.1.html">Af1.1</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/af1.2.html">Af1.2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/af1.3.html">Af1.3</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/af1.4.html">Af1.4</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/cloud.html">Af1 cloud data access</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/download.html">Af1 data downloads</a></li>
<li class="toctree-l1"><a class="reference internal" href="../af1/api.html">Af1 API</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Anopheles minimus</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../amin1/intro.html">Amin1.0 (Vector Observatory - Asia Project <em>Anopheles minimus</em> Phase 1 Data Release)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../amin1/cloud.html">Amin1.0 cloud data access</a></li>
<li class="toctree-l2"><a class="reference internal" href="../amin1/download.html">Amin1.0 data downloads</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../amin1/api.html">Amin1 API</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Anopheles dirus</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../adir1/adir1.0.html">Adir1.0 (Vector Observatory - Asia Project <em>Anopheles dirus</em> Phase 1 Data Release)</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../adir1/cloud.html">Adir1 cloud data access</a></li>
<li class="toctree-l2"><a class="reference internal" href="../adir1/download.html">Adir1.0 data downloads</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../adir1/api.html">Adir1 API</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Partners and community</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../studies-ag1000g.html">Ag1000G partner studies</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Release notes</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../release-notes.html">Release notes</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/malariagen/vector-data" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/malariagen/vector-data/issues/new?title=Issue%20on%20page%20%2Fag3/ag3.0.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/ag3/ag3.0.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Ag3.0 (Ag1000G phase 3)</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#terms-of-use">Terms of use</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#partner-studies">Partner studies</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#population-sampling">Population sampling</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#whole-genome-sequencing-and-variant-calling">Whole-genome sequencing and variant calling</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-hosting">Data hosting</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#sample-sets">Sample sets</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#further-reading">Further reading</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="ag3-0-ag1000g-phase-3">
<h1>Ag3.0 (Ag1000G phase 3)<a class="headerlink" href="#ag3-0-ag1000g-phase-3" title="Link to this heading">#</a></h1>
<p>The <a class="reference external" href="https://www.malariagen.net/project/ag1000g/"><em>Anopheles gambiae</em> 1000 Genomes Project (Ag1000G)</a> is a collaborative project using whole-genome sequencing to study genetic variation and evolution in natural populations of mosquitoes in the <em>Anopheles gambiae</em> species complex.</p>
<p>This page provides an introduction to open data resources released as part of the third phase of the Ag1000G project, known as <code class="docutils literal notranslate"><span class="pre">Ag3.0</span></code> for short. We hope the data from <code class="docutils literal notranslate"><span class="pre">Ag3.0</span></code> will be a valuable resource for research and surveillance of malaria vectors. If you have any questions about this guide or how to use the data, please <a class="reference external" href="https://github.com/malariagen/vector-public-data/discussions/new">start a new discussion</a> on the malariagen/vector-open-data repo on GitHub. If you find any bugs, please <a class="reference external" href="https://github.com/malariagen/vector-public-data/issues/new/choose">raise an issue</a>.</p>
<section id="terms-of-use">
<h2>Terms of use<a class="headerlink" href="#terms-of-use" title="Link to this heading">#</a></h2>
<p>Data from <code class="docutils literal notranslate"><span class="pre">Ag3.0</span></code> are released openly and can be downloaded and analysed for any purpose. The data have been released prior to publication by the Ag1000G Consortium, and are currently subject to a publication embargo described further in the <a class="reference external" href="https://www.malariagen.net/data/our-approach-sharing-data/ag1000g-terms-of-use/">Ag1000G terms of use</a>. If you have any questions about the terms of use, please email support&#64;malariagen.net.</p>
</section>
<section id="partner-studies">
<h2>Partner studies<a class="headerlink" href="#partner-studies" title="Link to this heading">#</a></h2>
<p>The Ag1000G project is coordinated by a consortium of partners from a range of different research institutions and countries. This includes consortium members who are carrying out independent research studies in malaria-endemic regions, and who have contributed mosquito specimens or mosquito DNA samples collected in the course of their own research. In total, 26 studies contributed samples to <code class="docutils literal notranslate"><span class="pre">Ag3.0</span></code>, including wild-caught specimens from 19 countries. For further information about these contributing studies, the researchers involved, and the collection sites and methods, see the <a class="reference internal" href="../studies-ag1000g.html"><span class="doc">Ag1000G partner studies</span></a> page.</p>
</section>
<section id="population-sampling">
<h2>Population sampling<a class="headerlink" href="#population-sampling" title="Link to this heading">#</a></h2>
<p><code class="docutils literal notranslate"><span class="pre">Ag3.0</span></code> includes data from 3,081 individual mosquitoes, including 2,784 mosquitoes collected from natural populations in 19 countries. Three species are represented within the cohort: <em>Anopheles gambiae</em>, <em>Anopheles coluzzii</em> and <em>Anopheles arabiensis</em>. The map below provides an overview of the collection locations and the numbers of samples broken down by species.</p>
<a class="bg-primary reference internal image-reference" href="../_images/ag3-map.png"><img alt="Ag3 map of sampling sites" class="bg-primary align-center" src="../_images/ag3-map.png" style="width: 700px;" /></a>
<p>In addition to these wild-caught samples, a further 297 samples are included from 15 lab crosses.</p>
</section>
<section id="whole-genome-sequencing-and-variant-calling">
<h2>Whole-genome sequencing and variant calling<a class="headerlink" href="#whole-genome-sequencing-and-variant-calling" title="Link to this heading">#</a></h2>
<p>All samples in <code class="docutils literal notranslate"><span class="pre">Ag3.0</span></code> have been sequenced individually to high coverage using Illumina technology at the Wellcome Sanger Institute. These sequence data have then been analysed to identify genetic variants such as single nucleotide polymorphisms (SNPs). After variant calling, both the samples and the variants have been through a range of quality control analyses, to ensure the data are of high quality. Both the raw sequence data and the curated variant calls are openly available for download and analysis.</p>
<p>For further information about the sequencing and variant calling methods used, please see the <a class="reference internal" href="methods.html"><span class="doc std std-doc">methods page</span></a>.</p>
</section>
<section id="data-hosting">
<h2>Data hosting<a class="headerlink" href="#data-hosting" title="Link to this heading">#</a></h2>
<p>Data from <code class="docutils literal notranslate"><span class="pre">Ag3.0</span></code> are hosted by several different services.</p>
<p>Raw sequence reads, sequence read alignments and SNP calls are available for download from the European Nucleotide Archive (ENA). Further information on how to find and download these data is provided in the <a class="reference internal" href="download.html"><span class="doc std std-doc">data download guide</span></a>.</p>
<p>The SNP data have also been uploaded to Google Cloud, and can be analysed directly within the cloud without having to download or copy any data, including via free interactive computing services such as <a class="reference external" href="https://colab.research.google.com/">Google Colab</a>. Further information about analysing these data in the cloud is provided in the <a class="reference internal" href="cloud.html"><span class="doc std std-doc">cloud data access guide</span></a>.</p>
</section>
<section id="sample-sets">
<h2>Sample sets<a class="headerlink" href="#sample-sets" title="Link to this heading">#</a></h2>
<p>The samples included in <code class="docutils literal notranslate"><span class="pre">Ag3.0</span></code> have been organised into 28 sample sets. Each of these sample sets corresponds to a set of mosquito specimens from a contributing study. Depending on your objectives, you may want to access data from only specific sample sets, or all sample sets. Here is a list of the sample sets included:</p>
<div class="cell tag_remove-input docutils container">
</div>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<script type="application/javascript">(function(root) {
  function now() {
    return new Date();
  }

  const force = true;

  if (typeof root._bokeh_onload_callbacks === "undefined" || force === true) {
    root._bokeh_onload_callbacks = [];
    root._bokeh_is_loading = undefined;
  }

const JS_MIME_TYPE = 'application/javascript';
  const HTML_MIME_TYPE = 'text/html';
  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';
  const CLASS_NAME = 'output_bokeh rendered_html';

  /**
   * Render data to the DOM node
   */
  function render(props, node) {
    const script = document.createElement("script");
    node.appendChild(script);
  }

  /**
   * Handle when an output is cleared or removed
   */
  function handleClearOutput(event, handle) {
    function drop(id) {
      const view = Bokeh.index.get_by_id(id)
      if (view != null) {
        view.model.document.clear()
        Bokeh.index.delete(view)
      }
    }

    const cell = handle.cell;

    const id = cell.output_area._bokeh_element_id;
    const server_id = cell.output_area._bokeh_server_id;

    // Clean up Bokeh references
    if (id != null) {
      drop(id)
    }

    if (server_id !== undefined) {
      // Clean up Bokeh references
      const cmd_clean = "from bokeh.io.state import curstate; print(curstate().uuid_to_server['" + server_id + "'].get_sessions()[0].document.roots[0]._id)";
      cell.notebook.kernel.execute(cmd_clean, {
        iopub: {
          output: function(msg) {
            const id = msg.content.text.trim()
            drop(id)
          }
        }
      });
      // Destroy server and session
      const cmd_destroy = "import bokeh.io.notebook as ion; ion.destroy_server('" + server_id + "')";
      cell.notebook.kernel.execute(cmd_destroy);
    }
  }

  /**
   * Handle when a new output is added
   */
  function handleAddOutput(event, handle) {
    const output_area = handle.output_area;
    const output = handle.output;

    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only
    if ((output.output_type != "display_data") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {
      return
    }

    const toinsert = output_area.element.find("." + CLASS_NAME.split(' ')[0]);

    if (output.metadata[EXEC_MIME_TYPE]["id"] !== undefined) {
      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];
      // store reference to embed id on output_area
      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE]["id"];
    }
    if (output.metadata[EXEC_MIME_TYPE]["server_id"] !== undefined) {
      const bk_div = document.createElement("div");
      bk_div.innerHTML = output.data[HTML_MIME_TYPE];
      const script_attrs = bk_div.children[0].attributes;
      for (let i = 0; i < script_attrs.length; i++) {
        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);
        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent
      }
      // store reference to server id on output_area
      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE]["server_id"];
    }
  }

  function register_renderer(events, OutputArea) {

    function append_mime(data, metadata, element) {
      // create a DOM node to render to
      const toinsert = this.create_output_subarea(
        metadata,
        CLASS_NAME,
        EXEC_MIME_TYPE
      );
      this.keyboard_manager.register_events(toinsert);
      // Render to node
      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};
      render(props, toinsert[toinsert.length - 1]);
      element.append(toinsert);
      return toinsert
    }

    /* Handle when an output is cleared or removed */
    events.on('clear_output.CodeCell', handleClearOutput);
    events.on('delete.Cell', handleClearOutput);

    /* Handle when a new output is added */
    events.on('output_added.OutputArea', handleAddOutput);

    /**
     * Register the mime type and append_mime function with output_area
     */
    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {
      /* Is output safe? */
      safe: true,
      /* Index of renderer in `output_area.display_order` */
      index: 0
    });
  }

  // register the mime type if in Jupyter Notebook environment and previously unregistered
  if (root.Jupyter !== undefined) {
    const events = require('base/js/events');
    const OutputArea = require('notebook/js/outputarea').OutputArea;

    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {
      register_renderer(events, OutputArea);
    }
  }
  if (typeof (root._bokeh_timeout) === "undefined" || force === true) {
    root._bokeh_timeout = Date.now() + 5000;
    root._bokeh_failed_load = false;
  }

  const NB_LOAD_WARNING = {'data': {'text/html':
     "<div style='background-color: #fdd'>\n"+
     "<p>\n"+
     "BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \n"+
     "may be due to a slow or bad network connection. Possible fixes:\n"+
     "</p>\n"+
     "<ul>\n"+
     "<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\n"+
     "<li>use INLINE resources instead, as so:</li>\n"+
     "</ul>\n"+
     "<code>\n"+
     "from bokeh.resources import INLINE\n"+
     "output_notebook(resources=INLINE)\n"+
     "</code>\n"+
     "</div>"}};

  function display_loaded() {
    const el = document.getElementById(null);
    if (el != null) {
      el.textContent = "BokehJS is loading...";
    }
    if (root.Bokeh !== undefined) {
      if (el != null) {
        el.textContent = "BokehJS " + root.Bokeh.version + " successfully loaded.";
      }
    } else if (Date.now() < root._bokeh_timeout) {
      setTimeout(display_loaded, 100)
    }
  }

  function run_callbacks() {
    try {
      root._bokeh_onload_callbacks.forEach(function(callback) {
        if (callback != null)
          callback();
      });
    } finally {
      delete root._bokeh_onload_callbacks
    }
    console.debug("Bokeh: all callbacks have finished");
  }

  function load_libs(css_urls, js_urls, callback) {
    if (css_urls == null) css_urls = [];
    if (js_urls == null) js_urls = [];

    root._bokeh_onload_callbacks.push(callback);
    if (root._bokeh_is_loading > 0) {
      console.debug("Bokeh: BokehJS is being loaded, scheduling callback at", now());
      return null;
    }
    if (js_urls == null || js_urls.length === 0) {
      run_callbacks();
      return null;
    }
    console.debug("Bokeh: BokehJS not loaded, scheduling load and callback at", now());
    root._bokeh_is_loading = css_urls.length + js_urls.length;

    function on_load() {
      root._bokeh_is_loading--;
      if (root._bokeh_is_loading === 0) {
        console.debug("Bokeh: all BokehJS libraries/stylesheets loaded");
        run_callbacks()
      }
    }

    function on_error(url) {
      console.error("failed to load " + url);
    }

    for (let i = 0; i < css_urls.length; i++) {
      const url = css_urls[i];
      const element = document.createElement("link");
      element.onload = on_load;
      element.onerror = on_error.bind(null, url);
      element.rel = "stylesheet";
      element.type = "text/css";
      element.href = url;
      console.debug("Bokeh: injecting link tag for BokehJS stylesheet: ", url);
      document.body.appendChild(element);
    }

    for (let i = 0; i < js_urls.length; i++) {
      const url = js_urls[i];
      const element = document.createElement('script');
      element.onload = on_load;
      element.onerror = on_error.bind(null, url);
      element.async = false;
      element.src = url;
      console.debug("Bokeh: injecting script tag for BokehJS library: ", url);
      document.head.appendChild(element);
    }
  };

  function inject_raw_css(css) {
    const element = document.createElement("style");
    element.appendChild(document.createTextNode(css));
    document.body.appendChild(element);
  }

  const js_urls = ["https://cdn.bokeh.org/bokeh/release/bokeh-3.3.0.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.3.0.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.3.0.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.3.0.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.3.0.min.js"];
  const css_urls = [];

  const inline_js = [    function(Bokeh) {
      Bokeh.set_log_level("info");
    },
function(Bokeh) {
    }
  ];

  function run_inline_js() {
    if (root.Bokeh !== undefined || force === true) {
          for (let i = 0; i < inline_js.length; i++) {
      inline_js[i].call(root, root.Bokeh);
    }
} else if (Date.now() < root._bokeh_timeout) {
      setTimeout(run_inline_js, 100);
    } else if (!root._bokeh_failed_load) {
      console.log("Bokeh: BokehJS failed to load within specified timeout.");
      root._bokeh_failed_load = true;
    } else if (force !== true) {
      const cell = $(document.getElementById(null)).parents('.cell').data().cell;
      cell.output_area.append_execute_result(NB_LOAD_WARNING)
    }
  }

  if (root._bokeh_is_loading === 0) {
    console.debug("Bokeh: BokehJS loaded, going straight to plotting");
    run_inline_js();
  } else {
    load_libs(css_urls, js_urls, function() {
      console.debug("Bokeh: BokehJS plotting callback run at", now());
      run_inline_js();
    });
  }
}(window));</script></div>
</div>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_count</th>
    </tr>
    <tr>
      <th>sample_set</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AG1000G-AO</th>
      <td>81</td>
    </tr>
    <tr>
      <th>AG1000G-BF-A</th>
      <td>181</td>
    </tr>
    <tr>
      <th>AG1000G-BF-B</th>
      <td>102</td>
    </tr>
    <tr>
      <th>AG1000G-BF-C</th>
      <td>13</td>
    </tr>
    <tr>
      <th>AG1000G-CD</th>
      <td>76</td>
    </tr>
    <tr>
      <th>AG1000G-CF</th>
      <td>73</td>
    </tr>
    <tr>
      <th>AG1000G-CI</th>
      <td>80</td>
    </tr>
    <tr>
      <th>AG1000G-CM-A</th>
      <td>303</td>
    </tr>
    <tr>
      <th>AG1000G-CM-B</th>
      <td>97</td>
    </tr>
    <tr>
      <th>AG1000G-CM-C</th>
      <td>44</td>
    </tr>
    <tr>
      <th>AG1000G-FR</th>
      <td>23</td>
    </tr>
    <tr>
      <th>AG1000G-GA-A</th>
      <td>69</td>
    </tr>
    <tr>
      <th>AG1000G-GH</th>
      <td>100</td>
    </tr>
    <tr>
      <th>AG1000G-GM-A</th>
      <td>74</td>
    </tr>
    <tr>
      <th>AG1000G-GM-B</th>
      <td>31</td>
    </tr>
    <tr>
      <th>AG1000G-GM-C</th>
      <td>174</td>
    </tr>
    <tr>
      <th>AG1000G-GN-A</th>
      <td>45</td>
    </tr>
    <tr>
      <th>AG1000G-GN-B</th>
      <td>185</td>
    </tr>
    <tr>
      <th>AG1000G-GQ</th>
      <td>10</td>
    </tr>
    <tr>
      <th>AG1000G-GW</th>
      <td>101</td>
    </tr>
    <tr>
      <th>AG1000G-KE</th>
      <td>86</td>
    </tr>
    <tr>
      <th>AG1000G-ML-A</th>
      <td>60</td>
    </tr>
    <tr>
      <th>AG1000G-ML-B</th>
      <td>71</td>
    </tr>
    <tr>
      <th>AG1000G-MW</th>
      <td>41</td>
    </tr>
    <tr>
      <th>AG1000G-MZ</th>
      <td>74</td>
    </tr>
    <tr>
      <th>AG1000G-TZ</th>
      <td>300</td>
    </tr>
    <tr>
      <th>AG1000G-UG</th>
      <td>290</td>
    </tr>
    <tr>
      <th>AG1000G-X</th>
      <td>669</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>The sample set identifiers all start with “AG1000G-” followed by the two-letter code of the country from which samples were collected (e.g., “AO” is Angola). Where there are multiple sample sets from the same country, these have been given alphabetical suffixes, e.g., “AG1000G-BF-A”, “AG1000G-BF-B” and “AG1000G-BF-C” are three sample sets from Burkina Faso.</p>
<p>These country code suffixes are just a convenience to help remember which sample sets contain which data, please see the sample metadata for more precise location information. Note also that sample set AG1000G-GN-B contains samples from both Guinea and Mali.</p>
<p>Here is a more detailed breakdown of the samples contained within each sample set, summarised by country, year of collection, and species:</p>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th></th>
      <th></th>
      <th>taxon</th>
      <th>arabiensis</th>
      <th>coluzzii</th>
      <th>gambiae</th>
      <th>gcx1</th>
      <th>gcx2</th>
      <th>gcx3</th>
      <th>unassigned</th>
    </tr>
    <tr>
      <th>study_id</th>
      <th>sample_set</th>
      <th>country</th>
      <th>year</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AG1000G-AO</th>
      <th>AG1000G-AO</th>
      <th>Angola</th>
      <th>2009</th>
      <td>0</td>
      <td>81</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th rowspan="2" valign="top">AG1000G-BF-1</th>
      <th>AG1000G-BF-A</th>
      <th>Burkina Faso</th>
      <th>2012</th>
      <td>0</td>
      <td>82</td>
      <td>99</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-BF-B</th>
      <th>Burkina Faso</th>
      <th>2014</th>
      <td>3</td>
      <td>53</td>
      <td>46</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-BF-2</th>
      <th>AG1000G-BF-C</th>
      <th>Burkina Faso</th>
      <th>2004</th>
      <td>0</td>
      <td>0</td>
      <td>13</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-CD</th>
      <th>AG1000G-CD</th>
      <th>Democratic Republic of the Congo</th>
      <th>2015</th>
      <td>0</td>
      <td>0</td>
      <td>76</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th rowspan="2" valign="top">AG1000G-CF</th>
      <th rowspan="2" valign="top">AG1000G-CF</th>
      <th rowspan="2" valign="top">Central African Republic</th>
      <th>1993</th>
      <td>0</td>
      <td>5</td>
      <td>2</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1994</th>
      <td>0</td>
      <td>13</td>
      <td>53</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-CI</th>
      <th>AG1000G-CI</th>
      <th>Cote d'Ivoire</th>
      <th>2012</th>
      <td>0</td>
      <td>80</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-CM-1</th>
      <th>AG1000G-CM-A</th>
      <th>Cameroon</th>
      <th>2009</th>
      <td>0</td>
      <td>0</td>
      <td>303</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-CM-2</th>
      <th>AG1000G-CM-B</th>
      <th>Cameroon</th>
      <th>2005</th>
      <td>0</td>
      <td>7</td>
      <td>90</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-CM-3</th>
      <th>AG1000G-CM-C</th>
      <th>Cameroon</th>
      <th>2013</th>
      <td>2</td>
      <td>19</td>
      <td>23</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-FR</th>
      <th>AG1000G-FR</th>
      <th>Mayotte</th>
      <th>2011</th>
      <td>0</td>
      <td>0</td>
      <td>23</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-GA-1</th>
      <th>AG1000G-GA-A</th>
      <th>Gabon</th>
      <th>2000</th>
      <td>0</td>
      <td>0</td>
      <td>69</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-GH</th>
      <th>AG1000G-GH</th>
      <th>Ghana</th>
      <th>2012</th>
      <td>0</td>
      <td>64</td>
      <td>36</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-GM-1</th>
      <th>AG1000G-GM-A</th>
      <th>Gambia, The</th>
      <th>2011</th>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>68</td>
      <td>6</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-GM-2</th>
      <th>AG1000G-GM-B</th>
      <th>Gambia, The</th>
      <th>2006</th>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>9</td>
      <td>22</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-GM-3</th>
      <th>AG1000G-GM-C</th>
      <th>Gambia, The</th>
      <th>2012</th>
      <td>0</td>
      <td>0</td>
      <td>2</td>
      <td>0</td>
      <td>172</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th rowspan="3" valign="top">AG1000G-GN-ML</th>
      <th>AG1000G-GN-A</th>
      <th>Guinea</th>
      <th>2012</th>
      <td>0</td>
      <td>4</td>
      <td>41</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th rowspan="2" valign="top">AG1000G-GN-B</th>
      <th>Guinea</th>
      <th>2012</th>
      <td>0</td>
      <td>7</td>
      <td>84</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>Mali</th>
      <th>2012</th>
      <td>0</td>
      <td>27</td>
      <td>65</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>2</td>
    </tr>
    <tr>
      <th>AG1000G-GQ</th>
      <th>AG1000G-GQ</th>
      <th>Equatorial Guinea</th>
      <th>2002</th>
      <td>0</td>
      <td>0</td>
      <td>10</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-GW</th>
      <th>AG1000G-GW</th>
      <th>Guinea-Bissau</th>
      <th>2010</th>
      <td>0</td>
      <td>0</td>
      <td>8</td>
      <td>93</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th rowspan="3" valign="top">AG1000G-KE</th>
      <th rowspan="3" valign="top">AG1000G-KE</th>
      <th rowspan="3" valign="top">Kenya</th>
      <th>2000</th>
      <td>0</td>
      <td>0</td>
      <td>19</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2007</th>
      <td>3</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2012</th>
      <td>10</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>54</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-ML-1</th>
      <th>AG1000G-ML-A</th>
      <th>Mali</th>
      <th>2014</th>
      <td>0</td>
      <td>27</td>
      <td>33</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-ML-2</th>
      <th>AG1000G-ML-B</th>
      <th>Mali</th>
      <th>2004</th>
      <td>2</td>
      <td>36</td>
      <td>33</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-MW</th>
      <th>AG1000G-MW</th>
      <th>Malawi</th>
      <th>2015</th>
      <td>41</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th rowspan="2" valign="top">AG1000G-MZ</th>
      <th rowspan="2" valign="top">AG1000G-MZ</th>
      <th rowspan="2" valign="top">Mozambique</th>
      <th>2003</th>
      <td>0</td>
      <td>0</td>
      <td>3</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2004</th>
      <td>0</td>
      <td>0</td>
      <td>71</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th rowspan="3" valign="top">AG1000G-TZ</th>
      <th rowspan="3" valign="top">AG1000G-TZ</th>
      <th rowspan="3" valign="top">Tanzania</th>
      <th>2012</th>
      <td>87</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2013</th>
      <td>1</td>
      <td>0</td>
      <td>32</td>
      <td>0</td>
      <td>0</td>
      <td>10</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2015</th>
      <td>137</td>
      <td>0</td>
      <td>32</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-UG</th>
      <th>AG1000G-UG</th>
      <th>Uganda</th>
      <th>2012</th>
      <td>82</td>
      <td>0</td>
      <td>207</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>AG1000G-X</th>
      <th>AG1000G-X</th>
      <th>Lab Cross</th>
      <th>-1</th>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>297</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Note that there are also multiple sampling sites represented within some sample sets.</p>
</section>
<section id="further-reading">
<h2>Further reading<a class="headerlink" href="#further-reading" title="Link to this heading">#</a></h2>
<p>Hopefully this page has provided a useful introduction to the <code class="docutils literal notranslate"><span class="pre">Ag3.0</span></code> data resource. If you would like to start working with these data, please visit the <a class="reference internal" href="cloud.html"><span class="doc std std-doc">cloud data access guide</span></a> or the <a class="reference internal" href="download.html"><span class="doc std std-doc">data download guide</span></a> or continue browsing the other documentation on this site.</p>
<p>If you have any questions about the data and how to use them, please do get in touch by <a class="reference external" href="https://github.com/malariagen/vector-data/discussions/new">starting a new discussion</a> on the malariagen/vector-data repo on GitHub.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "conda-env-global-global-mgenv-6.0.6-py"
        },
        kernelOptions: {
            name: "conda-env-global-global-mgenv-6.0.6-py",
            path: "./ag3"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'conda-env-global-global-mgenv-6.0.6-py'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="../vobs/vobs-data-access.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Vector Observatory Data Access</p>
      </div>
    </a>
    <a class="right-next"
       href="ag3.1.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Ag3.1</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#terms-of-use">Terms of use</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#partner-studies">Partner studies</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#population-sampling">Population sampling</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#whole-genome-sequencing-and-variant-calling">Whole-genome sequencing and variant calling</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-hosting">Data hosting</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#sample-sets">Sample sets</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#further-reading">Further reading</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By MalariaGEN
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2021.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>