
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Ag3 API &#8212; MalariaGEN vector data user guide</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "malariagen/vector-data");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Amin1.0" href="../amin1/intro.html" />
    <link rel="prev" title="Exploring copy number variation (CNV)" href="examples/cnv-explore.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/logo.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">MalariaGEN vector data user guide</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../landing-page.html">
                    MalariaGEN vector data user guide
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Anopheles gambiae complex
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="intro.html">
   Ag3.0 (Ag1000G phase 3)
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="cloud.html">
     Ag3.0 cloud data access
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="download.html">
     Ag3.0 data downloads
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="methods.html">
     Ag3.0 sequencing and variant calling methods
    </a>
   </li>
   <li class="toctree-l2 has-children">
    <a class="reference internal" href="examples.html">
     Ag3.0 analysis examples
    </a>
    <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
    <label for="toctree-checkbox-2">
     <i class="fas fa-chevron-down">
     </i>
    </label>
    <ul>
     <li class="toctree-l3">
      <a class="reference internal" href="examples/collection-sites.html">
       Mosquito collection sites
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="examples/snps.html">
       Single nucleotide polymorphisms (SNPs)
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="examples/snp-effects.html">
       Genes and SNP effects
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="examples/pca.html">
       Investigating population structure with PCA
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="examples/cnv-explore.html">
       Exploring copy number variation (CNV)
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Ag3 API
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Anopheles minimus
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../amin1/intro.html">
   Amin1.0
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../amin1/cloud.html">
     Amin1.0 cloud data access
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../amin1/download.html">
     Amin1.0 data downloads
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../amin1/api.html">
   Amin1 API
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Partners and community
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../studies-ag1000g.html">
   Ag1000G partner studies
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Release notes
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../release-notes.html">
   Release notes
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-repository-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Source repositories">
      <i class="fab fa-github"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://github.com/malariagen/vector-data"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Source repository"
>
  

<span class="headerbtn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="headerbtn__text-container">repository</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/malariagen/vector-data/issues/new?title=Issue%20on%20page%20%2Fag3/api.html&body=Your%20issue%20content%20here."
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Open an issue"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="headerbtn__text-container">open issue</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="../_sources/ag3/api.md"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.md</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#ag3">
   Ag3()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-sets">
   sample_sets()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-metadata">
   sample_metadata()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-cohorts">
   sample_cohorts()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#count-samples">
   count_samples()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-samples-interactive-map">
   plot_samples_interactive_map()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cross-metadata">
   cross_metadata()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#species-calls">
   species_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#genome-sequence">
   genome_sequence()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#geneset">
   geneset()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-calls">
   snp_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-sites">
   snp_sites()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-genotypes">
   snp_genotypes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-filters">
   site_filters()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#is-accessible">
   is_accessible()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-effects">
   snp_effects()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-annotations">
   site_annotations()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-hmm">
   cnv_hmm()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-coverage-calls">
   cnv_coverage_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-discordant-read-calls">
   cnv_discordant_read_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv">
   gene_cnv()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#haplotypes">
   haplotypes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-allele-frequencies">
   snp_allele_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aa-allele-frequencies">
   aa_allele_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv-frequencies">
   gene_cnv_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-heatmap">
   plot_frequencies_heatmap()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-time-series">
   plot_frequencies_time_series()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-interactive-map">
   plot_frequencies_interactive_map()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-map-markers">
   plot_frequencies_map_markers()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-allele-frequencies-advanced">
   snp_allele_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aa-allele-frequencies-advanced">
   aa_allele_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv-frequencies-advanced">
   gene_cnv_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-genes">
   plot_genes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-transcript">
   plot_transcript()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-cnv-hmm-coverage">
   plot_cnv_hmm_coverage()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-cnv-hmm-heatmap">
   plot_cnv_hmm_heatmap()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#resolve-region">
   resolve_region()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#igv">
   igv()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#view-alignments">
   view_alignments()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#wgs-data-catalog">
   wgs_data_catalog()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-allele-counts">
   snp_allele_counts()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#pca">
   pca()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-pca-variance">
   plot_pca_variance()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-pca-coords">
   plot_pca_coords()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-pca-coords-3d">
   plot_pca_coords_3d()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-pca-snps">
   plot_pca_snps()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aim-variants">
   aim_variants()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aim-calls">
   aim_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-aim-heatmap">
   plot_aim_heatmap()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cohort-diversity-stats">
   cohort_diversity_stats()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#diversity-stats">
   diversity_stats()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-diversity-stats">
   plot_diversity_stats()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-heterozygosity">
   plot_heterozygosity()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#roh-hmm">
   roh_hmm()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-roh">
   plot_roh()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#h12-calibration">
   h12_calibration()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-h12-calibration">
   plot_h12_calibration()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#h12-gwss">
   h12_gwss()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-h12-gwss">
   plot_h12_gwss()
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Ag3 API</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#ag3">
   Ag3()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-sets">
   sample_sets()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-metadata">
   sample_metadata()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-cohorts">
   sample_cohorts()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#count-samples">
   count_samples()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-samples-interactive-map">
   plot_samples_interactive_map()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cross-metadata">
   cross_metadata()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#species-calls">
   species_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#genome-sequence">
   genome_sequence()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#geneset">
   geneset()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-calls">
   snp_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-sites">
   snp_sites()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-genotypes">
   snp_genotypes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-filters">
   site_filters()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#is-accessible">
   is_accessible()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-effects">
   snp_effects()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-annotations">
   site_annotations()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-hmm">
   cnv_hmm()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-coverage-calls">
   cnv_coverage_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-discordant-read-calls">
   cnv_discordant_read_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv">
   gene_cnv()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#haplotypes">
   haplotypes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-allele-frequencies">
   snp_allele_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aa-allele-frequencies">
   aa_allele_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv-frequencies">
   gene_cnv_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-heatmap">
   plot_frequencies_heatmap()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-time-series">
   plot_frequencies_time_series()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-interactive-map">
   plot_frequencies_interactive_map()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-map-markers">
   plot_frequencies_map_markers()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-allele-frequencies-advanced">
   snp_allele_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aa-allele-frequencies-advanced">
   aa_allele_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv-frequencies-advanced">
   gene_cnv_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-genes">
   plot_genes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-transcript">
   plot_transcript()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-cnv-hmm-coverage">
   plot_cnv_hmm_coverage()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-cnv-hmm-heatmap">
   plot_cnv_hmm_heatmap()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#resolve-region">
   resolve_region()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#igv">
   igv()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#view-alignments">
   view_alignments()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#wgs-data-catalog">
   wgs_data_catalog()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-allele-counts">
   snp_allele_counts()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#pca">
   pca()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-pca-variance">
   plot_pca_variance()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-pca-coords">
   plot_pca_coords()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-pca-coords-3d">
   plot_pca_coords_3d()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-pca-snps">
   plot_pca_snps()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aim-variants">
   aim_variants()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aim-calls">
   aim_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-aim-heatmap">
   plot_aim_heatmap()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cohort-diversity-stats">
   cohort_diversity_stats()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#diversity-stats">
   diversity_stats()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-diversity-stats">
   plot_diversity_stats()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-heterozygosity">
   plot_heterozygosity()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#roh-hmm">
   roh_hmm()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-roh">
   plot_roh()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#h12-calibration">
   h12_calibration()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-h12-calibration">
   plot_h12_calibration()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#h12-gwss">
   h12_gwss()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-h12-gwss">
   plot_h12_gwss()
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="ag3-api">
<h1>Ag3 API<a class="headerlink" href="#ag3-api" title="Permalink to this headline">#</a></h1>
<p>This page provides documentation for functions in the <a class="reference external" href="https://github.com/malariagen/malariagen-data-python">malariagen_data</a> Python package for accessing <em>Anopheles gambiae</em> data.</p>
<section id="ag3">
<h2>Ag3()<a class="headerlink" href="#ag3" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3">
<span class="sig-prename descclassname"><span class="pre">malariagen_data.</span></span><span class="sig-name descname"><span class="pre">Ag3</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bokeh_output_notebook=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">results_cache=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log=&lt;_io.TextIOWrapper</span> <span class="pre">name='&lt;stdout&gt;'</span> <span class="pre">mode='w'</span> <span class="pre">encoding='utf-8'&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">debug=False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_progress=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">check_location=True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts_analysis=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species_analysis=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis=None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">**kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3" title="Permalink to this definition">#</a></dt>
<dd><p>Provides access to data from Ag3.x releases.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>url</strong><span class="classifier">str</span></dt><dd><p>Base path to data. Give “gs://vo_agam_release/” to use Google Cloud
Storage, or a local path on your file system if data have been
downloaded.</p>
</dd>
<dt><strong>cohorts_analysis</strong><span class="classifier">str</span></dt><dd><p>Cohort analysis version.</p>
</dd>
<dt><strong>species_analysis</strong><span class="classifier">{“aim_20200422”, “pca_20200422”}, optional</span></dt><dd><p>Species analysis version.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters analysis version.</p>
</dd>
<dt><strong>bokeh_output_notebook</strong><span class="classifier">bool, optional</span></dt><dd><p>If True (default), configure bokeh to output plots to the notebook.</p>
</dd>
<dt><strong>results_cache</strong><span class="classifier">str, optional</span></dt><dd><p>Path to directory on local file system to save results.</p>
</dd>
<dt><strong>log</strong><span class="classifier">str or stream, optional</span></dt><dd><p>File path or stream output for logging messages.</p>
</dd>
<dt><strong>debug</strong><span class="classifier">bool, optional</span></dt><dd><p>Set to True to enable debug level logging.</p>
</dd>
<dt><strong>show_progress</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, show a progress bar during longer-running computations.</p>
</dd>
<dt><strong>check_location</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, use ipinfo to check the location of the client system.</p>
</dd>
<dt><strong>**kwargs</strong></dt><dd><p>Passed through to fsspec when setting up file system access.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Examples</p>
<p>Access data from Google Cloud Storage (default):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">malariagen_data</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">ag3</span> <span class="o">=</span> <span class="n">malariagen_data</span><span class="o">.</span><span class="n">Ag3</span><span class="p">()</span>
</pre></div>
</div>
<p>Access data downloaded to a local file system:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">ag3</span> <span class="o">=</span> <span class="n">malariagen_data</span><span class="o">.</span><span class="n">Ag3</span><span class="p">(</span><span class="s2">&quot;/local/path/to/vo_agam_release/&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>Access data from Google Cloud Storage, with caching on the local file system
in a directory named “gcs_cache”:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">ag3</span> <span class="o">=</span> <span class="n">malariagen_data</span><span class="o">.</span><span class="n">Ag3</span><span class="p">(</span>
<span class="gp">... </span>    <span class="s2">&quot;simplecache::gs://vo_agam_release&quot;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">simplecache</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">cache_storage</span><span class="o">=</span><span class="s2">&quot;gcs_cache&quot;</span><span class="p">),</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>Set up caching of some longer-running computations on the local file system,
in a directory named “results_cache”:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">ag3</span> <span class="o">=</span> <span class="n">malariagen_data</span><span class="o">.</span><span class="n">Ag3</span><span class="p">(</span><span class="n">results_cache</span><span class="o">=</span><span class="s2">&quot;results_cache&quot;</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

</section>
<section id="sample-sets">
<h2>sample_sets()<a class="headerlink" href="#sample-sets" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.sample_sets">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">sample_sets</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">release</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.sample_sets" title="Permalink to this definition">#</a></dt>
<dd><p>Access a dataframe of sample sets.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>release</strong><span class="classifier">str, optional</span></dt><dd><p>Release identifier. Give “3.0” to access the Ag1000G phase 3 data
release.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of sample sets, one row per sample set.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="sample-metadata">
<h2>sample_metadata()<a class="headerlink" href="#sample-metadata" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.sample_metadata">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">sample_metadata</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.sample_metadata" title="Permalink to this definition">#</a></dt>
<dd><p>Access sample metadata for one or more sample sets.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df_samples</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of sample metadata, one row per sample.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="sample-cohorts">
<h2>sample_cohorts()<a class="headerlink" href="#sample-cohorts" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.sample_cohorts">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">sample_cohorts</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.sample_cohorts" title="Permalink to this definition">#</a></dt>
<dd><p>Access cohorts metadata for one or more sample sets.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of cohort metadata, one row per sample.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="count-samples">
<h2>count_samples()<a class="headerlink" href="#count-samples" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.count_samples">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">count_samples</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">index</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">('country',</span> <span class="pre">'admin1_iso',</span> <span class="pre">'admin1_name',</span> <span class="pre">'admin2_name',</span> <span class="pre">'year')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">columns</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'taxon'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.count_samples" title="Permalink to this definition">#</a></dt>
<dd><p>Create a pivot table showing numbers of samples available by space,
time and taxon.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>index</strong><span class="classifier">str or tuple of str</span></dt><dd><p>Sample metadata columns to use for the index.</p>
</dd>
<dt><strong>columns</strong><span class="classifier">str or tuple of str</span></dt><dd><p>Sample metadata columns to use for the columns.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>Pivot table of sample counts.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-samples-interactive-map">
<h2>plot_samples_interactive_map()<a class="headerlink" href="#plot-samples-interactive-map" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_samples_interactive_map">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_samples_interactive_map</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">basemap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">center</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(-</span> <span class="pre">2,</span> <span class="pre">20)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">zoom</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_samples</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_samples_interactive_map" title="Permalink to this definition">#</a></dt>
<dd><p>Plot an interactive map showing sampling locations using ipyleaflet.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>basemap</strong><span class="classifier">dict</span></dt><dd><p>Basemap description coming from ipyleaflet.basemaps.</p>
</dd>
<dt><strong>center</strong><span class="classifier">tuple of int, optional</span></dt><dd><p>Location to center the map.</p>
</dd>
<dt><strong>zoom</strong><span class="classifier">int, optional</span></dt><dd><p>Initial zoom level.</p>
</dd>
<dt><strong>min_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Minimum number of samples required to show a marker for a given
location.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>samples_map</strong><span class="classifier">ipyleaflet.Map</span></dt><dd><p>Ipyleaflet map widget.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="cross-metadata">
<h2>cross_metadata()<a class="headerlink" href="#cross-metadata" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.cross_metadata">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">cross_metadata</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.cross_metadata" title="Permalink to this definition">#</a></dt>
<dd><p>Load a dataframe containing metadata about samples in colony crosses,
including which samples are parents or progeny in which crosses.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of sample metadata for colony crosses.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="species-calls">
<h2>species_calls()<a class="headerlink" href="#species-calls" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.species_calls">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">species_calls</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.species_calls" title="Permalink to this definition">#</a></dt>
<dd><p>Access species calls for one or more sample sets.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”] or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of species calls for one or more sample sets, one row
per sample.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="genome-sequence">
<h2>genome_sequence()<a class="headerlink" href="#genome-sequence" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.genome_sequence">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">genome_sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.genome_sequence" title="Permalink to this definition">#</a></dt>
<dd><p>Access the reference genome sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>d</strong><span class="classifier">dask.array.Array</span></dt><dd><p>An array of nucleotides giving the reference genome sequence for the
given contig.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="geneset">
<h2>geneset()<a class="headerlink" href="#geneset" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.geneset">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">geneset</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">*</span></span><span class="n"><span class="pre">args</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.geneset" title="Permalink to this definition">#</a></dt>
<dd><p>Deprecated, this method has been renamed to genome_features().</p>
</dd></dl>

</section>
<section id="snp-calls">
<h2>snp_calls()<a class="headerlink" href="#snp-calls" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_calls">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_calls</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_class</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_calls" title="Permalink to this definition">#</a></dt>
<dd><p>Access SNP sites, site filters and genotype calls.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_class</strong><span class="classifier">str, optional</span></dt><dd><p>Select sites belonging to one of the following classes: CDS_DEG_4,
(4-fold degenerate coding sites), CDS_DEG_2_SIMPLE (2-fold simple
degenerate coding sites), CDS_DEG_0 (non-degenerate coding sites),
INTRON_SHORT (introns shorter than 100 bp), INTRON_LONG (introns
longer than 200 bp), INTRON_SPLICE_5PRIME (intron within 2 bp of
5’ splice site), INTRON_SPLICE_3PRIME (intron within 2 bp of 3’
splice site), UTR_5PRIME (5’ untranslated region), UTR_3PRIME (3’
untranslated region), INTERGENIC (intergenic, more than 10 kbp from
a gene).</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
<dt><strong>cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>If provided, randomly down-sample to the given cohort size.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Random seed used for down-sampling.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset containing SNP sites, site filters and genotype calls.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="snp-sites">
<h2>snp_sites()<a class="headerlink" href="#snp-sites" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_sites">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_sites</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">field</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_sites" title="Permalink to this definition">#</a></dt>
<dd><p>Access SNP site data (positions and alleles).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>field</strong><span class="classifier">{“POS”, “REF”, “ALT”}</span></dt><dd><p>Array to access.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>d</strong><span class="classifier">dask.array.Array</span></dt><dd><p>An array of either SNP positions, reference alleles or alternate
alleles.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="snp-genotypes">
<h2>snp_genotypes()<a class="headerlink" href="#snp-genotypes" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_genotypes">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_genotypes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">field</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'GT'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_genotypes" title="Permalink to this definition">#</a></dt>
<dd><p>Access SNP genotypes and associated data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>field</strong><span class="classifier">{“GT”, “GQ”, “AD”, “MQ”}</span></dt><dd><p>Array to access.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>d</strong><span class="classifier">dask.array.Array</span></dt><dd><p>An array of either genotypes (GT), genotype quality (GQ), allele
depths (AD) or mapping quality (MQ) values.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="site-filters">
<h2>site_filters()<a class="headerlink" href="#site-filters" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.site_filters">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">site_filters</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">field</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'filter_pass'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.site_filters" title="Permalink to this definition">#</a></dt>
<dd><p>Access SNP site filters.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Mask to use.</p>
</dd>
<dt><strong>field</strong><span class="classifier">str, optional</span></dt><dd><p>Array to access.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>d</strong><span class="classifier">dask.array.Array</span></dt><dd><p>An array of boolean values identifying sites that pass the filters.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="is-accessible">
<h2>is_accessible()<a class="headerlink" href="#is-accessible" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.is_accessible">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">is_accessible</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.is_accessible" title="Permalink to this definition">#</a></dt>
<dd><p>Compute genome accessibility array.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>a</strong><span class="classifier">numpy.ndarray</span></dt><dd><p>An array of boolean values identifying accessible genome sites.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="snp-effects">
<h2>snp_effects()<a class="headerlink" href="#snp-effects" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_effects">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_effects</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_effects" title="Permalink to this definition">#</a></dt>
<dd><p>Compute variant effects for a gene transcript.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Gene transcript ID (AgamP4.12), e.g., “AGAP004707-RA”.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}, optional</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of all possible SNP variants and their effects, one row
per variant.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="site-annotations">
<h2>site_annotations()<a class="headerlink" href="#site-annotations" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.site_annotations">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">site_annotations</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'auto'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.site_annotations" title="Permalink to this definition">#</a></dt>
<dd><p>Load site annotations.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of site annotations.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="cnv-hmm">
<h2>cnv_hmm()<a class="headerlink" href="#cnv-hmm" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.cnv_hmm">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">cnv_hmm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_coverage_variance</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.cnv_hmm" title="Permalink to this definition">#</a></dt>
<dd><p>Access CNV HMM data from CNV calling.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>max_coverage_variance</strong><span class="classifier">float, optional</span></dt><dd><p>Remove samples if coverage variance exceeds this value.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of CNV HMM calls and associated data.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="cnv-coverage-calls">
<h2>cnv_coverage_calls()<a class="headerlink" href="#cnv-coverage-calls" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.cnv_coverage_calls">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">cnv_coverage_calls</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_set</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">analysis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.cnv_coverage_calls" title="Permalink to this definition">#</a></dt>
<dd><p>Access CNV HMM data from genome-wide CNV discovery and filtering.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>sample_set</strong><span class="classifier">str</span></dt><dd><p>Sample set identifier.</p>
</dd>
<dt><strong>analysis</strong><span class="classifier">{‘gamb_colu’, ‘arab’, ‘crosses’}</span></dt><dd><p>Name of CNV analysis.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of CNV alleles and genotypes.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="cnv-discordant-read-calls">
<h2>cnv_discordant_read_calls()<a class="headerlink" href="#cnv-discordant-read-calls" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.cnv_discordant_read_calls">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">cnv_discordant_read_calls</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.cnv_discordant_read_calls" title="Permalink to this definition">#</a></dt>
<dd><p>Access CNV discordant read calls data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>contig</strong><span class="classifier">str or list of str</span></dt><dd><p>Chromosome arm, e.g., “3R”. Multiple values can be provided
as a list, in which case data will be concatenated, e.g., [“2R”,
“3R”].</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of CNV alleles and genotypes.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="gene-cnv">
<h2>gene_cnv()<a class="headerlink" href="#gene-cnv" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.gene_cnv">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">gene_cnv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_coverage_variance</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.gene_cnv" title="Permalink to this definition">#</a></dt>
<dd><p>Compute modal copy number by gene, from HMM data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or
a release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>max_coverage_variance</strong><span class="classifier">float, optional</span></dt><dd><p>Remove samples if coverage variance exceeds this value.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of modal copy number per gene and associated data.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="haplotypes">
<h2>haplotypes()<a class="headerlink" href="#haplotypes" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.haplotypes">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">haplotypes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">analysis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.haplotypes" title="Permalink to this definition">#</a></dt>
<dd><p>Access haplotype data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>analysis</strong><span class="classifier">{“arab”, “gamb_colu”, “gamb_colu_arab”}</span></dt><dd><p>Which phasing analysis to use. If analysing only An. arabiensis, the
“arab” analysis is best. If analysing only An. gambiae and An.
coluzzii, the “gamb_colu” analysis is best. Otherwise, use the
“gamb_colu_arab” analysis.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
<dt><strong>cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>If provided, randomly down-sample to the given cohort size.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Random seed used for down-sampling.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of haplotypes and associated data.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="snp-allele-frequencies">
<h2>snp_allele_frequencies()<a class="headerlink" href="#snp-allele-frequencies" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_allele_frequencies">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_allele_frequencies</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_invariant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">effects</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_allele_frequencies" title="Permalink to this definition">#</a></dt>
<dd><p>Compute per variant allele frequencies for a gene transcript.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Gene transcript ID (AgamP4.12), e.g., “AGAP004707-RD”.</p>
</dd>
<dt><strong>cohorts</strong><span class="classifier">str or dict</span></dt><dd><p>If a string, gives the name of a predefined cohort set, e.g., one of
{“admin1_month”, “admin1_year”, “admin2_month”, “admin2_year”}.
If a dict, should map cohort labels to sample queries, e.g.,
<cite>{“bf_2012_col”: “country == ‘Burkina Faso’ and year == 2012 and
taxon == ‘coluzzii’”}</cite>.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int</span></dt><dd><p>Minimum cohort size. Any cohorts below this size are omitted.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>drop_invariant</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, variants with no alternate allele calls in any cohorts are
dropped from the result.</p>
</dd>
<dt><strong>effects</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, add SNP effect columns.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of SNP frequencies, one row per variant.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>Cohorts with fewer samples than min_cohort_size will be excluded from
output.</p>
</dd></dl>

</section>
<section id="aa-allele-frequencies">
<h2>aa_allele_frequencies()<a class="headerlink" href="#aa-allele-frequencies" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.aa_allele_frequencies">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">aa_allele_frequencies</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_invariant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.aa_allele_frequencies" title="Permalink to this definition">#</a></dt>
<dd><p>Compute per amino acid allele frequencies for a gene transcript.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Gene transcript ID (AgamP4.12), e.g., “AGAP004707-RA”.</p>
</dd>
<dt><strong>cohorts</strong><span class="classifier">str or dict</span></dt><dd><p>If a string, gives the name of a predefined cohort set, e.g., one of
{“admin1_month”, “admin1_year”, “admin2_month”, “admin2_year”}.
If a dict, should map cohort labels to sample queries, e.g.,
<cite>{“bf_2012_col”: “country == ‘Burkina Faso’ and year == 2012 and
taxon == ‘coluzzii’”}</cite>.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int</span></dt><dd><p>Minimum cohort size, below which allele frequencies are not
calculated for cohorts.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>drop_invariant</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, variants with no alternate allele calls in any cohorts are
dropped from the result.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of amino acid allele frequencies, one row per
replacement.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>Cohorts with fewer samples than min_cohort_size will be excluded from
output.</p>
</dd></dl>

</section>
<section id="gene-cnv-frequencies">
<h2>gene_cnv_frequencies()<a class="headerlink" href="#gene-cnv-frequencies" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.gene_cnv_frequencies">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">gene_cnv_frequencies</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_invariant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_coverage_variance</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.gene_cnv_frequencies" title="Permalink to this definition">#</a></dt>
<dd><p>Compute modal copy number by gene, then compute the frequency of
amplifications and deletions in one or more cohorts, from HMM data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>cohorts</strong><span class="classifier">str or dict</span></dt><dd><p>If a string, gives the name of a predefined cohort set, e.g., one of
{“admin1_month”, “admin1_year”, “admin2_month”, “admin2_year”}.
If a dict, should map cohort labels to sample queries, e.g.,
<cite>{“bf_2012_col”: “country == ‘Burkina Faso’ and year == 2012 and
taxon == ‘coluzzii’”}</cite>.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int</span></dt><dd><p>Minimum cohort size, below which cohorts are dropped.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>drop_invariant</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, drop any rows where there is no evidence of variation.</p>
</dd>
<dt><strong>max_coverage_variance</strong><span class="classifier">float, optional</span></dt><dd><p>Remove samples if coverage variance exceeds this value.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of CNV amplification (amp) and deletion (del)
frequencies in the specified cohorts, one row per gene and CNV type
(amp/del).</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-frequencies-heatmap">
<h2>plot_frequencies_heatmap()<a class="headerlink" href="#plot-frequencies-heatmap" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_frequencies_heatmap">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_frequencies_heatmap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">index</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'label'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_len</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_label</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Cohorts'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_label</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Variants'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">col_width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">40</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">row_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">text_auto</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'.0%'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aspect</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'auto'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">color_continuous_scale</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Reds'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_frequencies_heatmap" title="Permalink to this definition">#</a></dt>
<dd><p>Plot a heatmap from a pandas DataFrame of frequencies, e.g., output
from <cite>Ag3.snp_allele_frequencies()</cite> or <cite>Ag3.gene_cnv_frequencies()</cite>.
It’s recommended to filter the input DataFrame to just rows of interest,
i.e., fewer rows than <cite>max_len</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas DataFrame</span></dt><dd><p>A DataFrame of frequencies, e.g., output from
<cite>snp_allele_frequencies()</cite> or <cite>gene_cnv_frequencies()</cite>.</p>
</dd>
<dt><strong>index</strong><span class="classifier">str or list of str</span></dt><dd><p>One or more column headers that are present in the input dataframe.
This becomes the heatmap y-axis row labels. The column/s must
produce a unique index.</p>
</dd>
<dt><strong>max_len</strong><span class="classifier">int, optional</span></dt><dd><p>Displaying large styled dataframes may cause ipython notebooks to
crash.</p>
</dd>
<dt><strong>x_label</strong><span class="classifier">str, optional</span></dt><dd><p>This is the x-axis label that will be displayed on the heatmap.</p>
</dd>
<dt><strong>y_label</strong><span class="classifier">str, optional</span></dt><dd><p>This is the y-axis label that will be displayed on the heatmap.</p>
</dd>
<dt><strong>colorbar</strong><span class="classifier">bool, optional</span></dt><dd><p>If False, colorbar is not output.</p>
</dd>
<dt><strong>col_width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width per column in pixels (px).</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px), overrides col_width.</p>
</dd>
<dt><strong>row_height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height per row in pixels (px).</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height in pixels (px), overrides row_height.</p>
</dd>
<dt><strong>text_auto</strong><span class="classifier">str, optional</span></dt><dd><p>Formatting for frequency values.</p>
</dd>
<dt><strong>aspect</strong><span class="classifier">str, optional</span></dt><dd><p>Control the aspect ratio of the heatmap.</p>
</dd>
<dt><strong>color_continuous_scale</strong><span class="classifier">str, optional</span></dt><dd><p>Color scale to use.</p>
</dd>
<dt><strong>title</strong><span class="classifier">bool or str, optional</span></dt><dd><p>If True, attempt to use metadata from input dataset as a plot
title. Otherwise, use supplied value as a title.</p>
</dd>
<dt><strong>**kwargs</strong></dt><dd><p>Other parameters are passed through to px.imshow().</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">plotly.graph_objects.Figure</span></dt><dd></dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-frequencies-time-series">
<h2>plot_frequencies_time_series()<a class="headerlink" href="#plot-frequencies-time-series" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_frequencies_time_series">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_frequencies_time_series</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ds</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_frequencies_time_series" title="Permalink to this definition">#</a></dt>
<dd><p>Create a time series plot of variant frequencies using plotly.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of variant frequencies, such as returned by
<cite>Ag3.snp_allele_frequencies_advanced()</cite>,
<cite>Ag3.aa_allele_frequencies_advanced()</cite> or
<cite>Ag3.gene_cnv_frequencies_advanced()</cite>.</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of plot in pixels (px).</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Width of plot in pixels (px).</p>
</dd>
<dt><strong>title</strong><span class="classifier">bool or str, optional</span></dt><dd><p>If True, attempt to use metadata from input dataset as a plot
title. Otherwise, use supplied value as a title.</p>
</dd>
<dt><strong>**kwargs</strong></dt><dd><p>Passed through to <cite>px.line()</cite>.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">plotly.graph_objects.Figure</span></dt><dd><p>A plotly figure containing line graphs. The resulting figure will
have one panel per cohort, grouped into columns by taxon, and
grouped into rows by area. Markers and lines show frequencies of
variants.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-frequencies-interactive-map">
<h2>plot_frequencies_interactive_map()<a class="headerlink" href="#plot-frequencies-interactive-map" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_frequencies_interactive_map">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_frequencies_interactive_map</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ds</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">center</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(-</span> <span class="pre">2,</span> <span class="pre">20)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">zoom</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">epilogue</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_frequencies_interactive_map" title="Permalink to this definition">#</a></dt>
<dd><p>Create an interactive map with markers showing variant frequencies or
cohorts grouped by area (space), period (time) and taxon.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of variant frequencies, such as returned by
<cite>Ag3.snp_allele_frequencies_advanced()</cite>,
<cite>Ag3.aa_allele_frequencies_advanced()</cite> or
<cite>Ag3.gene_cnv_frequencies_advanced()</cite>.</p>
</dd>
<dt><strong>center</strong><span class="classifier">tuple of int, optional</span></dt><dd><p>Location to center the map.</p>
</dd>
<dt><strong>zoom</strong><span class="classifier">int, optional</span></dt><dd><p>Initial zoom level.</p>
</dd>
<dt><strong>title</strong><span class="classifier">bool or str, optional</span></dt><dd><p>If True, attempt to use metadata from input dataset as a plot
title. Otherwise, use supplied value as a title.</p>
</dd>
<dt><strong>epilogue</strong><span class="classifier">bool or str, optional</span></dt><dd><p>Additional text to display below the map.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>out</strong><span class="classifier">ipywidgets.Widget</span></dt><dd><p>An interactive map with widgets for selecting which variant, taxon
and time period to display.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-frequencies-map-markers">
<h2>plot_frequencies_map_markers()<a class="headerlink" href="#plot-frequencies-map-markers" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_frequencies_map_markers">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_frequencies_map_markers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">m</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ds</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">taxon</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">period</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clear</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_frequencies_map_markers" title="Permalink to this definition">#</a></dt>
<dd><p>Plot markers on a map showing variant frequencies for cohorts grouped
by area (space), period (time) and taxon.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>m</strong><span class="classifier">ipyleaflet.Map</span></dt><dd><p>The map on which to add the markers.</p>
</dd>
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of variant frequencies, such as returned by
<cite>Ag3.snp_allele_frequencies_advanced()</cite>,
<cite>Ag3.aa_allele_frequencies_advanced()</cite> or
<cite>Ag3.gene_cnv_frequencies_advanced()</cite>.</p>
</dd>
<dt><strong>variant</strong><span class="classifier">int or str</span></dt><dd><p>Index or label of variant to plot.</p>
</dd>
<dt><strong>taxon</strong><span class="classifier">str</span></dt><dd><p>Taxon to show markers for.</p>
</dd>
<dt><strong>period</strong><span class="classifier">pd.Period</span></dt><dd><p>Time period to show markers for.</p>
</dd>
<dt><strong>clear</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, clear all layers (except the base layer) from the map
before adding new markers.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="snp-allele-frequencies-advanced">
<h2>snp_allele_frequencies_advanced()<a class="headerlink" href="#snp-allele-frequencies-advanced" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_allele_frequencies_advanced">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_allele_frequencies_advanced</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">area_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">period_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_invariant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nobs_mode</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'called'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ci_method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'wilson'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_allele_frequencies_advanced" title="Permalink to this definition">#</a></dt>
<dd><p>Group samples by taxon, area (space) and period (time), then compute
SNP allele counts and frequencies.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Gene transcript ID (AgamP4.12), e.g., “AGAP004707-RD”.</p>
</dd>
<dt><strong>area_by</strong><span class="classifier">str</span></dt><dd><p>Column name in the sample metadata to use to group samples
spatially. E.g., use “admin1_iso” or “admin1_name” to group by level
1 administrative divisions, or use “admin2_name” to group by level 2
administrative divisions.</p>
</dd>
<dt><strong>period_by</strong><span class="classifier">{“year”, “quarter”, “month”}</span></dt><dd><p>Length of time to group samples temporally.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>Minimum cohort size. Any cohorts below this size are omitted.</p>
</dd>
<dt><strong>drop_invariant</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, variants with no alternate allele calls in any cohorts are
dropped from the result.</p>
</dd>
<dt><strong>variant_query</strong><span class="classifier">str, optional</span></dt><dd></dd>
<dt><strong>site_mask</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>nobs_mode</strong><span class="classifier">{“called”, “fixed”}</span></dt><dd><p>Method for calculating the denominator when computing frequencies.
If “called” then use the number of called alleles, i.e., number of
samples with non-missing genotype calls multiplied by 2. If “fixed”
then use the number of samples multiplied by 2.</p>
</dd>
<dt><strong>ci_method</strong><span class="classifier">{“normal”, “agresti_coull”, “beta”, “wilson”, “binom_test”}, optional</span></dt><dd><p>Method to use for computing confidence intervals, passed through to
<cite>statsmodels.stats.proportion.proportion_confint</cite>.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>The resulting dataset contains data has dimensions “cohorts” and
“variants”. Variables prefixed with “cohort” are 1-dimensional
arrays with data about the cohorts, such as the area, period, taxon
and cohort size. Variables prefixed with “variant” are
1-dimensional arrays with data about the variants, such as the
contig, position, reference and alternate alleles. Variables
prefixed with “event” are 2-dimensional arrays with the allele
counts and frequency calculations.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="aa-allele-frequencies-advanced">
<h2>aa_allele_frequencies_advanced()<a class="headerlink" href="#aa-allele-frequencies-advanced" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.aa_allele_frequencies_advanced">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">aa_allele_frequencies_advanced</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">area_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">period_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nobs_mode</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'called'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ci_method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'wilson'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.aa_allele_frequencies_advanced" title="Permalink to this definition">#</a></dt>
<dd><p>Group samples by taxon, area (space) and period (time), then compute
amino acid change allele counts and frequencies.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Gene transcript ID (AgamP4.12), e.g., “AGAP004707-RD”.</p>
</dd>
<dt><strong>area_by</strong><span class="classifier">str</span></dt><dd><p>Column name in the sample metadata to use to group samples spatially.
E.g., use “admin1_iso” or “admin1_name” to group by level 1
administrative divisions, or use “admin2_name” to group by level 2
administrative divisions.</p>
</dd>
<dt><strong>period_by</strong><span class="classifier">{“year”, “quarter”, “month”}</span></dt><dd><p>Length of time to group samples temporally.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>Minimum cohort size. Any cohorts below this size are omitted.</p>
</dd>
<dt><strong>variant_query</strong><span class="classifier">str, optional</span></dt><dd></dd>
<dt><strong>site_mask</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>nobs_mode</strong><span class="classifier">{“called”, “fixed”}</span></dt><dd><p>Method for calculating the denominator when computing frequencies.
If “called” then use the number of called alleles, i.e., number of
samples with non-missing genotype calls multiplied by 2. If “fixed”
then use the number of samples multiplied by 2.</p>
</dd>
<dt><strong>ci_method</strong><span class="classifier">{“normal”, “agresti_coull”, “beta”, “wilson”, “binom_test”}, optional</span></dt><dd><p>Method to use for computing confidence intervals, passed through to
<cite>statsmodels.stats.proportion.proportion_confint</cite>.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>The resulting dataset contains data has dimensions “cohorts” and
“variants”. Variables prefixed with “cohort” are 1-dimensional
arrays with data about the cohorts, such as the area, period, taxon
and cohort size. Variables prefixed with “variant” are 1-dimensional
arrays with data about the variants, such as the contig, position,
reference and alternate alleles. Variables prefixed with “event” are
2-dimensional arrays with the allele counts and frequency
calculations.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="gene-cnv-frequencies-advanced">
<h2>gene_cnv_frequencies_advanced()<a class="headerlink" href="#gene-cnv-frequencies-advanced" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.gene_cnv_frequencies_advanced">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">gene_cnv_frequencies_advanced</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">area_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">period_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_invariant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_coverage_variance</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ci_method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'wilson'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.gene_cnv_frequencies_advanced" title="Permalink to this definition">#</a></dt>
<dd><p>Group samples by taxon, area (space) and period (time), then compute
gene CNV counts and frequencies.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>area_by</strong><span class="classifier">str</span></dt><dd><p>Column name in the sample metadata to use to group samples spatially.
E.g., use “admin1_iso” or “admin1_name” to group by level 1
administrative divisions, or use “admin2_name” to group by level 2
administrative divisions.</p>
</dd>
<dt><strong>period_by</strong><span class="classifier">{“year”, “quarter”, “month”}</span></dt><dd><p>Length of time to group samples temporally.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>Minimum cohort size. Any cohorts below this size are omitted.</p>
</dd>
<dt><strong>variant_query</strong><span class="classifier">str, optional</span></dt><dd></dd>
<dt><strong>drop_invariant</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, drop any rows where there is no evidence of variation.</p>
</dd>
<dt><strong>max_coverage_variance</strong><span class="classifier">float, optional</span></dt><dd><p>Remove samples if coverage variance exceeds this value.</p>
</dd>
<dt><strong>ci_method</strong><span class="classifier">{“normal”, “agresti_coull”, “beta”, “wilson”, “binom_test”}, optional</span></dt><dd><p>Method to use for computing confidence intervals, passed through to
<cite>statsmodels.stats.proportion.proportion_confint</cite>.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>The resulting dataset contains data has dimensions “cohorts” and
“variants”. Variables prefixed with “cohort” are 1-dimensional
arrays with data about the cohorts, such as the area, period, taxon
and cohort size. Variables prefixed with “variant” are 1-dimensional
arrays with data about the variants, such as the contig, position,
reference and alternate alleles. Variables prefixed with “event” are
2-dimensional arrays with the allele counts and frequency
calculations.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-genes">
<h2>plot_genes()<a class="headerlink" href="#plot-genes" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_genes">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_genes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">toolbar_location</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'above'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_range</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Genes'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_genes" title="Permalink to this definition">#</a></dt>
<dd><p>Plot a genes track, using bokeh.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region</strong><span class="classifier">str or Region</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height in pixels (px).</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If true, show the plot.</p>
</dd>
<dt><strong>toolbar_location</strong><span class="classifier">str, optional</span></dt><dd><p>Location of bokeh toolbar.</p>
</dd>
<dt><strong>x_range</strong><span class="classifier">bokeh.models.Range1d, optional</span></dt><dd><p>X axis range (for linking to other tracks).</p>
</dd>
<dt><strong>title</strong><span class="classifier">str, optional</span></dt><dd><p>Plot title.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-transcript">
<h2>plot_transcript()<a class="headerlink" href="#plot-transcript" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_transcript">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_transcript</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_range</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">toolbar_location</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'above'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_transcript" title="Permalink to this definition">#</a></dt>
<dd><p>Plot a transcript, using bokeh.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Transcript identifier, e.g., “AGAP004707-RD”.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height in pixels (px).</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If true, show the plot.</p>
</dd>
<dt><strong>toolbar_location</strong><span class="classifier">str, optional</span></dt><dd><p>Location of bokeh toolbar.</p>
</dd>
<dt><strong>x_range</strong><span class="classifier">bokeh.models.Range1d, optional</span></dt><dd><p>X axis range (for linking to other tracks).</p>
</dd>
<dt><strong>title</strong><span class="classifier">str, optional</span></dt><dd><p>Plot title.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-cnv-hmm-coverage">
<h2>plot_cnv_hmm_coverage()<a class="headerlink" href="#plot-cnv-hmm-coverage" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_cnv_hmm_coverage">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_cnv_hmm_coverage</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_set</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_max</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'auto'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">track_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">170</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">circle_kwargs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">line_kwargs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_cnv_hmm_coverage" title="Permalink to this definition">#</a></dt>
<dd><p>Plot CNV HMM data for a single sample, together with a genes track,
using bokeh.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample</strong><span class="classifier">str or int</span></dt><dd><p>Sample identifier or index within sample set.</p>
</dd>
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>sample_set</strong><span class="classifier">str, optional</span></dt><dd><p>Sample set identifier.</p>
</dd>
<dt><strong>y_max</strong><span class="classifier">str or int, optional</span></dt><dd><p>Maximum Y axis value.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>track_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of CNV HMM track in pixels (px).</p>
</dd>
<dt><strong>genes_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of genes track in pixels (px).</p>
</dd>
<dt><strong>circle_kwargs</strong><span class="classifier">dict, optional</span></dt><dd><p>Passed through to bokeh circle() function.</p>
</dd>
<dt><strong>line_kwargs</strong><span class="classifier">dict, optional</span></dt><dd><p>Passed through to bokeh line() function.</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If true, show the plot.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-cnv-hmm-heatmap">
<h2>plot_cnv_hmm_heatmap()<a class="headerlink" href="#plot-cnv-hmm-heatmap" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_cnv_hmm_heatmap">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_cnv_hmm_heatmap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_coverage_variance</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">row_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">7</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">track_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_cnv_hmm_heatmap" title="Permalink to this definition">#</a></dt>
<dd><p>Plot CNV HMM data for multiple samples as a heatmap, with a genes
track, using bokeh.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>max_coverage_variance</strong><span class="classifier">float, optional</span></dt><dd><p>Remove samples if coverage variance exceeds this value.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>row_height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height per row (sample) in pixels (px).</p>
</dd>
<dt><strong>track_height</strong><span class="classifier">int, optional</span></dt><dd><p>Absolute plot height for HMM track in pixels (px), overrides
row_height.</p>
</dd>
<dt><strong>genes_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of genes track in pixels (px).</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If true, show the plot.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="resolve-region">
<h2>resolve_region()<a class="headerlink" href="#resolve-region" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.resolve_region">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">resolve_region</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.resolve_region" title="Permalink to this definition">#</a></dt>
<dd><p>Convert a genome region into a standard data structure.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>out</strong><span class="classifier">Region</span></dt><dd><p>A named tuple with attributes contig, start and end.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="igv">
<h2>igv()<a class="headerlink" href="#igv" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.igv">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">igv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tracks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.igv" title="Permalink to this definition">#</a></dt>
<dd><p>Create an IGV browser and display it within the notebook.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or Region</strong></dt><dd><p>Genomic region defined with coordinates, e.g., “2L:2422600-2422700”.</p>
</dd>
<dt><strong>tracks</strong><span class="classifier">list of dict, optional</span></dt><dd><p>Configuration for any additional tracks.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>browser</strong><span class="classifier">igv_notebook.Browser</span></dt><dd></dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="view-alignments">
<h2>view_alignments()<a class="headerlink" href="#view-alignments" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.view_alignments">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">view_alignments</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">visibility_window</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20000</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.view_alignments" title="Permalink to this definition">#</a></dt>
<dd><p>Launch IGV and view sequence read alignments and SNP genotypes from
the given sample.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or Region</strong></dt><dd><p>Genomic region defined with coordinates, e.g., “2L:2422600-2422700”.</p>
</dd>
<dt><strong>sample</strong><span class="classifier">str</span></dt><dd><p>Sample identifier, e.g., “AR0001-C”.</p>
</dd>
<dt><strong>visibility_window</strong><span class="classifier">int, optional</span></dt><dd><p>Zoom level in base pairs at which alignment and SNP data will become
visible.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>Only samples from the Ag3.0 release are currently supported.</p>
</dd></dl>

</section>
<section id="wgs-data-catalog">
<h2>wgs_data_catalog()<a class="headerlink" href="#wgs-data-catalog" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.wgs_data_catalog">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">wgs_data_catalog</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_set</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.wgs_data_catalog" title="Permalink to this definition">#</a></dt>
<dd><p>Load a data catalog providing URLs for downloading BAM, VCF and Zarr
files for samples in a given sample set.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample_set</strong><span class="classifier">str</span></dt><dd><p>Sample set identifier.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>One row per sample, columns provide URLs.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="snp-allele-counts">
<h2>snp_allele_counts()<a class="headerlink" href="#snp-allele-counts" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_allele_counts">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_allele_counts</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_class</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_allele_counts" title="Permalink to this definition">#</a></dt>
<dd><p>Compute SNP allele counts. This returns the number of times each
SNP allele was observed in the selected samples.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region</strong><span class="classifier">str or Region</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_class</strong><span class="classifier">str, optional</span></dt><dd><p>Select sites belonging to one of the following classes: CDS_DEG_4,
(4-fold degenerate coding sites), CDS_DEG_2_SIMPLE (2-fold simple
degenerate coding sites), CDS_DEG_0 (non-degenerate coding sites),
INTRON_SHORT (introns shorter than 100 bp), INTRON_LONG (introns
longer than 200 bp), INTRON_SPLICE_5PRIME (intron within 2 bp of
5’ splice site), INTRON_SPLICE_3PRIME (intron within 2 bp of 3’
splice site), UTR_5PRIME (5’ untranslated region), UTR_3PRIME (3’
untranslated region), INTERGENIC (intergenic, more than 10 kbp from
a gene).</p>
</dd>
<dt><strong>cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>If provided, randomly down-sample to the given cohort size before
computing allele counts.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Random seed used for down-sampling.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ac</strong><span class="classifier">np.ndarray</span></dt><dd><p>A numpy array of shape (n_variants, 4), where the first column has
the reference allele (0) counts, the second column has the first
alternate allele (1) counts, the third column has the second
alternate allele (2) counts, and the fourth column has the third
alternate allele (3) counts.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>This computation may take some time to run, depending on your computing
environment. Results of this computation will be cached and re-used if
the <cite>results_cache</cite> parameter was set when instantiating the Ag3 class.</p>
</dd></dl>

</section>
<section id="pca">
<h2>pca()<a class="headerlink" href="#pca" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.pca">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">pca</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_snps</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">thin_offset</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'gamb_colu_arab'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_minor_ac</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_missing_an</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_components</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.pca" title="Permalink to this definition">#</a></dt>
<dd><p>Run a principal components analysis (PCA) using biallelic SNPs from
the selected genome region and samples.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>n_snps</strong><span class="classifier">int</span></dt><dd><p>The desired number of SNPs to use when running the analysis.
SNPs will be evenly thinned to approximately this number.</p>
</dd>
<dt><strong>thin_offset</strong><span class="classifier">int, optional</span></dt><dd><p>Starting index for SNP thinning. Change this to repeat the analysis
using a different set of SNPs.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>min_minor_ac</strong><span class="classifier">int, optional</span></dt><dd><p>The minimum minor allele count. SNPs with a minor allele count
below this value will be excluded prior to thinning.</p>
</dd>
<dt><strong>max_missing_an</strong><span class="classifier">int, optional</span></dt><dd><p>The maximum number of missing allele calls to accept. SNPs with
more than this value will be excluded prior to thinning. Set to 0
(default) to require no missing calls.</p>
</dd>
<dt><strong>n_components</strong><span class="classifier">int, optional</span></dt><dd><p>Number of components to return.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df_pca</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of sample metadata, with columns “PC1”, “PC2”, “PC3”,
etc., added.</p>
</dd>
<dt><strong>evr</strong><span class="classifier">np.ndarray</span></dt><dd><p>An array of explained variance ratios, one per component.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>This computation may take some time to run, depending on your computing
environment. Results of this computation will be cached and re-used if
the <cite>results_cache</cite> parameter was set when instantiating the Ag3 class.</p>
</dd></dl>

</section>
<section id="plot-pca-variance">
<h2>plot_pca_variance()<a class="headerlink" href="#plot-pca-variance" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_pca_variance">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_pca_variance</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">evr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">900</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">400</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_pca_variance" title="Permalink to this definition">#</a></dt>
<dd><p>Plot explained variance ratios from a principal components analysis
(PCA) using a plotly bar plot.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>evr</strong><span class="classifier">np.ndarray</span></dt><dd><p>An array of explained variance ratios, one per component.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height in pixels (px).</p>
</dd>
<dt><strong>**kwargs</strong></dt><dd><p>Passed through to px.bar().</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>A plotly figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-pca-coords">
<h2>plot_pca_coords()<a class="headerlink" href="#plot-pca-coords" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_pca_coords">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_pca_coords</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'PC1'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'PC2'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">symbol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">jitter_frac</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.02</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">900</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">600</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">marker_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_pca_coords" title="Permalink to this definition">#</a></dt>
<dd><p>Plot sample coordinates from a principal components analysis (PCA)
as a plotly scatter plot.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>data</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of sample metadata, with columns “PC1”, “PC2”, “PC3”,
etc., added.</p>
</dd>
<dt><strong>x</strong><span class="classifier">str, optional</span></dt><dd><p>Name of principal component to plot on the X axis.</p>
</dd>
<dt><strong>y</strong><span class="classifier">str, optional</span></dt><dd><p>Name of principal component to plot on the Y axis.</p>
</dd>
<dt><strong>color</strong><span class="classifier">str, optional</span></dt><dd><p>Name of column in the input dataframe to use to color the markers.</p>
</dd>
<dt><strong>symbol</strong><span class="classifier">str, optional</span></dt><dd><p>Name of column in the input dataframe to use to choose marker symbols.</p>
</dd>
<dt><strong>jitter_frac</strong><span class="classifier">float, optional</span></dt><dd><p>Randomly jitter points by this fraction of their range.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Random seed for jitter.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height in pixels (px).</p>
</dd>
<dt><strong>marker_size</strong><span class="classifier">int, optional</span></dt><dd><p>Marker size.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>A plotly figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-pca-coords-3d">
<h2>plot_pca_coords_3d()<a class="headerlink" href="#plot-pca-coords-3d" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_pca_coords_3d">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_pca_coords_3d</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'PC1'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'PC2'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">z</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'PC3'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">symbol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">jitter_frac</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.02</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">900</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">600</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">marker_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_pca_coords_3d" title="Permalink to this definition">#</a></dt>
<dd><p>Plot sample coordinates from a principal components analysis (PCA)
as a plotly 3D scatter plot.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>data</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of sample metadata, with columns “PC1”, “PC2”, “PC3”,
etc., added.</p>
</dd>
<dt><strong>x</strong><span class="classifier">str, optional</span></dt><dd><p>Name of principal component to plot on the X axis.</p>
</dd>
<dt><strong>y</strong><span class="classifier">str, optional</span></dt><dd><p>Name of principal component to plot on the Y axis.</p>
</dd>
<dt><strong>z</strong><span class="classifier">str, optional</span></dt><dd><p>Name of principal component to plot on the Z axis.</p>
</dd>
<dt><strong>color</strong><span class="classifier">str, optional</span></dt><dd><p>Name of column in the input dataframe to use to color the markers.</p>
</dd>
<dt><strong>symbol</strong><span class="classifier">str, optional</span></dt><dd><p>Name of column in the input dataframe to use to choose marker symbols.</p>
</dd>
<dt><strong>jitter_frac</strong><span class="classifier">float, optional</span></dt><dd><p>Randomly jitter points by this fraction of their range.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Random seed for jitter.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height in pixels (px).</p>
</dd>
<dt><strong>marker_size</strong><span class="classifier">int, optional</span></dt><dd><p>Marker size.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>A plotly figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-pca-snps">
<h2>plot_pca_snps()<a class="headerlink" href="#plot-pca-snps" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_snps">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_snps</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'gamb_colu_arab'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">track_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">80</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">120</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_snps</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">200000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_snps" title="Permalink to this definition">#</a></dt>
<dd><p>Plot SNPs in a given genome region. SNPs are shown as rectangles,
with segregating and non-segregating SNPs positioned on different levels,
and coloured by site filter.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Width of plot in pixels (px).</p>
</dd>
<dt><strong>track_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of SNPs track in pixels (px).</p>
</dd>
<dt><strong>genes_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of genes track in pixels (px).</p>
</dd>
<dt><strong>max_snps</strong><span class="classifier">int, optional</span></dt><dd><p>Maximum number of SNPs to show.</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, show the plot.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="aim-variants">
<h2>aim_variants()<a class="headerlink" href="#aim-variants" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.aim_variants">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">aim_variants</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aims</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.aim_variants" title="Permalink to this definition">#</a></dt>
<dd><p>Open ancestry informative marker variants.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>aims</strong><span class="classifier">{‘gamb_vs_colu’, ‘gambcolu_vs_arab’}</span></dt><dd><p>Which ancestry informative markers to use.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset containing AIM positions and discriminating alleles.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="aim-calls">
<h2>aim_calls()<a class="headerlink" href="#aim-calls" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.aim_calls">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">aim_calls</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aims</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.aim_calls" title="Permalink to this definition">#</a></dt>
<dd><p>Access ancestry informative marker SNP sites, alleles and genotype
calls.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>aims</strong><span class="classifier">{‘gamb_vs_colu’, ‘gambcolu_vs_arab’}</span></dt><dd><p>Which ancestry informative markers to use.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset containing AIM SNP sites, alleles and genotype calls.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-aim-heatmap">
<h2>plot_aim_heatmap()<a class="headerlink" href="#plot-aim-heatmap" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_aim_heatmap">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_aim_heatmap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aims</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sort</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">row_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colors</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'T10'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xgap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ygap</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_aim_heatmap" title="Permalink to this definition">#</a></dt>
<dd><p>Plot a heatmap of ancestry-informative marker (AIM) genotypes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>aims</strong><span class="classifier">{‘gamb_vs_colu’, ‘gambcolu_vs_arab’}</span></dt><dd><p>Which ancestry informative markers to use.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>sort</strong><span class="classifier">bool, optional</span></dt><dd><p>If true (default), sort the samples by the total fraction of AIM
alleles for the second species in the comparison.</p>
</dd>
<dt><strong>row_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height per sample in px.</p>
</dd>
<dt><strong>colors</strong><span class="classifier">str, optional</span></dt><dd><p>Choose your favourite color palette.</p>
</dd>
<dt><strong>xgap</strong><span class="classifier">float, optional</span></dt><dd><p>Creates lines between columns (variants).</p>
</dd>
<dt><strong>ygap</strong><span class="classifier">float, optional</span></dt><dd><p>Creates lines between rows (samples).</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">plotly.graph_objects.Figure</span></dt><dd></dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="cohort-diversity-stats">
<h2>cohort_diversity_stats()<a class="headerlink" href="#cohort-diversity-stats" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.cohort_diversity_stats">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">cohort_diversity_stats</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cohort</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohort_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_class</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_jack</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">200</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">confidence_level</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.95</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.cohort_diversity_stats" title="Permalink to this definition">#</a></dt>
<dd><p>Compute genetic diversity summary statistics for a cohort of
individuals.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>cohort</strong><span class="classifier">str or (str, str)</span></dt><dd><p>Either a string giving one of the predefined cohort labels, or a
pair of strings giving a custom cohort label and a sample query to
select samples in the cohort.</p>
</dd>
<dt><strong>cohort_size</strong><span class="classifier">int</span></dt><dd><p>Number of individuals to use for computation of summary statistics.
If the cohort is larger than this size, it will be randomly
down-sampled.</p>
</dd>
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_class</strong><span class="classifier">str, optional</span></dt><dd><p>Select sites belonging to one of the following classes: CDS_DEG_4,
(4-fold degenerate coding sites), CDS_DEG_2_SIMPLE (2-fold simple
degenerate coding sites), CDS_DEG_0 (non-degenerate coding sites),
INTRON_SHORT (introns shorter than 100 bp), INTRON_LONG (introns
longer than 200 bp), INTRON_SPLICE_5PRIME (intron within 2 bp of
5’ splice site), INTRON_SPLICE_3PRIME (intron within 2 bp of 3’
splice site), UTR_5PRIME (5’ untranslated region), UTR_3PRIME (3’
untranslated region), INTERGENIC (intergenic, more than 10 kbp from
a gene).</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Seed for random number generator.</p>
</dd>
<dt><strong>n_jack</strong><span class="classifier">int, optional</span></dt><dd><p>Number of blocks to divide the data into for the block jackknife
estimation of confidence intervals. N.B., larger is not necessarily
better.</p>
</dd>
<dt><strong>confidence_level</strong><span class="classifier">float, optional</span></dt><dd><p>Confidence level to use for confidence interval calculation. 0.95
means 95% confidence interval.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>stats</strong><span class="classifier">pandas.Series</span></dt><dd><p>A series with summary statistics and their confidence intervals.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="diversity-stats">
<h2>diversity_stats()<a class="headerlink" href="#diversity-stats" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.diversity_stats">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">diversity_stats</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cohorts</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohort_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_class</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_jack</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">200</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">confidence_level</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.95</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.diversity_stats" title="Permalink to this definition">#</a></dt>
<dd><p>Compute genetic diversity summary statistics for multiple cohorts.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>cohorts</strong><span class="classifier">str or dict</span></dt><dd><p>Either a string giving one of the predefined cohort columns, or a
dictionary mapping cohort labels to sample queries.</p>
</dd>
<dt><strong>cohort_size</strong><span class="classifier">int</span></dt><dd><p>Number of individuals to use for computation of summary statistics.
If the cohort is larger than this size, it will be randomly
down-sampled.</p>
</dd>
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_class</strong><span class="classifier">str, optional</span></dt><dd><p>Select sites belonging to one of the following classes: CDS_DEG_4,
(4-fold degenerate coding sites), CDS_DEG_2_SIMPLE (2-fold simple
degenerate coding sites), CDS_DEG_0 (non-degenerate coding sites),
INTRON_SHORT (introns shorter than 100 bp), INTRON_LONG (introns
longer than 200 bp), INTRON_SPLICE_5PRIME (intron within 2 bp of
5’ splice site), INTRON_SPLICE_3PRIME (intron within 2 bp of 3’
splice site), UTR_5PRIME (5’ untranslated region), UTR_3PRIME (3’
untranslated region), INTERGENIC (intergenic, more than 10 kbp from
a gene).</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Seed for random number generator.</p>
</dd>
<dt><strong>n_jack</strong><span class="classifier">int, optional</span></dt><dd><p>Number of blocks to divide the data into for the block jackknife
estimation of confidence intervals. N.B., larger is not necessarily
better.</p>
</dd>
<dt><strong>confidence_level</strong><span class="classifier">float, optional</span></dt><dd><p>Confidence level to use for confidence interval calculation. 0.95
means 95% confidence interval.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df_stats</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A DataFrame where each row provides summary statistics and their
confidence intervals for a single cohort.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-diversity-stats">
<h2>plot_diversity_stats()<a class="headerlink" href="#plot-diversity-stats" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_diversity_stats">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_diversity_stats</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_stats</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">color</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bar_plot_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">450</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bar_width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scatter_plot_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scatter_plot_width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">template</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'plotly_white'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_kwargs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_diversity_stats" title="Permalink to this definition">#</a></dt>
<dd><p>Plot diversity statistics.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df_stats</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>Output from diversity_stats().</p>
</dd>
<dt><strong>color</strong><span class="classifier">str, optional</span></dt><dd><p>Column to color by.</p>
</dd>
<dt><strong>bar_plot_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of bar plots in pixels (px).</p>
</dd>
<dt><strong>bar_width</strong><span class="classifier">int, optional</span></dt><dd><p>Width per bar in pixels (px).</p>
</dd>
<dt><strong>scatter_plot_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of scatter plot in pixels (px).</p>
</dd>
<dt><strong>scatter_plot_width</strong><span class="classifier">int, optional</span></dt><dd><p>Width of scatter plot in pixels (px).</p>
</dd>
<dt><strong>template</strong><span class="classifier">str, optional</span></dt><dd><p>Plotly template.</p>
</dd>
<dt><strong>plot_kwargs</strong><span class="classifier">dict, optional</span></dt><dd><p>Extra plotting parameters</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-heterozygosity">
<h2>plot_heterozygosity()<a class="headerlink" href="#plot-heterozygosity" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_heterozygosity">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_heterozygosity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_set</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_max</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.03</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">track_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">170</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">circle_kwargs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_heterozygosity" title="Permalink to this definition">#</a></dt>
<dd><p>Plot windowed heterozygosity for a single sample over a genome
region.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample</strong><span class="classifier">str or int</span></dt><dd><p>Sample identifier or index within sample set.</p>
</dd>
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>window_size</strong><span class="classifier">int</span></dt><dd><p>Number of sites per window.</p>
</dd>
<dt><strong>sample_set</strong><span class="classifier">str, optional</span></dt><dd><p>Sample set identifier. Not needed if sample parameter gives a sample
identifier.</p>
</dd>
<dt><strong>y_max</strong><span class="classifier">float, optional</span></dt><dd><p>Y axis limit.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>track_height</strong><span class="classifier">int, optional</span></dt><dd><p>Heterozygosity track height in pixels (px).</p>
</dd>
<dt><strong>genes_height</strong><span class="classifier">int, optional</span></dt><dd><p>Genes track height in pixels (px).</p>
</dd>
<dt><strong>circle_kwargs</strong><span class="classifier">dict, optional</span></dt><dd><p>Passed through to bokeh circle() function.</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If true, show the plot.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="roh-hmm">
<h2>roh_hmm()<a class="headerlink" href="#roh-hmm" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.roh_hmm">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">roh_hmm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_set</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phet_roh</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.001</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phet_nonroh</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(0.003,</span> <span class="pre">0.01)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transition</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.001</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.roh_hmm" title="Permalink to this definition">#</a></dt>
<dd><p>Infer runs of homozygosity for a single sample over a genome region.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample</strong><span class="classifier">str or int</span></dt><dd><p>Sample identifier or index within sample set.</p>
</dd>
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>window_size</strong><span class="classifier">int</span></dt><dd><p>Number of sites per window.</p>
</dd>
<dt><strong>sample_set</strong><span class="classifier">str, optional</span></dt><dd><p>Sample set identifier. Not needed if sample parameter gives a sample
identifier.</p>
</dd>
<dt><strong>phet_roh: float, optional</strong></dt><dd><p>Probability of observing a heterozygote in a ROH.</p>
</dd>
<dt><strong>phet_nonroh: tuple of floats, optional</strong></dt><dd><p>One or more probabilities of observing a heterozygote outside a ROH.</p>
</dd>
<dt><strong>transition: float, optional</strong></dt><dd><p>Probability of moving between states. A larger window size may call
for a larger transitional probability.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df_roh</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A DataFrame where each row provides data about a single run of
homozygosity.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-roh">
<h2>plot_roh()<a class="headerlink" href="#plot-roh" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_roh">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_roh</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_set</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phet_roh</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.001</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phet_nonroh</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(0.003,</span> <span class="pre">0.01)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transition</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.001</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_max</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.03</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">heterozygosity_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">170</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">roh_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">circle_kwargs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_roh" title="Permalink to this definition">#</a></dt>
<dd><p>Plot windowed heterozygosity and inferred runs of homozygosity for a
single sample over a genome region.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample</strong><span class="classifier">str or int</span></dt><dd><p>Sample identifier or index within sample set.</p>
</dd>
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>window_size</strong><span class="classifier">int</span></dt><dd><p>Number of sites per window.</p>
</dd>
<dt><strong>sample_set</strong><span class="classifier">str, optional</span></dt><dd><p>Sample set identifier. Not needed if sample parameter gives a sample
identifier.</p>
</dd>
<dt><strong>phet_roh: float, optional</strong></dt><dd><p>Probability of observing a heterozygote in a ROH.</p>
</dd>
<dt><strong>phet_nonroh: tuple of floats, optional</strong></dt><dd><p>One or more probabilities of observing a heterozygote outside a ROH.</p>
</dd>
<dt><strong>transition: float, optional</strong></dt><dd><p>Probability of moving between states. A larger window size may call
for a larger transitional probability.</p>
</dd>
<dt><strong>y_max</strong><span class="classifier">float, optional</span></dt><dd><p>Y axis limit.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>heterozygosity_height</strong><span class="classifier">int, optional</span></dt><dd><p>Heterozygosity track height in pixels (px).</p>
</dd>
<dt><strong>roh_height</strong><span class="classifier">int, optional</span></dt><dd><p>ROH track height in pixels (px).</p>
</dd>
<dt><strong>genes_height</strong><span class="classifier">int, optional</span></dt><dd><p>Genes track height in pixels (px).</p>
</dd>
<dt><strong>circle_kwargs</strong><span class="classifier">dict, optional</span></dt><dd><p>Passed through to bokeh circle() function.</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If true, show the plot.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="h12-calibration">
<h2>h12_calibration()<a class="headerlink" href="#h12-calibration" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.h12_calibration">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">h12_calibration</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">analysis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_sizes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(100,</span> <span class="pre">200,</span> <span class="pre">500,</span> <span class="pre">1000,</span> <span class="pre">2000,</span> <span class="pre">5000,</span> <span class="pre">10000,</span> <span class="pre">20000)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.h12_calibration" title="Permalink to this definition">#</a></dt>
<dd><p>Generate h12 GWSS calibration data for different window sizes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>contig: str</strong></dt><dd><p>Chromosome arm (e.g., “2L”)</p>
</dd>
<dt><strong>analysis</strong><span class="classifier">{“arab”, “gamb_colu”, “gamb_colu_arab”}</span></dt><dd><p>Which phasing analysis to use. If analysing only An. arabiensis, the
“arab” analysis is best. If analysing only An. gambiae and An.
coluzzii, the “gamb_colu” analysis is best. Otherwise, use the
“gamb_colu_arab” analysis.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>If provided, randomly down-sample to the given cohort size.</p>
</dd>
<dt><strong>window_sizes</strong><span class="classifier">int or list of int, optional</span></dt><dd><p>The sizes of windows used to calculate h12 over. Multiple window
sizes should be used to calibrate the optimal size for h12 analysis.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Random seed used for down-sampling.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>calibration runs</strong><span class="classifier">list of numpy.ndarrays</span></dt><dd><p>A list of h12 calibration run arrays for each window size, containing
values and percentiles.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-h12-calibration">
<h2>plot_h12_calibration()<a class="headerlink" href="#plot-h12-calibration" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_h12_calibration">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_h12_calibration</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">analysis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_sizes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(100,</span> <span class="pre">200,</span> <span class="pre">500,</span> <span class="pre">1000,</span> <span class="pre">2000,</span> <span class="pre">5000,</span> <span class="pre">10000,</span> <span class="pre">20000)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_h12_calibration" title="Permalink to this definition">#</a></dt>
<dd><p>Plot h12 GWSS calibration data for different window sizes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>contig: str</strong></dt><dd><p>Chromosome arm (e.g., “2L”)</p>
</dd>
<dt><strong>analysis</strong><span class="classifier">{“arab”, “gamb_colu”, “gamb_colu_arab”}</span></dt><dd><p>Which phasing analysis to use. If analysing only An. arabiensis, the
“arab” analysis is best. If analysing only An. gambiae and An.
coluzzii, the “gamb_colu” analysis is best. Otherwise, use the
“gamb_colu_arab” analysis.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>If provided, randomly down-sample to the given cohort size.</p>
</dd>
<dt><strong>window_sizes</strong><span class="classifier">int or list of int, optional</span></dt><dd><p>The sizes of windows used to calculate h12 over. Multiple window
sizes should be used to calibrate the optimal size for h12 analysis.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Random seed used for down-sampling.</p>
</dd>
<dt><strong>title</strong><span class="classifier">str, optional</span></dt><dd><p>If provided, title string is used to label plot.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">figure</span></dt><dd><p>A plot showing h12 calibration run percentiles for different window
sizes.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="h12-gwss">
<h2>h12_gwss()<a class="headerlink" href="#h12-gwss" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.h12_gwss">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">h12_gwss</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">analysis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.h12_gwss" title="Permalink to this definition">#</a></dt>
<dd><p>Run h12 GWSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>contig: str</strong></dt><dd><p>Chromosome arm (e.g., “2L”)</p>
</dd>
<dt><strong>analysis</strong><span class="classifier">{“arab”, “gamb_colu”, “gamb_colu_arab”}</span></dt><dd><p>Which phasing analysis to use. If analysing only An. arabiensis, the
“arab” analysis is best. If analysing only An. gambiae and An.
coluzzii, the “gamb_colu” analysis is best. Otherwise, use the
“gamb_colu_arab” analysis.</p>
</dd>
<dt><strong>window_size</strong><span class="classifier">int</span></dt><dd><p>The size of windows used to calculate h12 over.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>If provided, randomly down-sample to the given cohort size.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Random seed used for down-sampling.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>x</strong><span class="classifier">numpy.ndarray</span></dt><dd><p>An array containing the window centre point genomic positions.</p>
</dd>
<dt><strong>h12</strong><span class="classifier">numpy.ndarray</span></dt><dd><p>An array with h12 statistic values for each window.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="plot-h12-gwss">
<h2>plot_h12_gwss()<a class="headerlink" href="#plot-h12-gwss" title="Permalink to this headline">#</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_h12_gwss">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_h12_gwss</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">analysis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">random_seed</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">42</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">track_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">170</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_h12_gwss" title="Permalink to this definition">#</a></dt>
<dd><p>Plot h12 GWSS data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>contig: str</strong></dt><dd><p>Chromosome arm (e.g., “2L”)</p>
</dd>
<dt><strong>analysis</strong><span class="classifier">{“arab”, “gamb_colu”, “gamb_colu_arab”}</span></dt><dd><p>Which phasing analysis to use. If analysing only An. arabiensis, the
“arab” analysis is best. If analysing only An. gambiae and An.
coluzzii, the “gamb_colu” analysis is best. Otherwise, use the
“gamb_colu_arab” analysis.</p>
</dd>
<dt><strong>window_size</strong><span class="classifier">int</span></dt><dd><p>The size of windows used to calculate h12 over.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>If provided, randomly down-sample to the given cohort size.</p>
</dd>
<dt><strong>random_seed</strong><span class="classifier">int, optional</span></dt><dd><p>Random seed used for down-sampling.</p>
</dd>
<dt><strong>title</strong><span class="classifier">str, optional</span></dt><dd><p>If provided, title string is used to label plot.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>track_height</strong><span class="classifier">int. optional</span></dt><dd><p>GWSS track height in pixels (px).</p>
</dd>
<dt><strong>genes_height</strong><span class="classifier">int. optional</span></dt><dd><p>Gene track height in pixels (px).</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">figure</span></dt><dd><p>A plot showing windowed h12 statistic with gene track on x-axis.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./ag3"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="examples/cnv-explore.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Exploring copy number variation (CNV)</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="../amin1/intro.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Amin1.0</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By MalariaGEN<br/>
  
      &copy; Copyright 2021.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>