
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Ag3 API &#8212; MalariaGEN vector data user guide</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet">
  <link href="../_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "malariagen/vector-data");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Amin1.0" href="../amin1/intro.html" />
    <link rel="prev" title="Exploring copy number variation (CNV)" href="examples/cnv-explore.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/logo.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">MalariaGEN vector data user guide</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../landing-page.html">
   MalariaGEN vector data user guide
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Anopheles gambiae complex
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="intro.html">
   Ag3.0 (Ag1000G phase 3)
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="cloud.html">
     Ag3.0 cloud data access
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="download.html">
     Ag3.0 data downloads
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="methods.html">
     Ag3.0 sequencing and variant calling methods
    </a>
   </li>
   <li class="toctree-l2 has-children">
    <a class="reference internal" href="examples.html">
     Ag3.0 analysis examples
    </a>
    <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
    <label for="toctree-checkbox-2">
     <i class="fas fa-chevron-down">
     </i>
    </label>
    <ul>
     <li class="toctree-l3">
      <a class="reference internal" href="examples/collection-sites.html">
       Mosquito collection sites
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="examples/snps.html">
       Single nucleotide polymorphisms (SNPs)
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="examples/snp-effects.html">
       Genes and SNP effects
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="examples/pca.html">
       Investigating population structure with PCA
      </a>
     </li>
     <li class="toctree-l3">
      <a class="reference internal" href="examples/cnv-explore.html">
       Exploring copy number variation (CNV)
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Ag3 API
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Anopheles minimus
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../amin1/intro.html">
   Amin1.0
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../amin1/cloud.html">
     Amin1.0 cloud data access
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../amin1/download.html">
     Amin1.0 data downloads
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../amin1/api.html">
   Amin1 API
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Partners and community
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../studies-ag1000g.html">
   Ag1000G partner studies
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Release notes
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../release-notes.html">
   Release notes
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/ag3/api.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
                onclick="printPdf(this)" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/malariagen/vector-data"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/malariagen/vector-data/issues/new?title=Issue%20on%20page%20%2Fag3/api.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#ag3">
   Ag3()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-sets">
   sample_sets()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-metadata">
   sample_metadata()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-cohorts">
   sample_cohorts()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cross-metadata">
   cross_metadata()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#species-calls">
   species_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#genome-sequence">
   genome_sequence()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#geneset">
   geneset()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-calls">
   snp_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-sites">
   snp_sites()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-genotypes">
   snp_genotypes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-filters">
   site_filters()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#is-accessible">
   is_accessible()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-effects">
   snp_effects()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-annotations">
   site_annotations()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-hmm">
   cnv_hmm()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-coverage-calls">
   cnv_coverage_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-discordant-read-calls">
   cnv_discordant_read_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv">
   gene_cnv()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#haplotypes">
   haplotypes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-allele-frequencies">
   snp_allele_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aa-allele-frequencies">
   aa_allele_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv-frequencies">
   gene_cnv_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-heatmap">
   plot_frequencies_heatmap()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-time-series">
   plot_frequencies_time_series()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-interactive-map">
   plot_frequencies_interactive_map()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-map-markers">
   plot_frequencies_map_markers()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-allele-frequencies-advanced">
   snp_allele_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aa-allele-frequencies-advanced">
   aa_allele_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv-frequencies-advanced">
   gene_cnv_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-genes">
   plot_genes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-transcript">
   plot_transcript()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-cnv-hmm-coverage">
   plot_cnv_hmm_coverage()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-cnv-hmm-heatmap">
   plot_cnv_hmm_heatmap()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#resolve-region">
   resolve_region()
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Ag3 API</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#ag3">
   Ag3()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-sets">
   sample_sets()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-metadata">
   sample_metadata()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-cohorts">
   sample_cohorts()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cross-metadata">
   cross_metadata()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#species-calls">
   species_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#genome-sequence">
   genome_sequence()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#geneset">
   geneset()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-calls">
   snp_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-sites">
   snp_sites()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-genotypes">
   snp_genotypes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-filters">
   site_filters()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#is-accessible">
   is_accessible()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-effects">
   snp_effects()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-annotations">
   site_annotations()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-hmm">
   cnv_hmm()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-coverage-calls">
   cnv_coverage_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-discordant-read-calls">
   cnv_discordant_read_calls()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv">
   gene_cnv()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#haplotypes">
   haplotypes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-allele-frequencies">
   snp_allele_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aa-allele-frequencies">
   aa_allele_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv-frequencies">
   gene_cnv_frequencies()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-heatmap">
   plot_frequencies_heatmap()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-time-series">
   plot_frequencies_time_series()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-interactive-map">
   plot_frequencies_interactive_map()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-frequencies-map-markers">
   plot_frequencies_map_markers()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-allele-frequencies-advanced">
   snp_allele_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aa-allele-frequencies-advanced">
   aa_allele_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#gene-cnv-frequencies-advanced">
   gene_cnv_frequencies_advanced()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-genes">
   plot_genes()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-transcript">
   plot_transcript()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-cnv-hmm-coverage">
   plot_cnv_hmm_coverage()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#plot-cnv-hmm-heatmap">
   plot_cnv_hmm_heatmap()
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#resolve-region">
   resolve_region()
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="tex2jax_ignore mathjax_ignore section" id="ag3-api">
<h1>Ag3 API<a class="headerlink" href="#ag3-api" title="Permalink to this headline">¶</a></h1>
<p>This page provides documentation for functions in the <a class="reference external" href="https://github.com/malariagen/malariagen-data-python">malariagen_data</a> Python package for accessing <em>Anopheles gambiae</em> data.</p>
<div class="section" id="ag3">
<h2>Ag3()<a class="headerlink" href="#ag3" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3">
<span class="sig-prename descclassname"><span class="pre">malariagen_data.</span></span><span class="sig-name descname"><span class="pre">Ag3</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3" title="Permalink to this definition">¶</a></dt>
<dd><p>Provides access to data from Ag 3 releases.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>url</strong><span class="classifier">str</span></dt><dd><p>Base path to data. Give “gs://vo_agam_release/” to use Google Cloud
Storage, or a local path on your file system if data have been
downloaded.</p>
</dd>
<dt><strong>**kwargs</strong></dt><dd><p>Passed through to fsspec when setting up file system access.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Examples</p>
<p>Access data from Google Cloud Storage (default):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">malariagen_data</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">ag3</span> <span class="o">=</span> <span class="n">malariagen_data</span><span class="o">.</span><span class="n">Ag3</span><span class="p">()</span>
</pre></div>
</div>
<p>Access data downloaded to a local file system:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">ag3</span> <span class="o">=</span> <span class="n">malariagen_data</span><span class="o">.</span><span class="n">Ag3</span><span class="p">(</span><span class="s2">&quot;/local/path/to/vo_agam_release/&quot;</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

</div>
<div class="section" id="sample-sets">
<h2>sample_sets()<a class="headerlink" href="#sample-sets" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.sample_sets">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">sample_sets</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">release</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.sample_sets" title="Permalink to this definition">¶</a></dt>
<dd><p>Access a dataframe of sample sets.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>release</strong><span class="classifier">str, optional</span></dt><dd><p>Release identifier. Give “3.0” to access the Ag1000G phase 3 data
release.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of sample sets, one row per sample set.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="sample-metadata">
<h2>sample_metadata()<a class="headerlink" href="#sample-metadata" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.sample_metadata">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">sample_metadata</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'aim_20200422'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'20211101'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.sample_metadata" title="Permalink to this definition">¶</a></dt>
<dd><p>Access sample metadata for one or more sample sets.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>species_analysis</strong><span class="classifier">{“aim_20200422”, “pca_20200422”}, optional</span></dt><dd><p>Include species calls in metadata.</p>
</dd>
<dt><strong>cohorts_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Cohort analysis identifier (date of analysis), optional,  default is
the latest version. Includes sample cohort calls in metadata.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of sample metadata, one row per sample.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="sample-cohorts">
<h2>sample_cohorts()<a class="headerlink" href="#sample-cohorts" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.sample_cohorts">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">sample_cohorts</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'20211101'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.sample_cohorts" title="Permalink to this definition">¶</a></dt>
<dd><p>Access cohorts metadata for one or more sample sets.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of sample set
identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a release identifier (e.g.,
“3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>cohorts_analysis</strong><span class="classifier">str</span></dt><dd><p>Cohort analysis identifier (date of analysis), default is the latest version.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of cohort metadata, one row per sample.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="cross-metadata">
<h2>cross_metadata()<a class="headerlink" href="#cross-metadata" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.cross_metadata">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">cross_metadata</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.cross_metadata" title="Permalink to this definition">¶</a></dt>
<dd><p>Load a dataframe containing metadata about samples in colony crosses, including
which samples are parents or progeny in which crosses.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of sample metadata for colony crosses.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="species-calls">
<h2>species_calls()<a class="headerlink" href="#species-calls" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.species_calls">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">species_calls</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'aim_20200422'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.species_calls" title="Permalink to this definition">¶</a></dt>
<dd><p>Access species calls for one or more sample sets.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”] or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>analysis</strong><span class="classifier">{“aim_20200422”, “pca_20200422”}</span></dt><dd><p>Species calling analysis.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of species calls for one or more sample sets, one row
per sample.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="genome-sequence">
<h2>genome_sequence()<a class="headerlink" href="#genome-sequence" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.genome_sequence">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">genome_sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.genome_sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Access the reference genome sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>d</strong><span class="classifier">dask.array.Array</span></dt><dd><p>An array of nucleotides giving the reference genome sequence for the
given contig.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="geneset">
<h2>geneset()<a class="headerlink" href="#geneset" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.geneset">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">geneset</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">attributes</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">('ID',</span> <span class="pre">'Parent',</span> <span class="pre">'Name',</span> <span class="pre">'description')</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.geneset" title="Permalink to this definition">¶</a></dt>
<dd><p>Access genome feature annotations (AgamP4.12).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>attributes</strong><span class="classifier">list of str, optional</span></dt><dd><p>Attribute keys to unpack into columns. Provide “*” to unpack all
attributes.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of genome annotations, one row per feature.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="snp-calls">
<h2>snp_calls()<a class="headerlink" href="#snp-calls" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_calls">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_calls</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_calls" title="Permalink to this definition">¶</a></dt>
<dd><p>Access SNP sites, site filters and genotype calls.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of sample set
identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a release identifier (e.g.,
“3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str</span></dt><dd><p>Site filters analysis version.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr chunks.
Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset containing SNP sites, site filters and genotype calls.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="snp-sites">
<h2>snp_sites()<a class="headerlink" href="#snp-sites" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_sites">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_sites</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">field</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_sites" title="Permalink to this definition">¶</a></dt>
<dd><p>Access SNP site data (positions and alleles).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>field</strong><span class="classifier">{“POS”, “REF”, “ALT”}</span></dt><dd><p>Array to access.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str</span></dt><dd><p>Site filters analysis version.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>d</strong><span class="classifier">dask.array.Array</span></dt><dd><p>An array of either SNP positions, reference alleles or alternate
alleles.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="snp-genotypes">
<h2>snp_genotypes()<a class="headerlink" href="#snp-genotypes" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_genotypes">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_genotypes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">field</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'GT'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_genotypes" title="Permalink to this definition">¶</a></dt>
<dd><p>Access SNP genotypes and associated data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>field</strong><span class="classifier">{“GT”, “GQ”, “AD”, “MQ”}</span></dt><dd><p>Array to access.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters analysis version.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>d</strong><span class="classifier">dask.array.Array</span></dt><dd><p>An array of either genotypes (GT), genotype quality (GQ), allele
depths (AD) or mapping quality (MQ) values.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="site-filters">
<h2>site_filters()<a class="headerlink" href="#site-filters" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.site_filters">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">site_filters</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">field</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'filter_pass'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.site_filters" title="Permalink to this definition">¶</a></dt>
<dd><p>Access SNP site filters.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Mask to use.</p>
</dd>
<dt><strong>field</strong><span class="classifier">str, optional</span></dt><dd><p>Array to access.</p>
</dd>
<dt><strong>analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters analysis version.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>d</strong><span class="classifier">dask.array.Array</span></dt><dd><p>An array of boolean values identifying sites that pass the filters.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="is-accessible">
<h2>is_accessible()<a class="headerlink" href="#is-accessible" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.is_accessible">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">is_accessible</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.is_accessible" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute genome accessibility array.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters analysis version.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>a</strong><span class="classifier">numpy.ndarray</span></dt><dd><p>An array of boolean values identifying accessible genome sites.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="snp-effects">
<h2>snp_effects()<a class="headerlink" href="#snp-effects" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_effects">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_effects</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_effects" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute variant effects for a gene transcript.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Gene transcript ID (AgamP4.12), e.g., “AGAP004707-RA”.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}, optional</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters analysis version.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of all possible SNP variants and their effects, one row
per variant.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="site-annotations">
<h2>site_annotations()<a class="headerlink" href="#site-annotations" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.site_annotations">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">site_annotations</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">field</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.site_annotations" title="Permalink to this definition">¶</a></dt>
<dd><p>Load site annotations.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>field</strong><span class="classifier">str</span></dt><dd><p>One of “codon_degeneracy”, “codon_nonsyn”, “codon_position”, “seq_cls”,
“seq_flen”, “seq_relpos_start”, “seq_relpos_stop”.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str</span></dt><dd><p>Site filters analysis version.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr chunks.
Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>d</strong><span class="classifier">dask.Array</span></dt><dd><p>An array of site annotations.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="cnv-hmm">
<h2>cnv_hmm()<a class="headerlink" href="#cnv-hmm" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.cnv_hmm">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">cnv_hmm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.cnv_hmm" title="Permalink to this definition">¶</a></dt>
<dd><p>Access CNV HMM data from CNV calling.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr
chunks. Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of CNV HMM calls and associated data.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="cnv-coverage-calls">
<h2>cnv_coverage_calls()<a class="headerlink" href="#cnv-coverage-calls" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.cnv_coverage_calls">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">cnv_coverage_calls</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_set</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">analysis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.cnv_coverage_calls" title="Permalink to this definition">¶</a></dt>
<dd><p>Access CNV HMM data from genome-wide CNV discovery and filtering.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>sample_set</strong><span class="classifier">str</span></dt><dd><p>Sample set identifier.</p>
</dd>
<dt><strong>analysis</strong><span class="classifier">{‘gamb_colu’, ‘arab’, ‘crosses’}</span></dt><dd><p>Name of CNV analysis.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr chunks.
Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of CNV alleles and genotypes.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="cnv-discordant-read-calls">
<h2>cnv_discordant_read_calls()<a class="headerlink" href="#cnv-discordant-read-calls" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.cnv_discordant_read_calls">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">cnv_discordant_read_calls</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.cnv_discordant_read_calls" title="Permalink to this definition">¶</a></dt>
<dd><p>Access CNV discordant read calls data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>contig</strong><span class="classifier">str or list of str</span></dt><dd><p>Chromosome arm, e.g., “3R”. Multiple values can be provided
as a list, in which case data will be concatenated, e.g., [“2R”, “3R”].</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of sample set
identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a release identifier (e.g.,
“3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr chunks.
Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of CNV alleles and genotypes.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="gene-cnv">
<h2>gene_cnv()<a class="headerlink" href="#gene-cnv" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.gene_cnv">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">gene_cnv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.gene_cnv" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute modal copy number by gene, from HMM data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or
a release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of modal copy number per gene and associated data.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="haplotypes">
<h2>haplotypes()<a class="headerlink" href="#haplotypes" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.haplotypes">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">haplotypes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">analysis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inline_array</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chunks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'native'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.haplotypes" title="Permalink to this definition">¶</a></dt>
<dd><p>Access haplotype data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>analysis</strong><span class="classifier">{“arab”, “gamb_colu”, “gamb_colu_arab”}</span></dt><dd><p>Which phasing analysis to use. If analysing only An. arabiensis, the “arab” analysis
is best. If analysing only An. gambiae and An. coluzzii, the “gamb_colu” analysis is
best. Otherwise, use the “gamb_colu_arab” analysis.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of sample set
identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a release identifier (e.g.,
“3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>inline_array</strong><span class="classifier">bool, optional</span></dt><dd><p>Passed through to dask.array.from_array().</p>
</dd>
<dt><strong>chunks</strong><span class="classifier">str, optional</span></dt><dd><p>If ‘auto’ let dask decide chunk size. If ‘native’ use native zarr chunks.
Also, can be a target size, e.g., ‘200 MiB’.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of haplotypes and associated data.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="snp-allele-frequencies">
<h2>snp_allele_frequencies()<a class="headerlink" href="#snp-allele-frequencies" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_allele_frequencies">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_allele_frequencies</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'20211101'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'aim_20200422'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_invariant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">effects</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_allele_frequencies" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute per variant allele frequencies for a gene transcript.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Gene transcript ID (AgamP4.12), e.g., “AGAP004707-RD”.</p>
</dd>
<dt><strong>cohorts</strong><span class="classifier">str or dict</span></dt><dd><p>If a string, gives the name of a predefined cohort set, e.g., one of
{“admin1_month”, “admin1_year”, “admin2_month”, “admin2_year”}.
If a dict, should map cohort labels to sample queries, e.g.,
<cite>{“bf_2012_col”: “country == ‘Burkina Faso’ and year == 2012 and taxon == ‘coluzzii’”}</cite>.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample metadata e.g.,
“taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>cohorts_analysis</strong><span class="classifier">str</span></dt><dd><p>Cohort analysis version, default is the latest version.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int</span></dt><dd><p>Minimum cohort size. Any cohorts below this size are omitted.</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters analysis version.</p>
</dd>
<dt><strong>species_analysis</strong><span class="classifier">{“aim_20200422”, “pca_20200422”}, optional</span></dt><dd><p>Species calls analysis version.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of sample set
identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a release identifier (e.g.,
“3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>drop_invariant</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, variants with no alternate allele calls in any cohorts are dropped from
the result.</p>
</dd>
<dt><strong>effects</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, add SNP effect columns.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of SNP frequencies, one row per variant.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>Cohorts with fewer samples than min_cohort_size will be excluded from output.</p>
</dd></dl>

</div>
<div class="section" id="aa-allele-frequencies">
<h2>aa_allele_frequencies()<a class="headerlink" href="#aa-allele-frequencies" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.aa_allele_frequencies">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">aa_allele_frequencies</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'20211101'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'aim_20200422'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_invariant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.aa_allele_frequencies" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute per amino acid allele frequencies for a gene transcript.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Gene transcript ID (AgamP4.12), e.g., “AGAP004707-RA”.</p>
</dd>
<dt><strong>cohorts</strong><span class="classifier">str or dict</span></dt><dd><p>If a string, gives the name of a predefined cohort set, e.g., one of
{“admin1_month”, “admin1_year”, “admin2_month”, “admin2_year”}.
If a dict, should map cohort labels to sample queries, e.g.,
<cite>{“bf_2012_col”: “country == ‘Burkina Faso’ and year == 2012 and taxon == ‘coluzzii’”}</cite>.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample metadata e.g.,
“taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>cohorts_analysis</strong><span class="classifier">str</span></dt><dd><p>Cohort analysis identifier (date of analysis), default is the latest version.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int</span></dt><dd><p>Minimum cohort size, below which allele frequencies are not calculated for cohorts.
Please note, NaNs will be returned for any cohorts with fewer samples than min_cohort_size,
these can be removed from the output dataframe using pandas df.dropna(axis=’columns’).</p>
</dd>
<dt><strong>site_mask</strong><span class="classifier">{“gamb_colu_arab”, “gamb_colu”, “arab”}</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters analysis version.</p>
</dd>
<dt><strong>species_analysis</strong><span class="classifier">{“aim_20200422”, “pca_20200422”}, optional</span></dt><dd><p>Include species calls in metadata.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of sample set
identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a release identifier (e.g.,
“3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>drop_invariant</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, variants with no alternate allele calls in any cohorts are dropped from
the result.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of amino acid allele frequencies, one row per replacement.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>Cohorts with fewer samples than min_cohort_size will be excluded from output.</p>
</dd></dl>

</div>
<div class="section" id="gene-cnv-frequencies">
<h2>gene_cnv_frequencies()<a class="headerlink" href="#gene-cnv-frequencies" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.gene_cnv_frequencies">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">gene_cnv_frequencies</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'20211101'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'aim_20200422'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_invariant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_coverage_variance</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.gene_cnv_frequencies" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute modal copy number by gene, then compute the frequency of
amplifications and deletions in one or more cohorts, from HMM data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>cohorts</strong><span class="classifier">str or dict</span></dt><dd><p>If a string, gives the name of a predefined cohort set, e.g., one of
{“admin1_month”, “admin1_year”, “admin2_month”, “admin2_year”}.
If a dict, should map cohort labels to sample queries, e.g.,
<cite>{“bf_2012_col”: “country == ‘Burkina Faso’ and year == 2012 and
taxon == ‘coluzzii’”}</cite>.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>cohorts_analysis</strong><span class="classifier">str</span></dt><dd><p>Cohort analysis identifier (date of analysis), default is the latest
version.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int</span></dt><dd><p>Minimum cohort size, below which cohorts are dropped.</p>
</dd>
<dt><strong>species_analysis</strong><span class="classifier">{“aim_20200422”, “pca_20200422”}, optional</span></dt><dd><p>Include species calls in metadata.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>drop_invariant</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, drop any rows where there is no evidence of variation.</p>
</dd>
<dt><strong>max_coverage_variance</strong><span class="classifier">float, optional</span></dt><dd><p>Remove samples if coverage variance exceeds this value.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas.DataFrame</span></dt><dd><p>A dataframe of CNV amplification (amp) and deletion (del) frequencies in the specified cohorts,
one row per gene and CNV type (amp/del).</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="plot-frequencies-heatmap">
<h2>plot_frequencies_heatmap()<a class="headerlink" href="#plot-frequencies-heatmap" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_frequencies_heatmap">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_frequencies_heatmap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">index</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'label'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_len</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_label</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Cohorts'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_label</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Variants'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colorbar</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">col_width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">40</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">row_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">text_auto</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'.0%'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aspect</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'auto'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">color_continuous_scale</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Reds'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_frequencies_heatmap" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot a heatmap from a pandas DataFrame of frequencies, e.g., output
from <cite>Ag3.snp_allele_frequencies()</cite> or <cite>Ag3.gene_cnv_frequencies()</cite>.
It’s recommended to filter the input DataFrame to just rows of interest,
i.e., fewer rows than <cite>max_len</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>df</strong><span class="classifier">pandas DataFrame</span></dt><dd><p>A DataFrame of frequencies, e.g., output from
<cite>snp_allele_frequencies()</cite> or <cite>gene_cnv_frequencies()</cite>.</p>
</dd>
<dt><strong>index</strong><span class="classifier">str or list of str</span></dt><dd><p>One or more column headers that are present in the input dataframe.
This becomes the heatmap y-axis row labels. The column/s must
produce a unique index.</p>
</dd>
<dt><strong>max_len</strong><span class="classifier">int, optional</span></dt><dd><p>Displaying large styled dataframes may cause ipython notebooks to
crash.</p>
</dd>
<dt><strong>x_label</strong><span class="classifier">str, optional</span></dt><dd><p>This is the x-axis label that will be displayed on the heatmap.</p>
</dd>
<dt><strong>y_label</strong><span class="classifier">str, optional</span></dt><dd><p>This is the y-axis label that will be displayed on the heatmap.</p>
</dd>
<dt><strong>colorbar</strong><span class="classifier">bool, optional</span></dt><dd><p>If False, colorbar is not output.</p>
</dd>
<dt><strong>col_width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width per column in pixels (px).</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px), overrides col_width.</p>
</dd>
<dt><strong>row_height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height per row in pixels (px).</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height in pixels (px), overrides row_height.</p>
</dd>
<dt><strong>text_auto</strong><span class="classifier">str, optional</span></dt><dd><p>Formatting for frequency values.</p>
</dd>
<dt><strong>aspect</strong><span class="classifier">str, optional</span></dt><dd><p>Control the aspect ratio of the heatmap.</p>
</dd>
<dt><strong>color_continuous_scale</strong><span class="classifier">str, optional</span></dt><dd><p>Color scale to use.</p>
</dd>
<dt><strong>title</strong><span class="classifier">bool or str, optional</span></dt><dd><p>If True, attempt to use metadata from input dataset as a plot
title. Otherwise, use supplied value as a title.</p>
</dd>
<dt><strong>**kwargs</strong></dt><dd><p>Other parameters are passed through to px.imshow().</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="plot-frequencies-time-series">
<h2>plot_frequencies_time_series()<a class="headerlink" href="#plot-frequencies-time-series" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_frequencies_time_series">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_frequencies_time_series</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ds</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_frequencies_time_series" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a time series plot of variant frequencies using plotly.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of variant frequencies, such as returned by <cite>Ag3.snp_allele_frequencies_advanced()</cite>,
<cite>Ag3.aa_allele_frequencies_advanced()</cite> or <cite>Ag3.gene_cnv_frequencies_advanced()</cite>.</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of plot in pixels.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Width of plot in pixels</p>
</dd>
<dt><strong>title</strong><span class="classifier">bool or str, optional</span></dt><dd><p>If True, attempt to use metadata from input dataset as a plot
title. Otherwise, use supplied value as a title.</p>
</dd>
<dt><strong>**kwargs</strong></dt><dd><p>Passed through to <cite>px.line()</cite>.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">plotly.graph_objects.Figure</span></dt><dd><p>A plotly figure containing line graphs. The resulting figure will have one
panel per cohort, grouped into columns by taxon, and grouped into rows by
area. Markers and lines show frequencies of variants.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="plot-frequencies-interactive-map">
<h2>plot_frequencies_interactive_map()<a class="headerlink" href="#plot-frequencies-interactive-map" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_frequencies_interactive_map">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_frequencies_interactive_map</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ds</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">center</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">(-</span> <span class="pre">2,</span> <span class="pre">20)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">zoom</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">epilogue</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'\n</span>&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160; <span class="pre">Variant</span> <span class="pre">frequencies</span> <span class="pre">are</span> <span class="pre">shown</span> <span class="pre">as</span> <span class="pre">coloured</span> <span class="pre">markers.</span> <span class="pre">Opacity</span> <span class="pre">of</span> <span class="pre">color\n</span>&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160; <span class="pre">denotes</span> <span class="pre">frequency.</span> <span class="pre">Click</span> <span class="pre">on</span> <span class="pre">a</span> <span class="pre">marker</span> <span class="pre">for</span> <span class="pre">more</span> <span class="pre">information.\n</span>&#160;&#160;&#160;&#160;&#160;&#160;&#160; <span class="pre">'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_frequencies_interactive_map" title="Permalink to this definition">¶</a></dt>
<dd><p>Create an interactive map with markers showing variant frequencies for cohorts
grouped by area (space), period (time) and taxon.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of variant frequencies, such as returned by <cite>Ag3.snp_allele_frequencies_advanced()</cite>,
<cite>Ag3.aa_allele_frequencies_advanced()</cite> or <cite>Ag3.gene_cnv_frequencies_advanced()</cite>.</p>
</dd>
<dt><strong>center</strong><span class="classifier">tuple of int, optional</span></dt><dd><p>Location to center the map.</p>
</dd>
<dt><strong>zoom</strong><span class="classifier">int, optional</span></dt><dd><p>Initial zoom level.</p>
</dd>
<dt><strong>title</strong><span class="classifier">bool or str, optional</span></dt><dd><p>If True, attempt to use metadata from input dataset as a plot
title. Otherwise, use supplied value as a title.</p>
</dd>
<dt><strong>epilogue</strong><span class="classifier">str, optional</span></dt><dd><p>Additional text to display below the map.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>out</strong><span class="classifier">ipywidgets.Widget</span></dt><dd><p>An interactive map with widgets for selecting which variant, taxon and
time period to display.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="plot-frequencies-map-markers">
<h2>plot_frequencies_map_markers()<a class="headerlink" href="#plot-frequencies-map-markers" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_frequencies_map_markers">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_frequencies_map_markers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">m</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ds</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">taxon</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">period</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">clear</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_frequencies_map_markers" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot markers on a map showing variant frequencies for cohorts grouped
by area (space), period (time) and taxon.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>m</strong><span class="classifier">ipyleaflet.Map</span></dt><dd><p>The map on which to add the markers.</p>
</dd>
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>A dataset of variant frequencies, such as returned by <cite>Ag3.snp_allele_frequencies_advanced()</cite>,
<cite>Ag3.aa_allele_frequencies_advanced()</cite> or <cite>Ag3.gene_cnv_frequencies_advanced()</cite>.</p>
</dd>
<dt><strong>variant</strong><span class="classifier">int or str</span></dt><dd><p>Index or label of variant to plot.</p>
</dd>
<dt><strong>taxon</strong><span class="classifier">str</span></dt><dd><p>Taxon to show markers for.</p>
</dd>
<dt><strong>period</strong><span class="classifier">pd.Period</span></dt><dd><p>Time period to show markers for.</p>
</dd>
<dt><strong>clear</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, clear all layers (except the base layer) from the map before
adding new markers.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="snp-allele-frequencies-advanced">
<h2>snp_allele_frequencies_advanced()<a class="headerlink" href="#snp-allele-frequencies-advanced" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.snp_allele_frequencies_advanced">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">snp_allele_frequencies_advanced</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">area_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">period_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_invariant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nobs_mode</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'called'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ci_method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'wilson'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'20211101'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'aim_20200422'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.snp_allele_frequencies_advanced" title="Permalink to this definition">¶</a></dt>
<dd><p>Group samples by taxon, area (space) and period (time), then compute
SNP allele counts and frequencies.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Gene transcript ID (AgamP4.12), e.g., “AGAP004707-RD”.</p>
</dd>
<dt><strong>area_by</strong><span class="classifier">str</span></dt><dd><p>Column name in the sample metadata to use to group samples
spatially. E.g., use “admin1_iso” or “admin1_name” to group by level 1
administrative divisions, or use “admin2_name” to group by level 2
administrative divisions.</p>
</dd>
<dt><strong>period_by</strong><span class="classifier">{“year”, “quarter”, “month”}</span></dt><dd><p>Length of time to group samples temporally.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>Minimum cohort size. Any cohorts below this size are omitted.</p>
</dd>
<dt><strong>drop_invariant</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, variants with no alternate allele calls in any cohorts are
dropped from the result.</p>
</dd>
<dt><strong>variant_query</strong><span class="classifier">str, optional</span></dt><dd></dd>
<dt><strong>site_mask</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>nobs_mode</strong><span class="classifier">{“called”, “fixed”}</span></dt><dd><p>Method for calculating the denominator when computing frequencies.
If “called” then use the number of called alleles, i.e., number of
samples with non-missing genotype calls multiplied by 2. If “fixed”
then use the number of samples multiplied by 2.</p>
</dd>
<dt><strong>ci_method</strong><span class="classifier">{“normal”, “agresti_coull”, “beta”, “wilson”, “binom_test”}, optional</span></dt><dd><p>Method to use for computing confidence intervals, passed through to
<cite>statsmodels.stats.proportion.proportion_confint</cite>.</p>
</dd>
<dt><strong>cohorts_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Cohort analysis version, default is the latest version.</p>
</dd>
<dt><strong>species_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Species calls analysis version.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters analysis version.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>The resulting dataset contains data has dimensions “cohorts” and “variants”.
Variables prefixed with “cohort” are 1-dimensional arrays with data about
the cohorts, such as the area, period, taxon and cohort size. Variables
prefixed with “variant” are 1-dimensional arrays with data about the variants,
such as the contig, position, reference and alternate alleles. Variables prefixed
with “event” are 2-dimensional arrays with the allele counts and frequency
calculations.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="aa-allele-frequencies-advanced">
<h2>aa_allele_frequencies_advanced()<a class="headerlink" href="#aa-allele-frequencies-advanced" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.aa_allele_frequencies_advanced">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">aa_allele_frequencies_advanced</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">area_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">period_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_mask</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">nobs_mode</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'called'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ci_method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'wilson'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'20211101'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'aim_20200422'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">site_filters_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'dt_20200416'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.aa_allele_frequencies_advanced" title="Permalink to this definition">¶</a></dt>
<dd><p>Group samples by taxon, area (space) and period (time), then compute amino acid
change allele counts and frequencies.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Gene transcript ID (AgamP4.12), e.g., “AGAP004707-RD”.</p>
</dd>
<dt><strong>area_by</strong><span class="classifier">str</span></dt><dd><p>Column name in the sample metadata to use to group samples spatially. E.g.,
use “admin1_iso” or “admin1_name” to group by level 1 administrative divisions,
or use “admin2_name” to group by level 2 administrative divisions.</p>
</dd>
<dt><strong>period_by</strong><span class="classifier">{“year”, “quarter”, “month”}</span></dt><dd><p>Length of time to group samples temporally.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of sample set
identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a release identifier (e.g.,
“3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample metadata e.g.,
“taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>Minimum cohort size. Any cohorts below this size are omitted.</p>
</dd>
<dt><strong>variant_query</strong><span class="classifier">str, optional</span></dt><dd></dd>
<dt><strong>site_mask</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters mask to apply.</p>
</dd>
<dt><strong>nobs_mode</strong><span class="classifier">{“called”, “fixed”}</span></dt><dd><p>Method for calculating the denominator when computing frequencies. If “called”
then use the number of called alleles, i.e., number of samples with non-missing
genotype calls multiplied by 2. If “fixed” then use the number of samples
multiplied by 2.</p>
</dd>
<dt><strong>ci_method</strong><span class="classifier">{“normal”, “agresti_coull”, “beta”, “wilson”, “binom_test”}, optional</span></dt><dd><p>Method to use for computing confidence intervals, passed through to
<cite>statsmodels.stats.proportion.proportion_confint</cite>.</p>
</dd>
<dt><strong>cohorts_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Cohort analysis version, default is the latest version.</p>
</dd>
<dt><strong>species_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Species calls analysis version.</p>
</dd>
<dt><strong>site_filters_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Site filters analysis version.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>The resulting dataset contains data has dimensions “cohorts” and “variants”.
Variables prefixed with “cohort” are 1-dimensional arrays with data about
the cohorts, such as the area, period, taxon and cohort size. Variables
prefixed with “variant” are 1-dimensional arrays with data about the variants,
such as the contig, position, reference and alternate alleles. Variables prefixed
with “event” are 2-dimensional arrays with the allele counts and frequency
calculations.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="gene-cnv-frequencies-advanced">
<h2>gene_cnv_frequencies_advanced()<a class="headerlink" href="#gene-cnv-frequencies-advanced" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.gene_cnv_frequencies_advanced">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">gene_cnv_frequencies_advanced</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">area_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">period_by</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_cohort_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_invariant</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_coverage_variance</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ci_method</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'wilson'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'20211101'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'aim_20200422'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.gene_cnv_frequencies_advanced" title="Permalink to this definition">¶</a></dt>
<dd><p>Group samples by taxon, area (space) and period (time), then compute gene CNV counts
and frequencies.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str or list of str or Region or list of Region</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”), genomic
region defined with coordinates (e.g., “2L:44989425-44998059”) or a
named tuple with genomic location <cite>Region(contig, start, end)</cite>.
Multiple values can be provided as a list, in which case data will
be concatenated, e.g., [“3R”, “3L”].</p>
</dd>
<dt><strong>area_by</strong><span class="classifier">str</span></dt><dd><p>Column name in the sample metadata to use to group samples spatially.
E.g., use “admin1_iso” or “admin1_name” to group by level 1
administrative divisions, or use “admin2_name” to group by level 2
administrative divisions.</p>
</dd>
<dt><strong>period_by</strong><span class="classifier">{“year”, “quarter”, “month”}</span></dt><dd><p>Length of time to group samples temporally.</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>min_cohort_size</strong><span class="classifier">int, optional</span></dt><dd><p>Minimum cohort size. Any cohorts below this size are omitted.</p>
</dd>
<dt><strong>variant_query</strong><span class="classifier">str, optional</span></dt><dd></dd>
<dt><strong>drop_invariant</strong><span class="classifier">bool, optional</span></dt><dd><p>If True, drop any rows where there is no evidence of variation.</p>
</dd>
<dt><strong>max_coverage_variance</strong><span class="classifier">float, optional</span></dt><dd><p>Remove samples if coverage variance exceeds this value.</p>
</dd>
<dt><strong>ci_method</strong><span class="classifier">{“normal”, “agresti_coull”, “beta”, “wilson”, “binom_test”}, optional</span></dt><dd><p>Method to use for computing confidence intervals, passed through to
<cite>statsmodels.stats.proportion.proportion_confint</cite>.</p>
</dd>
<dt><strong>cohorts_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Cohort analysis version, default is the latest version.</p>
</dd>
<dt><strong>species_analysis</strong><span class="classifier">str, optional</span></dt><dd><p>Species calls analysis version.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>ds</strong><span class="classifier">xarray.Dataset</span></dt><dd><p>The resulting dataset contains data has dimensions “cohorts” and
“variants”. Variables prefixed with “cohort” are 1-dimensional
arrays with data about the cohorts, such as the area, period, taxon
and cohort size. Variables prefixed with “variant” are 1-dimensional
arrays with data about the variants, such as the contig, position,
reference and alternate alleles. Variables prefixed with “event” are
2-dimensional arrays with the allele counts and frequency
calculations.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="plot-genes">
<h2>plot_genes()<a class="headerlink" href="#plot-genes" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_genes">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_genes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">120</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">toolbar_location</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'above'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_range</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Genes'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_genes" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot a genes track, using bokeh.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height in pixels (px).</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If true, show the plot.</p>
</dd>
<dt><strong>toolbar_location</strong><span class="classifier">str, optional</span></dt><dd><p>Location of bokeh toolbar.</p>
</dd>
<dt><strong>x_range</strong><span class="classifier">bokeh.models.Range1d, optional</span></dt><dd><p>X axis range (for linking to other tracks).</p>
</dd>
<dt><strong>title</strong><span class="classifier">str, optional</span></dt><dd><p>Plot title.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="plot-transcript">
<h2>plot_transcript()<a class="headerlink" href="#plot-transcript" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_transcript">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_transcript</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">700</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">120</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x_range</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">toolbar_location</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'above'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_transcript" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot a transcript, using bokeh.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>transcript</strong><span class="classifier">str</span></dt><dd><p>Transcript identifier, e.g., “AGAP004707-RD”.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height in pixels (px).</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If true, show the plot.</p>
</dd>
<dt><strong>toolbar_location</strong><span class="classifier">str, optional</span></dt><dd><p>Location of bokeh toolbar.</p>
</dd>
<dt><strong>x_range</strong><span class="classifier">bokeh.models.Range1d, optional</span></dt><dd><p>X axis range (for linking to other tracks).</p>
</dd>
<dt><strong>title</strong><span class="classifier">str, optional</span></dt><dd><p>Plot title.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="plot-cnv-hmm-coverage">
<h2>plot_cnv_hmm_coverage()<a class="headerlink" href="#plot-cnv-hmm-coverage" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_cnv_hmm_coverage">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_cnv_hmm_coverage</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_set</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_max</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'auto'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">track_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">170</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">circle_kwargs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">line_kwargs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_cnv_hmm_coverage" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot CNV HMM data for a single sample, together with a genes track,
using bokeh.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>sample</strong><span class="classifier">str or int</span></dt><dd><p>Sample identifier or index within sample set.</p>
</dd>
<dt><strong>sample_set</strong><span class="classifier">str</span></dt><dd><p>Sample set identifier.</p>
</dd>
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or genomic
region defined with coordinates (e.g., “2L:44989425-44998059”).</p>
</dd>
<dt><strong>y_max</strong><span class="classifier">str or int, optional</span></dt><dd><p>Maximum Y axis value.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>track_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of CNV HMM track in pixels (px).</p>
</dd>
<dt><strong>genes_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of genes track in pixels (px).</p>
</dd>
<dt><strong>circle_kwargs</strong><span class="classifier">dict, optional</span></dt><dd><p>Passed through to bokeh circle() function.</p>
</dd>
<dt><strong>line_kwargs</strong><span class="classifier">dict, optional</span></dt><dd><p>Passed through to bokeh line() function.</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If true, show the plot.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="plot-cnv-hmm-heatmap">
<h2>plot_cnv_hmm_heatmap()<a class="headerlink" href="#plot-cnv-hmm-heatmap" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.plot_cnv_hmm_heatmap">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">plot_cnv_hmm_heatmap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_sets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_query</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">800</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">row_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">track_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genes_height</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">species_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'aim_20200422'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cohorts_analysis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'20211101'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.plot_cnv_hmm_heatmap" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot CNV HMM data for multiple samples as a heatmap, with a genes
track, using bokeh.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region</strong><span class="classifier">str</span></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or
genomic region defined with coordinates (e.g.,
“2L:44989425-44998059”).</p>
</dd>
<dt><strong>sample_sets</strong><span class="classifier">str or list of str, optional</span></dt><dd><p>Can be a sample set identifier (e.g., “AG1000G-AO”) or a list of
sample set identifiers (e.g., [“AG1000G-BF-A”, “AG1000G-BF-B”]) or a
release identifier (e.g., “3.0”) or a list of release identifiers.</p>
</dd>
<dt><strong>sample_query</strong><span class="classifier">str, optional</span></dt><dd><p>A pandas query string which will be evaluated against the sample
metadata e.g., “taxon == ‘coluzzii’ and country == ‘Burkina Faso’”.</p>
</dd>
<dt><strong>width</strong><span class="classifier">int, optional</span></dt><dd><p>Plot width in pixels (px).</p>
</dd>
<dt><strong>row_height</strong><span class="classifier">int, optional</span></dt><dd><p>Plot height per row (sample) in pixels (px).</p>
</dd>
<dt><strong>track_height</strong><span class="classifier">int, optional</span></dt><dd><p>Absolute plot height for HMM track in pixels (px), overrides row_height.</p>
</dd>
<dt><strong>genes_height</strong><span class="classifier">int, optional</span></dt><dd><p>Height of genes track in pixels (px).</p>
</dd>
<dt><strong>show</strong><span class="classifier">bool, optional</span></dt><dd><p>If true, show the plot.</p>
</dd>
<dt><strong>species_analysis</strong><span class="classifier">{“aim_20200422”, “pca_20200422”}, optional</span></dt><dd><p>Include species calls in metadata.</p>
</dd>
<dt><strong>cohorts_analysis</strong><span class="classifier">str</span></dt><dd><p>Cohort analysis identifier (date of analysis), default is the latest
version.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>fig</strong><span class="classifier">Figure</span></dt><dd><p>Bokeh figure.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="resolve-region">
<h2>resolve_region()<a class="headerlink" href="#resolve-region" title="Permalink to this headline">¶</a></h2>
<dl class="py method">
<dt class="sig sig-object py" id="malariagen_data.Ag3.resolve_region">
<span class="sig-prename descclassname"><span class="pre">Ag3.</span></span><span class="sig-name descname"><span class="pre">resolve_region</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#malariagen_data.Ag3.resolve_region" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert a genome region into a standard data structure.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>region: str</strong></dt><dd><p>Chromosome arm (e.g., “2L”), gene name (e.g., “AGAP007280”) or genomic
region defined with coordinates (e.g., “2L:44989425-44998059”).</p>
</dd>
</dl>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>out</strong><span class="classifier">Region</span></dt><dd><p>A named tuple with attributes contig, start and end.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./ag3"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="examples/cnv-explore.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Exploring copy number variation (CNV)</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="../amin1/intro.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Amin1.0</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By MalariaGEN<br/>
    
        &copy; Copyright 2021.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>