
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Ag3 data downloads &#8212; MalariaGEN vector data user guide</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "malariagen/vector-data");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ag3 methods" href="methods.html" />
    <link rel="prev" title="Ag3 cloud data access" href="cloud.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/logo.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">MalariaGEN vector data user guide</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../landing-page.html">
   MalariaGEN vector data user guide
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Ag3 (Ag1000G phase 3)
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   Introduction to Ag3
  </a>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="data.html">
   Ag3 data availability and methods
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="cloud.html">
     Ag3 cloud data access
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     Ag3 data downloads
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="methods.html">
     Ag3 methods
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="examples.html">
   Ag3 analysis examples
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="examples/collection-sites.html">
     Mosquito collection sites
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="examples/snps.html">
     Single nucleotide polymorphisms (SNPs)
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="examples/snp-effects.html">
     Genes and SNP effects
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="examples/pca.html">
     Investigating population structure with PCA
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="examples/cnv-explore.html">
     Exploring copy number variation (CNV)
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Partners and community
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../studies-ag1000g.html">
   Ag1000G partner studies
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/ag3/download.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/malariagen/vector-data"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/malariagen/vector-data/issues/new?title=Issue%20on%20page%20%2Fag3/download.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://gke.mybinder.org/v2/gh/malariagen/vector-data/master?urlpath=lab/tree/docs/ag3/download.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="../_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        <a class="colab-button" href="https://colab.research.google.com/github/malariagen/vector-data/blob/master/docs/ag3/download.ipynb"><button type="button" class="btn btn-secondary topbarbtn"
                title="Launch Colab" data-toggle="tooltip" data-placement="left"><img class="colab-button-logo"
                    src="../_static/images/logo_colab.png"
                    alt="Interact on Colab">Colab</button></a>
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#data-hosting">
   Data hosting
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-sets">
   Sample sets
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-metadata">
   Sample metadata
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#specimen-collection-metadata">
     Specimen collection metadata
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#species-calls">
     Species calls
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#raw-sequence-reads-fastq-format">
   Raw sequence reads (FASTQ format)
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sequence-read-alignments-bam-format">
   Sequence read alignments (BAM format)
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-calls-vcf-format">
   SNP calls (VCF format)
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#snp-genotypes">
     SNP genotypes
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#site-filters">
     Site filters
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-calls-zarr-format">
   SNP calls (Zarr format)
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#snp-sites-and-alleles">
     SNP sites and alleles
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#id1">
     Site filters
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#id2">
     SNP genotypes
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#copy-number-variation-cnv-data">
   Copy number variation (CNV) data
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cnv-hmm">
     CNV HMM
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cnv-coverage-calls">
     CNV coverage calls
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cnv-discordant-read-calls">
     CNV discordant read calls
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#accessing-downloaded-data-from-python">
   Accessing downloaded data from Python
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#feedback-and-suggestions">
   Feedback and suggestions
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="ag3-data-downloads">
<h1>Ag3 data downloads<a class="headerlink" href="#ag3-data-downloads" title="Permalink to this headline">¶</a></h1>
<p>This notebook provides information about how to download data from the <a class="reference internal" href="intro.html"><span class="doc std std-doc"><em>Anopheles gambiae</em> 1000 Genomes project (Ag1000G) phase 3</span></a>. This includes sample metadata, raw sequence reads, sequence read alignments, and single nucleotide polymorphism (SNP) calls.</p>
<p>Code examples that are intended to be run via a Linux command line are prefixed with an exclamation mark (!). If you are running these commands directly from a terminal, remove the exclamation mark.</p>
<p>Examples in this notebook assume you are downloading data to a local folder within your home directory at the path <code class="docutils literal notranslate"><span class="pre">~/vo_agam_release/</span></code>. Change this if you want to download to a different folder on the local file system.</p>
<div class="section" id="data-hosting">
<h2>Data hosting<a class="headerlink" href="#data-hosting" title="Permalink to this headline">¶</a></h2>
<p>Ag3 data are hosted by several different services.</p>
<p>Raw sequence reads in FASTQ format and sequence read alignments in BAM format are hosted by the European Nucleotide Archive (ENA). This guide provides examples of downloading data from ENA via FTP using the <code class="docutils literal notranslate"><span class="pre">wget</span></code> command line tool, but please note that there are several other options for downloading data, see the <a class="reference external" href="https://ena-docs.readthedocs.io/en/latest/retrieval/file-download.html">ENA documentation on how to download data files</a> for more information.</p>
<p>Sample metadata in CSV format and SNP calls in VCF and Zarr formats are hosted on Google Cloud Storage (GCS) in the <code class="docutils literal notranslate"><span class="pre">vo_agam_release</span></code> bucket, which is a multi-region bucket located in the United States. All data hosted on GCS are publicly accessible and do not require any authentication to access. This guide provides examples of downloading data from GCS to a local computer using the <code class="docutils literal notranslate"><span class="pre">wget</span></code> and <code class="docutils literal notranslate"><span class="pre">gsutil</span></code> command line tools. For more information about <code class="docutils literal notranslate"><span class="pre">gsutil</span></code>, see the <a class="reference external" href="https://cloud.google.com/storage/docs/gsutil">gsutil tool documentation</a>.</p>
</div>
<div class="section" id="sample-sets">
<h2>Sample sets<a class="headerlink" href="#sample-sets" title="Permalink to this headline">¶</a></h2>
<p>Data in this release are organised into 28 sample sets. Each of these sample sets corresponds to a set of mosquito specimens contributed by a collaborating study. Depending on your objectives, you may want to download data from only specific sample sets, or all sample sets. For convenience there is a tab-delimited manifest file listing all sample sets in the release. Here is a direct download link for the sample set manifest:</p>
<ul class="simple">
<li><p>https://storage.googleapis.com/vo_agam_release/v3/manifest.tsv</p></li>
</ul>
<p>The sample set manifest can also be downloaded via <code class="docutils literal notranslate"><span class="pre">gsutil</span></code> to a directory on the local file system, e.g.:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>mkdir -pv ~/vo_agam_release/v3/
<span class="o">!</span>gsutil cp gs://vo_agam_release/v3/manifest.tsv ~/vo_agam_release/v3/
</pre></div>
</div>
</div>
</div>
<p>Here are the file contents:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>cat ~/vo_agam_release/v3/manifest.tsv
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>sample_set	sample_count
AG1000G-AO	81
AG1000G-BF-A	181
AG1000G-BF-B	102
AG1000G-BF-C	13
AG1000G-CD	76
AG1000G-CF	73
AG1000G-CI	80
AG1000G-CM-A	303
AG1000G-CM-B	97
AG1000G-CM-C	44
AG1000G-FR	23
AG1000G-GA-A	69
AG1000G-GH	100
AG1000G-GM-A	74
AG1000G-GM-B	31
AG1000G-GM-C	174
AG1000G-GN-A	45
AG1000G-GN-B	185
AG1000G-GQ	10
AG1000G-GW	101
AG1000G-KE	86
AG1000G-ML-A	60
AG1000G-ML-B	71
AG1000G-MW	41
AG1000G-MZ	74
AG1000G-TZ	300
AG1000G-UG	290
AG1000G-X	297
</pre></div>
</div>
</div>
</div>
<p>For more information about these sample sets, see the section on sample sets in the <a class="reference internal" href="intro.html"><span class="doc std std-doc">introduction to Ag1000G phase 3</span></a>.</p>
</div>
<div class="section" id="sample-metadata">
<h2>Sample metadata<a class="headerlink" href="#sample-metadata" title="Permalink to this headline">¶</a></h2>
<p>Data about the samples that were sequenced to generate this data resource are available, including the time and place of collection, the gender of the specimen, and our call regarding the species of the specimen.</p>
<div class="section" id="specimen-collection-metadata">
<h3>Specimen collection metadata<a class="headerlink" href="#specimen-collection-metadata" title="Permalink to this headline">¶</a></h3>
<p>Specimen collection metadata can be downloaded from GCS. E.g., here is the download link for the sample metadata for sample set AG1000G-BF-A:</p>
<ul class="simple">
<li><p>https://storage.googleapis.com/vo_agam_release/v3/metadata/general/AG1000G-BF-A/samples.meta.csv</p></li>
</ul>
<p>Sample metadata for all sample sets can also be downloaded using <code class="docutils literal notranslate"><span class="pre">gsutil</span></code>:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>mkdir -pv ~/vo_agam_release/v3/metadata/
<span class="o">!</span>gsutil -m rsync -r gs://vo_agam_release/v3/metadata/ ~/vo_agam_release/v3/metadata/
</pre></div>
</div>
</div>
</div>
<p>Here are the first few rows of the sample metadata for sample set AG1000G-BF-A:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>head ~/vo_agam_release/v3/metadata/general/AG1000G-BF-A/samples.meta.csv
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>sample_id,partner_sample_id,contributor,country,location,year,month,latitude,longitude,sex_call
AB0085-Cx,BF2-4,Austin Burt,Burkina Faso,Pala,2012,7,11.150,-4.235,F
AB0086-Cx,BF2-6,Austin Burt,Burkina Faso,Pala,2012,7,11.150,-4.235,F
AB0087-C,BF3-3,Austin Burt,Burkina Faso,Bana,2012,7,11.233,-4.472,F
AB0088-C,BF3-5,Austin Burt,Burkina Faso,Bana,2012,7,11.233,-4.472,F
AB0089-Cx,BF3-8,Austin Burt,Burkina Faso,Bana,2012,7,11.233,-4.472,F
AB0090-C,BF3-10,Austin Burt,Burkina Faso,Bana,2012,7,11.233,-4.472,F
AB0091-C,BF3-12,Austin Burt,Burkina Faso,Bana,2012,7,11.233,-4.472,F
AB0092-C,BF3-13,Austin Burt,Burkina Faso,Bana,2012,7,11.233,-4.472,F
AB0094-Cx,BF3-17,Austin Burt,Burkina Faso,Bana,2012,7,11.233,-4.472,F
</pre></div>
</div>
</div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">sample_id</span></code> columns gives the sample identifier used throughout all Ag1000G analyses.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">country</span></code>, <code class="docutils literal notranslate"><span class="pre">location</span></code>, <code class="docutils literal notranslate"><span class="pre">latitude</span></code> and <code class="docutils literal notranslate"><span class="pre">longitude</span></code> columns give the location where the specimen was collected.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">year</span></code> and <code class="docutils literal notranslate"><span class="pre">month</span></code> columns give the approximate date when the specimen was collected.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">sex_call</span></code> column gives the gender as determined from the sequence data.</p>
</div>
<div class="section" id="species-calls">
<h3>Species calls<a class="headerlink" href="#species-calls" title="Permalink to this headline">¶</a></h3>
<p>We have made a call for each specimen as to which species it belongs to (<em>Anopheles gambiae</em>, <em>Anopheles coluzzii</em>, <em>Anopheles arabiensis</em>) based on the genotypes of the samples. These calls were made from the sequence data, and there are cases where the species is not easy to determine. We report species calls using two methods, principal components analysis (PCA) and ancestry informative markers (AIMs).</p>
<p>Species calls can be downloaded from GCS, e.g., for sample set AG1000G-BF-A:</p>
<ul class="simple">
<li><p>PCA species calls - https://storage.googleapis.com/vo_agam_release/v3/metadata/species_calls_20200422/AG1000G-BF-A/samples.species_pca.csv</p></li>
<li><p>AIM species calls - https://storage.googleapis.com/vo_agam_release/v3/metadata/species_calls_20200422/AG1000G-BF-A/samples.species_aim.csv</p></li>
</ul>
<p>Alternatively if you ran the <code class="docutils literal notranslate"><span class="pre">gsutil</span> <span class="pre">rsync</span></code> command above to download sample metadata then this file will already be present on your local file system.</p>
<p>Here are the first few rows of the AIM species calls for sample set AG1000G-BF-A:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>head ~/vo_agam_release/v3/metadata/species_calls_20200422/AG1000G-BF-A/samples.species_aim.csv
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>sample_id,aim_fraction_colu,aim_fraction_arab,species_gambcolu_arabiensis,species_gambiae_coluzzii
AB0085-Cx,0.024,0.002,gamb_colu,gambiae
AB0086-Cx,0.038,0.002,gamb_colu,gambiae
AB0087-C,0.982,0.002,gamb_colu,coluzzii
AB0088-C,0.990,0.002,gamb_colu,coluzzii
AB0089-Cx,0.975,0.002,gamb_colu,coluzzii
AB0090-C,0.977,0.002,gamb_colu,coluzzii
AB0091-C,0.974,0.002,gamb_colu,coluzzii
AB0092-C,0.978,0.002,gamb_colu,coluzzii
AB0094-Cx,0.986,0.002,gamb_colu,coluzzii
</pre></div>
</div>
</div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">species_gambcolu_arabiensis</span></code> column provides a call as to whether the specimen is arabiensis or not (gamb_colu).</p>
<p>The <code class="docutils literal notranslate"><span class="pre">species_gambiae_coluzzii</span></code> column applies to samples that are not arabiensis, and differentiates gambiae versus coluzzii.</p>
</div>
</div>
<div class="section" id="raw-sequence-reads-fastq-format">
<h2>Raw sequence reads (FASTQ format)<a class="headerlink" href="#raw-sequence-reads-fastq-format" title="Permalink to this headline">¶</a></h2>
<p>The raw sequence reads used in this data release can be downloaded from ENA. Note that for most samples there were multiple sequencing runs, and hence there are usually multiple ENA run accessions per sample. For most samples there were 3 sequencing runs, but some samples have 4 and some have a single sequencing run.</p>
<p>To find the ENA run accessions for a given sample, first download the catalog of run accessions:</p>
<ul class="simple">
<li><p>https://storage.googleapis.com/vo_agam_release/v3/metadata/ena_runs.csv</p></li>
</ul>
<p>Alternatively if you ran the <code class="docutils literal notranslate"><span class="pre">gsutil</span> <span class="pre">rsync</span></code> command above to download sample metadata then this file will already be present on your local file system. Inspect the file:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>head ~/vo_agam_release/v3/metadata/ena_runs.csv
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>sample_id,ena_run
AR0001-C,ERR347035
AR0001-C,ERR347047
AR0001-C,ERR352136
AR0002-C,ERR328585
AR0002-C,ERR323844
AR0002-C,ERR328597
AR0004-C,ERR343648
AR0004-C,ERR343636
AR0004-C,ERR343468
</pre></div>
</div>
</div>
</div>
<p>For example, the sequence reads for sample AR0001-C are available from three ENA accessions: ERR347035, ERR347047 and ERR352136. To download the sequence reads, visit the ENA website and search for these accessions. E.g., links to download sequence reads for run ERR352136 are available from this web page: https://www.ebi.ac.uk/ena/browser/view/ERR352136. To download the FASTQ files for this run via <code class="docutils literal notranslate"><span class="pre">wget</span></code>:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR352/ERR352136/ERR352136_1.fastq.gz
<span class="o">!</span>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR352/ERR352136/ERR352136_2.fastq.gz
</pre></div>
</div>
</div>
</div>
<p>Note that FASTQ files are relatively large, several GB per sample, so they may take a long time to download, and may require a substantial amount of disk space on your local system.</p>
</div>
<div class="section" id="sequence-read-alignments-bam-format">
<h2>Sequence read alignments (BAM format)<a class="headerlink" href="#sequence-read-alignments-bam-format" title="Permalink to this headline">¶</a></h2>
<p>Analysis-ready sequence read alignments are available in BAM format for all samples in the release and can be downloaded from ENA. A catalog file mapping sample identifiers to ENA accessions is available at this link:</p>
<ul class="simple">
<li><p>https://storage.googleapis.com/vo_agam_release/v3/metadata/ena_alignments.csv</p></li>
</ul>
<p>Alternatively if you ran the <code class="docutils literal notranslate"><span class="pre">gsutil</span> <span class="pre">rsync</span></code> command above to download sample metadata then this file will already be present on your local file system. Here are the first few rows:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>head ~/vo_agam_release/v3/metadata/ena_alignments.csv
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>sample_id,ena_analysis
AR0001-C,ERZ1695275
AR0002-C,ERZ1695276
AR0004-C,ERZ1695277
AR0006-C,ERZ1695278
AR0007-C,ERZ1695279
AR0008-C,ERZ1695280
AR0009-C,ERZ1695281
AR0010-Cx,ERZ1695282
AR0011-C,ERZ1695283
</pre></div>
</div>
</div>
</div>
<p>Each row in this file provides a mapping from Ag1000G sample identifiers to ENA analysis accessions. To find links for downloading the data, visit the ENA website and search for the corresponding analysis accession. E.g., the analysis-ready BAM file for sample AR0001-C can be downloaded from this web page: https://www.ebi.ac.uk/ena/browser/view/ERZ1695275. To download the BAM file via <code class="docutils literal notranslate"><span class="pre">wget</span></code>:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>wget ftp://ftp.sra.ebi.ac.uk/vol1/ERZ169/ERZ1695275/AR0001-C.bam
</pre></div>
</div>
</div>
</div>
<p>Note that BAM files are relatively large, approximately 10G per sample, so they may take a long time to download, and may require a substantial amount of disk space on your local system.</p>
</div>
<div class="section" id="snp-calls-vcf-format">
<h2>SNP calls (VCF format)<a class="headerlink" href="#snp-calls-vcf-format" title="Permalink to this headline">¶</a></h2>
<div class="section" id="snp-genotypes">
<h3>SNP genotypes<a class="headerlink" href="#snp-genotypes" title="Permalink to this headline">¶</a></h3>
<p>SNP genotypes for individual mosquitoes in VCF format are available for download from GCS. A VCF file is available for each individual sample. To download a VCF file for a given sample, you will need the sample identifier and the sample set in which the sample belongs. For example, to download the VCF file for sample AR0001-C in sample set AG1000G-AO, run the following:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>wget --no-clobber https://storage.googleapis.com/vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/AR0001-C.vcf.gz
<span class="o">!</span>wget --no-clobber https://storage.googleapis.com/vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/AR0001-C.vcf.gz.tbi
</pre></div>
</div>
</div>
</div>
<p>Alternatively, VCF files for all samples in a given sample set can be downloaded using the <code class="docutils literal notranslate"><span class="pre">gsutil</span> <span class="pre">rsync</span></code> command. E.g., to download all VCFs for sample set AG1000G-AO, run the following:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>!mkdir -pv ~/vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/
!gsutil -m rsync -r \
    gs://vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/ \
    ~/vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/
</pre></div>
</div>
</div>
</div>
<p>Note that each of these VCF files is around 3 Gb, so downloading may take some time, and sufficient local storage will be needed.</p>
<p>Each of these VCF files is an “all sites” VCF file, meaning that genotypes have been called at all genomic positions where the reference nucleotide is not “N”, regardless of whether variation is observed in the given sample. This means that VCFs from multiple samples can be merged easily to create a multi-sample VCF, which may be required for certain analyses. For example, the code below downloads VCFs for three samples, then uses the <a class="reference external" href="http://samtools.github.io/bcftools/bcftools.html#merge">bcftools merge</a> command to merge the calls for chromosome arm 3R, up to 1 Mbp:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>wget --no-clobber https://storage.googleapis.com/vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/AR0001-C.vcf.gz
<span class="o">!</span>wget --no-clobber https://storage.googleapis.com/vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/AR0001-C.vcf.gz.tbi
<span class="o">!</span>wget --no-clobber https://storage.googleapis.com/vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/AR0002-C.vcf.gz
<span class="o">!</span>wget --no-clobber https://storage.googleapis.com/vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/AR0002-C.vcf.gz.tbi
<span class="o">!</span>wget --no-clobber https://storage.googleapis.com/vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/AR0004-C.vcf.gz
<span class="o">!</span>wget --no-clobber https://storage.googleapis.com/vo_agam_release/v3/snp_genotypes/per_sample/AG1000G-AO/vcf/AR0004-C.vcf.gz.tbi
<span class="o">!</span>bcftools merge --output-type z --regions 3R:1-1000000 --output merged.vcf.gz AR0001-C.vcf.gz AR0002-C.vcf.gz AR0004-C.vcf.gz 
</pre></div>
</div>
</div>
</div>
<p>If you are just interested in analysing variants within a given set of samples, you might like to filter the merged VCF to remove non-variant sites and alleles, e.g., using <a class="reference external" href="http://samtools.github.io/bcftools/bcftools.html#view">bcftools view</a>:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>bcftools view --output-type z --output-file merged_variant.vcf.gz --min-ac <span class="m">1</span>:nonmajor --trim-alt-alleles merged.vcf.gz
</pre></div>
</div>
</div>
</div>
<p>SNP calls in VCF format will also shortly be available from the European Variation Archive (EVA), check back soon for more information.</p>
<!--

SNP calls in VCF format are also available from the European Variation Archive (EVA). A catalog file mapping sample identifiers to EVA accessions is available at this link:

* https://storage.googleapis.com/vo_agam_release/v3/metadata/eva_snp_genotypes.csv (@@TODO)

Alternatively if you ran the `gsutil rsync` command above to download sample metadata then this file will already be present on your local file system. Inspect the file:

!head ~/vo_agam_release/v3/metadata/ena_snp_genotypes.csv

Each row in this file provides a mapping from Ag1000G sample identifiers to EVA analysis accessions. To find links for downloading the data, visit the EVA website and search for the corresponding analysis accession. E.g., the VCF file for sample @@TODO can be downloaded from this web page: @@TODO

--></div>
<div class="section" id="site-filters">
<h3>Site filters<a class="headerlink" href="#site-filters" title="Permalink to this headline">¶</a></h3>
<p>SNP calling is not always reliable, and we have created some site filters to allow excluding low quality SNPs. We have created some sites-only VCF files with site filter information in the <code class="docutils literal notranslate"><span class="pre">FILTER</span></code> column. These VCF files are hosted on GCS.</p>
<p>Because different species may have different genome accessibility issues, we have created three separate site filters:</p>
<ul class="simple">
<li><p>The “gamb_colu” site filter is designed for working only with samples that are not <em>An. arabiensis</em>.</p></li>
<li><p>The “arab” filter is designed for when only working with samples that are <em>An. arabiensis</em>.</p></li>
<li><p>The “gamb_colu_arab” filter is suitable for when analysing samples of any species together.</p></li>
</ul>
<p>Each filter is available as a set of VCF files, one per chromosome arm. E.g., here is the direct download link for the gamb_colu_arab filters on chromosome arm 3R:</p>
<ul class="simple">
<li><p>https://storage.googleapis.com/vo_agam_release/v3/site_filters/dt_20200416/vcf/gamb_colu_arab/3R_sitefilters.vcf.gz</p></li>
</ul>
<p>Alternatively, all site filters VCFs can be downloaded using <code class="docutils literal notranslate"><span class="pre">gsutil</span></code>, e.g.:</p>
<!--

@@TODO describe how to use site filters VCFs with the genotypes VCF.

--><div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>!mkdir -pv ~/vo_agam_release/v3/site_filters/dt_20200416/vcf/
!gsutil -m rsync -r \
    gs://vo_agam_release/v3/site_filters/dt_20200416/vcf/ \
    ~/vo_agam_release/v3/site_filters/dt_20200416/vcf/
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="snp-calls-zarr-format">
<h2>SNP calls (Zarr format)<a class="headerlink" href="#snp-calls-zarr-format" title="Permalink to this headline">¶</a></h2>
<p>SNP data are also available in Zarr format, which can be convenient and efficient to use for certain types of analysis. These data can be analysed directly in the cloud without downloading to the local system, see the <a class="reference internal" href="cloud.html"><span class="doc std std-doc">Ag3 cloud data access guide</span></a> for more information. The data can also be downloaded to your own system for local analysis if that is more convenient. Below are examples of how to download the Zarr data to your local system.</p>
<p>The data are organised into several Zarr hierarchies.</p>
<div class="section" id="snp-sites-and-alleles">
<h3>SNP sites and alleles<a class="headerlink" href="#snp-sites-and-alleles" title="Permalink to this headline">¶</a></h3>
<p>Data on the genomic positions (sites) and reference and alternate alleles that were genotyped can be downloaded as follows:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>!mkdir -pv ~/vo_agam_release/v3/snp_genotypes/all/sites/
!gsutil -m rsync -r \
    gs://vo_agam_release/v3/snp_genotypes/all/sites/ \
    ~/vo_agam_release/v3/snp_genotypes/all/sites/
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="id1">
<h3>Site filters<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<p>SNP calling is not always reliable, and we have created some site filters to allow excluding low quality SNPs. To download site filters data in Zarr format, excluding some parts of the data that you probably won’t need:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>!mkdir -pv ~/vo_agam_release/v3/site_filters/
!gsutil -m rsync -r \
    -x &#39;.*vcf.*|.*crosses_stats.*|.*[MG]Q10.*|.*[MG]Q30.*|.*[MG]Q_mean.*|.*[MG]Q_std.*|.*/lo_.*|.*/hi_.*|.*no_cov.*|.*allele_consistency.*|.*heterozygosity.*&#39; \
    gs://vo_agam_release/v3/site_filters/ \
    ~/vo_agam_release/v3/site_filters/
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="id2">
<h3>SNP genotypes<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h3>
<p>SNP genotypes are available for each sample set separately. E.g., to download SNP genotypes in Zarr format for sample set AG1000G-BF-A, excluding some data you probably won’t need:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>!mkdir -pv ~/vo_agam_release/v3/snp_genotypes/all/AG1000G-BF-A/
!gsutil -m rsync -r \
        -x &#39;.*/calldata/(AD|GQ|MQ)/.*&#39; \
        gs://vo_agam_release/v3/snp_genotypes/all/AG1000G-BF-A/ \
        ~/vo_agam_release/v3/snp_genotypes/all/AG1000G-BF-A/
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="copy-number-variation-cnv-data">
<h2>Copy number variation (CNV) data<a class="headerlink" href="#copy-number-variation-cnv-data" title="Permalink to this headline">¶</a></h2>
<p>Data on copy number variation within the Ag3 cohort are available as three separate data types:</p>
<ul class="simple">
<li><p><strong>HMM</strong> – Genome-wide inferences of copy number state within each individual mosquito in 300 bp non-overlapping windows.</p></li>
<li><p><strong>Coverage calls</strong> – Genome-wide copy number variant calls, derived from the HMM outputs by analysing contiguous regions of elevated copy number state then clustering of variants across individuals based on breakpoint proximity.</p></li>
<li><p><strong>Discordant read calls</strong> – Copy number variant calls at selected insecticide resistance loci, based on analysis of read alignments at CNV breakpoints.</p></li>
</ul>
<p>For more information on the methods used to generate these data, see the <a class="reference internal" href="methods.html"><span class="doc std std-doc">variant-calling methods</span></a> page.</p>
<p>For each of these data types, data can be downloaded from Google Cloud Storage, and are available in either VCF or Zarr format.</p>
<div class="section" id="cnv-hmm">
<h3>CNV HMM<a class="headerlink" href="#cnv-hmm" title="Permalink to this headline">¶</a></h3>
<p>The HMM inferences of copy number state are available in VCF, Zarr and text formats, and are organised by sample set.</p>
<p>For example, the VCF file for sample set AG1000G-AO can be downloaded from the following URL:</p>
<ul class="simple">
<li><p>https://storage.googleapis.com/vo_agam_release/v3/cnv/AG1000G-AO/hmm/vcf/AG1000G-AO_cnv_hmm.vcf.gz</p></li>
</ul>
<p>VCF files for all samples sets can be downloaded via gsutil as follows:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># create a local directory to hold downloaded CNV data</span>
<span class="o">!</span>mkdir -pv ~/vo_agam_release/v3/cnv/
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># download the HMM data in VCF format for all sample sets
!gsutil -m rsync -r \
    -x &#39;.*/coverage_calls/.*|.*/discordant_read_calls/.*|.*/hmm/zarr/.*|.*/hmm/per_sample/.*&#39; \
    gs://vo_agam_release/v3/cnv/ ~/vo_agam_release/v3/cnv/
</pre></div>
</div>
</div>
</div>
<p>Zarr files for all sample sets can be downloaded as follows:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># download HMM data in Zarr format for all sample sets
!gsutil -m rsync -r \
    -x &#39;.*/coverage_calls/.*|.*/discordant_read_calls/.*|.*/hmm/vcf/.*|.*/hmm/per_sample/.*&#39; \
    gs://vo_agam_release/v3/cnv/ ~/vo_agam_release/v3/cnv/
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="cnv-coverage-calls">
<h3>CNV coverage calls<a class="headerlink" href="#cnv-coverage-calls" title="Permalink to this headline">¶</a></h3>
<p>Coverage-based CNV calls are available in VCF and Zarr formats, and are organised by sample set. Additionally the coverage calls were performed separately for <em>An. arabiensis</em> and (<em>An. gambiae</em> + <em>An. coluzzii</em>), and so are subdivided into “arab” and “gamb_colu” datasets, depending on which species are present in a given sample set.</p>
<p>Note that some samples were excluded from coverage calling because of high coverage variance.</p>
<p>For example, the VCF file for sample set AG1000G-AO and the gamb_colu callset can be downloaded from:</p>
<ul class="simple">
<li><p>https://storage.googleapis.com/vo_agam_release/v3/cnv/AG1000G-AO/coverage_calls/gamb_colu/vcf/AG1000G-AO_gamb_colu_cnv_coverage_calls.vcf.gz</p></li>
</ul>
<p>VCF files for all sample sets can be downloaded with:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># download coverage calls in VCF format for all sample sets
!gsutil -m rsync -r \
    -x &#39;.*/hmm/.*|.*/discordant_read_calls/.*|.*/coverage_calls/.*/zarr/.*&#39; \
    gs://vo_agam_release/v3/cnv/ ~/vo_agam_release/v3/cnv/
</pre></div>
</div>
</div>
</div>
<p>Zarr files for all sample sets can be downloaded with:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># download coverage calls in Zarr format for all sample sets
!gsutil -m rsync -r \
    -x &#39;.*/hmm/.*|.*/discordant_read_calls/.*|.*/coverage_calls/.*/vcf/.*&#39; \
    gs://vo_agam_release/v3/cnv/ ~/vo_agam_release/v3/cnv/
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="cnv-discordant-read-calls">
<h3>CNV discordant read calls<a class="headerlink" href="#cnv-discordant-read-calls" title="Permalink to this headline">¶</a></h3>
<p>CNV calls for selected insecticide resistance loci are available in VCF and Zarr formats, and are organised by sample set.</p>
<p>For example, the VCF file for sample set AG1000G-AO can be downloaded from:</p>
<ul class="simple">
<li><p>https://storage.googleapis.com/vo_agam_release/v3/cnv/AG1000G-AO/discordant_read_calls/vcf/AG1000G-AO_cnv_discordant_read_calls.vcf.gz</p></li>
</ul>
<p>VCF and Zarr files for all sample sets can be downloaded with:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># download discordant read calls for all sample sets
!gsutil -m rsync -r \
    -x &#39;.*/hmm/.*|.*/coverage_calls/.*&#39; \
    gs://vo_agam_release/v3/cnv/ ~/vo_agam_release/v3/cnv/
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="accessing-downloaded-data-from-python">
<h2>Accessing downloaded data from Python<a class="headerlink" href="#accessing-downloaded-data-from-python" title="Permalink to this headline">¶</a></h2>
<p>There are a wide variety of tools available for analysing the data from Ag1000G phase 3 once downloaded to your local system. If you would like to ask about possible approaches for a given analysis, please feel free to <a class="reference external" href="https://github.com/malariagen/vector-public-data/discussions/new">start a new discussion</a> on the malariagen/vector-public-data repo on GitHub.</p>
<p>Within the MalariaGEN vector genomics team we primarily use the Python programming language for analysing the SNP data, making use of software packages in the Scientific Python / PyData ecosystem. This section gives some simple examples illustrating how to use these tools to read downloaded data.</p>
<p>These examples use data from two sample sets, AG1000G-BF-A and AG1000G-BF-B.</p>
<p>Firstly, here are all the commands again to download the data needed to run these examples (requires about 10G of local storage):</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span># download sample set manifest
!mkdir -pv ~/vo_agam_release/v3/
!gsutil cp gs://vo_agam_release/v3/manifest.tsv ~/vo_agam_release/v3/

# download sample metadata
!mkdir -pv ~/vo_agam_release/v3/metadata/
!gsutil -m rsync -r gs://vo_agam_release/v3/metadata/ ~/vo_agam_release/v3/metadata/

# download sites data
!mkdir -pv ~/vo_agam_release/v3/snp_genotypes/all/sites/
!gsutil -m rsync -r \
    gs://vo_agam_release/v3/snp_genotypes/all/sites/ \
    ~/vo_agam_release/v3/snp_genotypes/all/sites/

# download site filters data
!mkdir -pv ~/vo_agam_release/v3/site_filters/
!gsutil -m rsync -r \
    -x &#39;.*vcf.*|.*crosses_stats.*|.*[MG]Q10.*|.*[MG]Q30.*|.*[MG]Q_mean.*|.*[MG]Q_std.*|.*/lo_.*|.*/hi_.*|.*no_cov.*|.*allele_consistency.*|.*heterozygosity.*&#39; \
    gs://vo_agam_release/v3/site_filters/ \
    ~/vo_agam_release/v3/site_filters/

# download SNP genotype data for selected sample sets
!mkdir -pv ~/vo_agam_release/v3/snp_genotypes/all/AG1000G-BF-A/
!gsutil -m rsync -r \
        -x &#39;.*/calldata/(AD|GQ|MQ)/.*&#39; \
        gs://vo_agam_release/v3/snp_genotypes/all/AG1000G-BF-A/ \
        ~/vo_agam_release/v3/snp_genotypes/all/AG1000G-BF-A/
!mkdir -pv ~/vo_agam_release/v3/snp_genotypes/all/AG1000G-BF-B/
!gsutil -m rsync -r \
        -x &#39;.*/calldata/(AD|GQ|MQ)/.*&#39; \
        gs://vo_agam_release/v3/snp_genotypes/all/AG1000G-BF-B/ \
        ~/vo_agam_release/v3/snp_genotypes/all/AG1000G-BF-B/
    
# download CNV HMM data for all sample sets
!mkdir -pv ~/vo_agam_release/v3/cnv/
!gsutil -m rsync -r \
    -x &#39;.*/coverage_calls/.*|.*/discordant_read_calls/.*|.*/hmm/vcf/.*|.*/hmm/per_sample/.*&#39; \
    gs://vo_agam_release/v3/cnv/ ~/vo_agam_release/v3/cnv/
</pre></div>
</div>
</div>
</div>
<p>The following Python packages need to be installed on the local system: numpy, dask, zarr, gcsfs, fsspec, scikit-allel. These packages can be installed via pip or conda. E.g.:</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>pip install -q malariagen_data
</pre></div>
</div>
</div>
</div>
<p>To make accessing these data more convenient, we’ve also created a <a class="reference external" href="https://github.com/malariagen/malariagen-data-python">malariagen_data Python package</a>, which is available from PyPI. This is experimental so please let us know if you find any bugs or have any suggestions. The <code class="docutils literal notranslate"><span class="pre">malariagen_data</span></code> package can be installed via pip, e.g.:</p>
<p>Set up access to the Ag1000G phase 3 data, at the local path where data were downloaded:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">malariagen_data</span>
<span class="n">ag3</span> <span class="o">=</span> <span class="n">malariagen_data</span><span class="o">.</span><span class="n">Ag3</span><span class="p">(</span><span class="s2">&quot;~/vo_agam_release/&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>Examples of reading/opening various files:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read sample set manifest into a pandas dataframe</span>
<span class="n">df_sample_sets</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">sample_sets</span><span class="p">()</span>
<span class="n">df_sample_sets</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_set</th>
      <th>sample_count</th>
      <th>release</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>AG1000G-AO</td>
      <td>81</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>1</th>
      <td>AG1000G-BF-A</td>
      <td>181</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>2</th>
      <td>AG1000G-BF-B</td>
      <td>102</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>3</th>
      <td>AG1000G-BF-C</td>
      <td>13</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>AG1000G-CD</td>
      <td>76</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>5</th>
      <td>AG1000G-CF</td>
      <td>73</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>6</th>
      <td>AG1000G-CI</td>
      <td>80</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>7</th>
      <td>AG1000G-CM-A</td>
      <td>303</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>8</th>
      <td>AG1000G-CM-B</td>
      <td>97</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>9</th>
      <td>AG1000G-CM-C</td>
      <td>44</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>10</th>
      <td>AG1000G-FR</td>
      <td>23</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>11</th>
      <td>AG1000G-GA-A</td>
      <td>69</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>12</th>
      <td>AG1000G-GH</td>
      <td>100</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>13</th>
      <td>AG1000G-GM-A</td>
      <td>74</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>14</th>
      <td>AG1000G-GM-B</td>
      <td>31</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>15</th>
      <td>AG1000G-GM-C</td>
      <td>174</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>16</th>
      <td>AG1000G-GN-A</td>
      <td>45</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>17</th>
      <td>AG1000G-GN-B</td>
      <td>185</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>18</th>
      <td>AG1000G-GQ</td>
      <td>10</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>19</th>
      <td>AG1000G-GW</td>
      <td>101</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>20</th>
      <td>AG1000G-KE</td>
      <td>86</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>21</th>
      <td>AG1000G-ML-A</td>
      <td>60</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>22</th>
      <td>AG1000G-ML-B</td>
      <td>71</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>23</th>
      <td>AG1000G-MW</td>
      <td>41</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>24</th>
      <td>AG1000G-MZ</td>
      <td>74</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>25</th>
      <td>AG1000G-TZ</td>
      <td>300</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>26</th>
      <td>AG1000G-UG</td>
      <td>290</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>27</th>
      <td>AG1000G-X</td>
      <td>297</td>
      <td>v3</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read sample metadata for AG1000G-BF-A and AG1000G-BF-B into a pandas dataframe</span>
<span class="n">df_samples</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">sample_metadata</span><span class="p">(</span><span class="n">sample_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;AG1000G-BF-A&#39;</span><span class="p">,</span> <span class="s1">&#39;AG1000G-BF-B&#39;</span><span class="p">])</span>
<span class="n">df_samples</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>partner_sample_id</th>
      <th>contributor</th>
      <th>country</th>
      <th>location</th>
      <th>year</th>
      <th>month</th>
      <th>latitude</th>
      <th>longitude</th>
      <th>sex_call</th>
      <th>sample_set</th>
      <th>release</th>
      <th>aim_fraction_colu</th>
      <th>aim_fraction_arab</th>
      <th>species_gambcolu_arabiensis</th>
      <th>species_gambiae_coluzzii</th>
      <th>species</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>AB0085-Cx</td>
      <td>BF2-4</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Pala</td>
      <td>2012</td>
      <td>7</td>
      <td>11.150</td>
      <td>-4.235</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.024</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>1</th>
      <td>AB0086-Cx</td>
      <td>BF2-6</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Pala</td>
      <td>2012</td>
      <td>7</td>
      <td>11.150</td>
      <td>-4.235</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.038</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>2</th>
      <td>AB0087-C</td>
      <td>BF3-3</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Bana</td>
      <td>2012</td>
      <td>7</td>
      <td>11.233</td>
      <td>-4.472</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.982</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>coluzzii</td>
      <td>coluzzii</td>
    </tr>
    <tr>
      <th>3</th>
      <td>AB0088-C</td>
      <td>BF3-5</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Bana</td>
      <td>2012</td>
      <td>7</td>
      <td>11.233</td>
      <td>-4.472</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.990</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>coluzzii</td>
      <td>coluzzii</td>
    </tr>
    <tr>
      <th>4</th>
      <td>AB0089-Cx</td>
      <td>BF3-8</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Bana</td>
      <td>2012</td>
      <td>7</td>
      <td>11.233</td>
      <td>-4.472</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.975</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>coluzzii</td>
      <td>coluzzii</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>278</th>
      <td>AB0533-C</td>
      <td>BF13-18</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Souroukoudinga</td>
      <td>2014</td>
      <td>7</td>
      <td>11.235</td>
      <td>-4.535</td>
      <td>F</td>
      <td>AG1000G-BF-B</td>
      <td>v3</td>
      <td>0.021</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>279</th>
      <td>AB0536-C</td>
      <td>BF13-31</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Souroukoudinga</td>
      <td>2014</td>
      <td>7</td>
      <td>11.235</td>
      <td>-4.535</td>
      <td>F</td>
      <td>AG1000G-BF-B</td>
      <td>v3</td>
      <td>0.025</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>280</th>
      <td>AB0537-C</td>
      <td>BF13-32</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Souroukoudinga</td>
      <td>2014</td>
      <td>7</td>
      <td>11.235</td>
      <td>-4.535</td>
      <td>F</td>
      <td>AG1000G-BF-B</td>
      <td>v3</td>
      <td>0.029</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>281</th>
      <td>AB0538-C</td>
      <td>BF13-33</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Souroukoudinga</td>
      <td>2014</td>
      <td>7</td>
      <td>11.235</td>
      <td>-4.535</td>
      <td>F</td>
      <td>AG1000G-BF-B</td>
      <td>v3</td>
      <td>0.018</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>282</th>
      <td>AB0408-C</td>
      <td>BF14-20</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Bana</td>
      <td>2014</td>
      <td>7</td>
      <td>11.233</td>
      <td>-4.472</td>
      <td>F</td>
      <td>AG1000G-BF-B</td>
      <td>v3</td>
      <td>0.986</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>coluzzii</td>
      <td>coluzzii</td>
    </tr>
  </tbody>
</table>
<p>283 rows × 17 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># inspect number of samples by species</span>
<span class="n">df_samples</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s2">&quot;species&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">size</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>species
arabiensis                         3
coluzzii                         135
gambiae                          144
intermediate_gambiae_coluzzii      1
dtype: int64
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># access SNP positions and alleles for chromosome arm 3R as dask arrays</span>
<span class="n">pos</span><span class="p">,</span> <span class="n">ref</span><span class="p">,</span> <span class="n">alt</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_sites</span><span class="p">(</span><span class="s2">&quot;3R&quot;</span><span class="p">)</span>
<span class="n">pos</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 208.91 MB </td> <td> 2.10 MB </td></tr>
    <tr><th> Shape </th><td> (52226568,) </td> <td> (524288,) </td></tr>
    <tr><th> Count </th><td> 101 Tasks </td><td> 100 Chunks </td></tr>
    <tr><th> Type </th><td> int32 </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="6" y1="0" x2="6" y2="25" />
  <line x1="12" y1="0" x2="12" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="25" y1="0" x2="25" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="43" y1="0" x2="43" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="56" y1="0" x2="56" y2="25" />
  <line x1="62" y1="0" x2="62" y2="25" />
  <line x1="68" y1="0" x2="68" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="81" y1="0" x2="81" y2="25" />
  <line x1="87" y1="0" x2="87" y2="25" />
  <line x1="93" y1="0" x2="93" y2="25" />
  <line x1="101" y1="0" x2="101" y2="25" />
  <line x1="107" y1="0" x2="107" y2="25" />
  <line x1="113" y1="0" x2="113" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >52226568</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ref</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 52.23 MB </td> <td> 524.29 kB </td></tr>
    <tr><th> Shape </th><td> (52226568,) </td> <td> (524288,) </td></tr>
    <tr><th> Count </th><td> 101 Tasks </td><td> 100 Chunks </td></tr>
    <tr><th> Type </th><td> |S1 </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="6" y1="0" x2="6" y2="25" />
  <line x1="12" y1="0" x2="12" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="25" y1="0" x2="25" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="43" y1="0" x2="43" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="56" y1="0" x2="56" y2="25" />
  <line x1="62" y1="0" x2="62" y2="25" />
  <line x1="68" y1="0" x2="68" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="81" y1="0" x2="81" y2="25" />
  <line x1="87" y1="0" x2="87" y2="25" />
  <line x1="93" y1="0" x2="93" y2="25" />
  <line x1="101" y1="0" x2="101" y2="25" />
  <line x1="107" y1="0" x2="107" y2="25" />
  <line x1="113" y1="0" x2="113" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >52226568</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">alt</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 156.68 MB </td> <td> 1.57 MB </td></tr>
    <tr><th> Shape </th><td> (52226568, 3) </td> <td> (524288, 3) </td></tr>
    <tr><th> Count </th><td> 101 Tasks </td><td> 100 Chunks </td></tr>
    <tr><th> Type </th><td> |S1 </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="75" height="170" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="25" y2="0" style="stroke-width:2" />
  <line x1="0" y1="6" x2="25" y2="6" />
  <line x1="0" y1="12" x2="25" y2="12" />
  <line x1="0" y1="18" x2="25" y2="18" />
  <line x1="0" y1="25" x2="25" y2="25" />
  <line x1="0" y1="31" x2="25" y2="31" />
  <line x1="0" y1="37" x2="25" y2="37" />
  <line x1="0" y1="43" x2="25" y2="43" />
  <line x1="0" y1="50" x2="25" y2="50" />
  <line x1="0" y1="56" x2="25" y2="56" />
  <line x1="0" y1="62" x2="25" y2="62" />
  <line x1="0" y1="68" x2="25" y2="68" />
  <line x1="0" y1="75" x2="25" y2="75" />
  <line x1="0" y1="81" x2="25" y2="81" />
  <line x1="0" y1="87" x2="25" y2="87" />
  <line x1="0" y1="93" x2="25" y2="93" />
  <line x1="0" y1="101" x2="25" y2="101" />
  <line x1="0" y1="107" x2="25" y2="107" />
  <line x1="0" y1="113" x2="25" y2="113" />
  <line x1="0" y1="120" x2="25" y2="120" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="120" style="stroke-width:2" />
  <line x1="25" y1="0" x2="25" y2="120" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="12.706308" y="140.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" >3</text>
  <text x="45.412617" y="60.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,45.412617,60.000000)">52226568</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read first 10 SNP positions into a numpy array</span>
<span class="n">p</span> <span class="o">=</span> <span class="n">pos</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">p</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10], dtype=int32)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read first 10 SNP reference alleles</span>
<span class="n">r</span> <span class="o">=</span> <span class="n">ref</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">r</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([b&#39;C&#39;, b&#39;C&#39;, b&#39;T&#39;, b&#39;C&#39;, b&#39;T&#39;, b&#39;A&#39;, b&#39;C&#39;, b&#39;G&#39;, b&#39;T&#39;, b&#39;T&#39;],
      dtype=&#39;|S1&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read first 10 SNP alternate alleles</span>
<span class="n">a</span> <span class="o">=</span> <span class="n">alt</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">a</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([[b&#39;A&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;G&#39;],
       [b&#39;C&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;T&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;G&#39;]], dtype=&#39;|S1&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># access gamb_colu_arab site filters for chromosome arm 3R as a dask array</span>
<span class="n">filter_pass</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">site_filters</span><span class="p">(</span><span class="s2">&quot;3R&quot;</span><span class="p">,</span> <span class="n">mask</span><span class="o">=</span><span class="s2">&quot;gamb_colu_arab&quot;</span><span class="p">)</span>
<span class="n">filter_pass</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 52.23 MB </td> <td> 300.00 kB </td></tr>
    <tr><th> Shape </th><td> (52226568,) </td> <td> (300000,) </td></tr>
    <tr><th> Count </th><td> 176 Tasks </td><td> 175 Chunks </td></tr>
    <tr><th> Type </th><td> bool </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="6" y1="0" x2="6" y2="25" />
  <line x1="12" y1="0" x2="12" y2="25" />
  <line x1="18" y1="0" x2="18" y2="25" />
  <line x1="24" y1="0" x2="24" y2="25" />
  <line x1="31" y1="0" x2="31" y2="25" />
  <line x1="37" y1="0" x2="37" y2="25" />
  <line x1="44" y1="0" x2="44" y2="25" />
  <line x1="50" y1="0" x2="50" y2="25" />
  <line x1="56" y1="0" x2="56" y2="25" />
  <line x1="63" y1="0" x2="63" y2="25" />
  <line x1="69" y1="0" x2="69" y2="25" />
  <line x1="75" y1="0" x2="75" y2="25" />
  <line x1="82" y1="0" x2="82" y2="25" />
  <line x1="88" y1="0" x2="88" y2="25" />
  <line x1="95" y1="0" x2="95" y2="25" />
  <line x1="101" y1="0" x2="101" y2="25" />
  <line x1="107" y1="0" x2="107" y2="25" />
  <line x1="113" y1="0" x2="113" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >52226568</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># read filter values for first 10 SNPs (True means the site passes filters)</span>
<span class="n">f</span> <span class="o">=</span> <span class="n">filter_pass</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">f</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([False, False, False, False, False, False, False, False, False,
       False])
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># access SNP genotypes for AG1000G-BF-A and AG1000G-BF-B as a dask array</span>
<span class="n">gt</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_genotypes</span><span class="p">(</span><span class="s2">&quot;3R&quot;</span><span class="p">,</span> <span class="n">sample_sets</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;AG1000G-BF-A&quot;</span><span class="p">,</span> <span class="s2">&quot;AG1000G-BF-B&quot;</span><span class="p">])</span>
<span class="n">gt</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 29.56 GB </td> <td> 30.00 MB </td></tr>
    <tr><th> Shape </th><td> (52226568, 283, 2) </td> <td> (300000, 50, 2) </td></tr>
    <tr><th> Count </th><td> 2452 Tasks </td><td> 1225 Chunks </td></tr>
    <tr><th> Type </th><td> int8 </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="4" x2="80" y2="75" />
  <line x1="10" y1="8" x2="80" y2="79" />
  <line x1="10" y1="13" x2="80" y2="84" />
  <line x1="10" y1="16" x2="80" y2="86" />
  <line x1="10" y1="20" x2="80" y2="91" />
  <line x1="10" y1="25" x2="80" y2="95" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="13" y1="3" x2="13" y2="29" />
  <line x1="17" y1="7" x2="17" y2="32" />
  <line x1="20" y1="10" x2="20" y2="36" />
  <line x1="24" y1="14" x2="24" y2="40" />
  <line x1="28" y1="18" x2="28" y2="44" />
  <line x1="32" y1="22" x2="32" y2="47" />
  <line x1="35" y1="25" x2="35" y2="51" />
  <line x1="39" y1="29" x2="39" y2="55" />
  <line x1="43" y1="33" x2="43" y2="58" />
  <line x1="47" y1="37" x2="47" y2="62" />
  <line x1="50" y1="40" x2="50" y2="66" />
  <line x1="54" y1="44" x2="54" y2="70" />
  <line x1="58" y1="48" x2="58" y2="73" />
  <line x1="61" y1="51" x2="61" y2="77" />
  <line x1="65" y1="55" x2="65" y2="81" />
  <line x1="69" y1="59" x2="69" y2="85" />
  <line x1="73" y1="63" x2="73" y2="88" />
  <line x1="76" y1="66" x2="76" y2="92" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="13" y1="3" x2="39" y2="3" />
  <line x1="17" y1="7" x2="42" y2="7" />
  <line x1="20" y1="10" x2="46" y2="10" />
  <line x1="24" y1="14" x2="50" y2="14" />
  <line x1="28" y1="18" x2="54" y2="18" />
  <line x1="32" y1="22" x2="57" y2="22" />
  <line x1="35" y1="25" x2="61" y2="25" />
  <line x1="39" y1="29" x2="65" y2="29" />
  <line x1="43" y1="33" x2="68" y2="33" />
  <line x1="47" y1="37" x2="72" y2="37" />
  <line x1="50" y1="40" x2="76" y2="40" />
  <line x1="54" y1="44" x2="80" y2="44" />
  <line x1="58" y1="48" x2="83" y2="48" />
  <line x1="61" y1="51" x2="87" y2="51" />
  <line x1="65" y1="55" x2="91" y2="55" />
  <line x1="69" y1="59" x2="95" y2="59" />
  <line x1="73" y1="63" x2="98" y2="63" />
  <line x1="76" y1="66" x2="102" y2="66" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="75" x2="106" y2="75" />
  <line x1="80" y1="79" x2="106" y2="79" />
  <line x1="80" y1="84" x2="106" y2="84" />
  <line x1="80" y1="86" x2="106" y2="86" />
  <line x1="80" y1="91" x2="106" y2="91" />
  <line x1="80" y1="95" x2="106" y2="95" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,126.000852,83.294544)">283</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">52226568</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<p>Genotypes are stored as a three-dimensional array, where the first dimension corresponds to genomic positions, the second dimension is samples, and the third dimension is ploidy (2). Values coded as integers, where -1 represents a missing value, 0 represents the reference allele, and 1, 2, and 3 represent alternate alleles.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># e.g., read genotypes for the first 5 SNPs and the first 3 samples</span>
<span class="n">g</span> <span class="o">=</span> <span class="n">gt</span><span class="p">[:</span><span class="mi">5</span><span class="p">,</span> <span class="p">:</span><span class="mi">3</span><span class="p">,</span> <span class="p">:]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">g</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([[[0, 0],
        [0, 0],
        [0, 0]],

       [[0, 0],
        [0, 0],
        [0, 0]],

       [[0, 0],
        [0, 0],
        [0, 0]],

       [[0, 0],
        [0, 0],
        [0, 0]],

       [[0, 0],
        [0, 0],
        [0, 0]]], dtype=int8)
</pre></div>
</div>
</div>
</div>
<p>Here’s an example computation to count the number of segregating SNPs on chromosome arm 3R that also pass gamb_colu_arab site filters:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># import scikit-allel</span>
<span class="kn">import</span> <span class="nn">allel</span>

<span class="c1"># import dask </span>
<span class="kn">import</span> <span class="nn">dask.array</span> <span class="k">as</span> <span class="nn">da</span>
<span class="kn">from</span> <span class="nn">dask.diagnostics.progress</span> <span class="kn">import</span> <span class="n">ProgressBar</span>

<span class="c1"># import numpy</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>

<span class="c1"># choose chromosome arm</span>
<span class="n">contig</span> <span class="o">=</span> <span class="s2">&quot;3R&quot;</span>

<span class="c1"># choose site filter mask</span>
<span class="n">mask</span> <span class="o">=</span> <span class="s2">&quot;gamb_colu_arab&quot;</span>

<span class="c1"># choose sample sets</span>
<span class="n">sample_sets</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;AG1000G-BF-A&quot;</span><span class="p">,</span> <span class="s2">&quot;AG1000G-BF-B&quot;</span><span class="p">]</span>

<span class="c1"># locate pass sites</span>
<span class="n">loc_pass</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">site_filters</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">,</span> <span class="n">mask</span><span class="o">=</span><span class="n">mask</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>

<span class="c1"># perform an allele count over genotypes</span>
<span class="n">gt</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_genotypes</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">,</span> <span class="n">sample_sets</span><span class="o">=</span><span class="n">sample_sets</span><span class="p">)</span>
<span class="n">gt</span> <span class="o">=</span> <span class="n">allel</span><span class="o">.</span><span class="n">GenotypeDaskArray</span><span class="p">(</span><span class="n">gt</span><span class="p">)</span>
<span class="n">ac</span> <span class="o">=</span> <span class="n">gt</span><span class="o">.</span><span class="n">count_alleles</span><span class="p">(</span><span class="n">max_allele</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span>

<span class="c1"># locate segregating sites</span>
<span class="n">loc_seg</span> <span class="o">=</span> <span class="n">ac</span><span class="o">.</span><span class="n">is_segregating</span><span class="p">()</span>

<span class="c1"># count segregating and pass sites</span>
<span class="n">n_pass_seg</span> <span class="o">=</span> <span class="n">da</span><span class="o">.</span><span class="n">count_nonzero</span><span class="p">(</span><span class="n">loc_pass</span> <span class="o">&amp;</span> <span class="n">loc_seg</span><span class="p">)</span>

<span class="c1"># run the computation</span>
<span class="k">with</span> <span class="n">ProgressBar</span><span class="p">():</span>
    <span class="n">n_pass_seg</span> <span class="o">=</span> <span class="n">n_pass_seg</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>

<span class="n">n_pass_seg</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[########################################] | 100% Completed | 10.0s
</pre></div>
</div>
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>11226723
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="feedback-and-suggestions">
<h2>Feedback and suggestions<a class="headerlink" href="#feedback-and-suggestions" title="Permalink to this headline">¶</a></h2>
<p>If there are particular analyses you would like to run, or if you have other suggestions for useful documentation we could add to this site, we would love to know, please get in touch via the <a class="reference external" href="https://github.com/malariagen/vector-data/discussions">malariagen/vector-data GitHub discussion board</a>.</p>
<hr class="docutils" />
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./ag3"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
            



<div class='prev-next-bottom'>
    
    <div id="prev">
        <a class="left-prev" href="cloud.html" title="previous page">
            <i class="prevnext-label fas fa-angle-left"></i>
            <div class="prevnext-info">
                <p class="prevnext-label">previous</p>
                <p class="prevnext-title">Ag3 cloud data access</p>
            </div>
        </a>
    </div>
     <div id="next">
        <a class="right-next" href="methods.html" title="next page">
            <div class="prevnext-info">
                <p class="prevnext-label">next</p>
                <p class="prevnext-title">Ag3 methods</p>
            </div>
            <i class="prevnext-label fas fa-angle-right"></i>
        </a>
     </div>

</div>
        
        </div>
    </div>
    <footer class="footer">
    <div class="container">
      <p>
        
          By MalariaGEN<br/>
        
            &copy; Copyright 2021.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>