
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Introduction to Ag1000G phase 3 (Ag3) &#8212; MalariaGEN vector data user guide</title>
    
  <link rel="stylesheet" href="../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ag3 cloud data access guide" href="cloud.html" />
    <link rel="prev" title="MalariaGEN vector data user guide" href="../landing-page.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  <img src="../_static/logo.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">MalariaGEN vector data user guide</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../landing-page.html">
   MalariaGEN vector data user guide
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Ag1000G phase 3 (Ag3)
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Introduction to Ag1000G phase 3 (Ag3)
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="cloud.html">
   Ag3 cloud data access guide
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="download.html">
   Ag3 data download guide
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/ag3/intro.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

        <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/malariagen/vector-data"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/malariagen/vector-data/issues/new?title=Issue%20on%20page%20%2Fag3/intro.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://gke.mybinder.org/v2/gh/malariagen/vector-data/master?urlpath=tree/docs/ag3/intro.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="../_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        <a class="colab-button" href="https://colab.research.google.com/github/malariagen/vector-data/blob/master/docs/ag3/intro.ipynb"><button type="button" class="btn btn-secondary topbarbtn"
                title="Launch Colab" data-toggle="tooltip" data-placement="left"><img class="colab-button-logo"
                    src="../_static/images/logo_colab.png"
                    alt="Interact on Colab">Colab</button></a>
        
        
    </div>
</div>

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#terms-of-use">
   Terms of use
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#contributing-studies">
   Contributing studies
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#population-sampling">
   Population sampling
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#whole-genome-sequencing-and-variant-calling">
   Whole-genome sequencing and variant calling
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#data-hosting">
   Data hosting
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-sets">
   Sample sets
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#further-reading">
   Further reading
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="introduction-to-ag1000g-phase-3-ag3">
<h1>Introduction to Ag1000G phase 3 (Ag3)<a class="headerlink" href="#introduction-to-ag1000g-phase-3-ag3" title="Permalink to this headline">¶</a></h1>
<p>The <a class="reference external" href="https://www.malariagen.net/ag1000g"><em>Anopheles gambiae</em> 1000 Genomes Project (Ag1000G)</a> is a collaborative project using whole-genome sequencing to study genetic variation and evolution in natural populations of mosquitoes in the <em>Anopheles gambiae</em> species complex.</p>
<p>This page provides an introduction to open data resources released as part of the third phase of the Ag1000G project, known as “Ag3” for short. We hope the data from Ag3 will be a valuable resource for research and surveillance of malaria vectors. If you have any questions about this guide or how to use the data, please <a class="reference external" href="https://github.com/malariagen/vector-public-data/discussions/new">start a new discussion</a> on the malariagen/vector-open-data repo on GitHub. If you find any bugs, please <a class="reference external" href="https://github.com/malariagen/vector-public-data/issues/new/choose">raise an issue</a>.</p>
<div class="section" id="terms-of-use">
<h2>Terms of use<a class="headerlink" href="#terms-of-use" title="Permalink to this headline">¶</a></h2>
<p>Data from Ag3 are released openly and can be downloaded and analysed for any purpose. The data have been released prior to publication by the Ag1000G Consortium, and are currently subject to a publication embargo described further in the <a class="reference external" href="https://www.malariagen.net/data/terms-use/ag1000g-terms-use">Ag1000G terms of use</a>. If you have any questions about the terms of use, please email data&#64;malariagen.net.</p>
</div>
<div class="section" id="contributing-studies">
<h2>Contributing studies<a class="headerlink" href="#contributing-studies" title="Permalink to this headline">¶</a></h2>
<p>The Ag1000G project is coordinated by a consortium of partners from a range
of different research institutions and countries. This includes consortium members
who are carrying out independent research studies in malaria-endemic regions, and
who have contributed mosquito specimens or mosquito DNA samples collected in
the course of their own research. In total, 26 studies contributed samples to Ag3, including wild-caught specimens from 19 countries.</p>
<p>For further information about these contributing studies, the researchers involved, and the collection sites and methods, please see the <a class="reference external" href="https://storage.googleapis.com/vo_agam_release/v3/ag1000g-phase3-contributing-studies.pdf">contributing studies document</a>.</p>
</div>
<div class="section" id="population-sampling">
<h2>Population sampling<a class="headerlink" href="#population-sampling" title="Permalink to this headline">¶</a></h2>
<p>Ag3 includes data from 3,081 individual mosquitoes, including 2,784 mosquitoes collected from natural populations in 19 countries. Three species are represented within the cohort: <em>Anopheles gambiae</em>, <em>Anopheles coluzzii</em> and <em>Anopheles arabiensis</em>. The map below provides an overview of the collection locations and the numbers of samples broken down by species.</p>
<a class="bg-primary reference internal image-reference" href="../_images/ag3-map.png"><img alt="Ag3 map of sampling sites" class="bg-primary align-center" src="../_images/ag3-map.png" style="width: 700px;" /></a>
<p>In addition to these wild-caught samples, a further 297 samples are included from 15 lab crosses.</p>
</div>
<div class="section" id="whole-genome-sequencing-and-variant-calling">
<h2>Whole-genome sequencing and variant calling<a class="headerlink" href="#whole-genome-sequencing-and-variant-calling" title="Permalink to this headline">¶</a></h2>
<p>All samples in Ag3 have been sequenced individually to high coverage using Illumina technology at the Wellcome Sanger Institute. These sequence data have then been analysed to identify genetic variants such as single nucleotide polymorphisms (SNPs). After variant calling, both the samples and the variants have been through a range of quality control analyses, to ensure the data are of high quality. Both the raw sequence data and the curated variant calls are openly available for download and analysis.</p>
<p>For further information about the sequencing and variant calling methods used, please see the <a class="reference external" href="https://storage.googleapis.com/vo_agam_release/v3/ag1000g-phase3-snp-calling-methods.pdf">SNP calling methods document</a>.</p>
</div>
<div class="section" id="data-hosting">
<h2>Data hosting<a class="headerlink" href="#data-hosting" title="Permalink to this headline">¶</a></h2>
<p>Data from Ag3 are hosted by several different services.</p>
<p>Raw sequence reads, sequence read alignments and SNP calls are available for download from the European Nucleotide Archive (ENA). Further information on how to find and download these data is provided in the <a class="reference internal" href="download.html"><span class="doc std std-doc">data download guide</span></a>.</p>
<p>The SNP data have also been uploaded to Google Cloud, and can be analysed directly within the cloud without having to download or copy any data, including via free interactive computing services such as <a class="reference external" href="https://gke.mybinder.org/">MyBinder</a> and <a class="reference external" href="https://colab.research.google.com/">Google Colab</a>. Further information about analysing these data in the cloud is provided in the <a class="reference internal" href="cloud.html"><span class="doc std std-doc">cloud data access guide</span></a>.</p>
</div>
<div class="section" id="sample-sets">
<h2>Sample sets<a class="headerlink" href="#sample-sets" title="Permalink to this headline">¶</a></h2>
<p>The samples included in Ag3 have been organised into 26 sample sets. Each of these sample sets corresponds to a set of mosquito specimens from a contributing study. Depending on your objectives, you may want to access data from only specific sample sets, or all sample sets. Here is a list of the sample sets included in Ag3:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">malariagen_data</span>
<span class="n">ag3</span> <span class="o">=</span> <span class="n">malariagen_data</span><span class="o">.</span><span class="n">Ag3</span><span class="p">(</span><span class="s2">&quot;gs://vo_agam_release/&quot;</span><span class="p">)</span>
<span class="n">df_sample_sets</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">sample_sets</span><span class="p">()</span>
<span class="n">df_sample_sets</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_set</th>
      <th>sample_count</th>
      <th>release</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>AG1000G-AO</td>
      <td>81</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>1</th>
      <td>AG1000G-BF-A</td>
      <td>181</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>2</th>
      <td>AG1000G-BF-B</td>
      <td>102</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>3</th>
      <td>AG1000G-BF-C</td>
      <td>13</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>AG1000G-CD</td>
      <td>76</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>5</th>
      <td>AG1000G-CF</td>
      <td>73</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>6</th>
      <td>AG1000G-CI</td>
      <td>80</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>7</th>
      <td>AG1000G-CM-A</td>
      <td>303</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>8</th>
      <td>AG1000G-CM-B</td>
      <td>97</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>9</th>
      <td>AG1000G-CM-C</td>
      <td>44</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>10</th>
      <td>AG1000G-FR</td>
      <td>23</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>11</th>
      <td>AG1000G-GA-A</td>
      <td>69</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>12</th>
      <td>AG1000G-GH</td>
      <td>100</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>13</th>
      <td>AG1000G-GM-A</td>
      <td>74</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>14</th>
      <td>AG1000G-GM-B</td>
      <td>31</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>15</th>
      <td>AG1000G-GM-C</td>
      <td>174</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>16</th>
      <td>AG1000G-GN-A</td>
      <td>45</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>17</th>
      <td>AG1000G-GN-B</td>
      <td>185</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>18</th>
      <td>AG1000G-GQ</td>
      <td>10</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>19</th>
      <td>AG1000G-GW</td>
      <td>101</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>20</th>
      <td>AG1000G-KE</td>
      <td>86</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>21</th>
      <td>AG1000G-ML-A</td>
      <td>60</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>22</th>
      <td>AG1000G-ML-B</td>
      <td>71</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>23</th>
      <td>AG1000G-MW</td>
      <td>41</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>24</th>
      <td>AG1000G-MZ</td>
      <td>74</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>25</th>
      <td>AG1000G-TZ</td>
      <td>300</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>26</th>
      <td>AG1000G-UG</td>
      <td>290</td>
      <td>v3</td>
    </tr>
    <tr>
      <th>27</th>
      <td>AG1000G-X</td>
      <td>297</td>
      <td>v3</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>The sample set identifiers all start with “AG1000G-” followed by the two-letter code of the country from which samples were collected (e.g., “AO” is Angola). Where there are multiple sample sets from the same country, these have been given alphabetical suffixes, e.g., “AG1000G-BF-A”, “AG1000G-BF-B” and “AG1000G-BF-C” are three sample sets from Burkina Faso.</p>
<p>These country code suffixes are just a convenience to help remember which sample sets contain which data, please see the sample metadata for more precise location information. Note also that sample set AG1000G-GN-B contains samples from both Guinea and Mali.</p>
<p>Here is a more detailed breakdown of the samples contained within each sample set, summarised by country, year of collection, and species:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_samples</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">sample_metadata</span><span class="p">()</span>
<span class="n">df_summary</span> <span class="o">=</span> <span class="n">df_samples</span><span class="o">.</span><span class="n">pivot_table</span><span class="p">(</span>
    <span class="n">index</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;sample_set&quot;</span><span class="p">,</span> <span class="s2">&quot;country&quot;</span><span class="p">,</span> <span class="s2">&quot;year&quot;</span><span class="p">],</span> 
    <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;species&quot;</span><span class="p">],</span>
    <span class="n">values</span><span class="o">=</span><span class="s2">&quot;sample_id&quot;</span><span class="p">,</span> 
    <span class="n">aggfunc</span><span class="o">=</span><span class="nb">len</span><span class="p">,</span>
    <span class="n">fill_value</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">df_summary</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th></th>
      <th>species</th>
      <th>arabiensis</th>
      <th>coluzzii</th>
      <th>gambiae</th>
      <th>intermediate_arabiensis_gambiae</th>
      <th>intermediate_gambiae_coluzzii</th>
    </tr>
    <tr>
      <th>sample_set</th>
      <th>country</th>
      <th>year</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>AG1000G-AO</th>
      <th>Angola</th>
      <th>2009</th>
      <td>0</td>
      <td>81</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-BF-A</th>
      <th>Burkina Faso</th>
      <th>2012</th>
      <td>0</td>
      <td>82</td>
      <td>98</td>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>AG1000G-BF-B</th>
      <th>Burkina Faso</th>
      <th>2014</th>
      <td>3</td>
      <td>53</td>
      <td>46</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-BF-C</th>
      <th>Burkina Faso</th>
      <th>2004</th>
      <td>0</td>
      <td>0</td>
      <td>13</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-CD</th>
      <th>Democratic Republic of Congo</th>
      <th>2015</th>
      <td>0</td>
      <td>0</td>
      <td>76</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th rowspan="2" valign="top">AG1000G-CF</th>
      <th rowspan="2" valign="top">Central African Republic</th>
      <th>1993</th>
      <td>0</td>
      <td>5</td>
      <td>2</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1994</th>
      <td>0</td>
      <td>13</td>
      <td>53</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-CI</th>
      <th>Cote d'Ivoire</th>
      <th>2012</th>
      <td>0</td>
      <td>80</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-CM-A</th>
      <th>Cameroon</th>
      <th>2009</th>
      <td>0</td>
      <td>0</td>
      <td>303</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-CM-B</th>
      <th>Cameroon</th>
      <th>2005</th>
      <td>0</td>
      <td>7</td>
      <td>90</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-CM-C</th>
      <th>Cameroon</th>
      <th>2013</th>
      <td>2</td>
      <td>19</td>
      <td>23</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-FR</th>
      <th>Mayotte</th>
      <th>2011</th>
      <td>0</td>
      <td>0</td>
      <td>23</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-GA-A</th>
      <th>Gabon</th>
      <th>2000</th>
      <td>0</td>
      <td>0</td>
      <td>69</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-GH</th>
      <th>Ghana</th>
      <th>2012</th>
      <td>0</td>
      <td>64</td>
      <td>36</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-GM-A</th>
      <th>Gambia, The</th>
      <th>2011</th>
      <td>0</td>
      <td>5</td>
      <td>58</td>
      <td>0</td>
      <td>11</td>
    </tr>
    <tr>
      <th>AG1000G-GM-B</th>
      <th>Gambia, The</th>
      <th>2012</th>
      <td>0</td>
      <td>16</td>
      <td>9</td>
      <td>0</td>
      <td>6</td>
    </tr>
    <tr>
      <th>AG1000G-GM-C</th>
      <th>Gambia, The</th>
      <th>2012</th>
      <td>0</td>
      <td>148</td>
      <td>2</td>
      <td>0</td>
      <td>24</td>
    </tr>
    <tr>
      <th>AG1000G-GN-A</th>
      <th>Guinea</th>
      <th>2012</th>
      <td>0</td>
      <td>4</td>
      <td>40</td>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th rowspan="2" valign="top">AG1000G-GN-B</th>
      <th>Guinea</th>
      <th>2012</th>
      <td>0</td>
      <td>7</td>
      <td>83</td>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>Mali</th>
      <th>2012</th>
      <td>0</td>
      <td>28</td>
      <td>65</td>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>AG1000G-GQ</th>
      <th>Equatorial Guinea</th>
      <th>2002</th>
      <td>0</td>
      <td>0</td>
      <td>10</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-GW</th>
      <th>Guinea-Bissau</th>
      <th>2010</th>
      <td>0</td>
      <td>0</td>
      <td>29</td>
      <td>0</td>
      <td>72</td>
    </tr>
    <tr>
      <th rowspan="3" valign="top">AG1000G-KE</th>
      <th rowspan="3" valign="top">Kenya</th>
      <th>2000</th>
      <td>0</td>
      <td>0</td>
      <td>19</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2007</th>
      <td>3</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2012</th>
      <td>10</td>
      <td>0</td>
      <td>9</td>
      <td>0</td>
      <td>45</td>
    </tr>
    <tr>
      <th>AG1000G-ML-A</th>
      <th>Mali</th>
      <th>2014</th>
      <td>0</td>
      <td>27</td>
      <td>33</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-ML-B</th>
      <th>Mali</th>
      <th>2004</th>
      <td>2</td>
      <td>36</td>
      <td>33</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>AG1000G-MW</th>
      <th>Malawi</th>
      <th>2015</th>
      <td>41</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th rowspan="2" valign="top">AG1000G-MZ</th>
      <th rowspan="2" valign="top">Mozambique</th>
      <th>2003</th>
      <td>0</td>
      <td>0</td>
      <td>3</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2004</th>
      <td>0</td>
      <td>0</td>
      <td>71</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th rowspan="3" valign="top">AG1000G-TZ</th>
      <th rowspan="3" valign="top">Tanzania</th>
      <th>2012</th>
      <td>87</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2013</th>
      <td>1</td>
      <td>0</td>
      <td>36</td>
      <td>0</td>
      <td>6</td>
    </tr>
    <tr>
      <th>2015</th>
      <td>137</td>
      <td>0</td>
      <td>32</td>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>AG1000G-UG</th>
      <th>Uganda</th>
      <th>2012</th>
      <td>82</td>
      <td>0</td>
      <td>207</td>
      <td>1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>Note that there are also multiple sampling sites represented within some sample sets.</p>
</div>
<div class="section" id="further-reading">
<h2>Further reading<a class="headerlink" href="#further-reading" title="Permalink to this headline">¶</a></h2>
<p>Hopefully this page has provided a useful introduction to the Ag3 data resource. If you would like to start working with these data, please visit the <span class="xref myst">cloud data access guide</span> or the <span class="xref myst">data download guide</span> or continue browsing the other documentation on this site.</p>
<p>If you have any questions about the data and how to use them, please do get in touch by <a class="reference external" href="https://github.com/malariagen/vector-public-data/discussions/new">starting a new discussion</a> on the malariagen/vector-open-data repo on GitHub.</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./ag3"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="../landing-page.html" title="previous page">MalariaGEN vector data user guide</a>
    <a class='right-next' id="next-link" href="cloud.html" title="next page">Ag3 cloud data access guide</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By MalariaGEN<br/>
        
            &copy; Copyright 2021.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>