
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Ag3 sequencing and variant calling methods &#8212; MalariaGEN vector data user guide</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "malariagen/vector-data");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ag3 API" href="api.html" />
    <link rel="prev" title="Ag3 data downloads" href="download.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/logo.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">MalariaGEN vector data user guide</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../landing-page.html">
                    MalariaGEN vector data user guide
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Malaria Vector Genome Observatory
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../vobs/vobs-data-access.html">
   Vector Observatory Data Access
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Anopheles gambiae complex
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.0.html">
   Ag3.0 (Ag1000G phase 3)
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.1.html">
   Ag3.1
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.2.html">
   Ag3.2
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.3.html">
   Ag3.3
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.4.html">
   Ag3.4
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.5.html">
   Ag3.5
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.6.html">
   Ag3.6
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.7.html">
   Ag3.7
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.8.html">
   Ag3.8
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.9.html">
   Ag3.9
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.10.html">
   Ag3.10
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="ag3.11.html">
   Ag3.11
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="cloud.html">
   Ag3 cloud data access
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="download.html">
   Ag3 data downloads
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Ag3 sequencing and variant calling methods
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="api.html">
   Ag3 API
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="analysis.html">
   Ag3 analysis
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Anopheles funestus
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/af1.0.html">
   Af1.0  (
   <em>
    Anopheles funestus
   </em>
   Project Phase 1 Data Release)
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/af1.1.html">
   Af1.1
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/af1.2.html">
   Af1.2
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/af1.3.html">
   Af1.3
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/af1.4.html">
   Af1.4
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/cloud.html">
   Af1 cloud data access
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/download.html">
   Af1 data downloads
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/api.html">
   Af1 API
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Anopheles minimus
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../amin1/intro.html">
   Amin1.0
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../amin1/cloud.html">
     Amin1.0 cloud data access
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../amin1/download.html">
     Amin1.0 data downloads
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../amin1/api.html">
   Amin1 API
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Partners and community
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../studies-ag1000g.html">
   Ag1000G partner studies
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Release notes
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../release-notes.html">
   Release notes
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-repository-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Source repositories">
      <i class="fab fa-github"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://github.com/malariagen/vector-data"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Source repository"
>
  

<span class="headerbtn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="headerbtn__text-container">repository</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/malariagen/vector-data/issues/new?title=Issue%20on%20page%20%2Fag3/methods.html&body=Your%20issue%20content%20here."
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Open an issue"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="headerbtn__text-container">open issue</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="../_sources/ag3/methods.md"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.md</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#partner-studies-and-population-sampling">
   Partner studies and population sampling
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#whole-genome-sequencing">
   Whole-genome sequencing
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#alignment-and-snp-calling">
   Alignment and SNP calling
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-qc">
   Sample QC
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#coverage">
     Coverage
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cross-contamination">
     Cross-contamination
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#technical-replicates">
     Technical replicates
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#population-outliers-and-anomaly-detection">
     Population outliers and anomaly detection
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#colony-crosses">
     Colony crosses
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#sex-calling">
     Sex calling
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#species-assignment">
   Species assignment
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#species-calling-via-ancestry-informative-markers-aims">
     Species calling via ancestry-informative markers (AIMs)
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#species-calling-via-pca">
     Species calling via PCA
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-filtering">
   Site filtering
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#site-filters-for-use-with-an-gambiae-and-or-an-coluzzii">
     Site filters for use with
     <em>
      An. gambiae
     </em>
     and/or
     <em>
      An. coluzzii
     </em>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#site-filters-for-use-with-an-arabiensis">
     Site filters for use with
     <em>
      An. arabiensis
     </em>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#site-filters-for-joint-analyses-of-all-three-species">
     Site filters for joint analyses of all three species
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-calling">
   CNV calling
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#calculation-and-normalization-of-coverage">
     Calculation and normalization of coverage
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#hmm-inference-of-copy-number-state">
     HMM inference of copy number state
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genome-wide-cnv-discovery-and-filtering-coverage-calls">
     Genome-wide CNV discovery and filtering (“coverage calls”)
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#identifying-cnv-alleles-at-insecticide-resistance-loci-discordant-read-calls">
     Identifying CNV alleles at insecticide resistance loci (“discordant read calls”)
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#haplotype-phasing">
   Haplotype phasing
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#acknowledgments">
   Acknowledgments
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#appendices">
   Appendices
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#summary-statistics-used-as-input-to-site-filter-models">
     Summary statistics used as input to site filter models
    </a>
   </li>
  </ul>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Ag3 sequencing and variant calling methods</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#partner-studies-and-population-sampling">
   Partner studies and population sampling
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#whole-genome-sequencing">
   Whole-genome sequencing
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#alignment-and-snp-calling">
   Alignment and SNP calling
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sample-qc">
   Sample QC
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#coverage">
     Coverage
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cross-contamination">
     Cross-contamination
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#technical-replicates">
     Technical replicates
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#population-outliers-and-anomaly-detection">
     Population outliers and anomaly detection
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#colony-crosses">
     Colony crosses
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#sex-calling">
     Sex calling
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#species-assignment">
   Species assignment
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#species-calling-via-ancestry-informative-markers-aims">
     Species calling via ancestry-informative markers (AIMs)
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#species-calling-via-pca">
     Species calling via PCA
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-filtering">
   Site filtering
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#site-filters-for-use-with-an-gambiae-and-or-an-coluzzii">
     Site filters for use with
     <em>
      An. gambiae
     </em>
     and/or
     <em>
      An. coluzzii
     </em>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#site-filters-for-use-with-an-arabiensis">
     Site filters for use with
     <em>
      An. arabiensis
     </em>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#site-filters-for-joint-analyses-of-all-three-species">
     Site filters for joint analyses of all three species
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cnv-calling">
   CNV calling
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#calculation-and-normalization-of-coverage">
     Calculation and normalization of coverage
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#hmm-inference-of-copy-number-state">
     HMM inference of copy number state
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genome-wide-cnv-discovery-and-filtering-coverage-calls">
     Genome-wide CNV discovery and filtering (“coverage calls”)
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#identifying-cnv-alleles-at-insecticide-resistance-loci-discordant-read-calls">
     Identifying CNV alleles at insecticide resistance loci (“discordant read calls”)
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#haplotype-phasing">
   Haplotype phasing
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#acknowledgments">
   Acknowledgments
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#appendices">
   Appendices
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#summary-statistics-used-as-input-to-site-filter-models">
     Summary statistics used as input to site filter models
    </a>
   </li>
  </ul>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="ag3-sequencing-and-variant-calling-methods">
<h1>Ag3 sequencing and variant calling methods<a class="headerlink" href="#ag3-sequencing-and-variant-calling-methods" title="Permalink to this headline">#</a></h1>
<p>This page contains a description of methods used in phase 3 of the <em>Anopheles gambiae</em> 1000 Genomes Project (Ag1000G).</p>
<section id="partner-studies-and-population-sampling">
<h2>Partner studies and population sampling<a class="headerlink" href="#partner-studies-and-population-sampling" title="Permalink to this headline">#</a></h2>
<p>Ag1000G phase 3 includes data from 3,081 individual mosquitoes, including 2,784 mosquitoes collected from natural populations in 19 countries.
Three species are represented within the cohort: <em>An. gambiae</em>, <em>An. coluzzii</em> and <em>An. arabiensis</em>.
Further details of studies which contributed these samples and specimen collection locations and methods are provided in the <a class="reference internal" href="../studies-ag1000g.html"><span class="doc">Ag1000G partner studies</span></a> page.</p>
</section>
<section id="whole-genome-sequencing">
<h2>Whole-genome sequencing<a class="headerlink" href="#whole-genome-sequencing" title="Permalink to this headline">#</a></h2>
<p>All library preparation and sequencing was performed at the Wellcome
Sanger Institute.
Paired-end multiplex libraries were prepared using
the manufacturer’s protocol, with the exception that genomic DNA was
fragmented using Covaris Adaptive Focused Acoustics rather than
nebulization.
Multiplexes comprised 12 tagged individual mosquitoes and three lanes of sequencing were generated for each multiplex to
even out variations in yield between sequencing runs.
Cluster generation and sequencing were undertaken according to the
manufacturer’s protocol for paired-end sequence reads with insert size in the range 100–200 bp.
4,693 individual mosquitoes were sequenced in total, of which 3,130 were sequenced using the Illumina HiSeq 2000 platform and 1,563 were sequenced using the Illumina HiSeq X platform.
All individuals were sequenced to a target coverage of 30×.
The HiSeq 2000 sequencing runs generated 100 bp paired-end reads, and the HiSeq X sequencing runs generated 150 bp paired-end reads.
Library preparation and sequencing was performed at the Wellcome Sanger Institute by the Sample Logistics, Sequencing and Informatics facilities.</p>
</section>
<section id="alignment-and-snp-calling">
<h2>Alignment and SNP calling<a class="headerlink" href="#alignment-and-snp-calling" title="Permalink to this headline">#</a></h2>
<p>Reads were aligned to the AgamP4 reference genome using BWA version
0.7.15. Indel realignment was performed using GATK version 3.7-0
<code class="docutils literal notranslate"><span class="pre">RealignerTargetCreator</span></code> and <code class="docutils literal notranslate"><span class="pre">IndelRealigner</span></code>. Single nucleotide
polymorphisms were called using GATK version 3.7-0
<code class="docutils literal notranslate"><span class="pre">UnifiedGenotyper</span></code>. Genotypes were called for each sample
independently, in genotyping mode, given all possible alleles at all
genomic sites where the reference base was not “N”. Coverage was
capped at 250× by random down-sampling. Complete specifications of the
<a class="reference external" href="https://github.com/malariagen/pipelines/blob/v0.0.4/docs/specs/short-read-alignment-vector.md">alignment</a>
and
<a class="reference external" href="https://github.com/malariagen/pipelines/blob/v0.0.4/docs/specs/snp-genotyping-vector.md">genotyping</a>
pipelines are available from the <a class="reference external" href="https://github.com/malariagen/pipelines">malariagen/pipelines GitHub
repository</a>. Open source WDL
implementations of the
<a class="reference external" href="https://github.com/malariagen/pipelines/tree/v0.0.4/pipelines/short-read-alignment-vector">alignment</a>
and
<a class="reference external" href="https://github.com/malariagen/pipelines/tree/v0.0.4/pipelines/SNP-genotyping-vector">genotyping</a>
pipelines are also available from GitHub. Following successful
completion of these pipelines, samples entered the sample quality
control (QC) process described below. Alignment and SNP calling was performed by the MalariaGEN Resource Centre Team. Open source pipeline implementations were developed by the Broad Institute Data Engineering Team.</p>
</section>
<section id="sample-qc">
<h2>Sample QC<a class="headerlink" href="#sample-qc" title="Permalink to this headline">#</a></h2>
<p>The following subsections describe analyses performed to identify and exclude samples from the final dataset. Sample QC was performed by the MalariaGEN Resource Centre Team.</p>
<section id="coverage">
<h3>Coverage<a class="headerlink" href="#coverage" title="Permalink to this headline">#</a></h3>
<p>For each sample, depth of coverage was computed at all genome
positions. Samples were excluded if median coverage across all
chromosomes was less than 10×, or if less than 50% of the reference
genome was covered by at least 1×.</p>
</section>
<section id="cross-contamination">
<h3>Cross-contamination<a class="headerlink" href="#cross-contamination" title="Permalink to this headline">#</a></h3>
<p>To identify samples affected by cross-contamination, we implemented
the model for detecting contamination in NGS alignments described in
<a class="reference external" href="https://doi.org/10.1016/j.ajhg.2012.09.004">Jun et
al. (2012)</a>. Briefly, the
method estimates the likelihood of the observed alternate and
reference allele counts under different contamination fractions, given
approximate population allele frequencies. Population allele
frequencies were estimated from the <a class="reference external" href="https://www.malariagen.net/resource/27">Ag1000G phase 2 data
release</a>. The model computes a
maximum likelihood value for a parameter <em>alpha</em> representing
percentage contamination. Samples were excluded if <em>alpha</em> was 4.5% or
greater.</p>
</section>
<section id="technical-replicates">
<h3>Technical replicates<a class="headerlink" href="#technical-replicates" title="Permalink to this headline">#</a></h3>
<p>A number of samples were sequenced more than once within this project
phase (technical replicates). To create a final dataset without any
replicates suitable for population genetic analysis, we performed an
analysis to confirm all technical replicates, and to choose the sample
within each replicate with the best sequencing data. We computed
pairwise genetic distance between all sample pairs within a sample
set. The distance metric used was city block distance between genotype
allele counts, to allow for handling of multiallelic SNPs.  So, e.g.,
distance between genotypes of 0/1 and 0/1 is 0, distance between 0/0
and 0/1 is 2, distance between 0/1 and 1/2 is 2, distance between 0/0
and 1/1 is 4, etc. For each pair of samples, distance was averaged
over all sites where both samples had a non-missing genotype
call. Computations were initially carried out on a down-sampled set of
10 x 100,000 contiguous genomic sites, to be computationally
feasible. Where a pair of samples fell beneath a conservative
threshold of 0.012, the genetic distance was then recomputed across
all genomic sites (i.e., without down-sampling).  For each pair of
samples that were expected to be technical replicates according to our
metadata records, we excluded both members of the pair if genetic
distance was above 0.006.  Where an expected replicate pair had
genetic distance below 0.006, we retained only one sample in the pair.
We also identified and excluded both samples in any pair where genetic
distance was below 0.006, but the samples were not expected to be
replicates.</p>
</section>
<section id="population-outliers-and-anomaly-detection">
<h3>Population outliers and anomaly detection<a class="headerlink" href="#population-outliers-and-anomaly-detection" title="Permalink to this headline">#</a></h3>
<p>We used principal component analysis (PCA) to identify and exclude
individual samples that were population outliers. SNPs were
down-sampled to use 100,000 segregating non-singleton sites from
chromosomes 3R and 3L, to avoid regions complicated by known
introgression loci or paracentric inversions. PCA was computed using
scikit-allel version 1.2.0. We iteratively identified and excluded any
individual samples that were outliers along a single principal
component. We then identified and excluded any individual samples or
small sample groups that clustered together with other samples in a
way that was not plausible given metadata regarding their collection
location.</p>
</section>
<section id="colony-crosses">
<h3>Colony crosses<a class="headerlink" href="#colony-crosses" title="Permalink to this headline">#</a></h3>
<p>Samples in the AG1000G-X sample set were parents and progeny from
colony crosses and were subject to a slightly different set of QC
steps. For each cross, we performed an analysis of Mendelian
inheritance and consistency to confirm the true parents and the
validity of the cross. Not all crosses were able to be successfully
resolved, and samples that were not in a resolved cross were excluded.
From the samples originally submitted in the AG1000G-X sample set, 297
samples from 15 crosses were retained for release. We did not include
the colony crosses in the population outlier analysis due to their
relatedness.</p>
</section>
<section id="sex-calling">
<h3>Sex calling<a class="headerlink" href="#sex-calling" title="Permalink to this headline">#</a></h3>
<p>We called the sex of all samples based on the modal coverage ratio
between the X chromosome and the autosomal chromosome arm 3R. The
sample was classed as male where the coverage ratio was between
0.4-0.6, and female between 0.8-1.2. Where the ratio was outside these
limits, the sample was excluded. One of the sample sets from The
Gambia, AG1000G-GM-B, included whole-genome amplified (WGA) samples
which displayed some skew in their coverage ratios, which meant that
sex could not be called via the same process. These samples received a
sex call where possible, but no samples were excluded based on
uncertain sex call.</p>
</section>
</section>
<section id="species-assignment">
<h2>Species assignment<a class="headerlink" href="#species-assignment" title="Permalink to this headline">#</a></h2>
<p>We assigned a species to each individual that passed sample QC using
their genomic data, via two independent methods: ancestry-informative
markers (AIMs) and principal components analysis (PCA).</p>
<section id="species-calling-via-ancestry-informative-markers-aims">
<h3>Species calling via ancestry-informative markers (AIMs)<a class="headerlink" href="#species-calling-via-ancestry-informative-markers-aims" title="Permalink to this headline">#</a></h3>
<p>We derive AIMs between <em>An. arabiensis</em> and <em>An. gambiae</em> using
publicly available data from the <em>Anopheles</em> 16 genomes project
(<a class="reference external" href="https://doi.org/10.1126/science.1258522">Neafsey et
al. 2015</a>). Whole genome
SNP calls for 12 <em>An. arabiensis</em> and 38 <em>An. gambiae</em> individuals
were used. Alleles were mapped onto the same alternate allele space,
and allele frequencies were computed for both species. Sites that were
multiallelic in either group were excluded, as well as sites where any
genotypes were missing. 565,329 SNPs were identified as potentially
informative, where no shared alleles were present between
groups. These were spread throughout the genome, but were concentrated
on the X chromosome (63.2%), particularly around the Xag inversion. We
randomly down-sampled these SNPs to a set of 50,000 AIMs, then
computed the fraction of alleles at these SNPs that were
arabiensis-like for each individual in the Ag1000G phase 3 cohort.
Given the relatively small number of <em>An. arabiensis</em> samples in the
16 genomes project, it was clear that a significant proportion of
putative AIMs were not likely to be truly informative across the
broader sampling in Ag1000G phase 3. Individuals in Ag1000G were
classed as <em>An. arabiensis</em> where a fraction &gt;0.6 of alleles were
arabiensis-like.</p>
<p>To resolve the non-arabiensis individuals into <em>An. gambiae</em> and
<em>An. coluzzii</em>, we applied the AIMs previously used in <a class="reference external" href="https://www.malariagen.net/resource/27">Ag1000G phase
2</a>. For each individual, we
computed the fraction of coluzzii-like alleles at these
AIMs. Individuals were called as <em>An. gambiae</em> where this fraction was
&lt;0.12 and <em>An. coluzzii</em> where this fraction was &gt;0.9, with
individuals in between classed as intermediate.</p>
</section>
<section id="species-calling-via-pca">
<h3>Species calling via PCA<a class="headerlink" href="#species-calling-via-pca" title="Permalink to this headline">#</a></h3>
<p>To provide a complementary view of species assignments, we also used
the results of the principal components analysis of Chromosome 3
computed during the outlier analysis described above. Based on a
comparison with the AIM species calls, it was apparent that the first
two principal components could be used to assign species. Individuals
where PC1 &gt;150 were called as <em>An. arabiensis</em>.  Individuals where PC1
&lt;0 and PC2 &gt;-7 were called as <em>An. gambiae</em>.  Individuals where PC1 &lt;0
and PC2 &lt;-24 were called as <em>An. coluzzii</em>.  All other individuals
were called as intermediate. The results of the PCA and AIM species
calls were highly concordant in most sample sets, except for the Far
West (Guinea-Bissau, The Gambia) and Far East (Kenya,
Tanzania). Further investigation is required to resolve the species
status of these individuals.</p>
</section>
</section>
<section id="site-filtering">
<h2>Site filtering<a class="headerlink" href="#site-filtering" title="Permalink to this headline">#</a></h2>
<p>We developed filters that identify genomic sites where SNP calling and
genotyping is likely to be less reliable in one or more mosquito
species. To guide the design and calibration of the site filters, we
made use of the 15 colony crosses included in Ag1000G phase 3. Each
cross comprises two parents and up to 20 progeny, and thus it is
possible to identify sites where genotypes in one or more progeny are
not consistent with Mendelian inheritance (Mendelian errors). A small
number of Mendelian errors may be due to <em>de novo</em> mutation, but the
vast majority of Mendelian errors are likely to be due to errors in
sequencing, alignment or SNP calling. The general approach we took
was to use Mendelian consistency to identify sets of positive and
negative training sites, then used these to train a machine learning
model that classified all genomic sites as either PASS or FAIL. Site filtering was performed by the MalariaGEN Resource Centre Team.</p>
<section id="site-filters-for-use-with-an-gambiae-and-or-an-coluzzii">
<h3>Site filters for use with <em>An. gambiae</em> and/or <em>An. coluzzii</em><a class="headerlink" href="#site-filters-for-use-with-an-gambiae-and-or-an-coluzzii" title="Permalink to this headline">#</a></h3>
<p>All the 15 crosses involved <em>An. gambiae</em> and/or <em>An. coluzzii</em>
parents, and none of the crosses involved <em>An. arabiensis</em>, so we used
the crosses to first develop site filters suitable for use with
<em>An. gambiae</em> and/or <em>An. coluzzii</em>. Hereafter we refer to these
filters as the “gamb_colu” site filters. Five of the 15 crosses were
held out for validation, so performance could be evaluated
objectively. Sites were assigned to the positive training set where
all genotypes across all 10 crosses were called, and no Mendelian
errors were observed. Sites were assigned to negative training set
where one or more Mendelian errors were observed in any cross. All
other sites were not considered eligible for inclusion in model
training. A balanced training set was then generated containing
100,000 autosomal sites from each of the positive and negative
training sets.</p>
<p>The inputs to the machine learning model were a set of per-site
summary statistics computed from the sequence read alignments and SNP
genotypes across all wild-caught <em>An. gambiae</em> and <em>An. coluzzii</em>
individuals. These input summary statistics are described further in
the appendix. Male individuals were excluded from the summary
statistic calculations, so that the model could also be applied
without modification to the X chromosome. We used these summary
statistics, together with the positive and negative training sites, to
train a decision tree model. We initially trained a set of trees with
different hyperparameter values, exploring the depth of trees, and the
number of samples allowed at a terminal node. Each of these trees was
evaluated on an unbalanced set of sites randomly sampled from the
whole genome (2% of all sites, without replacement). Leaves of these
trees contained different proportions of positive and negative
training sites, and by increasing the cutoff for these proportions
required to label a leaf as PASS, we were able to compute the area
under the receiver operating curve (AUROC) for each set of
hyperparameter values. The best performing hyperparameter set based on
AUROC was selected as the final model, and the leaf classification
cutoff used was optimised based on the Youden statistic. The resulting
model was a decision tree of depth 8, where leaves were assigned to
PASS where &gt; 0.533 of training data in that leaf were positive
training sites. All sites in the genome were then assigned to PASS or
FAIL via this model.</p>
<p>The 5 remaining cross pedigrees were used to perform a final
evaluation of the approach. For each of these crosses, we computed the
Mendelian error rate (fraction of variants with one or more Mendelian
errors among progeny) before and after applying the site filters, to
provide five independent evaluation results. We also evaluated
performance on the X chromosome using heterozygote calls in males as
an error indicator. The fraction of variants with a heterozygous
genotype call in or more males was computed before and after applying
site filters. Male error rates were estimated from genotype calls with
a minimum Genotype Quality (GQ) value of 30. Performance of the
decision tree model was better than the hand-crafted site filters
created during the previous project phase (Ag1000G phase 2), with
lower Mendelian error rates, and a larger number of sites passing the
filter. Full performance metrics will be reported in a future
publication.</p>
</section>
<section id="site-filters-for-use-with-an-arabiensis">
<h3>Site filters for use with <em>An. arabiensis</em><a class="headerlink" href="#site-filters-for-use-with-an-arabiensis" title="Permalink to this headline">#</a></h3>
<p>To generate site filters for use with <em>An. arabiensis</em>, we recomputed
site summary statistics using only wild-caught <em>An. arabiensis</em>
individuals, then applied the decision tree model described
above. These filters, which we refer to as the “arab”” site filters,
are appropriate when working with <em>An. arabiensis</em> samples only.</p>
</section>
<section id="site-filters-for-joint-analyses-of-all-three-species">
<h3>Site filters for joint analyses of all three species<a class="headerlink" href="#site-filters-for-joint-analyses-of-all-three-species" title="Permalink to this headline">#</a></h3>
<p>We created site filters suitable for joint analysis of individuals
from all three species by taking the intersection of the “gamb_colu”
and the “arab” site filters. We refer to these filters as the
“gamb_colu_arab”” site filters.</p>
</section>
</section>
<section id="cnv-calling">
<h2>CNV calling<a class="headerlink" href="#cnv-calling" title="Permalink to this headline">#</a></h2>
<p>Copy number variant (CNV) calling methods followed those described in
<a class="reference external" href="https://genome.cshlp.org/content/29/8/1250.full#sec-8">Lucas et
al. (2019)</a>
with some adaptations to accommodate the different mosquito species
being analysed, described further below. CNV calling was performed by the Liverpool School of Tropical Medicine.</p>
<section id="calculation-and-normalization-of-coverage">
<h3>Calculation and normalization of coverage<a class="headerlink" href="#calculation-and-normalization-of-coverage" title="Permalink to this headline">#</a></h3>
<p>For each individual, we used
<a class="reference external" href="https://github.com/pysam-developers/pysam">pysam</a> to count the number
of aligned reads (coverage) in non-overlapping 300 bp windows over the
nuclear genome. The position of each read was considered to be its
alignment start point; thus, each read was only counted once.</p>
<p>Sequencing coverage can be biased by variation in local nucleotide
composition. To account for this, we computed a normalized coverage
from the read counts based on the expected coverage of each window
given its GC content (<a class="reference external" href="https://doi.org/10.1101/gr.114876.110">Abyzov et
al. 2011</a>). For each 300 bp
window we computed the percentage of (G+C) nucleotides to the nearest
percentage point within the reference sequence and then divided the
read counts in each window by the median read count over all autosomal
windows with the same (G+C) percentage. To minimize the impact of copy
number variation when calculating these normalizing constants, we
excluded windows from the calculation of mean read counts with &lt;90%
sites passing site filters or with &gt;50% reads aligned with zero
mapping quality. The normalized coverage values were then multiplied
by a factor of 2, so that genome regions with a normal diploid copy
number should have an expected normalized coverage of 2.</p>
<p>Before examining the normalized coverage data for evidence of copy
number variation, we applied two filters to exclude windows for which
coverage may be an unreliable indicator of copy number. The first
filter removed windows in which &gt;50% of reads were aligned with
mapping quality 0, which indicates that a read is mapped ambiguously
and could be mapped equally well to a different genomic location. This
filter was calculated separately for <em>An. arabiensis</em> and
(<em>An. gambiae</em> + <em>An. coluzzii</em>). The second filter removed windows
for which the percentage (G+C) content was extreme and rarely
represented within the accessible reference sequence, that is, fewer
than 100 accessible windows with the same (G+C) percentage, because
the small number of windows makes the calculation of a (G+C)
normalizing constant unreliable. Windows retained for analysis were
referred to as “filtered windows”. The filtered windows were computed
separately for <em>An. arabiensis</em> and for (<em>An. gambiae</em> +
<em>An. coluzzii</em>), based on the AIM species calls, to account for
differences between the species.  One individual with an intermediate
species call between these two groups were excluded.</p>
</section>
<section id="hmm-inference-of-copy-number-state">
<h3>HMM inference of copy number state<a class="headerlink" href="#hmm-inference-of-copy-number-state" title="Permalink to this headline">#</a></h3>
<p>To infer the most likely copy number state (CN) at each window in each
individual, we applied a Gaussian hidden Markov model (HMM) to the
individual’s normalized windowed coverage data, following a similar
approach to <a class="reference external" href="https://doi.org/10.1101/gr.203711.115">Miles et
al. (2016)</a> and <a class="reference external" href="https://doi.org/10.1126/science.aam6393">Leffler et
al. (2017)</a>. The HMM was
implemented using the <code class="docutils literal notranslate"><span class="pre">GaussianHMM</span></code> function from
<a class="reference external" href="https://hmmlearn.readthedocs.io/en/latest/">hmmlearn</a>. The HMM
contained 13 hidden states (<em>c</em>), representing CN from 0 to 12 in
increments of 1, allowing the detection of up to 6-fold amplication of
a genetic region (the normal diploid complement of two copies of a
genetic region is represented by a CN of 2, a single duplication on
one chromosome is represented by 3, and so on). The Gaussian emission
probability distribution for each copy number state <em>n</em> had a mean
<em>c<sub>n</sub></em> (<em>c<sub>n</sub></em> = <em>n</em>), with variance <em>v<sub>n</sub></em>
= 0.01 + <em>a<sub>n</sub>c<sub>n</sub></em>, where <em>a<sub>n</sub></em> is the
variance in normalised coverage for all autosomal windows, excluding
the top 1% of windows with the highest normalised coverage. We
determined the variance empirically for each individual because
variance in coverage can differ between individuals, presumably due to
stochastic variation in library preparation and/or sequencing
runs. Following <a class="reference external" href="https://genome.cshlp.org/content/29/8/1250.full#sec-8">Lucas et
al. (2019)</a> we
set the HMM transition probability <em>t</em> = 0.00001. After parameter
calibration, we fitted a Gaussian HMM to normalised windowed coverage
data for each individual, to obtain a predicted copy number state
within each window.</p>
</section>
<section id="genome-wide-cnv-discovery-and-filtering-coverage-calls">
<h3>Genome-wide CNV discovery and filtering (“coverage calls”)<a class="headerlink" href="#genome-wide-cnv-discovery-and-filtering-coverage-calls" title="Permalink to this headline">#</a></h3>
<p>Using the results of the HMM, we obtained a set of CNV calls for each
individual by locating contiguous runs of at least five windows with
amplified copy number (CN &gt; 2, or CN &gt; 1 for Chromosome X in
males). This set of per-individual CNV calls was filtered by computing
likelihoods for each CNV call for both the copy number state predicted
by the HMM and for a null model of copy number = 2, and removing CNV
calls for which the likelihood ratio was &lt;1000.</p>
<p>From the per-individual CNV call set, we created a merged set CNV
calls. We first removed individuals with high coverage variance, where
the variance in normalized coverage was greater than 0.2, because high
variance could lead to erratic CNV calls. We then clustered the CNV
calls across individuals, merging two CNV calls into the same variant
if their breakpoints (inferred from the change in CN state) occurred
within one 300 bp window of each other. Merged CNVs were annotated to
record a confidence interval for the start and end breakpoints,
calculated as the 5-95 percentiles within each cluster. A filter
annotation was added to the final output VCF file where this
confidence interval was greater than 1200 bp for either the start or
end breakpoint.</p>
<p>The CNV merging process was performed separately for wild-caught
<em>An. arabiensis</em>, wild-caught (<em>An. gambiae</em> + <em>An. coluzzii</em>), and
the crosses, producing three sets of “CNV coverage calls”.</p>
</section>
<section id="identifying-cnv-alleles-at-insecticide-resistance-loci-discordant-read-calls">
<h3>Identifying CNV alleles at insecticide resistance loci (“discordant read calls”)<a class="headerlink" href="#identifying-cnv-alleles-at-insecticide-resistance-loci-discordant-read-calls" title="Permalink to this headline">#</a></h3>
<p>We characterized in detail the different duplication events (CNV
alleles) at six loci containing genes of particular interest
(Cyp6aa1–Cyp6p2, Gstu4–Gste3, Cyp6m2–Cyp6m4, Cyp6z3–Cyp6z1, Cyp9k1,
Ace1) using their unique patterns of discordant read pairs and reads
crossing the CNV breakpoint (“breakpoint reads”). We manually
inspected the six regions of interest in all individuals to identify
patterns of discordant and breakpoint reads (“diagnostic reads”)
consistently associated with changes in coverage. The start and end
point of each CNV allele could usually be precisely determined by the
breakpoint reads and was otherwise determined by discordant read pairs
or the point of change in coverage. Once the diagnostic reads were
identified for a CNV allele, we recorded the presence of that allele
in all samples with at least four supporting diagnostic reads.</p>
</section>
</section>
<section id="haplotype-phasing">
<h2>Haplotype phasing<a class="headerlink" href="#haplotype-phasing" title="Permalink to this headline">#</a></h2>
<p>Genotypes at biallelic SNPs passing site filters were phased into haplotypes using a combination of read-backed and statistical phasing. Read-backed phasing was performed for each individual using <a class="reference external" href="https://whatshap.readthedocs.io/en/latest/">WhatsHap</a> version 1.0. Statistical phasing was then performed using <a class="reference external" href="https://odelaneau.github.io/shapeit4/">SHAPEIT4</a> version 4.2.1. An <a class="reference external" href="https://github.com/malariagen/pipelines/tree/master/pipelines/phasing-vector/gcp">open source WDL implementation of this phasing pipeline</a> is available from the <a class="reference external" href="https://github.com/malariagen/pipelines">malariagen/pipelines GitHub repository</a>. The phasing pipeline was developed by the Broad Institute Data Engineering Team and run using the <a class="reference external" href="https://app.terra.bio/">Terra platform</a>.</p>
<p>To allow for a range of different analyses, three different haplotype reference panels were constructed, each using a different subset of samples and applying different site filters:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">gamb_colu_arab</span></code> - This haplotype reference panel included all wild-caught samples, and phased biallelic SNPs passing the <code class="docutils literal notranslate"><span class="pre">gamb_colu_arab</span></code> site filters.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">gamb_colu</span></code> - This haplotype reference panel included all wild-caught samples assigned as either <em>An. gambiae</em>, <em>An. coluzzii</em> or intermediate <em>An. gambiae/An. coluzzii</em> via the AIM species calling method, and phased biallelic SNPs passing the <code class="docutils literal notranslate"><span class="pre">gamb_colu</span></code> site filters.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">arab</span></code> - This haplotype reference panel included all wild-caught samples assigned as <em>An. arabiensis</em> via the AIM species calling method, and phased biallelic SNPs passing the <code class="docutils literal notranslate"><span class="pre">arab</span></code> site filters.</p></li>
</ul>
</section>
<section id="acknowledgments">
<h2>Acknowledgments<a class="headerlink" href="#acknowledgments" title="Permalink to this headline">#</a></h2>
<p>We would like to thank the staff of the Wellcome Sanger Institute
Sample Logistics, Sequencing and Informatics facilities for their
contributions to the production of this data release.</p>
<p>We would like to thank the members of the Data Engineering team of the
Broad Institute of Harvard and MIT for their work on open source
implementations of the alignment, SNP calling and haplotype phasing pipelines used in Ag1000G phase 3.</p>
</section>
<section id="appendices">
<h2>Appendices<a class="headerlink" href="#appendices" title="Permalink to this headline">#</a></h2>
<section id="summary-statistics-used-as-input-to-site-filter-models">
<h3>Summary statistics used as input to site filter models<a class="headerlink" href="#summary-statistics-used-as-input-to-site-filter-models" title="Permalink to this headline">#</a></h3>
<ul class="simple">
<li><p><strong>No Coverage</strong> – Number of samples with no coverage whatsoever at
the given position.</p></li>
<li><p><strong>Low Coverage</strong> – Number of samples with low depth of coverage at
the given position (less than half modal coverage for whole
chromosome).</p></li>
<li><p><strong>Low Coverage GC normalised</strong> – Number of samples with low depth
of GC-normalised coverage at the given position (less than half
modal GC-normalised coverage for whole chromosome).</p></li>
<li><p><strong>High Coverage</strong> – Number of samples with high depth of coverage
at the given position (more than twice modal coverage for the whole
chromosome).</p></li>
<li><p><strong>High Coverage GC normed</strong> – Number of samples with high depth of
GC-normalised coverage at the given position (more than twice modal
GC-normalised coverage for the whole chromosome).</p></li>
<li><p><strong>Mean GQ</strong> – Mean of genotype quality across samples.</p></li>
<li><p><strong>Median GQ</strong> – Median of genotype quality across samples.</p></li>
<li><p><strong>Variance GQ</strong> – Variance of genotype quality across samples.</p></li>
<li><p><strong>Low GQ 30</strong> – Number of samples with a genotype quality &lt;30.</p></li>
<li><p><strong>Low GQ 10</strong> – Number of samples with a genotype quality &lt;10.</p></li>
<li><p><strong>Low MQ</strong> – Number of samples with root-mean-square mapping
quality &lt;30.</p></li>
<li><p><strong>Median MQ</strong> – Median of root-mean-square mapping quality across
samples.</p></li>
<li><p><strong>Mean MQ</strong> – Mean of root-mean-square mapping quality across
samples.</p></li>
<li><p><strong>Variance MQ</strong> – Variance of root-mean-square mapping quality
across samples.</p></li>
<li><p><strong>Error Fraction</strong> – Fraction of samples with an allele not called
in the genotype.</p></li>
<li><p><strong>Allele Balance Het</strong> – Product of binomial probability of read
counts at het calls.</p></li>
</ul>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./ag3"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="download.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Ag3 data downloads</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="api.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Ag3 API</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By MalariaGEN<br/>
  
      &copy; Copyright 2021.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>