
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Single nucleotide polymorphisms (SNPs) &#8212; MalariaGEN vector data user guide</title>
    
  <link href="../../_static/css/theme.css" rel="stylesheet" />
  <link href="../../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/sphinx-book-theme.e8f53015daec13862f6db5e763c41738.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/togglebutton.js"></script>
    <script src="../../_static/clipboard.min.js"></script>
    <script src="../../_static/copybutton.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "malariagen/vector-data");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Genes and SNP effects" href="snp-effects.html" />
    <link rel="prev" title="Mosquito collection sites" href="collection-sites.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../../index.html">
      
      <img src="../../_static/logo.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">MalariaGEN vector data user guide</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../landing-page.html">
   MalariaGEN vector data user guide
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Ag3 (Ag1000G phase 3)
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../intro.html">
   Introduction to Ag3
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cloud.html">
   Ag3 cloud data access
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../download.html">
   Ag3 data downloads
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../methods.html">
   Ag3 sequencing and variant calling methods
  </a>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="../examples.html">
   Ag3 analysis examples
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="collection-sites.html">
     Mosquito collection sites
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     Single nucleotide polymorphisms (SNPs)
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="snp-effects.html">
     Genes and SNP effects
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="pca.html">
     Investigating population structure with PCA
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="cnv-explore.html">
     Exploring copy number variation (CNV)
    </a>
   </li>
  </ul>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../_sources/ag3/examples/snps.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/malariagen/vector-data"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/malariagen/vector-data/issues/new?title=Issue%20on%20page%20%2Fag3/examples/snps.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://gke.mybinder.org/v2/gh/malariagen/vector-data/master?urlpath=lab/tree/docs/ag3/examples/snps.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="../../_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        <a class="colab-button" href="https://colab.research.google.com/github/malariagen/vector-data/blob/master/docs/ag3/examples/snps.ipynb"><button type="button" class="btn btn-secondary topbarbtn"
                title="Launch Colab" data-toggle="tooltip" data-placement="left"><img class="colab-button-logo"
                    src="../../_static/images/logo_colab.png"
                    alt="Interact on Colab">Colab</button></a>
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#setup">
   Setup
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#preface-navigating-the-reference-genome">
   Preface: navigating the reference genome
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snp-positions-and-alleles">
   SNP positions and alleles
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#genotypes">
   Genotypes
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#allele-counts">
   Allele counts
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#site-filters">
   Site filters
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#locating-snps">
   Locating SNPs
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#locating-a-single-snp">
     Locating a single SNP
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#locating-snps-in-a-genome-region">
     Locating SNPs in a genome region
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#locating-samples">
   Locating samples
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#locating-a-single-sample">
     Locating a single sample
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#locating-multiple-samples">
     Locating multiple samples
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#selecting-sites-and-samples">
   Selecting sites and samples
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#differences-from-previous-data-releases">
   Differences from previous data releases
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genotyping-at-all-sites">
     Genotyping at all sites
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#predefined-alternate-alleles">
     Predefined alternate alleles
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#id1">
     Site filters
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#further-reading">
   Further reading
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#api-docs">
   API docs
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="single-nucleotide-polymorphisms-snps">
<h1>Single nucleotide polymorphisms (SNPs)<a class="headerlink" href="#single-nucleotide-polymorphisms-snps" title="Permalink to this headline">¶</a></h1>
<p>This notebook looks in more detail at how single nucleotide polymorphisms (SNPs) are represented in the Ag1000G phase 3 (Ag3) data release. We’ll also look at the different site filters that we’ve created to help select high quality SNPs for robust population genetic analyses. As always, this notebook is executable, click the launch icon (<i class="fas fa-rocket"></i>) at the top of the page and select a cloud computing service if you’d like to try running it for yourself.</p>
<div class="section" id="setup">
<h2>Setup<a class="headerlink" href="#setup" title="Permalink to this headline">¶</a></h2>
<p>Install some packages. You can skip this step if running on MyBinder, these will be already installed.</p>
<div class="cell tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>!pip install -q \
    zarr==2.6.1 \
    fsspec==0.8.7 \
    gcsfs==0.7.2 \
    dask==2021.03.0 \
    xarray==0.18.0 \
    scikit-allel==1.3.5 \
    malariagen_data==0.8.0 \
    seaborn matplotlib
</pre></div>
</div>
</div>
</div>
<p>Import the packages we’ll need.</p>
<div class="cell tag_thebe-init docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">Counter</span>
<span class="kn">from</span> <span class="nn">functools</span> <span class="kn">import</span> <span class="n">lru_cache</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">dask</span>
<span class="kn">import</span> <span class="nn">dask.array</span> <span class="k">as</span> <span class="nn">da</span>
<span class="c1"># silence some dask warnings</span>
<span class="n">dask</span><span class="o">.</span><span class="n">config</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="o">**</span><span class="p">{</span><span class="s1">&#39;array.slicing.split_large_chunks&#39;</span><span class="p">:</span> <span class="kc">True</span><span class="p">})</span>
<span class="kn">import</span> <span class="nn">matplotlib</span> <span class="k">as</span> <span class="nn">mpl</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="o">%</span><span class="k">matplotlib</span> inline
<span class="o">%</span><span class="k">config</span> InlineBackend.figure_format = &#39;retina&#39;
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">allel</span>
<span class="kn">from</span> <span class="nn">dask.diagnostics.progress</span> <span class="kn">import</span> <span class="n">ProgressBar</span>
<span class="kn">import</span> <span class="nn">malariagen_data</span>
</pre></div>
</div>
</div>
</div>
<p>Set up access to the Ag3 data.</p>
<div class="cell tag_thebe-init docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ag3</span> <span class="o">=</span> <span class="n">malariagen_data</span><span class="o">.</span><span class="n">Ag3</span><span class="p">(</span><span class="s2">&quot;gs://vo_agam_release/&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="preface-navigating-the-reference-genome">
<h2>Preface: navigating the reference genome<a class="headerlink" href="#preface-navigating-the-reference-genome" title="Permalink to this headline">¶</a></h2>
<p>In case you’re not already familiar with the <em>Anopheles gambiae</em> genome, here’s a quick primer on how it’s structured and how we refer to specific sites (positions) within the genome.</p>
<p>For Ag1000G analyses we use the AgamP4 reference genome. This genome is comprised of five main sequences:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ag3</span><span class="o">.</span><span class="n">contigs</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(&#39;2R&#39;, &#39;2L&#39;, &#39;3R&#39;, &#39;3L&#39;, &#39;X&#39;)
</pre></div>
</div>
</div>
</div>
<p>Each of these identifiers (e.g., “2R”) refers to a sequence of nucleotides, i.e., a sequence of the letters “A”, “C”, “T” and “G”. Let’s see how long these sequences are.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">contig</span> <span class="ow">in</span> <span class="n">ag3</span><span class="o">.</span><span class="n">contigs</span><span class="p">:</span>
    <span class="n">seq</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">genome_sequence</span><span class="p">(</span><span class="n">contig</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">contig</span><span class="si">}</span><span class="s2">: </span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span><span class="si">:</span><span class="s2">,</span><span class="si">}</span><span class="s2"> bp&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2R: 61,545,105 bp
2L: 49,364,325 bp
3R: 53,200,684 bp
3L: 41,963,435 bp
X: 24,393,108 bp
</pre></div>
</div>
</div>
</div>
<p>Let’s take a peek at the first 100 letters in one of the sequences (2R).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">seq</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">genome_sequence</span><span class="p">(</span><span class="s2">&quot;2R&quot;</span><span class="p">)</span>
<span class="n">seq</span><span class="p">[:</span><span class="mi">100</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span><span class="o">.</span><span class="n">tobytes</span><span class="p">()</span><span class="o">.</span><span class="n">decode</span><span class="p">()</span><span class="o">.</span><span class="n">upper</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;CTCTAAACATTAATAAAACCAAATACATGATCATATCAAACAAAAATAATCAAGAACAGCTGTGTATTTCAATAAATCAAGAACATATAGACAATACTTC&#39;
</pre></div>
</div>
</div>
</div>
<p>Here “bp” is short for base pair, which is the unit of length of nucleotide sequences. I.e., if a sequence is 10 bp then it is 10 nucleotides (letters) long.</p>
<p>In the case of <em>Anopheles gambiae</em>, these sequences correspond to chromosome arms. “2R” and “2L” are the two parts (arms) of Chromosome 2. “3R” and “3L” are the arms of Chromosome 3. “X” is the X chromosome. The X chromosome is a sex chromosome, present in two copies in females and one copy in males. Chromosomes 2 and 3 are autosomes, meaning that both males and females carry two copies.</p>
<p>To refer to a specific site (position) within a sequence, we typically use 1-based coordinates. E.g., “2R:10” refers to the tenth nucleotide (letter) in the 2R sequence. We can access this nucleotide via Python, but when indexing in Python we need to remember to use zero-based indices. E.g., access the nucleotide at position 10 on chromosome arm 2R:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">seq</span><span class="p">[</span><span class="mi">9</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span><span class="o">.</span><span class="n">tobytes</span><span class="p">()</span><span class="o">.</span><span class="n">decode</span><span class="p">()</span><span class="o">.</span><span class="n">upper</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;T&#39;
</pre></div>
</div>
</div>
</div>
<p>This nucleotide is a “T” (thymine).</p>
<p>Note that there are also some “gaps” in the reference genome, where we don’t know what the nucleotide is. These are represented in the sequence with the letter “N”. Let’s take a look at how common these are, for chromosome arm 2R:</p>
<div class="cell tag_thebe-init docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">count_nucleotides</span><span class="p">(</span><span class="n">contig</span><span class="p">):</span>
    <span class="n">seq</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">genome_sequence</span><span class="p">(</span><span class="n">contig</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
    <span class="k">return</span> <span class="n">Counter</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span><span class="o">.</span><span class="n">most_common</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">count_nucleotides</span><span class="p">(</span><span class="s2">&quot;2R&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[(b&#39;T&#39;, 14379951),
 (b&#39;A&#39;, 14377315),
 (b&#39;C&#39;, 12197820),
 (b&#39;G&#39;, 12183243),
 (b&#39;t&#39;, 2077134),
 (b&#39;a&#39;, 2060370),
 (b&#39;g&#39;, 1430905),
 (b&#39;c&#39;, 1425715),
 (b&#39;n&#39;, 1412612),
 (b&#39;N&#39;, 40)]
</pre></div>
</div>
</div>
</div>
<p>There are two things to notice here. First, there are both upper case and lower case letters. For most purposes you can ignore this, e.g., both “T” and “t” mean the same thing. Second, there are 1,412,652 nucleotides which are unknown (either “N” or “n”).</p>
<p>To gain a bit more intuition for how the genome is structured and the relative sizes of the different sequences, let’s make a plot of the five chromosome arms, summarising the relative abundance of the different nucleotides.</p>
<div class="cell tag_thebe-init docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">plot_sequence_composition</span><span class="p">(</span><span class="n">seq_id</span><span class="p">,</span> <span class="n">window_size</span><span class="o">=</span><span class="mi">100_000</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="kc">None</span><span class="p">):</span>
    
    <span class="c1"># load reference sequence</span>
    <span class="n">seq</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">genome_sequence</span><span class="p">(</span><span class="n">seq_id</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
    
    <span class="k">if</span> <span class="n">ax</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="c1"># make the figure size relative to largest contig</span>
        <span class="n">figw</span> <span class="o">=</span> <span class="mi">11</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span> <span class="o">/</span> <span class="nb">len</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">genome_sequence</span><span class="p">(</span><span class="s2">&quot;2R&quot;</span><span class="p">))</span>
        <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="n">figw</span><span class="p">,</span> <span class="mi">1</span><span class="p">))</span>
 
    <span class="c1"># convert to upper-case</span>
    <span class="n">seq</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">char</span><span class="o">.</span><span class="n">upper</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span>

    <span class="c1"># locate nucleotides</span>
    <span class="n">is_a</span> <span class="o">=</span> <span class="n">seq</span> <span class="o">==</span> <span class="sa">b</span><span class="s1">&#39;A&#39;</span>
    <span class="n">is_c</span> <span class="o">=</span> <span class="n">seq</span> <span class="o">==</span> <span class="sa">b</span><span class="s1">&#39;C&#39;</span>
    <span class="n">is_g</span> <span class="o">=</span> <span class="n">seq</span> <span class="o">==</span> <span class="sa">b</span><span class="s1">&#39;G&#39;</span>
    <span class="n">is_t</span> <span class="o">=</span> <span class="n">seq</span> <span class="o">==</span> <span class="sa">b</span><span class="s1">&#39;T&#39;</span>
    <span class="n">is_n</span> <span class="o">=</span> <span class="n">seq</span> <span class="o">==</span> <span class="sa">b</span><span class="s1">&#39;N&#39;</span>
    <span class="c1"># check there&#39;s nothing unexpected</span>
    <span class="n">is_other</span> <span class="o">=</span> <span class="o">~</span><span class="n">is_a</span> <span class="o">&amp;</span> <span class="o">~</span><span class="n">is_c</span> <span class="o">&amp;</span> <span class="o">~</span><span class="n">is_g</span> <span class="o">&amp;</span> <span class="o">~</span><span class="n">is_t</span> <span class="o">&amp;</span> <span class="o">~</span><span class="n">is_n</span>
    <span class="k">assert</span> <span class="n">np</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">is_other</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span>

    <span class="c1"># construct windows</span>
    <span class="n">bins</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">seq</span><span class="p">),</span> <span class="n">window_size</span><span class="p">)</span>

    <span class="c1"># count nucleotides in each window</span>
    <span class="n">h_a</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">histogram</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nonzero</span><span class="p">(</span><span class="n">is_a</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="n">bins</span><span class="p">)</span>
    <span class="n">h_c</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">histogram</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nonzero</span><span class="p">(</span><span class="n">is_c</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="n">bins</span><span class="p">)</span>
    <span class="n">h_g</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">histogram</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nonzero</span><span class="p">(</span><span class="n">is_g</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="n">bins</span><span class="p">)</span>
    <span class="n">h_t</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">histogram</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nonzero</span><span class="p">(</span><span class="n">is_t</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="n">bins</span><span class="p">)</span>
    <span class="n">h_n</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">histogram</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">nonzero</span><span class="p">(</span><span class="n">is_n</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="n">bins</span><span class="p">)</span>

    <span class="c1"># plot frequence of nucleotides within each bin</span>
    <span class="n">left</span> <span class="o">=</span> <span class="n">bins</span><span class="p">[:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
    <span class="n">bottom</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">width</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">diff</span><span class="p">(</span><span class="n">bins</span><span class="p">)</span>
    <span class="n">palette</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">color_palette</span><span class="p">(</span><span class="s1">&#39;colorblind&#39;</span><span class="p">)</span>
    <span class="n">colors</span> <span class="o">=</span> <span class="p">[</span><span class="n">palette</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="p">[</span><span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">8</span><span class="p">]]</span> <span class="o">+</span> <span class="p">[</span><span class="s1">&#39;k&#39;</span><span class="p">]</span>
    <span class="k">for</span> <span class="n">h</span><span class="p">,</span> <span class="n">c</span><span class="p">,</span> <span class="n">l</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">([</span><span class="n">h_a</span><span class="p">,</span> <span class="n">h_t</span><span class="p">,</span> <span class="n">h_g</span><span class="p">,</span> <span class="n">h_c</span><span class="p">,</span> <span class="n">h_n</span><span class="p">],</span> <span class="n">colors</span><span class="p">,</span> <span class="s1">&#39;ATGCN&#39;</span><span class="p">):</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">bar</span><span class="p">(</span><span class="n">left</span><span class="p">,</span> <span class="n">h</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="n">width</span><span class="p">,</span> <span class="n">bottom</span><span class="o">=</span><span class="n">bottom</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">c</span><span class="p">,</span> <span class="n">align</span><span class="o">=</span><span class="s1">&#39;edge&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="n">l</span><span class="p">)</span>
        <span class="n">bottom</span> <span class="o">+=</span> <span class="n">h</span>
        
    <span class="c1"># tidy up plot</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">seq</span><span class="p">))</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">())</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticklabels</span><span class="p">([</span><span class="s1">&#39;0%&#39;</span><span class="p">,</span> <span class="s1">&#39;100%&#39;</span><span class="p">])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">seq_id</span><span class="p">)</span>
    <span class="c1"># convert X axis to Mbp</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">mpl</span><span class="o">.</span><span class="n">ticker</span><span class="o">.</span><span class="n">FuncFormatter</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">,</span> <span class="n">p</span><span class="p">:</span> <span class="n">x</span><span class="o">//</span><span class="mf">1e6</span><span class="p">))</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Position (Mbp)&quot;</span><span class="p">)</span>
    
    <span class="c1"># add centromere/telomere annotation</span>
    <span class="k">if</span> <span class="n">seq_id</span> <span class="ow">in</span> <span class="p">{</span><span class="s1">&#39;2L&#39;</span><span class="p">,</span> <span class="s1">&#39;3L&#39;</span><span class="p">}:</span>
        <span class="n">ltxt</span> <span class="o">=</span> <span class="s2">&quot;centromere&quot;</span>
        <span class="n">rtxt</span> <span class="o">=</span> <span class="s2">&quot;telomere&quot;</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">ltxt</span> <span class="o">=</span> <span class="s2">&quot;telomere&quot;</span>
        <span class="n">rtxt</span> <span class="o">=</span> <span class="s2">&quot;centromere&quot;</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="n">ltxt</span><span class="p">,</span> <span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span> <span class="n">xycoords</span><span class="o">=</span><span class="s2">&quot;axes fraction&quot;</span><span class="p">,</span> 
                <span class="n">xytext</span><span class="o">=</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">2</span><span class="p">),</span> <span class="n">textcoords</span><span class="o">=</span><span class="s1">&#39;offset points&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="n">rtxt</span><span class="p">,</span> <span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span> <span class="n">xycoords</span><span class="o">=</span><span class="s2">&quot;axes fraction&quot;</span><span class="p">,</span> 
                <span class="n">xytext</span><span class="o">=</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">2</span><span class="p">),</span> <span class="n">textcoords</span><span class="o">=</span><span class="s1">&#39;offset points&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>
    
    <span class="c1"># add legend</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="s1">&#39;center left&#39;</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mf">.5</span><span class="p">),</span> 
              <span class="n">prop</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">family</span><span class="o">=</span><span class="s1">&#39;monospace&#39;</span><span class="p">),</span> <span class="n">ncol</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
              <span class="n">frameon</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
        
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_sequence_composition</span><span class="p">(</span><span class="s1">&#39;2R&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_20_0.png" src="../../_images/snps_20_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_sequence_composition</span><span class="p">(</span><span class="s1">&#39;2L&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_21_0.png" src="../../_images/snps_21_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_sequence_composition</span><span class="p">(</span><span class="s1">&#39;3R&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_22_0.png" src="../../_images/snps_22_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_sequence_composition</span><span class="p">(</span><span class="s1">&#39;3L&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_23_0.png" src="../../_images/snps_23_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_sequence_composition</span><span class="p">(</span><span class="s1">&#39;X&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_24_0.png" src="../../_images/snps_24_0.png" />
</div>
</div>
<p>A few points to note about the plots above:</p>
<ol class="simple">
<li><p>All of the chromosome arms are made up of A, C, T and G in roughly equal proportion, although A and T are a little more common than C and G.</p></li>
<li><p>We annotated each chromosome arm with “<a class="reference external" href="https://www.genome.gov/genetics-glossary/Centromere">centromere</a>” and “<a class="reference external" href="https://www.genome.gov/genetics-glossary/Telomere">telomere</a>”. You probably don’t need to know what these mean, but it is worth knowing that the sequences at these two different ends of the chromosome tend to be quite different, especially towards the centromere.</p></li>
<li><p>Gaps (N) are relatively uncommon, except towards the centromere of all chromosome arms where they are more common. This is typically because sequences are more repetitive towards the centromeres and so assembling the genome is more difficult.</p></li>
<li><p>We made the size of each figure proportional to the length of the sequence. This was to help give a feeling for the relative size of the different chromosome arms.</p></li>
<li><p>We converted the X-axis coordinates to “Mbp” which means mega-base pairs, i.e., millions of letters.</p></li>
</ol>
<p>Now we’re acquainted with the reference genome, let’s look at nucleotide variation.</p>
</div>
<div class="section" id="snp-positions-and-alleles">
<h2>SNP positions and alleles<a class="headerlink" href="#snp-positions-and-alleles" title="Permalink to this headline">¶</a></h2>
<p>When studying genetic variation in natural populations, there are two fundamental questions:</p>
<ol class="simple">
<li><p>Where in the genome do we observe <strong>differences (i.e., nucleotide variation)</strong> between individual mosquitoes.</p></li>
<li><p>Where in the genome do we observe <strong>no differences (i.e., nucleotide conservation)</strong> between individuals.</p></li>
</ol>
<p>Sometimes this second question is overlooked, but both questions are important.</p>
<p>In Ag1000G phase 3 we have taken a slightly different approach to calling single nucleotide polymorphisms (SNPs - single letter differences) than in previous project phases. Because we are interested in both of the above questions, <strong>we have genotyped all individual mosquitoes at all genomic sites where the reference genotype nucleotide is known (i.e., not an “N”), regardless of whether any variation is present</strong>. This is sometimes known as an “all sites” analysis.</p>
<p>In order to simplify SNP calling across a large number of individuals, <strong>we also declared the set of all possible alternate (variant) SNP alleles ahead of time</strong>. Because we are analysing SNPs and we are not considering any other types of variation at this stage, then at each site in the genome there is one reference allele (i.e., the nucleotide in the reference genome) and three possible alternate alleles. E.g., at position 10 on chromosome arm 2R, the reference allele is “T”, and so there are three possible alternate alleles: “A”, “C” and “G”.</p>
<p>Let’s look at how these data are represented for chromosome arm 2R:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pos</span><span class="p">,</span> <span class="n">ref</span><span class="p">,</span> <span class="n">alt</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_sites</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">pos</span></code> array here holds the 1-based positions (coordinates) where the SNPs were called. E.g., here’s the first 10 SNP positions:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pos</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10], dtype=int32)
</pre></div>
</div>
</div>
</div>
<p>Note that this is simply the positions 1-10. However, these positions <strong>do not</strong> continue increasing by 1 all the way through the array, i.e., there are gaps, because of gaps in the reference genome. We can quickly visualise that, by plotting the distance between adjacent sites:</p>
<div class="cell tag_thebe-init docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">plot_gaps</span><span class="p">(</span><span class="n">contig</span><span class="p">):</span>
    <span class="n">pos</span><span class="p">,</span> <span class="n">_</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_sites</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">)</span>
    <span class="n">p</span> <span class="o">=</span> <span class="n">pos</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
    <span class="n">d</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">diff</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
    <span class="n">loc_gap</span> <span class="o">=</span> <span class="n">d</span> <span class="o">&gt;</span> <span class="mi">1</span>
    <span class="n">x</span> <span class="o">=</span> <span class="n">p</span><span class="p">[</span><span class="mi">1</span><span class="p">:][</span><span class="n">loc_gap</span><span class="p">]</span>
    <span class="n">y</span> <span class="o">=</span> <span class="n">d</span><span class="p">[</span><span class="n">loc_gap</span><span class="p">]</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">11</span><span class="p">,</span> <span class="mi">2</span><span class="p">))</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">.5</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;o&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">markersize</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">mfc</span><span class="o">=</span><span class="s1">&#39;none&#39;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;2R&quot;</span><span class="p">)</span>
    <span class="c1"># convert X axis to Mbp</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">mpl</span><span class="o">.</span><span class="n">ticker</span><span class="o">.</span><span class="n">FuncFormatter</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">,</span> <span class="n">p</span><span class="p">:</span> <span class="n">x</span><span class="o">//</span><span class="mf">1e6</span><span class="p">))</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Position (Mbp)&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Gap size (bp)&#39;</span><span class="p">)</span>
    
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_gaps</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_32_0.png" src="../../_images/snps_32_0.png" />
</div>
</div>
<p>Note also that the length of the <code class="docutils literal notranslate"><span class="pre">pos</span></code> array is less than the length of the reference genome sequence:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">len</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span> <span class="o">-</span> <span class="nb">len</span><span class="p">(</span><span class="n">pos</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1412652
</pre></div>
</div>
</div>
</div>
<p>This difference is the number of unknown nucleotides (“N”s) in the reference genome that we computed earlier. This is another way of confirming that we have called SNP genotypes at all genomic sites where the reference genome nucleotide is known.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">ref</span></code> array holds the reference alleles, i.e., the reference genome nucleotide for each SNP site. E.g., here’s the reference alleles at the first 10 SNP sites:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ref</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([b&#39;C&#39;, b&#39;T&#39;, b&#39;C&#39;, b&#39;T&#39;, b&#39;A&#39;, b&#39;A&#39;, b&#39;A&#39;, b&#39;C&#39;, b&#39;A&#39;, b&#39;T&#39;],
      dtype=&#39;|S1&#39;)
</pre></div>
</div>
</div>
</div>
<p>We can confirm that this corresponds to the nucleotides in the reference sequence by using the SNP positions to index the sequence (remembering to convert to zero-based coordinates):</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">seq</span><span class="p">[</span><span class="n">pos</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span> <span class="o">-</span> <span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([b&#39;C&#39;, b&#39;T&#39;, b&#39;c&#39;, b&#39;t&#39;, b&#39;a&#39;, b&#39;a&#39;, b&#39;a&#39;, b&#39;c&#39;, b&#39;a&#39;, b&#39;t&#39;],
      dtype=&#39;|S1&#39;)
</pre></div>
</div>
</div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">alt</span></code> array holds the predefined alternate alleles at each site. E.g., here are the alternate alleles for the first 10 SNP sites:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">alt</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([[b&#39;A&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;G&#39;],
       [b&#39;C&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;C&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;C&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;C&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;G&#39;]], dtype=&#39;|S1&#39;)
</pre></div>
</div>
</div>
</div>
<p>Each row represents a site, and the columns show the three possible alternate alleles.</p>
<p>The same applies if you are working with the data in VCF format. E.g., here’s the first 10 rows of the VCF file with SNP genotypes for sample AR0001-C (see the <a class="reference internal" href="../download.html"><span class="doc std std-doc">download guide</span></a> for how to get this file), note the three alternate (ALT) alleles at every site:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	AR0001-C</span>
<span class="mi">2</span><span class="n">R</span>	<span class="mi">1</span>	<span class="o">.</span>	<span class="n">C</span>	<span class="n">A</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="n">G</span>	<span class="mf">0.08</span>	<span class="o">.</span>	<span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">14.72</span>	<span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>	<span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">:</span><span class="mi">6</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">52</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">52</span><span class="p">,</span><span class="mi">52</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">52</span><span class="p">,</span><span class="mi">52</span><span class="p">,</span><span class="mi">52</span>
<span class="mi">2</span><span class="n">R</span>	<span class="mi">2</span>	<span class="o">.</span>	<span class="n">T</span>	<span class="n">A</span><span class="p">,</span><span class="n">C</span><span class="p">,</span><span class="n">G</span>	<span class="mf">0.08</span>	<span class="o">.</span>	<span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">13.44</span>	<span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>	<span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">6</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">6</span><span class="p">:</span><span class="mi">6</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span>
<span class="mi">2</span><span class="n">R</span>	<span class="mi">3</span>	<span class="o">.</span>	<span class="n">C</span>	<span class="n">A</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="n">G</span>	<span class="mf">0.08</span>	<span class="o">.</span>	<span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">12.44</span>	<span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>	<span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">7</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">7</span><span class="p">:</span><span class="mi">6</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span>
<span class="mi">2</span><span class="n">R</span>	<span class="mi">4</span>	<span class="o">.</span>	<span class="n">T</span>	<span class="n">A</span><span class="p">,</span><span class="n">C</span><span class="p">,</span><span class="n">G</span>	<span class="mf">0.08</span>	<span class="o">.</span>	<span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">12.44</span>	<span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>	<span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">7</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">7</span><span class="p">:</span><span class="mi">6</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span>
<span class="mi">2</span><span class="n">R</span>	<span class="mi">5</span>	<span class="o">.</span>	<span class="n">A</span>	<span class="n">C</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="n">G</span>	<span class="mf">0.08</span>	<span class="o">.</span>	<span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">22.81</span>	<span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>	<span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">9</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">9</span><span class="p">:</span><span class="mi">6</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span><span class="p">,</span><span class="mi">53</span>
<span class="mi">2</span><span class="n">R</span>	<span class="mi">6</span>	<span class="o">.</span>	<span class="n">A</span>	<span class="n">C</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="n">G</span>	<span class="mf">0.02</span>	<span class="o">.</span>	<span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">28.78</span>	<span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>	<span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">10</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">10</span><span class="p">:</span><span class="mi">12</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">113</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">113</span><span class="p">,</span><span class="mi">113</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">113</span><span class="p">,</span><span class="mi">113</span><span class="p">,</span><span class="mi">113</span>
<span class="mi">2</span><span class="n">R</span>	<span class="mi">7</span>	<span class="o">.</span>	<span class="n">A</span>	<span class="n">C</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="n">G</span>	<span class="mf">0.02</span>	<span class="o">.</span>	<span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">27.68</span>	<span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>	<span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">11</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">11</span><span class="p">:</span><span class="mi">12</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">118</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">118</span><span class="p">,</span><span class="mi">118</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">118</span><span class="p">,</span><span class="mi">118</span><span class="p">,</span><span class="mi">118</span>
<span class="mi">2</span><span class="n">R</span>	<span class="mi">8</span>	<span class="o">.</span>	<span class="n">C</span>	<span class="n">A</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="n">G</span>	<span class="mf">0.02</span>	<span class="o">.</span>	<span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">27.68</span>	<span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>	<span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">11</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">11</span><span class="p">:</span><span class="mi">12</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">130</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">130</span><span class="p">,</span><span class="mi">130</span><span class="p">,</span><span class="mi">12</span><span class="p">,</span><span class="mi">130</span><span class="p">,</span><span class="mi">130</span><span class="p">,</span><span class="mi">130</span>
<span class="mi">2</span><span class="n">R</span>	<span class="mi">9</span>	<span class="o">.</span>	<span class="n">A</span>	<span class="n">C</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="n">G</span>	<span class="mf">0.01</span>	<span class="o">.</span>	<span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">31.66</span>	<span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>	<span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">12</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">12</span><span class="p">:</span><span class="mi">15</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">15</span><span class="p">,</span><span class="mi">156</span><span class="p">,</span><span class="mi">15</span><span class="p">,</span><span class="mi">156</span><span class="p">,</span><span class="mi">156</span><span class="p">,</span><span class="mi">15</span><span class="p">,</span><span class="mi">156</span><span class="p">,</span><span class="mi">156</span><span class="p">,</span><span class="mi">156</span>
<span class="mi">2</span><span class="n">R</span>	<span class="mi">10</span>	<span class="o">.</span>	<span class="n">T</span>	<span class="n">A</span><span class="p">,</span><span class="n">C</span><span class="p">,</span><span class="n">G</span>	<span class="mf">0.01</span>	<span class="o">.</span>	<span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">,</span><span class="mf">0.00</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">34.67</span>	<span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>	<span class="mi">0</span><span class="o">/</span><span class="mi">0</span><span class="p">:</span><span class="mi">13</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">:</span><span class="mi">13</span><span class="p">:</span><span class="mi">18</span><span class="p">:</span><span class="mi">0</span><span class="p">,</span><span class="mi">18</span><span class="p">,</span><span class="mi">181</span><span class="p">,</span><span class="mi">18</span><span class="p">,</span><span class="mi">181</span><span class="p">,</span><span class="mi">181</span><span class="p">,</span><span class="mi">18</span><span class="p">,</span><span class="mi">181</span><span class="p">,</span><span class="mi">181</span><span class="p">,</span><span class="mi">181</span>
</pre></div>
</div>
<p>The most important thing to note about the SNP data in Ag3 is that <strong>not all of these alternate alleles will have been observed in the mosquitoes we sequenced.</strong></p>
<p>To determine which alternate alleles are observed in Ag3, we need to look at the SNP genotypes. Let’s take a look at how SNP genotypes are represented, then return to the question of which alternate alleles have been observed.</p>
</div>
<div class="section" id="genotypes">
<h2>Genotypes<a class="headerlink" href="#genotypes" title="Permalink to this headline">¶</a></h2>
<p>Here a SNP “genotype” means the pair of nucleotides (alleles) carried by an individual mosquito at a specific site in the genome. Note that this is a pair of alleles because mosquitoes are diploid like us, meaning that they carry two copies of their genome, one inherited from each parent (except for males on the X chromosome, which inherit one copy from their mother only).</p>
<p>Rather than storing the actual nucleotides for each genotype, it is conventional to use a numerical encoding of genotypes, where 0 stands for the reference allele, 1 stands for the first alternate allele, 2 stands for the second alternate allele, 3 stands for the third alternate allele, etc.</p>
<p>E.g., at site 2R:10 the reference allele is “T”. If an individual mosquito carries a homozygous reference genotype (both genome copies have a “T”), we represent this in Python as the pair of integers <code class="docutils literal notranslate"><span class="pre">(0,</span> <span class="pre">0)</span></code>. In a VCF file this is represented as the string “0/0”.</p>
<p>At 2R:10 we have arbitrarily declared that “A” is the first alternate allele. If an individual carries a heterozygous genotype, where one genome copy has “T” and the other has “A”, this is represented as the pair <code class="docutils literal notranslate"><span class="pre">(0,</span> <span class="pre">1)</span></code>. The ordering of this pair has no meaning.</p>
<p>Similarly, at 2R:10 we have arbitrarily declared that “C” is the second alternate allele and “G” is the third alternate allele. So, a genotype homozygous for the “C” allele would be represented as <code class="docutils literal notranslate"><span class="pre">(2,</span> <span class="pre">2)</span></code>, a genotype homozygous for “G” would be <code class="docutils literal notranslate"><span class="pre">(3,</span> <span class="pre">3)</span></code> and a genotype heterozygous for the “C” and “G” alleles would be <code class="docutils literal notranslate"><span class="pre">(2,</span> <span class="pre">3)</span></code>.</p>
<p>There’s one more thing to know. Sometimes a genotype is called as missing, meaning that there were no sequence reads for the given individual at a given site. In Python we represent this as a pair of negative numbers <code class="docutils literal notranslate"><span class="pre">(-1,</span> <span class="pre">-1)</span></code>. In VCF this is represented as the string “./.”.</p>
<p>Here are those example genotypes again, shown as a table with the different representations:</p>
<div class="cell_output">
<table border="1" class="dataframe">
<thead>
<tr>
<th>Reference allele</th>
<th>Alternate alleles</th>
<th>Genotype</th>
<th>Genotype (Python)</th>
<th>Genotype (VCF)</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<th rowspan="7">T</td>
<th rowspan="7">A,C,G</td>
<td>T/T</td>
<td><code>(0, 0)</code></td>
<td>0/0</td>
<td>homozygous reference</td>
</tr>
<tr>
<td>T/A</td>
<td><code>(0, 1)</code></td>
<td>0/1</td>
<td>heterozygous</td>
</tr>
<tr>
<td>A/A</td>
<td><code>(1, 1)</code></td>
<td>1/1</td>
<td>homozygous alternate</td>
</tr>
<tr>
<td>C/C</td>
<td><code>(2, 2)</code></td>
<td>2/2</td>
<td>homozygous alternate</td>
</tr>
<tr>
<td>G/G</td>
<td><code>(3, 3)</code></td>
<td>3/3</td>
<td>homozygous alternate</td>
</tr>
<tr>
<td>C/G</td>
<td><code>(2, 3)</code></td>
<td>2/3</td>
<td>heterozygous</td>
</tr>
<tr>
<td>missing</td>
<td><code>(-1, -1)</code></td>
<td>./.</td>
<td>missing</td>
</tr>
</tbody>
</table>
</div>
<br/>
<p>To see this in action, let’s access the SNP genotypes for chromosome arm 2R for all wild samples in Ag3.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_genotypes</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">,</span> <span class="n">sample_sets</span><span class="o">=</span><span class="s2">&quot;v3&quot;</span><span class="p">)</span>
<span class="n">gt</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(60132453, 3081, 2)
</pre></div>
</div>
</div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">gt</span></code> array has three dimensions. The first dimension corresponds to the genomic sites we’ve analysed (in this case, there are 60,132,453 sites on chromosome arm 2R). The second dimension (3,081) is the number of individual mosquitoes (a.k.a. samples) sequenced. The third dimension (2) represents the two alleles in each individual genotype.</p>
<p>So, e.g., let’s look up the genotype call at the 10th site in the 1st sample:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt</span><span class="p">[</span><span class="mi">9</span><span class="p">,</span> <span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([0, 0], dtype=int8)
</pre></div>
</div>
</div>
</div>
<p>This is a homozygous reference genotype.</p>
<p>Let’s try another, the 10th site and the 2,393rd sample.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt</span><span class="p">[</span><span class="mi">9</span><span class="p">,</span> <span class="mi">2392</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([0, 2], dtype=int8)
</pre></div>
</div>
</div>
</div>
<p>This is a heterozygous genotype, where the reference allele and the second alternate allele are both present.</p>
<p>Let’s look at one more genotype, the 10th site and the 249th sample.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt</span><span class="p">[</span><span class="mi">9</span><span class="p">,</span> <span class="mi">248</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([-1, -1], dtype=int8)
</pre></div>
</div>
</div>
</div>
<p>This is a missing genotype call.</p>
<p>When working with genotype arrays it can be useful to use the <a class="reference external" href="http://scikit-allel.readthedocs.org/">scikit-allel</a> package. Let’s wrap the <code class="docutils literal notranslate"><span class="pre">gt</span></code> variable, which is a general purpose <a class="reference external" href="https://docs.dask.org/en/latest/array.html">dask array</a>, with the scikit-allel <code class="docutils literal notranslate"><span class="pre">GenotypeDaskArray</span></code> class, which provides some useful functionality for analysing genotype data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt</span> <span class="o">=</span> <span class="n">allel</span><span class="o">.</span><span class="n">GenotypeDaskArray</span><span class="p">(</span><span class="n">gt</span><span class="p">)</span>
<span class="n">gt</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div class="allel allel-DisplayAs2D"><span>&lt;GenotypeDaskArray shape=(60132453, 3081, 2) dtype=int8&gt;</span><table><thead><tr><th></th><th style="text-align: center">0</th><th style="text-align: center">1</th><th style="text-align: center">2</th><th style="text-align: center">3</th><th style="text-align: center">4</th><th style="text-align: center">...</th><th style="text-align: center">3076</th><th style="text-align: center">3077</th><th style="text-align: center">3078</th><th style="text-align: center">3079</th><th style="text-align: center">3080</th></tr></thead><tbody><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">0</th><td style="text-align: center">0/0</td><td style="text-align: center">./.</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">1</th><td style="text-align: center">0/0</td><td style="text-align: center">./.</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">2</th><td style="text-align: center">0/0</td><td style="text-align: center">./.</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">...</th><td style="text-align: center" colspan="12">...</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">60132450</th><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">...</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">60132451</th><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">...</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">60132452</th><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">...</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td><td style="text-align: center">./.</td></tr></tbody></table></div></div></div>
</div>
<p>This wrapper class provided a display showing a preview of genotype calls for the first and last sites and samples, using the VCF-style representation of genotype calls (e.g., “0/0” is homozygous reference, etc.).</p>
<p>Once we have access to the genotype calls, we can run a variety of different computations. A common computation needed for many analyses is an allele count, which we’ll look at next.</p>
</div>
<div class="section" id="allele-counts">
<h2>Allele counts<a class="headerlink" href="#allele-counts" title="Permalink to this headline">¶</a></h2>
<p>An allele count involves scanning across all the genotypes at a given site and counting the number of times each allele is observed. I.e., how many 0s, 1s, 2s and 3s are observed. Let’s do an allele count for the first 10 sites on chromosome arm 2R.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ac</span> <span class="o">=</span> <span class="n">gt</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">count_alleles</span><span class="p">(</span><span class="n">max_allele</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">ac</span><span class="o">.</span><span class="n">displayall</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div class="allel allel-DisplayAs2D"><span>&lt;AlleleCountsArray shape=(10, 4) dtype=int32&gt;</span><table><thead><tr><th></th><th style="text-align: center">0</th><th style="text-align: center">1</th><th style="text-align: center">2</th><th style="text-align: center">3</th></tr></thead><tbody><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">0</th><td style="text-align: center">5871</td><td style="text-align: center">   2</td><td style="text-align: center">   1</td><td style="text-align: center">   0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">1</th><td style="text-align: center">5930</td><td style="text-align: center">   0</td><td style="text-align: center">   5</td><td style="text-align: center">   1</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">2</th><td style="text-align: center">5981</td><td style="text-align: center">   7</td><td style="text-align: center">   2</td><td style="text-align: center">   0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">3</th><td style="text-align: center">6033</td><td style="text-align: center">   1</td><td style="text-align: center">   2</td><td style="text-align: center">   0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">4</th><td style="text-align: center">6050</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">5</th><td style="text-align: center">6070</td><td style="text-align: center">   1</td><td style="text-align: center">   0</td><td style="text-align: center">   3</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">6</th><td style="text-align: center">6080</td><td style="text-align: center">   4</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">7</th><td style="text-align: center">6073</td><td style="text-align: center">   2</td><td style="text-align: center">   6</td><td style="text-align: center">  15</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">8</th><td style="text-align: center">6118</td><td style="text-align: center">   0</td><td style="text-align: center">   1</td><td style="text-align: center">   1</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">9</th><td style="text-align: center">6120</td><td style="text-align: center">   1</td><td style="text-align: center">   4</td><td style="text-align: center">   1</td></tr></tbody></table></div></div></div>
</div>
<p>In the <code class="docutils literal notranslate"><span class="pre">ac</span></code> array above, each row represents a site, and the four columns represent the four alleles. The values are the number of times each allele was observed at each site. So, now we can see which alleles have been observed in the Ag3 data.</p>
<p>E.g., for the 10th site, the reference allele (<code class="docutils literal notranslate"><span class="pre">0</span></code>) was observed 6120 times, the first alternate allele (<code class="docutils literal notranslate"><span class="pre">1</span></code>) was observed once, the second alternate allele (<code class="docutils literal notranslate"><span class="pre">2</span></code>) was observed four times, and the third alternate allele (<code class="docutils literal notranslate"><span class="pre">3</span></code>) was observed once. So, this is a site where in fact all four possible alleles (nucleotides) were observed in Ag3 samples. This is an example of a multiallelic site.</p>
<p>Let’s look at some other sites.</p>
<p>At the fifth site, only the reference allele was observed. This is an example of an invariant site. I.e., there is no variation there, all samples were homozygous for the reference allele or had a missing genotype call.</p>
<p>At the seventh site, two alleles were observed, the reference allele and the first alternate allele. This is an example of a biallelic site. Note that biallelic sites might involve any pair of alleles, not always <code class="docutils literal notranslate"><span class="pre">0</span></code> and <code class="docutils literal notranslate"><span class="pre">1</span></code>.</p>
<p>Let’s use this allele counts array to compute some properties of these 10 sites.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># how many alleles were observed</span>
<span class="n">ac</span><span class="o">.</span><span class="n">allelism</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([3, 3, 3, 3, 1, 3, 2, 4, 3, 4])
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># whether any alternate alleles were observed</span>
<span class="n">ac</span><span class="o">.</span><span class="n">is_variant</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([ True,  True,  True,  True, False,  True,  True,  True,  True,
        True])
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># whether more than one allele was observed</span>
<span class="n">ac</span><span class="o">.</span><span class="n">is_segregating</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([ True,  True,  True,  True, False,  True,  True,  True,  True,
        True])
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># whether exactly two alleles were observed</span>
<span class="n">ac</span><span class="o">.</span><span class="n">is_biallelic</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([False, False, False, False, False, False,  True, False, False,
       False])
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># total number of alleles called</span>
<span class="n">an_called</span> <span class="o">=</span> <span class="n">ac</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">an_called</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([5874, 5936, 5990, 6036, 6050, 6074, 6084, 6096, 6120, 6126])
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># total number of missing allele calls</span>
<span class="n">an_missing</span> <span class="o">=</span> <span class="p">(</span><span class="n">gt</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="o">*</span> <span class="mi">2</span><span class="p">)</span> <span class="o">-</span> <span class="n">an_called</span>
<span class="n">an_missing</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([288, 226, 172, 126, 112,  88,  78,  66,  42,  36])
</pre></div>
</div>
</div>
</div>
<p>These arrays can be useful when selecting which sites to use for a given analysis. For example, you might want to select only segregating sites, with at most a certain number of missing allele calls.</p>
<p>Now we’ve seen how SNP positions, alleles and genotypes are represented, let’s look at the site filters we’ve created for Ag3 to help select sites for robust population genetic analyses.</p>
</div>
<div class="section" id="site-filters">
<h2>Site filters<a class="headerlink" href="#site-filters" title="Permalink to this headline">¶</a></h2>
<p>When analysing genome variation, there are some sites in the genome where it is difficult to reliably ascertain nucleotide variation and call genotypes. This can be for a variety of reasons. For example, some genome regions are very repetitive, and short sequence reads cannot be aligned unambiguously. Another issue can be that some samples have structural variation in some regions of the genome, where different samples have different numbers of copies of a given sequence.</p>
<p>To identify and avoid sites like these, we have created site filters. These are Boolean masks which identify sites where SNP genotyping is reliable, based on the fact that most sequence reads are unambiguously aligned, and there is minimal evidence for structural variation.</p>
<p>Because we have multiple mosquito species included in the Ag3 cohort (<em>An. gambiae</em>, <em>An. coluzzii</em> and <em>An. arabiensis</em>) we have created three different site filters which are appropriate in different circumstances, depending on which samples you’re analysing. Here is a summary of the three site filters:</p>
<ul class="simple">
<li><p>“gamb_colu_arab” - These site filters are appropriate when performing a joint analysis of samples from any of the three species. They are, however, more stringent the other filters, which we’ll explore further below.</p></li>
<li><p>“gamb_colu” - These site filters are appropriate when analysing only samples that are not <em>An. arabiensis</em>.</p></li>
<li><p>“arab” - These sige filters are appropriate when analysing only samples that are <em>An. arabiensis</em>.</p></li>
</ul>
<p>For each of these filters, there is one Boolean mask per chromosome arm. The Boolean masks are True where a site passes the filter and False otherwise. E.g., here’s the <code class="docutils literal notranslate"><span class="pre">gamb_colu_arab</span></code> site filter for chromosome arm 2R:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">loc_pass</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">site_filters</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">,</span> <span class="n">mask</span><span class="o">=</span><span class="s2">&quot;gamb_colu_arab&quot;</span><span class="p">)</span>
<span class="n">loc_pass</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 60.13 MB </td> <td> 60.13 MB </td></tr>
    <tr><th> Shape </th><td> (60132453,) </td> <td> (60132453,) </td></tr>
    <tr><th> Count </th><td> 1 Tasks </td><td> 1 Chunks </td></tr>
    <tr><th> Type </th><td> bool </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >60132453</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># examine the first ten values</span>
<span class="n">loc_pass</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([False, False, False, False, False, False, False, False, False,
       False])
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># how many sites in total pass the filter?</span>
<span class="n">loc_pass</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>40561667
</pre></div>
</div>
</div>
</div>
<p>To get some intuition for what these site filters look like and how they differ from each other, let’s build a summary table showing how many sites pass in each chromosome arm. This may take a minute or two to run.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">count_filter_pass</span><span class="p">(</span><span class="n">contig</span><span class="p">,</span> <span class="n">mask</span><span class="p">):</span>
    <span class="n">loc_pass</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">site_filters</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">,</span> <span class="n">mask</span><span class="o">=</span><span class="n">mask</span><span class="p">)</span>
    <span class="n">n_pass</span> <span class="o">=</span> <span class="n">loc_pass</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
    <span class="k">return</span> <span class="n">n_pass</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">tabulate_filter_counts</span><span class="p">():</span>

    <span class="n">rows</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">contig</span> <span class="ow">in</span> <span class="n">ag3</span><span class="o">.</span><span class="n">contigs</span><span class="p">:</span>
        <span class="n">contig_length</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">genome_sequence</span><span class="p">(</span><span class="n">contig</span><span class="p">))</span>
        <span class="n">n_sites</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">snp_sites</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">,</span> <span class="n">field</span><span class="o">=</span><span class="s2">&quot;POS&quot;</span><span class="p">))</span>
        <span class="n">row</span> <span class="o">=</span> <span class="p">[</span><span class="n">contig</span><span class="p">,</span> <span class="n">contig_length</span><span class="p">,</span> <span class="n">n_sites</span><span class="p">]</span>
        <span class="k">for</span> <span class="n">mask</span> <span class="ow">in</span> <span class="s2">&quot;gamb_colu_arab&quot;</span><span class="p">,</span> <span class="s2">&quot;gamb_colu&quot;</span><span class="p">,</span> <span class="s2">&quot;arab&quot;</span><span class="p">:</span>
            <span class="n">n_pass</span> <span class="o">=</span> <span class="n">count_filter_pass</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">,</span> <span class="n">mask</span><span class="o">=</span><span class="n">mask</span><span class="p">)</span>
            <span class="n">pc_pass</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{:.1%}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">n_pass</span> <span class="o">/</span> <span class="n">n_sites</span><span class="p">)</span>
            <span class="n">row</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">pc_pass</span><span class="p">)</span>
        <span class="n">rows</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">row</span><span class="p">)</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="o">.</span><span class="n">from_records</span><span class="p">(</span>
        <span class="n">rows</span><span class="p">,</span> 
        <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;contig&quot;</span><span class="p">,</span> <span class="s2">&quot;contig length&quot;</span><span class="p">,</span> <span class="s2">&quot;no. sites&quot;</span><span class="p">,</span> <span class="s2">&quot;pass gamb_colu_arab&quot;</span><span class="p">,</span> <span class="s2">&quot;pass gamb_colu&quot;</span><span class="p">,</span> <span class="s2">&quot;pass arab&quot;</span><span class="p">]</span>
    <span class="p">)</span>
    <span class="k">return</span> <span class="n">df</span>

<span class="n">df_filter_counts</span> <span class="o">=</span> <span class="n">tabulate_filter_counts</span><span class="p">()</span>
<span class="n">df_filter_counts</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>contig</th>
      <th>contig length</th>
      <th>no. sites</th>
      <th>pass gamb_colu_arab</th>
      <th>pass gamb_colu</th>
      <th>pass arab</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>2R</td>
      <td>61545105</td>
      <td>60132453</td>
      <td>67.5%</td>
      <td>73.9%</td>
      <td>73.6%</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2L</td>
      <td>49364325</td>
      <td>48525747</td>
      <td>67.0%</td>
      <td>74.2%</td>
      <td>72.6%</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3R</td>
      <td>53200684</td>
      <td>52226568</td>
      <td>63.9%</td>
      <td>71.2%</td>
      <td>69.8%</td>
    </tr>
    <tr>
      <th>3</th>
      <td>3L</td>
      <td>41963435</td>
      <td>40758473</td>
      <td>63.5%</td>
      <td>70.4%</td>
      <td>69.7%</td>
    </tr>
    <tr>
      <th>4</th>
      <td>X</td>
      <td>24393108</td>
      <td>23385349</td>
      <td>48.0%</td>
      <td>70.0%</td>
      <td>53.3%</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>There’s a couple of things to note about these numbers:</p>
<ol class="simple">
<li><p>The percentage of sites passing the <code class="docutils literal notranslate"><span class="pre">gamb_colu_arab</span></code> filter is always lower than either the <code class="docutils literal notranslate"><span class="pre">gamb_colu</span></code> or <code class="docutils literal notranslate"><span class="pre">arab</span></code> filters. This is because the <code class="docutils literal notranslate"><span class="pre">gamb_colu_arab</span></code> filter is the intersection of the two other filters. I.e., a site passes the <code class="docutils literal notranslate"><span class="pre">gamb_colu_arab</span></code> filter if it passes <strong>both</strong> the <code class="docutils literal notranslate"><span class="pre">gamb_colu</span></code> and the <code class="docutils literal notranslate"><span class="pre">arab</span></code> filters.</p></li>
<li><p>The percentage of sites passing the <code class="docutils literal notranslate"><span class="pre">gamb_colu</span></code> and <code class="docutils literal notranslate"><span class="pre">arab</span></code> filters is similar on all chromosome arms, except for the X chromosome. This is because of a large inversion on the X chromosome known as Xag which is a fixed difference between <em>An. arabiensis</em> and the other two species, and where there is a higher level of divergence in <em>An. arabiensis</em> relative to the reference genome we’ve used.</p></li>
</ol>
<p>To gain a bit more intuition for these filters, let’s plot the fraction of genome sites that pass the site filters in moving windows over each of the chromosome arms.</p>
<div class="cell tag_thebe-init docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">plot_site_filters</span><span class="p">(</span><span class="n">contig</span><span class="p">,</span> <span class="n">window_size</span><span class="o">=</span><span class="mi">200_000</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="kc">None</span><span class="p">):</span>

    <span class="c1"># setup figure and axes</span>
    <span class="k">if</span> <span class="n">ax</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">contig_length</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">genome_sequence</span><span class="p">(</span><span class="n">contig</span><span class="p">))</span>
        <span class="n">figw</span> <span class="o">=</span> <span class="mi">11</span> <span class="o">*</span> <span class="n">contig_length</span> <span class="o">/</span> <span class="nb">len</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">genome_sequence</span><span class="p">(</span><span class="s2">&quot;2R&quot;</span><span class="p">))</span>
        <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="n">figw</span><span class="p">,</span> <span class="mi">2</span><span class="p">))</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">despine</span><span class="p">(</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">offset</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>

    <span class="c1"># setup X variable</span>
    <span class="n">pos</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_sites</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">,</span> <span class="n">field</span><span class="o">=</span><span class="s2">&quot;POS&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
    <span class="n">x</span> <span class="o">=</span> <span class="n">allel</span><span class="o">.</span><span class="n">moving_statistic</span><span class="p">(</span><span class="n">pos</span><span class="p">,</span> <span class="n">statistic</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="n">window_size</span><span class="p">)</span>

    <span class="c1"># setup Y variables and plot</span>
    <span class="k">for</span> <span class="n">mask</span> <span class="ow">in</span> <span class="s2">&quot;gamb_colu&quot;</span><span class="p">,</span> <span class="s2">&quot;arab&quot;</span><span class="p">,</span> <span class="s2">&quot;gamb_colu_arab&quot;</span><span class="p">:</span>
        <span class="n">loc_pass</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">site_filters</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">,</span> <span class="n">mask</span><span class="o">=</span><span class="n">mask</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
        <span class="n">y</span> <span class="o">=</span> <span class="n">allel</span><span class="o">.</span><span class="n">moving_statistic</span><span class="p">(</span><span class="n">loc_pass</span><span class="p">,</span> <span class="n">statistic</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="n">window_size</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span> <span class="o">*</span> <span class="mi">100</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="n">mask</span><span class="p">)</span>

    <span class="c1"># tidy plot</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="n">contig_length</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">100</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">contig</span><span class="p">)</span>
    <span class="c1"># convert X axis to Mbp</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">mpl</span><span class="o">.</span><span class="n">ticker</span><span class="o">.</span><span class="n">FuncFormatter</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">,</span> <span class="n">p</span><span class="p">:</span> <span class="n">x</span><span class="o">//</span><span class="mf">1e6</span><span class="p">))</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Position (Mbp)&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;% pass&#39;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="s1">&#39;upper left&#39;</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Filter&quot;</span><span class="p">,</span>
              <span class="n">frameon</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_site_filters</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_74_0.png" src="../../_images/snps_74_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_site_filters</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2L&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_75_0.png" src="../../_images/snps_75_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_site_filters</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;3R&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_76_0.png" src="../../_images/snps_76_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_site_filters</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;3L&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_77_0.png" src="../../_images/snps_77_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">plot_site_filters</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;X&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/snps_78_0.png" src="../../_images/snps_78_0.png" />
</div>
</div>
<p>The methods in the <code class="docutils literal notranslate"><span class="pre">malariagen_data.Ag3</span></code> class allow you to access data on sites or genotypes with one of these filters already applied. E.g., let’s access the SNP positions with the <code class="docutils literal notranslate"><span class="pre">gamb_colu_arab</span></code> filter applied.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pos</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_sites</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">,</span> <span class="n">site_mask</span><span class="o">=</span><span class="s2">&quot;gamb_colu_arab&quot;</span><span class="p">,</span> <span class="n">field</span><span class="o">=</span><span class="s2">&quot;POS&quot;</span><span class="p">)</span>
<span class="n">pos</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 162.25 MB </td> <td> 98.99 MB </td></tr>
    <tr><th> Shape </th><td> (40561667,) </td> <td> (24746331,) </td></tr>
    <tr><th> Count </th><td> 11 Tasks </td><td> 2 Chunks </td></tr>
    <tr><th> Type </th><td> int32 </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="170" height="75" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="120" y2="0" style="stroke-width:2" />
  <line x1="0" y1="25" x2="120" y2="25" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="25" style="stroke-width:2" />
  <line x1="73" y1="0" x2="73" y2="25" />
  <line x1="120" y1="0" x2="120" y2="25" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="60.000000" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >40561667</text>
  <text x="140.000000" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,140.000000,12.706308)">1</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<p>Notice the length of this array is 40,561,667 which we calculated earlier as the number of sites passing the <code class="docutils literal notranslate"><span class="pre">gamb_colu_arab</span></code> filter on chromosome arm 2R.</p>
<p>Similarly, here are the genotypes with the <code class="docutils literal notranslate"><span class="pre">gamb_colu_arab</span></code> site filter applied.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_genotypes</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">,</span> <span class="n">sample_sets</span><span class="o">=</span><span class="s2">&quot;v3&quot;</span><span class="p">,</span> <span class="n">site_mask</span><span class="o">=</span><span class="s2">&quot;gamb_colu_arab&quot;</span><span class="p">)</span>
<span class="n">gt</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 249.94 GB </td> <td> 103.66 MB </td></tr>
    <tr><th> Shape </th><td> (40561667, 3081, 2) </td> <td> (508154, 102, 2) </td></tr>
    <tr><th> Count </th><td> 27364 Tasks </td><td> 6240 Chunks </td></tr>
    <tr><th> Type </th><td> int8 </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="1" x2="80" y2="72" />
  <line x1="10" y1="3" x2="80" y2="73" />
  <line x1="10" y1="3" x2="80" y2="74" />
  <line x1="10" y1="4" x2="80" y2="75" />
  <line x1="10" y1="6" x2="80" y2="77" />
  <line x1="10" y1="7" x2="80" y2="78" />
  <line x1="10" y1="8" x2="80" y2="79" />
  <line x1="10" y1="9" x2="80" y2="80" />
  <line x1="10" y1="10" x2="80" y2="81" />
  <line x1="10" y1="12" x2="80" y2="83" />
  <line x1="10" y1="13" x2="80" y2="84" />
  <line x1="10" y1="14" x2="80" y2="85" />
  <line x1="10" y1="16" x2="80" y2="86" />
  <line x1="10" y1="17" x2="80" y2="87" />
  <line x1="10" y1="18" x2="80" y2="88" />
  <line x1="10" y1="19" x2="80" y2="90" />
  <line x1="10" y1="21" x2="80" y2="91" />
  <line x1="10" y1="22" x2="80" y2="93" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="13" y1="3" x2="13" y2="29" />
  <line x1="18" y1="8" x2="18" y2="33" />
  <line x1="22" y1="12" x2="22" y2="38" />
  <line x1="25" y1="15" x2="25" y2="41" />
  <line x1="30" y1="20" x2="30" y2="45" />
  <line x1="34" y1="24" x2="34" y2="50" />
  <line x1="38" y1="28" x2="38" y2="54" />
  <line x1="43" y1="33" x2="43" y2="58" />
  <line x1="46" y1="36" x2="46" y2="61" />
  <line x1="51" y1="41" x2="51" y2="66" />
  <line x1="54" y1="44" x2="54" y2="69" />
  <line x1="57" y1="47" x2="57" y2="72" />
  <line x1="60" y1="50" x2="60" y2="76" />
  <line x1="64" y1="54" x2="64" y2="79" />
  <line x1="68" y1="58" x2="68" y2="83" />
  <line x1="71" y1="61" x2="71" y2="86" />
  <line x1="74" y1="64" x2="74" y2="89" />
  <line x1="77" y1="67" x2="77" y2="93" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.41261651458248" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="13" y1="3" x2="39" y2="3" />
  <line x1="18" y1="8" x2="43" y2="8" />
  <line x1="22" y1="12" x2="48" y2="12" />
  <line x1="25" y1="15" x2="51" y2="15" />
  <line x1="30" y1="20" x2="55" y2="20" />
  <line x1="34" y1="24" x2="60" y2="24" />
  <line x1="38" y1="28" x2="64" y2="28" />
  <line x1="43" y1="33" x2="68" y2="33" />
  <line x1="46" y1="36" x2="71" y2="36" />
  <line x1="51" y1="41" x2="76" y2="41" />
  <line x1="54" y1="44" x2="79" y2="44" />
  <line x1="57" y1="47" x2="82" y2="47" />
  <line x1="60" y1="50" x2="86" y2="50" />
  <line x1="64" y1="54" x2="89" y2="54" />
  <line x1="68" y1="58" x2="93" y2="58" />
  <line x1="71" y1="61" x2="96" y2="61" />
  <line x1="74" y1="64" x2="99" y2="64" />
  <line x1="77" y1="67" x2="103" y2="67" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="72" x2="106" y2="72" />
  <line x1="80" y1="73" x2="106" y2="73" />
  <line x1="80" y1="74" x2="106" y2="74" />
  <line x1="80" y1="75" x2="106" y2="75" />
  <line x1="80" y1="77" x2="106" y2="77" />
  <line x1="80" y1="78" x2="106" y2="78" />
  <line x1="80" y1="79" x2="106" y2="79" />
  <line x1="80" y1="80" x2="106" y2="80" />
  <line x1="80" y1="81" x2="106" y2="81" />
  <line x1="80" y1="83" x2="106" y2="83" />
  <line x1="80" y1="84" x2="106" y2="84" />
  <line x1="80" y1="85" x2="106" y2="85" />
  <line x1="80" y1="86" x2="106" y2="86" />
  <line x1="80" y1="87" x2="106" y2="87" />
  <line x1="80" y1="88" x2="106" y2="88" />
  <line x1="80" y1="90" x2="106" y2="90" />
  <line x1="80" y1="91" x2="106" y2="91" />
  <line x1="80" y1="93" x2="106" y2="93" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,126.000852,83.294544)">3081</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">40561667</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
</div>
<div class="section" id="locating-snps">
<h2>Locating SNPs<a class="headerlink" href="#locating-snps" title="Permalink to this headline">¶</a></h2>
<div class="section" id="locating-a-single-snp">
<h3>Locating a single SNP<a class="headerlink" href="#locating-a-single-snp" title="Permalink to this headline">¶</a></h3>
<p>For some analyses you may have a particular genomic site you are investigating, and want to know what variation occurs at that site in the Ag3 data resource. To locate that site, a convenient approach is to use an index such as the scikit-allel <code class="docutils literal notranslate"><span class="pre">SortedIndex</span></code> class.</p>
<p>For example, let’s locate data for the site 2L:2422652.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pos</span><span class="p">,</span> <span class="n">ref</span><span class="p">,</span> <span class="n">alt</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_sites</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2L&quot;</span><span class="p">)</span>
<span class="n">pos</span> <span class="o">=</span> <span class="n">allel</span><span class="o">.</span><span class="n">SortedIndex</span><span class="p">(</span><span class="n">pos</span><span class="p">)</span>
<span class="n">pos</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div class="allel allel-DisplayAs1D"><span>&lt;SortedIndex shape=(48525747,) dtype=int32&gt;</span><table><thead><tr><th style="text-align: center">0</th><th style="text-align: center">1</th><th style="text-align: center">2</th><th style="text-align: center">3</th><th style="text-align: center">4</th><th style="text-align: center">...</th><th style="text-align: center">48525742</th><th style="text-align: center">48525743</th><th style="text-align: center">48525744</th><th style="text-align: center">48525745</th><th style="text-align: center">48525746</th></tr></thead><tbody><tr><td style="text-align: center">1</td><td style="text-align: center">2</td><td style="text-align: center">3</td><td style="text-align: center">4</td><td style="text-align: center">5</td><td style="text-align: center">...</td><td style="text-align: center">49364321</td><td style="text-align: center">49364322</td><td style="text-align: center">49364323</td><td style="text-align: center">49364324</td><td style="text-align: center">49364325</td></tr></tbody></table></div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># locate the position of interest</span>
<span class="n">loc_site</span> <span class="o">=</span> <span class="n">pos</span><span class="o">.</span><span class="n">locate_key</span><span class="p">(</span><span class="mi">2422652</span><span class="p">)</span>
<span class="n">loc_site</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2143836
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># check the position is correct</span>
<span class="n">pos</span><span class="p">[</span><span class="n">loc_site</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2422652
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># access the reference alleles</span>
<span class="n">ref</span><span class="p">[</span><span class="n">loc_site</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array(b&#39;A&#39;, dtype=&#39;|S1&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># access the alternate alleles</span>
<span class="n">alt</span><span class="p">[</span><span class="n">loc_site</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([b&#39;C&#39;, b&#39;T&#39;, b&#39;G&#39;], dtype=&#39;|S1&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># access the genotypes</span>
<span class="n">gt</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_genotypes</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2L&quot;</span><span class="p">,</span> <span class="n">sample_sets</span><span class="o">=</span><span class="s2">&quot;v3&quot;</span><span class="p">)</span>
<span class="n">gt_site</span> <span class="o">=</span> <span class="n">gt</span><span class="p">[</span><span class="n">loc_site</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">gt_site</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([[2, 2],
       [2, 2],
       [2, 2],
       ...,
       [0, 0],
       [0, 2],
       [0, 2]], dtype=int8)
</pre></div>
</div>
</div>
</div>
<p>From the output above we can see there are some samples with a <code class="docutils literal notranslate"><span class="pre">(2,</span> <span class="pre">2)</span></code> genotype, which correspond to the T/T genotype, and some samples with a <code class="docutils literal notranslate"><span class="pre">(0,</span> <span class="pre">2)</span></code> which corresponds to the A/T genotype.</p>
</div>
<div class="section" id="locating-snps-in-a-genome-region">
<h3>Locating SNPs in a genome region<a class="headerlink" href="#locating-snps-in-a-genome-region" title="Permalink to this headline">¶</a></h3>
<p>For other analyses you may want to extract data for a specific genome region, such as a gene. Again a good approach is to use an index. E.g., let’s locate data for the <em>Vgsc</em> gene, which spans the region 2L:2358158-2431617.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">loc_region</span> <span class="o">=</span> <span class="n">pos</span><span class="o">.</span><span class="n">locate_range</span><span class="p">(</span><span class="mi">2358158</span><span class="p">,</span> <span class="mi">2431617</span><span class="p">)</span>
<span class="n">loc_region</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>slice(2083135, 2152802, None)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># obtain SNP positions within selected region</span>
<span class="n">pos_region</span> <span class="o">=</span> <span class="n">pos</span><span class="p">[</span><span class="n">loc_region</span><span class="p">]</span>
<span class="n">pos_region</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div class="allel allel-DisplayAs1D"><span>&lt;SortedIndex shape=(69667,) dtype=int32&gt;</span><table><thead><tr><th style="text-align: center">0</th><th style="text-align: center">1</th><th style="text-align: center">2</th><th style="text-align: center">3</th><th style="text-align: center">4</th><th style="text-align: center">...</th><th style="text-align: center">69662</th><th style="text-align: center">69663</th><th style="text-align: center">69664</th><th style="text-align: center">69665</th><th style="text-align: center">69666</th></tr></thead><tbody><tr><td style="text-align: center">2358158</td><td style="text-align: center">2358159</td><td style="text-align: center">2358160</td><td style="text-align: center">2358161</td><td style="text-align: center">2358162</td><td style="text-align: center">...</td><td style="text-align: center">2431613</td><td style="text-align: center">2431614</td><td style="text-align: center">2431615</td><td style="text-align: center">2431616</td><td style="text-align: center">2431617</td></tr></tbody></table></div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># reference alleles within selected region</span>
<span class="n">ref_region</span> <span class="o">=</span> <span class="n">ref</span><span class="p">[</span><span class="n">loc_region</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">ref_region</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([b&#39;A&#39;, b&#39;T&#39;, b&#39;G&#39;, ..., b&#39;T&#39;, b&#39;G&#39;, b&#39;A&#39;], dtype=&#39;|S1&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># alternate alleles within selected region</span>
<span class="n">alt_region</span> <span class="o">=</span> <span class="n">alt</span><span class="p">[</span><span class="n">loc_region</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">alt_region</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([[b&#39;C&#39;, b&#39;T&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;T&#39;],
       ...,
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;G&#39;],
       [b&#39;A&#39;, b&#39;C&#39;, b&#39;T&#39;],
       [b&#39;C&#39;, b&#39;T&#39;, b&#39;G&#39;]], dtype=&#39;|S1&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># genotypes within selected region</span>
<span class="n">gt_region</span> <span class="o">=</span> <span class="n">gt</span><span class="p">[</span><span class="n">loc_region</span><span class="p">]</span>
<span class="c1"># wrap for convenience</span>
<span class="n">gt_region</span> <span class="o">=</span> <span class="n">allel</span><span class="o">.</span><span class="n">GenotypeDaskArray</span><span class="p">(</span><span class="n">gt_region</span><span class="p">)</span>
<span class="n">gt_region</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div class="allel allel-DisplayAs2D"><span>&lt;GenotypeDaskArray shape=(69667, 3081, 2) dtype=int8&gt;</span><table><thead><tr><th></th><th style="text-align: center">0</th><th style="text-align: center">1</th><th style="text-align: center">2</th><th style="text-align: center">3</th><th style="text-align: center">4</th><th style="text-align: center">...</th><th style="text-align: center">3076</th><th style="text-align: center">3077</th><th style="text-align: center">3078</th><th style="text-align: center">3079</th><th style="text-align: center">3080</th></tr></thead><tbody><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">0</th><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">1</th><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">2</th><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">...</th><td style="text-align: center" colspan="12">...</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">69664</th><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">69665</th><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">69666</th><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">...</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td><td style="text-align: center">0/0</td></tr></tbody></table></div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># allele counts</span>
<span class="n">ac_region</span> <span class="o">=</span> <span class="n">gt_region</span><span class="o">.</span><span class="n">count_alleles</span><span class="p">(</span><span class="n">max_allele</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">ac_region</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div class="allel allel-DisplayAs2D"><span>&lt;AlleleCountsArray shape=(69667, 4) dtype=int32&gt;</span><table><thead><tr><th></th><th style="text-align: center">0</th><th style="text-align: center">1</th><th style="text-align: center">2</th><th style="text-align: center">3</th></tr></thead><tbody><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">0</th><td style="text-align: center">6162</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">1</th><td style="text-align: center">6162</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">2</th><td style="text-align: center">6161</td><td style="text-align: center">   0</td><td style="text-align: center">   1</td><td style="text-align: center">   0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">...</th><td style="text-align: center" colspan="5">...</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">69664</th><td style="text-align: center">6162</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">69665</th><td style="text-align: center">6162</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td></tr><tr><th style="text-align: center; background-color: white; border-right: 1px solid black; ">69666</th><td style="text-align: center">6162</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td><td style="text-align: center">   0</td></tr></tbody></table></div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># how many segregating SNPs within our region of interest?</span>
<span class="n">ac_region</span><span class="o">.</span><span class="n">is_segregating</span><span class="p">()</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>35754
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="locating-samples">
<h2>Locating samples<a class="headerlink" href="#locating-samples" title="Permalink to this headline">¶</a></h2>
<p>In many cases you may want to access genotypes for an individual mosquito sample, or a group of samples. This usually requires linking the sample metadata to the genotypes. Let’s look at how to do that. First, load sample metadata.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_samples</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">sample_metadata</span><span class="p">(</span><span class="n">sample_sets</span><span class="o">=</span><span class="s2">&quot;v3&quot;</span><span class="p">)</span>
<span class="n">df_samples</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>partner_sample_id</th>
      <th>contributor</th>
      <th>country</th>
      <th>location</th>
      <th>year</th>
      <th>month</th>
      <th>latitude</th>
      <th>longitude</th>
      <th>sex_call</th>
      <th>sample_set</th>
      <th>release</th>
      <th>aim_fraction_colu</th>
      <th>aim_fraction_arab</th>
      <th>species_gambcolu_arabiensis</th>
      <th>species_gambiae_coluzzii</th>
      <th>species</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>AR0047-C</td>
      <td>LUA047</td>
      <td>Joao Pinto</td>
      <td>Angola</td>
      <td>Luanda</td>
      <td>2009</td>
      <td>4</td>
      <td>-8.884</td>
      <td>13.302</td>
      <td>F</td>
      <td>AG1000G-AO</td>
      <td>v3</td>
      <td>0.945</td>
      <td>0.001</td>
      <td>gamb_colu</td>
      <td>coluzzii</td>
      <td>coluzzii</td>
    </tr>
    <tr>
      <th>1</th>
      <td>AR0049-C</td>
      <td>LUA049</td>
      <td>Joao Pinto</td>
      <td>Angola</td>
      <td>Luanda</td>
      <td>2009</td>
      <td>4</td>
      <td>-8.884</td>
      <td>13.302</td>
      <td>F</td>
      <td>AG1000G-AO</td>
      <td>v3</td>
      <td>0.933</td>
      <td>0.001</td>
      <td>gamb_colu</td>
      <td>coluzzii</td>
      <td>coluzzii</td>
    </tr>
    <tr>
      <th>2</th>
      <td>AR0051-C</td>
      <td>LUA051</td>
      <td>Joao Pinto</td>
      <td>Angola</td>
      <td>Luanda</td>
      <td>2009</td>
      <td>4</td>
      <td>-8.884</td>
      <td>13.302</td>
      <td>F</td>
      <td>AG1000G-AO</td>
      <td>v3</td>
      <td>0.937</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>coluzzii</td>
      <td>coluzzii</td>
    </tr>
    <tr>
      <th>3</th>
      <td>AR0061-C</td>
      <td>LUA061</td>
      <td>Joao Pinto</td>
      <td>Angola</td>
      <td>Luanda</td>
      <td>2009</td>
      <td>4</td>
      <td>-8.884</td>
      <td>13.302</td>
      <td>F</td>
      <td>AG1000G-AO</td>
      <td>v3</td>
      <td>0.938</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>coluzzii</td>
      <td>coluzzii</td>
    </tr>
    <tr>
      <th>4</th>
      <td>AR0078-C</td>
      <td>LUA078</td>
      <td>Joao Pinto</td>
      <td>Angola</td>
      <td>Luanda</td>
      <td>2009</td>
      <td>4</td>
      <td>-8.884</td>
      <td>13.302</td>
      <td>F</td>
      <td>AG1000G-AO</td>
      <td>v3</td>
      <td>0.926</td>
      <td>0.001</td>
      <td>gamb_colu</td>
      <td>coluzzii</td>
      <td>coluzzii</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>3076</th>
      <td>AD0494-C</td>
      <td>80-2-o-16</td>
      <td>Martin Donnelly</td>
      <td>Lab Cross</td>
      <td>LSTM</td>
      <td>-1</td>
      <td>-1</td>
      <td>53.409</td>
      <td>-2.969</td>
      <td>F</td>
      <td>AG1000G-X</td>
      <td>v3</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3077</th>
      <td>AD0495-C</td>
      <td>80-2-o-17</td>
      <td>Martin Donnelly</td>
      <td>Lab Cross</td>
      <td>LSTM</td>
      <td>-1</td>
      <td>-1</td>
      <td>53.409</td>
      <td>-2.969</td>
      <td>M</td>
      <td>AG1000G-X</td>
      <td>v3</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3078</th>
      <td>AD0496-C</td>
      <td>80-2-o-18</td>
      <td>Martin Donnelly</td>
      <td>Lab Cross</td>
      <td>LSTM</td>
      <td>-1</td>
      <td>-1</td>
      <td>53.409</td>
      <td>-2.969</td>
      <td>M</td>
      <td>AG1000G-X</td>
      <td>v3</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3079</th>
      <td>AD0497-C</td>
      <td>80-2-o-19</td>
      <td>Martin Donnelly</td>
      <td>Lab Cross</td>
      <td>LSTM</td>
      <td>-1</td>
      <td>-1</td>
      <td>53.409</td>
      <td>-2.969</td>
      <td>F</td>
      <td>AG1000G-X</td>
      <td>v3</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3080</th>
      <td>AD0498-C</td>
      <td>80-2-o-20</td>
      <td>Martin Donnelly</td>
      <td>Lab Cross</td>
      <td>LSTM</td>
      <td>-1</td>
      <td>-1</td>
      <td>53.409</td>
      <td>-2.969</td>
      <td>M</td>
      <td>AG1000G-X</td>
      <td>v3</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
<p>3081 rows × 17 columns</p>
</div></div></div>
</div>
<p>Now access genotypes for whatever chromosome arm you are analysing, e.g., 2L.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_genotypes</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2L&quot;</span><span class="p">,</span> <span class="n">sample_sets</span><span class="o">=</span><span class="s2">&quot;v3&quot;</span><span class="p">)</span>
<span class="n">gt</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 299.02 GB </td> <td> 122.40 MB </td></tr>
    <tr><th> Shape </th><td> (48525747, 3081, 2) </td> <td> (600000, 102, 2) </td></tr>
    <tr><th> Count </th><td> 16972 Tasks </td><td> 5080 Chunks </td></tr>
    <tr><th> Type </th><td> int8 </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="1" x2="80" y2="72" />
  <line x1="10" y1="3" x2="80" y2="73" />
  <line x1="10" y1="3" x2="80" y2="74" />
  <line x1="10" y1="4" x2="80" y2="75" />
  <line x1="10" y1="6" x2="80" y2="77" />
  <line x1="10" y1="7" x2="80" y2="78" />
  <line x1="10" y1="8" x2="80" y2="79" />
  <line x1="10" y1="9" x2="80" y2="80" />
  <line x1="10" y1="10" x2="80" y2="81" />
  <line x1="10" y1="12" x2="80" y2="83" />
  <line x1="10" y1="13" x2="80" y2="84" />
  <line x1="10" y1="14" x2="80" y2="85" />
  <line x1="10" y1="16" x2="80" y2="86" />
  <line x1="10" y1="17" x2="80" y2="87" />
  <line x1="10" y1="18" x2="80" y2="88" />
  <line x1="10" y1="19" x2="80" y2="90" />
  <line x1="10" y1="21" x2="80" y2="91" />
  <line x1="10" y1="22" x2="80" y2="93" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="13" y1="3" x2="13" y2="28" />
  <line x1="16" y1="6" x2="16" y2="32" />
  <line x1="20" y1="10" x2="20" y2="36" />
  <line x1="24" y1="14" x2="24" y2="39" />
  <line x1="28" y1="18" x2="28" y2="43" />
  <line x1="32" y1="22" x2="32" y2="47" />
  <line x1="35" y1="25" x2="35" y2="50" />
  <line x1="39" y1="29" x2="39" y2="55" />
  <line x1="43" y1="33" x2="43" y2="59" />
  <line x1="47" y1="37" x2="47" y2="62" />
  <line x1="50" y1="40" x2="50" y2="65" />
  <line x1="54" y1="44" x2="54" y2="69" />
  <line x1="58" y1="48" x2="58" y2="73" />
  <line x1="61" y1="51" x2="61" y2="77" />
  <line x1="65" y1="55" x2="65" y2="81" />
  <line x1="68" y1="58" x2="68" y2="84" />
  <line x1="73" y1="63" x2="73" y2="88" />
  <line x1="77" y1="67" x2="77" y2="92" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.41261651458248" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="13" y1="3" x2="38" y2="3" />
  <line x1="16" y1="6" x2="42" y2="6" />
  <line x1="20" y1="10" x2="46" y2="10" />
  <line x1="24" y1="14" x2="49" y2="14" />
  <line x1="28" y1="18" x2="53" y2="18" />
  <line x1="32" y1="22" x2="57" y2="22" />
  <line x1="35" y1="25" x2="60" y2="25" />
  <line x1="39" y1="29" x2="65" y2="29" />
  <line x1="43" y1="33" x2="69" y2="33" />
  <line x1="47" y1="37" x2="72" y2="37" />
  <line x1="50" y1="40" x2="75" y2="40" />
  <line x1="54" y1="44" x2="79" y2="44" />
  <line x1="58" y1="48" x2="83" y2="48" />
  <line x1="61" y1="51" x2="87" y2="51" />
  <line x1="65" y1="55" x2="91" y2="55" />
  <line x1="68" y1="58" x2="94" y2="58" />
  <line x1="73" y1="63" x2="98" y2="63" />
  <line x1="77" y1="67" x2="102" y2="67" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="72" x2="106" y2="72" />
  <line x1="80" y1="73" x2="106" y2="73" />
  <line x1="80" y1="74" x2="106" y2="74" />
  <line x1="80" y1="75" x2="106" y2="75" />
  <line x1="80" y1="77" x2="106" y2="77" />
  <line x1="80" y1="78" x2="106" y2="78" />
  <line x1="80" y1="79" x2="106" y2="79" />
  <line x1="80" y1="80" x2="106" y2="80" />
  <line x1="80" y1="81" x2="106" y2="81" />
  <line x1="80" y1="83" x2="106" y2="83" />
  <line x1="80" y1="84" x2="106" y2="84" />
  <line x1="80" y1="85" x2="106" y2="85" />
  <line x1="80" y1="86" x2="106" y2="86" />
  <line x1="80" y1="87" x2="106" y2="87" />
  <line x1="80" y1="88" x2="106" y2="88" />
  <line x1="80" y1="90" x2="106" y2="90" />
  <line x1="80" y1="91" x2="106" y2="91" />
  <line x1="80" y1="93" x2="106" y2="93" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,126.000852,83.294544)">3081</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">48525747</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<p>The key point to know here is that the number of rows in the sample metadata dataframe and the number of columns in the genotype array are the same.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">len</span><span class="p">(</span><span class="n">df_samples</span><span class="p">)</span> <span class="o">==</span> <span class="n">gt</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>True
</pre></div>
</div>
</div>
</div>
<p>Also, the order of samples in the sample metadata is the same as the order of the columns in the genotype array. We can use this correspondance to look up samples and access their genotypes.</p>
<div class="section" id="locating-a-single-sample">
<h3>Locating a single sample<a class="headerlink" href="#locating-a-single-sample" title="Permalink to this headline">¶</a></h3>
<p>Let’s extract both sample metadata and genotypes for a single sample, e.g., sample AB0110-C.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">loc_sample</span> <span class="o">=</span> <span class="n">df_samples</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s2">&quot;sample_id == &#39;AB0110-C&#39;&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="n">loc_sample</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>100
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df_samples</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">loc_sample</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>sample_id                          AB0110-C
partner_sample_id                     BF9-2
contributor                     Austin Burt
country                        Burkina Faso
location                               Bana
year                                   2012
month                                     7
latitude                             11.233
longitude                            -4.472
sex_call                                  F
sample_set                     AG1000G-BF-A
release                                  v3
aim_fraction_colu                     0.974
aim_fraction_arab                     0.002
species_gambcolu_arabiensis       gamb_colu
species_gambiae_coluzzii           coluzzii
species                            coluzzii
Name: 100, dtype: object
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt_sample</span> <span class="o">=</span> <span class="n">gt</span><span class="p">[:,</span> <span class="n">loc_sample</span><span class="p">]</span>
<span class="n">gt_sample</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 97.05 MB </td> <td> 1.20 MB </td></tr>
    <tr><th> Shape </th><td> (48525747, 2) </td> <td> (600000, 2) </td></tr>
    <tr><th> Count </th><td> 17099 Tasks </td><td> 127 Chunks </td></tr>
    <tr><th> Type </th><td> int8 </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="75" height="170" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="0" y1="0" x2="25" y2="0" style="stroke-width:2" />
  <line x1="0" y1="5" x2="25" y2="5" />
  <line x1="0" y1="11" x2="25" y2="11" />
  <line x1="0" y1="18" x2="25" y2="18" />
  <line x1="0" y1="24" x2="25" y2="24" />
  <line x1="0" y1="31" x2="25" y2="31" />
  <line x1="0" y1="37" x2="25" y2="37" />
  <line x1="0" y1="43" x2="25" y2="43" />
  <line x1="0" y1="50" x2="25" y2="50" />
  <line x1="0" y1="57" x2="25" y2="57" />
  <line x1="0" y1="63" x2="25" y2="63" />
  <line x1="0" y1="68" x2="25" y2="68" />
  <line x1="0" y1="75" x2="25" y2="75" />
  <line x1="0" y1="82" x2="25" y2="82" />
  <line x1="0" y1="88" x2="25" y2="88" />
  <line x1="0" y1="94" x2="25" y2="94" />
  <line x1="0" y1="100" x2="25" y2="100" />
  <line x1="0" y1="107" x2="25" y2="107" />
  <line x1="0" y1="114" x2="25" y2="114" />
  <line x1="0" y1="120" x2="25" y2="120" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="0" y1="0" x2="0" y2="120" style="stroke-width:2" />
  <line x1="25" y1="0" x2="25" y2="120" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="12.706308" y="140.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="45.412617" y="60.000000" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,45.412617,60.000000)">48525747</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># compute some genotype counts for our sample of interest</span>
<span class="n">gt_sample</span> <span class="o">=</span> <span class="n">allel</span><span class="o">.</span><span class="n">GenotypeVector</span><span class="p">(</span><span class="n">gt_sample</span><span class="o">.</span><span class="n">compute</span><span class="p">())</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt_sample</span><span class="o">.</span><span class="n">count_missing</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3167911
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt_sample</span><span class="o">.</span><span class="n">count_hom_ref</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>43732318
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt_sample</span><span class="o">.</span><span class="n">count_het</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>816765
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">gt_sample</span><span class="o">.</span><span class="n">count_hom_alt</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>808753
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="locating-multiple-samples">
<h3>Locating multiple samples<a class="headerlink" href="#locating-multiple-samples" title="Permalink to this headline">¶</a></h3>
<p>We will also often want to locate a group of samples based on their metadata, e.g., all samples from a given country, collected in a certain year, of a certain species; and then analyse their genotypes. Let’s look at an example.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># select An. gambiae samples from Burkina Faso collected in 2012</span>
<span class="n">query</span> <span class="o">=</span> <span class="s2">&quot;(country == &#39;Burkina Faso&#39;) and (year == 2012) and (species == &#39;gambiae&#39;)&quot;</span>
<span class="n">loc_samples</span> <span class="o">=</span> <span class="n">df_samples</span><span class="o">.</span><span class="n">eval</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="o">.</span><span class="n">values</span>
<span class="n">loc_samples</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([False, False, False, ..., False, False, False])
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># examing their metadata</span>
<span class="n">df_samples_selected</span> <span class="o">=</span> <span class="n">df_samples</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">loc_samples</span><span class="p">]</span>
<span class="n">df_samples_selected</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>partner_sample_id</th>
      <th>contributor</th>
      <th>country</th>
      <th>location</th>
      <th>year</th>
      <th>month</th>
      <th>latitude</th>
      <th>longitude</th>
      <th>sex_call</th>
      <th>sample_set</th>
      <th>release</th>
      <th>aim_fraction_colu</th>
      <th>aim_fraction_arab</th>
      <th>species_gambcolu_arabiensis</th>
      <th>species_gambiae_coluzzii</th>
      <th>species</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>81</th>
      <td>AB0085-Cx</td>
      <td>BF2-4</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Pala</td>
      <td>2012</td>
      <td>7</td>
      <td>11.150</td>
      <td>-4.235</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.024</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>82</th>
      <td>AB0086-Cx</td>
      <td>BF2-6</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Pala</td>
      <td>2012</td>
      <td>7</td>
      <td>11.150</td>
      <td>-4.235</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.038</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>91</th>
      <td>AB0096-C</td>
      <td>BF4-4</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Bana</td>
      <td>2012</td>
      <td>7</td>
      <td>11.233</td>
      <td>-4.472</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.032</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>97</th>
      <td>AB0103-C</td>
      <td>BF6-7</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Bana</td>
      <td>2012</td>
      <td>7</td>
      <td>11.233</td>
      <td>-4.472</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.040</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>98</th>
      <td>AB0104-Cx</td>
      <td>BF6-8</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Bana</td>
      <td>2012</td>
      <td>7</td>
      <td>11.233</td>
      <td>-4.472</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.026</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>255</th>
      <td>AB0278-C</td>
      <td>BF12-29</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Pala</td>
      <td>2012</td>
      <td>7</td>
      <td>11.150</td>
      <td>-4.235</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.017</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>257</th>
      <td>AB0280-Cx</td>
      <td>BF12-31</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Pala</td>
      <td>2012</td>
      <td>7</td>
      <td>11.150</td>
      <td>-4.235</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.065</td>
      <td>0.003</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>258</th>
      <td>AB0281-Cx</td>
      <td>BF12-32</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Pala</td>
      <td>2012</td>
      <td>7</td>
      <td>11.150</td>
      <td>-4.235</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.029</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>260</th>
      <td>AB0283-C</td>
      <td>BF10-12</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Pala</td>
      <td>2012</td>
      <td>7</td>
      <td>11.150</td>
      <td>-4.235</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.028</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
    <tr>
      <th>261</th>
      <td>AB0284-C</td>
      <td>BF10-13</td>
      <td>Austin Burt</td>
      <td>Burkina Faso</td>
      <td>Pala</td>
      <td>2012</td>
      <td>7</td>
      <td>11.150</td>
      <td>-4.235</td>
      <td>F</td>
      <td>AG1000G-BF-A</td>
      <td>v3</td>
      <td>0.024</td>
      <td>0.002</td>
      <td>gamb_colu</td>
      <td>gambiae</td>
      <td>gambiae</td>
    </tr>
  </tbody>
</table>
<p>98 rows × 17 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># select genotypes</span>
<span class="n">gt_samples</span> <span class="o">=</span> <span class="n">gt</span><span class="p">[:,</span> <span class="n">loc_samples</span><span class="p">]</span>
<span class="n">gt_samples</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 9.51 GB </td> <td> 2.40 MB </td></tr>
    <tr><th> Shape </th><td> (48525747, 98, 2) </td> <td> (600000, 2, 2) </td></tr>
    <tr><th> Count </th><td> 23195 Tasks </td><td> 6223 Chunks </td></tr>
    <tr><th> Type </th><td> int8 </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="1" x2="80" y2="71" />
  <line x1="10" y1="2" x2="80" y2="73" />
  <line x1="10" y1="3" x2="80" y2="74" />
  <line x1="10" y1="5" x2="80" y2="75" />
  <line x1="10" y1="6" x2="80" y2="76" />
  <line x1="10" y1="7" x2="80" y2="78" />
  <line x1="10" y1="9" x2="80" y2="79" />
  <line x1="10" y1="10" x2="80" y2="80" />
  <line x1="10" y1="11" x2="80" y2="82" />
  <line x1="10" y1="12" x2="80" y2="83" />
  <line x1="10" y1="14" x2="80" y2="85" />
  <line x1="10" y1="15" x2="80" y2="86" />
  <line x1="10" y1="17" x2="80" y2="87" />
  <line x1="10" y1="18" x2="80" y2="89" />
  <line x1="10" y1="19" x2="80" y2="90" />
  <line x1="10" y1="21" x2="80" y2="91" />
  <line x1="10" y1="22" x2="80" y2="92" />
  <line x1="10" y1="23" x2="80" y2="94" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="13" y1="3" x2="13" y2="28" />
  <line x1="16" y1="6" x2="16" y2="32" />
  <line x1="20" y1="10" x2="20" y2="36" />
  <line x1="24" y1="14" x2="24" y2="39" />
  <line x1="28" y1="18" x2="28" y2="43" />
  <line x1="32" y1="22" x2="32" y2="47" />
  <line x1="35" y1="25" x2="35" y2="50" />
  <line x1="39" y1="29" x2="39" y2="55" />
  <line x1="43" y1="33" x2="43" y2="59" />
  <line x1="47" y1="37" x2="47" y2="62" />
  <line x1="50" y1="40" x2="50" y2="65" />
  <line x1="54" y1="44" x2="54" y2="69" />
  <line x1="58" y1="48" x2="58" y2="73" />
  <line x1="61" y1="51" x2="61" y2="77" />
  <line x1="65" y1="55" x2="65" y2="81" />
  <line x1="68" y1="58" x2="68" y2="84" />
  <line x1="73" y1="63" x2="73" y2="88" />
  <line x1="77" y1="67" x2="77" y2="92" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="13" y1="3" x2="38" y2="3" />
  <line x1="16" y1="6" x2="42" y2="6" />
  <line x1="20" y1="10" x2="46" y2="10" />
  <line x1="24" y1="14" x2="49" y2="14" />
  <line x1="28" y1="18" x2="53" y2="18" />
  <line x1="32" y1="22" x2="57" y2="22" />
  <line x1="35" y1="25" x2="60" y2="25" />
  <line x1="39" y1="29" x2="65" y2="29" />
  <line x1="43" y1="33" x2="69" y2="33" />
  <line x1="47" y1="37" x2="72" y2="37" />
  <line x1="50" y1="40" x2="75" y2="40" />
  <line x1="54" y1="44" x2="79" y2="44" />
  <line x1="58" y1="48" x2="83" y2="48" />
  <line x1="61" y1="51" x2="87" y2="51" />
  <line x1="65" y1="55" x2="91" y2="55" />
  <line x1="68" y1="58" x2="94" y2="58" />
  <line x1="73" y1="63" x2="98" y2="63" />
  <line x1="77" y1="67" x2="102" y2="67" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="71" x2="106" y2="71" />
  <line x1="80" y1="73" x2="106" y2="73" />
  <line x1="80" y1="74" x2="106" y2="74" />
  <line x1="80" y1="75" x2="106" y2="75" />
  <line x1="80" y1="76" x2="106" y2="76" />
  <line x1="80" y1="78" x2="106" y2="78" />
  <line x1="80" y1="79" x2="106" y2="79" />
  <line x1="80" y1="80" x2="106" y2="80" />
  <line x1="80" y1="82" x2="106" y2="82" />
  <line x1="80" y1="83" x2="106" y2="83" />
  <line x1="80" y1="85" x2="106" y2="85" />
  <line x1="80" y1="86" x2="106" y2="86" />
  <line x1="80" y1="87" x2="106" y2="87" />
  <line x1="80" y1="89" x2="106" y2="89" />
  <line x1="80" y1="90" x2="106" y2="90" />
  <line x1="80" y1="91" x2="106" y2="91" />
  <line x1="80" y1="92" x2="106" y2="92" />
  <line x1="80" y1="94" x2="106" y2="94" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#8B4903A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,126.000852,83.294544)">98</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">48525747</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<p>Note above that we now have a genotype array with 98 columns, which corresponds to the 98 samples we selected with our sample metadata query.</p>
</div>
</div>
<div class="section" id="selecting-sites-and-samples">
<h2>Selecting sites and samples<a class="headerlink" href="#selecting-sites-and-samples" title="Permalink to this headline">¶</a></h2>
<p>We can also combine site selections with sample selections, e.g.:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># obtain genotype at 2L:2422652 in sample AB0110-C</span>
<span class="n">gt</span><span class="p">[</span><span class="n">loc_site</span><span class="p">,</span> <span class="n">loc_sample</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>array([0, 2], dtype=int8)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># access genotypes in the Vgsc gene in An. gambiae samples from Burkina Faso collected in 2012</span>
<span class="n">gt_sel</span> <span class="o">=</span> <span class="n">gt</span><span class="p">[</span><span class="n">loc_region</span><span class="p">,</span> <span class="n">loc_samples</span><span class="p">]</span>
<span class="n">gt_sel</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table>
<tr>
<td>
<table>
  <thead>
    <tr><td> </td><th> Array </th><th> Chunk </th></tr>
  </thead>
  <tbody>
    <tr><th> Bytes </th><td> 13.65 MB </td> <td> 5.70 MB </td></tr>
    <tr><th> Shape </th><td> (69667, 98, 2) </td> <td> (52802, 54, 2) </td></tr>
    <tr><th> Count </th><td> 17056 Tasks </td><td> 4 Chunks </td></tr>
    <tr><th> Type </th><td> int8 </td><td> numpy.ndarray </td></tr>
  </tbody>
</table>
</td>
<td>
<svg width="156" height="146" style="stroke:rgb(0,0,0);stroke-width:1" >

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="10" y1="14" x2="80" y2="84" />
  <line x1="10" y1="25" x2="80" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="10" y2="25" style="stroke-width:2" />
  <line x1="27" y1="17" x2="27" y2="42" />
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="10" y1="0" x2="35" y2="0" style="stroke-width:2" />
  <line x1="27" y1="17" x2="52" y2="17" />
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="10" y1="0" x2="80" y2="70" style="stroke-width:2" />
  <line x1="35" y1="0" x2="106" y2="70" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Horizontal lines -->
  <line x1="80" y1="70" x2="106" y2="70" style="stroke-width:2" />
  <line x1="80" y1="84" x2="106" y2="84" />
  <line x1="80" y1="96" x2="106" y2="96" style="stroke-width:2" />

  <!-- Vertical lines -->
  <line x1="80" y1="70" x2="80" y2="96" style="stroke-width:2" />
  <line x1="106" y1="70" x2="106" y2="96" style="stroke-width:2" />

  <!-- Colored Rectangle -->
  <polygon points="80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013" style="fill:#ECB172A0;stroke-width:0"/>

  <!-- Text -->
  <text x="93.294544" y="116.000852" font-size="1.0rem" font-weight="100" text-anchor="middle" >2</text>
  <text x="126.000852" y="83.294544" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,126.000852,83.294544)">98</text>
  <text x="35.294118" y="80.706734" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,35.294118,80.706734)">69667</text>
</svg>
</td>
</tr>
</table></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># how many segregating sites now?</span>
<span class="n">ac</span> <span class="o">=</span> <span class="n">allel</span><span class="o">.</span><span class="n">GenotypeDaskArray</span><span class="p">(</span><span class="n">gt_sel</span><span class="p">)</span><span class="o">.</span><span class="n">count_alleles</span><span class="p">(</span><span class="n">max_allele</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">ac</span><span class="o">.</span><span class="n">is_segregating</span><span class="p">()</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>3858
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="differences-from-previous-data-releases">
<h2>Differences from previous data releases<a class="headerlink" href="#differences-from-previous-data-releases" title="Permalink to this headline">¶</a></h2>
<p>In case you have worked with previous data releases from the Ag1000G project, there are a couple of important differences to be aware of. These are touched on in the text above, but let’s expand a little here.</p>
<div class="section" id="genotyping-at-all-sites">
<h3>Genotyping at all sites<a class="headerlink" href="#genotyping-at-all-sites" title="Permalink to this headline">¶</a></h3>
<p>In Ag3 we have genotyped all samples at all genomic sites where the reference genome is not “N”, and those genotypes are included in the released data. This includes genotypes at sites where no variation from the reference genome was observed. This is different from previous releases, where we only released genotypes at sites where some variation was observed.</p>
<p>The reason why we made this change is because several important use cases for these data involve identifying genomic sites which are highly conserved, i.e., where samples do not vary at all. Because we now report genotypes at all sites, it is possible to identify these highly conserved sites with more confidence.</p>
</div>
<div class="section" id="predefined-alternate-alleles">
<h3>Predefined alternate alleles<a class="headerlink" href="#predefined-alternate-alleles" title="Permalink to this headline">¶</a></h3>
<p>In Ag3 we predefined the set of possible alternate SNP alleles at all sites in the genome, before we ran our SNP genotyping pipeline on the samples sequenced. This means that our data includes three alternate alleles at every site in the genome. Not all of these alleles will have been observed in the data, and determining which alleles are observed requires examination of the genotypes. This is different from previous releases, where alternate alleles were declared during the SNP calling process, and only observed alternate alleles were included in the data release.</p>
<p>The reason why we have made this change is because it simplifies the SNP genotyping pipeline. Because we fix the set of alleles ahead of time, all samples can be genotyped independently, and merging the resulting data is straightforward. This also makes it easier to integrate with new data in the future, because it allows merging Ag3 data with data from future samples where alleles may be observed that were not observed in the Ag3 cohort.</p>
</div>
<div class="section" id="id1">
<h3>Site filters<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<p>In Ag3 we have defined site filters, which are Boolean masks that apply to all sites in the genome, and which identify sites where SNP genotyping is reliable.These site filters replace the separate variant filters and genome accessibility masks defined in previous project phases.</p>
<p>The reason why we took this approach is that it is simpler. All sites are treated the same, regardless of whether variation is observed or not. Instead of having to work two different things (variant filters and genome accessibility masks) there is just a single site filter to work with.</p>
<p>Note that some software tools such as scikit-allel may require you to provide a genome accessibility mask, which is a Boolean array of the same length as the chromosomal sequence you’re analysing. In case you need this, the <code class="docutils literal notranslate"><span class="pre">Ag3</span></code> class provides a convenience method to obtain this from the site filters. E.g.:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">is_accessible</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">is_accessible</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">,</span> <span class="n">site_mask</span><span class="o">=</span><span class="s2">&quot;gamb_colu_arab&quot;</span><span class="p">)</span>
<span class="n">is_accessible</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(61545105,)
</pre></div>
</div>
</div>
</div>
<p>This is an array the same length as the chromosomal sequence, and is True where the reference genome is not an “N” and the site filter passes.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">seq</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">genome_sequence</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">seq</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(61545105,)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">len</span><span class="p">(</span><span class="n">seq</span><span class="p">)</span> <span class="o">==</span> <span class="nb">len</span><span class="p">(</span><span class="n">is_accessible</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>True
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">is_accessible</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>40561667
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">loc_pass</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">site_filters</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">,</span> <span class="n">mask</span><span class="o">=</span><span class="s2">&quot;gamb_colu_arab&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">loc_pass</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(60132453,)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">loc_pass</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>40561667
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pos</span> <span class="o">=</span> <span class="n">ag3</span><span class="o">.</span><span class="n">snp_sites</span><span class="p">(</span><span class="n">contig</span><span class="o">=</span><span class="s2">&quot;2R&quot;</span><span class="p">,</span> <span class="n">field</span><span class="o">=</span><span class="s2">&quot;POS&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
<span class="n">pos</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(60132453,)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">np</span><span class="o">.</span><span class="n">all</span><span class="p">(</span><span class="n">loc_pass</span> <span class="o">==</span> <span class="n">is_accessible</span><span class="p">[</span><span class="n">pos</span> <span class="o">-</span> <span class="mi">1</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>True
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="further-reading">
<h2>Further reading<a class="headerlink" href="#further-reading" title="Permalink to this headline">¶</a></h2>
<p>Hopefully the examples above have helped to give some sense of how to access and work with SNP data from Ag3. If you have any comments or questions, please <a class="reference external" href="https://github.com/malariagen/vector-data/discussions">get in touch</a>.</p>
<p>This notebook is part of the <a class="reference internal" href="../intro.html"><span class="doc std std-doc">Ag1000G phase 3 user guide</span></a>. See the menu at the left for more documentation and example analyses.</p>
</div>
<div class="section" id="api-docs">
<h2>API docs<a class="headerlink" href="#api-docs" title="Permalink to this headline">¶</a></h2>
<p>Here are the docstrings for the functions in the <code class="docutils literal notranslate"><span class="pre">malariagen_data</span></code> package we used in this notebook.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">genome_sequence</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Help on method genome_sequence in module malariagen_data.ag3:

genome_sequence(contig, inline_array=True, chunks=&#39;auto&#39;) method of malariagen_data.ag3.Ag3 instance
    Access the reference genome sequence.
    
    Parameters
    ----------
    contig : str
        Chromosome arm, e.g., &quot;3R&quot;.
    inline_array : bool, optional
        Passed through to dask.array.from_array().
    chunks : str, optional
        If &#39;auto&#39; let dask decide chunk size. If &#39;native&#39; use native zarr chunks.
        Also can be a target size, e.g., &#39;200 MiB&#39;.
    
    Returns
    -------
    d : dask.array.Array
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">sample_metadata</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Help on method sample_metadata in module malariagen_data.ag3:

sample_metadata(sample_sets=&#39;v3_wild&#39;, species_calls=(&#39;20200422&#39;, &#39;aim&#39;)) method of malariagen_data.ag3.Ag3 instance
    Access sample metadata for one or more sample sets.
    
    Parameters
    ----------
    sample_sets : str or list of str
        Can be a sample set identifier (e.g., &quot;AG1000G-AO&quot;) or a list of sample set
        identifiers (e.g., [&quot;AG1000G-BF-A&quot;, &quot;AG1000G-BF-B&quot;]) or a release identifier (e.g.,
        &quot;v3&quot;) or a list of release identifiers.
    species_calls : (str, str), optional
        Include species calls in metadata.
    
    Returns
    -------
    df : pandas.DataFrame
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">snp_sites</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Help on method snp_sites in module malariagen_data.ag3:

snp_sites(contig, field=None, site_mask=None, site_filters=&#39;dt_20200416&#39;, inline_array=True, chunks=&#39;auto&#39;) method of malariagen_data.ag3.Ag3 instance
    Access SNP site data (positions and alleles).
    
    Parameters
    ----------
    contig : str
        Chromosome arm, e.g., &quot;3R&quot;.
    field : {&quot;POS&quot;, &quot;REF&quot;, &quot;ALT&quot;}, optional
        Array to access. If not provided, all three arrays POS, REF, ALT will be returned as a
        tuple.
    site_mask : {&quot;gamb_colu_arab&quot;, &quot;gamb_colu&quot;, &quot;arab&quot;}
        Site filters mask to apply.
    site_filters : str
        Site filters analysis version.
    inline_array : bool, optional
        Passed through to dask.array.from_array().
    chunks : str, optional
        If &#39;auto&#39; let dask decide chunk size. If &#39;native&#39; use native zarr chunks.
        Also can be a target size, e.g., &#39;200 MiB&#39;.
    
    Returns
    -------
    d : dask.array.Array or tuple of dask.array.Array
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">site_filters</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Help on method site_filters in module malariagen_data.ag3:

site_filters(contig, mask, field=&#39;filter_pass&#39;, analysis=&#39;dt_20200416&#39;, inline_array=True, chunks=&#39;auto&#39;) method of malariagen_data.ag3.Ag3 instance
    Access SNP site filters.
    
    Parameters
    ----------
    contig : str
        Chromosome arm, e.g., &quot;3R&quot;.
    mask : {&quot;gamb_colu_arab&quot;, &quot;gamb_colu&quot;, &quot;arab&quot;}
        Mask to use.
    field : str, optional
        Array to access.
    analysis : str, optional
        Site filters analysis version.
    inline_array : bool, optional
        Passed through to dask.from_array().
    chunks : str, optional
        If &#39;auto&#39; let dask decide chunk size. If &#39;native&#39; use native zarr chunks.
        Also can be a target size, e.g., &#39;200 MiB&#39;.
    
    Returns
    -------
    d : dask.array.Array
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">snp_genotypes</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Help on method snp_genotypes in module malariagen_data.ag3:

snp_genotypes(contig, sample_sets=&#39;v3_wild&#39;, field=&#39;GT&#39;, site_mask=None, site_filters=&#39;dt_20200416&#39;, inline_array=True, chunks=&#39;auto&#39;) method of malariagen_data.ag3.Ag3 instance
    Access SNP genotypes and associated data.
    
    Parameters
    ----------
    contig : str
        Chromosome arm, e.g., &quot;3R&quot;.
    sample_sets : str or list of str
        Can be a sample set identifier (e.g., &quot;AG1000G-AO&quot;) or a list of sample set
        identifiers (e.g., [&quot;AG1000G-BF-A&quot;, &quot;AG1000G-BF-B&quot;]) or a release identifier (e.g.,
        &quot;v3&quot;) or a list of release identifiers.
    field : {&quot;GT&quot;, &quot;GQ&quot;, &quot;AD&quot;, &quot;MQ&quot;}
        Array to access.
    site_mask : {&quot;gamb_colu_arab&quot;, &quot;gamb_colu&quot;, &quot;arab&quot;}
        Site filters mask to apply.
    site_filters : str, optional
        Site filters analysis version.
    inline_array : bool, optional
        Passed through to dask.array.from_array().
    chunks : str, optional
        If &#39;auto&#39; let dask decide chunk size. If &#39;native&#39; use native zarr chunks.
        Also can be a target size, e.g., &#39;200 MiB&#39;.
    
    Returns
    -------
    d : dask.array.Array
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">help</span><span class="p">(</span><span class="n">ag3</span><span class="o">.</span><span class="n">snp_calls</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Help on method snp_calls in module malariagen_data.ag3:

snp_calls(contig, sample_sets=&#39;v3_wild&#39;, site_mask=None, site_filters=&#39;dt_20200416&#39;, inline_array=True, chunks=&#39;auto&#39;) method of malariagen_data.ag3.Ag3 instance
    Access SNP sites, site filters and genotype calls.
    
    Parameters
    ----------
    contig : str
        Chromosome arm, e.g., &quot;3R&quot;.
    sample_sets : str or list of str
        Can be a sample set identifier (e.g., &quot;AG1000G-AO&quot;) or a list of sample set
        identifiers (e.g., [&quot;AG1000G-BF-A&quot;, &quot;AG1000G-BF-B&quot;]) or a release identifier (e.g.,
        &quot;v3&quot;) or a list of release identifiers.
    site_mask : {&quot;gamb_colu_arab&quot;, &quot;gamb_colu&quot;, &quot;arab&quot;}
        Site filters mask to apply.
    site_filters : str
        Site filters analysis version.
    inline_array : bool, optional
        Passed through to dask.array.from_array().
    chunks : str, optional
        If &#39;auto&#39; let dask decide chunk size. If &#39;native&#39; use native zarr chunks.
        Also can be a target size, e.g., &#39;200 MiB&#39;.
    
    Returns
    -------
    ds : xarray.Dataset
</pre></div>
</div>
</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./ag3/examples"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
            



<div class='prev-next-bottom'>
    
    <div id="prev">
        <a class="left-prev" href="collection-sites.html" title="previous page">
            <i class="prevnext-label fas fa-angle-left"></i>
            <div class="prevnext-info">
                <p class="prevnext-label">previous</p>
                <p class="prevnext-title">Mosquito collection sites</p>
            </div>
        </a>
    </div>
     <div id="next">
        <a class="right-next" href="snp-effects.html" title="next page">
            <div class="prevnext-info">
                <p class="prevnext-label">next</p>
                <p class="prevnext-title">Genes and SNP effects</p>
            </div>
            <i class="prevnext-label fas fa-angle-right"></i>
        </a>
     </div>

</div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By MalariaGEN<br/>
        
            &copy; Copyright 2021.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>