
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Vector Observatory Data Access &#8212; MalariaGEN vector data user guide</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/myfile.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script kind="utterances">

    var commentsRunWhenDOMLoaded = cb => {
    if (document.readyState != 'loading') {
        cb()
    } else if (document.addEventListener) {
        document.addEventListener('DOMContentLoaded', cb)
    } else {
        document.attachEvent('onreadystatechange', function() {
        if (document.readyState == 'complete') cb()
        })
    }
}

var addUtterances = () => {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://utteranc.es/client.js";
    script.async = "async";

    script.setAttribute("repo", "malariagen/vector-data");
    script.setAttribute("issue-term", "pathname");
    script.setAttribute("theme", "github-light");
    script.setAttribute("label", "comment");
    script.setAttribute("crossorigin", "anonymous");

    sections = document.querySelectorAll("div.section");
    if (sections !== null) {
        section = sections[sections.length-1];
        section.appendChild(script);
    }
}
commentsRunWhenDOMLoaded(addUtterances);
</script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ag3.0 (Ag1000G phase 3)" href="../ag3/ag3.0.html" />
    <link rel="prev" title="MalariaGEN vector data user guide" href="../landing-page.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/logo.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">MalariaGEN vector data user guide</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../landing-page.html">
                    MalariaGEN vector data user guide
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Malaria Vector Genome Observatory
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Vector Observatory Data Access
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Anopheles gambiae complex
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.0.html">
   Ag3.0 (Ag1000G phase 3)
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.1.html">
   Ag3.1
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.2.html">
   Ag3.2
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.3.html">
   Ag3.3
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.4.html">
   Ag3.4
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.5.html">
   Ag3.5
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.6.html">
   Ag3.6
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.7.html">
   Ag3.7
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.8.html">
   Ag3.8
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.9.html">
   Ag3.9
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.10.html">
   Ag3.10
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/ag3.11.html">
   Ag3.11
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/cloud.html">
   Ag3 cloud data access
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/download.html">
   Ag3 data downloads
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/methods.html">
   Ag3 sequencing and variant calling methods
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/api.html">
   Ag3 API
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../ag3/analysis.html">
   Ag3 analysis
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Anopheles funestus
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/af1.0.html">
   Af1.0  (
   <em>
    Anopheles funestus
   </em>
   Project Phase 1 Data Release)
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/af1.1.html">
   Af1.1
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/af1.2.html">
   Af1.2
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/af1.3.html">
   Af1.3
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/af1.4.html">
   Af1.4
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/cloud.html">
   Af1 cloud data access
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/download.html">
   Af1 data downloads
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../af1/api.html">
   Af1 API
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Anopheles minimus
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../amin1/intro.html">
   Amin1.0
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../amin1/cloud.html">
     Amin1.0 cloud data access
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../amin1/download.html">
     Amin1.0 data downloads
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../amin1/api.html">
   Amin1 API
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Partners and community
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../studies-ag1000g.html">
   Ag1000G partner studies
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Release notes
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../release-notes.html">
   Release notes
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<div class="menu-dropdown menu-dropdown-launch-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Launch interactive content">
      <i class="fas fa-rocket"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://gke.mybinder.org/v2/gh/malariagen/vector-data/master?urlpath=lab/tree/docs/vobs/vobs-data-access.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Launch on Binder"
>
  

<span class="headerbtn__icon-container">
  
    <img src="../_static/images/logo_binder.svg">
  </span>
<span class="headerbtn__text-container">Binder</span>
</a>

      </li>
      
      <li>
        <a href="https://colab.research.google.com/github/malariagen/vector-data/blob/master/docs/vobs/vobs-data-access.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Launch on Colab"
>
  

<span class="headerbtn__icon-container">
  
    <img src="../_static/images/logo_colab.png">
  </span>
<span class="headerbtn__text-container">Colab</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-repository-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Source repositories">
      <i class="fab fa-github"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://github.com/malariagen/vector-data"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Source repository"
>
  

<span class="headerbtn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="headerbtn__text-container">repository</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/malariagen/vector-data/issues/new?title=Issue%20on%20page%20%2Fvobs/vobs-data-access.html&body=Your%20issue%20content%20here."
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Open an issue"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="headerbtn__text-container">open issue</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="../_sources/vobs/vobs-data-access.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.ipynb</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#terms-of-use">
   Terms of Use
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#fair-usage">
   Fair Usage
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#data-access">
   Data Access
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-1-make-sure-you-have-a-google-account">
     Step 1. Make sure you have a Google Account
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-2-fill-out-the-data-access-request-form">
     Step 2. Fill out the data access request form
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-3-ensure-you-are-using-the-latest-version-of-the-malariagen-data-python-package">
     Step 3. Ensure you are using the latest version of the
     <code class="docutils literal notranslate">
      <span class="pre">
       malariagen_data
      </span>
     </code>
     Python package
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-4-set-up-google-cloud-authentication-credentials">
     Step 4. Set up Google Cloud authentication credentials
    </a>
   </li>
  </ul>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Vector Observatory Data Access</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#terms-of-use">
   Terms of Use
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#fair-usage">
   Fair Usage
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#data-access">
   Data Access
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-1-make-sure-you-have-a-google-account">
     Step 1. Make sure you have a Google Account
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-2-fill-out-the-data-access-request-form">
     Step 2. Fill out the data access request form
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-3-ensure-you-are-using-the-latest-version-of-the-malariagen-data-python-package">
     Step 3. Ensure you are using the latest version of the
     <code class="docutils literal notranslate">
      <span class="pre">
       malariagen_data
      </span>
     </code>
     Python package
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#step-4-set-up-google-cloud-authentication-credentials">
     Step 4. Set up Google Cloud authentication credentials
    </a>
   </li>
  </ul>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="vector-observatory-data-access">
<h1>Vector Observatory Data Access<a class="headerlink" href="#vector-observatory-data-access" title="Permalink to this headline">#</a></h1>
<p>MalariaGEN data resources provide an integrated view of malaria vector genomes from across the globe. These data are available to everyone to benefit the science and surveillance of malaria. You can find more information on the vector data resources at <a class="reference external" href="https://www.malariagen.net/mosquito/">https://www.malariagen.net/mosquito/</a>.</p>
<p>Vector Observatory data are stored in Google Cloud Storage (GCS) in the US region. The current set-up requires users to request access and authenticate prior to accessing data.</p>
<section id="terms-of-use">
<h2>Terms of Use<a class="headerlink" href="#terms-of-use" title="Permalink to this headline">#</a></h2>
<p>Data in the Vector Observatory are organised into data releases. All data releases can be accessed for public health and educational purposes as soon as they are released. However, please note that data releases are subject to terms of use which may include an embargo on all public communications including academic publications. The terms of use for each data release can be found on the MalariaGEN website.</p>
</section>
<section id="fair-usage">
<h2>Fair Usage<a class="headerlink" href="#fair-usage" title="Permalink to this headline">#</a></h2>
<p>Vector Observatory data are currently stored in Google Cloud Storage (GCS) in the US region. Access to Vector Observatory data in Google Cloud is free for all users. However, large transfers of data outside of Google Cloud in the US region substantially increase our running costs, and so we ask users to adhere to the following fair usage policy. This will allow us to continue making the data freely available.</p>
<ul class="simple">
<li><p><strong>Data access from Google Colab -</strong> If you are using Google Colab to access data, please check if your allocated virtual machine (VM) is within the US region. If not, please request a new VM by selecting “Runtime &gt; Disconnect and delete runtime” from the Colab menu.</p></li>
<li><p><strong>Data access from other Google Cloud services -</strong> If you are using another Google Cloud service such as Vertex AI Workbench, or are using a third party service such as Terra or Coiled which uses VMs within Google Cloud, please ensure that VMs are provisioned within the US region.</p></li>
<li><p><strong>Data access from outside Google Cloud -</strong> If you are planning to access data from any computer or VM located outside of Google Cloud, please contact us at support&#64;malariagen.net. We can then advise on the most efficient methods for accessing data to both minimise our running costs and ensure you get the best performance.</p></li>
</ul>
<p>Please note that we monitor data access logs to detect any unexpected large data transfers outside of Google Cloud in the US region, and may temporarily suspend access to users performing large data transfers. If we do suspend access, we will reach out to you to see if we can help optimise your data access.</p>
</section>
<section id="data-access">
<h2>Data Access<a class="headerlink" href="#data-access" title="Permalink to this headline">#</a></h2>
<p>To access data from the Vector Observatory, you will need to follow these steps:</p>
<section id="step-1-make-sure-you-have-a-google-account">
<h3>Step 1. Make sure you have a Google Account<a class="headerlink" href="#step-1-make-sure-you-have-a-google-account" title="Permalink to this headline">#</a></h3>
<p>To allow us to configure data access permissions, you will need to provide us with an email address that is associated with a Google account. This could be a standard Google (i.e., Gmail) account, or alternatively it could be your work email address if your employer uses Google Workspace.</p>
</section>
<section id="step-2-fill-out-the-data-access-request-form">
<h3>Step 2. Fill out the data access request form<a class="headerlink" href="#step-2-fill-out-the-data-access-request-form" title="Permalink to this headline">#</a></h3>
<p>Please fill out and submit the following form:</p>
<blockquote>
<div><p><a class="reference external" href="https://forms.gle/kCqistorZyxaU4LP7">MalariaGEN cloud data access form</a></p>
</div></blockquote>
<p>All requests for data access will be granted subject to verification checks and agreement to reasonable use. This is to ensure that the data resources remain accessible to everyone. Submitting this form will allow us to configure storage permissions and monitor storage for excessive network usage in future.</p>
</section>
<section id="step-3-ensure-you-are-using-the-latest-version-of-the-malariagen-data-python-package">
<h3>Step 3. Ensure you are using the latest version of the <code class="docutils literal notranslate"><span class="pre">malariagen_data</span></code> Python package<a class="headerlink" href="#step-3-ensure-you-are-using-the-latest-version-of-the-malariagen-data-python-package" title="Permalink to this headline">#</a></h3>
<p>If you access data via the <code class="docutils literal notranslate"><span class="pre">malariagen_data</span></code> Python package, please upgrade to version 9.0 or higher. These versions will automatically use your authentication credentials when accessing data in Google Cloud.</p>
</section>
<section id="step-4-set-up-google-cloud-authentication-credentials">
<h3>Step 4. Set up Google Cloud authentication credentials<a class="headerlink" href="#step-4-set-up-google-cloud-authentication-credentials" title="Permalink to this headline">#</a></h3>
<p>If you are only accessing data via the <code class="docutils literal notranslate"><span class="pre">malariagen_data</span></code> Python package from within Google Colab, you can skip this step, because authentication credentials will be obtained automatically. If you have filled out the form but having issues authenticating in Google Colab, you can find a walkthrough video <a class="reference external" href="https://drive.google.com/file/d/1dOPimdsPabvOoWkKdomWl8v24LRvjiaj/view?usp=sharing">here</a>.</p>
<p>If you are accessing data from any other location, you will need to authenticate with Google Cloud. To do this, you will need to:</p>
<ol class="simple">
<li><p>Install the <a class="reference external" href="https://cloud.google.com/sdk/gcloud">Google Cloud CLI</a>. See the details in the Google Documentation <a class="reference external" href="https://cloud.google.com/sdk/docs/install-sdk">here</a>.</p></li>
<li><p>Check <code class="docutils literal notranslate"><span class="pre">gcloud</span></code> is installed correctly:</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>gcloud<span class="w"> </span><span class="nb">help</span>
</pre></div>
</div>
<ol class="simple">
<li><p>Authenticate using <code class="docutils literal notranslate"><span class="pre">gcloud</span></code>:</p></li>
</ol>
<ul class="simple">
<li><p>If you need to authenticate to use the <code class="docutils literal notranslate"><span class="pre">malariagen_data</span></code> package, you will need to use the following command:</p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>gcloud<span class="w"> </span>auth<span class="w"> </span>application-default<span class="w"> </span>login
</pre></div>
</div>
<ul class="simple">
<li><p>If you need to authenticate to access Google Cloud Storage from the command line using <code class="docutils literal notranslate"><span class="pre">gsutil</span></code>, you will need to use the following command:</p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>gcloud<span class="w"> </span>auth<span class="w"> </span>login
</pre></div>
</div>
<p>If you have any questions, please contact us at: <a class="reference external" href="mailto:support&#37;&#52;&#48;malariagen&#46;net">mailto:support<span>&#64;</span>malariagen<span>&#46;</span>net</a></p>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./vobs"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="../landing-page.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">MalariaGEN vector data user guide</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="../ag3/ag3.0.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Ag3.0 (Ag1000G phase 3)</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By MalariaGEN<br/>
  
      &copy; Copyright 2021.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>